<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Payne" />

<meta name="date" content="2024-10-20" />

<title>DE Genes by Disease Condition</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myocarditis snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DE Genes by Disease Condition</h1>
<h4 class="author">Emily Payne</h4>
<h4 class="date">2024-10-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-27
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Myocarditis snRNAseq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240909code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240909)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240909code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240909)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat
files/Myocarditis_RDS_04.10.2024.rds
</td>
<td style="text-align:left;">
data/01 Merged Seurat files/Myocarditis_RDS_04.10.2024.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis
snRNAseq/docs/Markers/DEGenes_condition pos. and neg. AM
</td>
<td style="text-align:left;">
docs/Markers/DEGenes_condition pos. and neg. AM
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/ego AM
20.10.2024
</td>
<td style="text-align:left;">
docs/Markers/ego AM 20.10.2024
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/ego CMP
20.10.2024
</td>
<td style="text-align:left;">
docs/Markers/ego CMP 20.10.2024
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/ego
Sarcoidosis 20.10.2024
</td>
<td style="text-align:left;">
docs/Markers/ego Sarcoidosis 20.10.2024
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/ego
Healthy 20.10.2024
</td>
<td style="text-align:left;">
docs/Markers/ego Healthy 20.10.2024
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree4daf12687e1b1561624f77778fa38396be0bb8detargetblank4daf126a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/4daf12687e1b1561624f77778fa38396be0bb8de" target="_blank">4daf126</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree4daf12687e1b1561624f77778fa38396be0bb8detargetblank4daf126a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/4daf12687e1b1561624f77778fa38396be0bb8de" target="_blank">4daf126</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/00 Inbox Documents and Pictures /
    Ignored:    data/01 Merged Seurat files/
    Ignored:    data/AM/
    Ignored:    data/Berlin/
    Ignored:    data/CM/
    Ignored:    data/Healthy/
    Ignored:    data/Sarcoidosis/

Untracked files:
    Untracked:  analysis/AM subgroups vs. Healthy.Rmd
    Untracked:  analysis/DE Genes by Disease Condition.Rmd
    Untracked:  analysis/DE genes AMsub vs. Healthy.Rmd
    Untracked:  analysis/DE genes AMsub vs. Healthy_V2.Rmd
    Untracked:  analysis/DE-Genes-by-Disease-Condition.Rmd
    Untracked:  analysis/Fibroblast subset V2 reclustering.Rmd
    Untracked:  analysis/Install.packages_MFZ.R
    Untracked:  analysis/MCP and TC reembedding.Rmd
    Untracked:  analysis/Macrophage DE subset V2.Rmd
    Untracked:  analysis/Macrophage Reembedding.Rmd
    Untracked:  analysis/Macrophage embedding.Rmd
    Untracked:  analysis/Macrophage heatmap.Rmd
    Untracked:  analysis/Merging/
    Untracked:  analysis/Myocarditis snRNAseq QC and merging_V2_Tchigh vs. low.Rmd
    Untracked:  analysis/Preliminary QC/
    Untracked:  analysis/T cell abundance DE Kopie.Rmd
    Untracked:  analysis/T cell abundance DE.Rmd
    Untracked:  analysis/T cell condition Differential expression.Rmd
    Untracked:  analysis/TC high vs. low condition.Rmd
    Untracked:  analysis/TC high vs. low.Rmd
    Untracked:  analysis/Tcell embedding.Rmd
    Untracked:  analysis/Tcell high vs. Tcell low.Rmd
    Untracked:  analysis/V1/

Unstaged changes:
    Modified:   analysis/Myocarditis snRNAseq.Rmd
    Deleted:    analysis/license.Rmd
    Deleted:    data/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(ExploreSCdataSeurat3)
library(runSeurat3)
library(Seurat)</code></pre>
<pre><code>Loading required package: SeuratObject</code></pre>
<pre><code>Loading required package: sp</code></pre>
<pre><code>&#39;SeuratObject&#39; was built under R 4.4.0 but the current version is 4.4.1; it is recomended
that you reinstall &#39;SeuratObject&#39; as the ABI for R may have changed</code></pre>
<pre><code>
Attaching package: &#39;SeuratObject&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, t</code></pre>
<pre class="r"><code>library(ggpubr)</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre class="r"><code>library(pheatmap)
library(SingleCellExperiment)</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: MatrixGenerics</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse, colCounts, colCummaxs,
    colCummins, colCumprods, colCumsums, colDiffs, colIQRDiffs, colIQRs, colLogSumExps,
    colMadDiffs, colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds, colSums2,
    colTabulates, colVarDiffs, colVars, colWeightedMads, colWeightedMeans,
    colWeightedMedians, colWeightedSds, colWeightedVars, rowAlls, rowAnyNAs, rowAnys,
    rowAvgsPerColSet, rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps, rowMadDiffs, rowMads,
    rowMaxs, rowMeans2, rowMedians, rowMins, rowOrderStats, rowProds, rowQuantiles,
    rowRanges, rowRanks, rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs,
    rowVars, rowWeightedMads, rowWeightedMeans, rowWeightedMedians, rowWeightedSds,
    rowWeightedVars</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:SeuratObject&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind, colnames, dirname,
    do.call, duplicated, eval, evalq, Filter, Find, get, grep, grepl, intersect,
    is.unsorted, lapply, Map, mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, Position, rank, rbind, Reduce, rownames, sapply, setdiff, table, tapply,
    union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:sp&#39;:

    %over%</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with &#39;browseVignettes()&#39;. To cite
    Bioconductor, see &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>
Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:MatrixGenerics&#39;:

    rowMedians</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    anyMissing, rowMedians</code></pre>
<pre><code>
Attaching package: &#39;SummarizedExperiment&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:Seurat&#39;:

    Assays</code></pre>
<pre><code>The following object is masked from &#39;package:SeuratObject&#39;:

    Assays</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    combine</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:matrixStats&#39;:

    count</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────────────────────────── tidyverse 2.0.0 ──
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ lubridate 1.9.3     ✔ tibble    3.2.1
✔ purrr     1.0.2     ✔ tidyr     1.3.1
✔ readr     2.1.5     </code></pre>
<pre><code>── Conflicts ────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ lubridate::%within%()    masks IRanges::%within%()
✖ dplyr::collapse()        masks IRanges::collapse()
✖ dplyr::combine()         masks Biobase::combine(), BiocGenerics::combine()
✖ dplyr::count()           masks matrixStats::count()
✖ dplyr::desc()            masks IRanges::desc()
✖ tidyr::expand()          masks S4Vectors::expand()
✖ dplyr::filter()          masks stats::filter()
✖ dplyr::first()           masks S4Vectors::first()
✖ dplyr::lag()             masks stats::lag()
✖ BiocGenerics::Position() masks ggplot2::Position(), base::Position()
✖ purrr::reduce()          masks GenomicRanges::reduce(), IRanges::reduce()
✖ dplyr::rename()          masks S4Vectors::rename()
✖ lubridate::second()      masks S4Vectors::second()
✖ lubridate::second&lt;-()    masks S4Vectors::second&lt;-()
✖ dplyr::slice()           masks IRanges::slice()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(viridis)</code></pre>
<pre><code>Loading required package: viridisLite</code></pre>
<pre class="r"><code>library(muscat)
library(circlize)</code></pre>
<pre><code>========================================
circlize version 0.4.16
CRAN page: https://cran.r-project.org/package=circlize
Github page: https://github.com/jokergoo/circlize
Documentation: https://jokergoo.github.io/circlize_book/book/

If you use it in published research, please cite:
Gu, Z. circlize implements and enhances circular visualization
  in R. Bioinformatics 2014.

This message can be suppressed by:
  suppressPackageStartupMessages(library(circlize))
========================================</code></pre>
<pre class="r"><code>library(destiny)</code></pre>
<pre><code>
Attaching package: &#39;destiny&#39;

The following object is masked from &#39;package:SummarizedExperiment&#39;:

    distance

The following object is masked from &#39;package:GenomicRanges&#39;:

    distance

The following object is masked from &#39;package:IRanges&#39;:

    distance</code></pre>
<pre class="r"><code>library(scater)</code></pre>
<pre><code>Loading required package: scuttle</code></pre>
<pre class="r"><code>library(metap)
library(multtest)
library(clusterProfiler)</code></pre>
<pre><code>
clusterProfiler v4.12.6 Learn more at https://yulab-smu.top/contribution-knowledge-mining/

Please cite:

S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L
Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024,
doi:10.1038/s41596-024-01020-z

Attaching package: &#39;clusterProfiler&#39;

The following object is masked from &#39;package:purrr&#39;:

    simplify

The following object is masked from &#39;package:IRanges&#39;:

    slice

The following object is masked from &#39;package:S4Vectors&#39;:

    rename

The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)</code></pre>
<pre><code>Loading required package: AnnotationDbi

Attaching package: &#39;AnnotationDbi&#39;

The following object is masked from &#39;package:clusterProfiler&#39;:

    select

The following object is masked from &#39;package:dplyr&#39;:

    select</code></pre>
<pre class="r"><code>library(msigdbr)
library(enrichplot)</code></pre>
<pre><code>
Attaching package: &#39;enrichplot&#39;

The following object is masked from &#39;package:ggpubr&#39;:

    color_palette</code></pre>
<pre class="r"><code>library(DOSE)</code></pre>
<pre><code>DOSE v3.30.5  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use DOSE in published research, please cite:
Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
<pre class="r"><code>library(grid)
library(gridExtra)</code></pre>
<pre><code>
Attaching package: &#39;gridExtra&#39;

The following object is masked from &#39;package:dplyr&#39;:

    combine

The following object is masked from &#39;package:Biobase&#39;:

    combine

The following object is masked from &#39;package:BiocGenerics&#39;:

    combine</code></pre>
<pre class="r"><code>library(ggupset)
library(VennDiagram)</code></pre>
<pre><code>Loading required package: futile.logger

Attaching package: &#39;VennDiagram&#39;

The following object is masked from &#39;package:ggpubr&#39;:

    rotate</code></pre>
<pre class="r"><code>library(NCmisc)
library(RColorBrewer)
library(EnhancedVolcano)</code></pre>
<pre><code>Loading required package: ggrepel</code></pre>
<pre class="r"><code>library(textshaping)
library(wordcloud)
library(pathview)</code></pre>
<pre><code>##############################################################################
Pathview is an open source software package distributed under GNU General
Public License version 3 (GPLv3). Details of GPLv3 is available at
http://www.gnu.org/licenses/gpl-3.0.html. Particullary, users are required to
formally cite the original Pathview paper (not just mention it) in publications
or products. For details, do citation(&quot;pathview&quot;) within R.

The pathview downloads and uses KEGG data. Non-academic uses may require a KEGG
license agreement (details at http://www.kegg.jp/kegg/legal.html).
##############################################################################</code></pre>
<pre class="r"><code>##load merged file 
seuratM &lt;- readRDS(&quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_RDS_04.10.2024.rds&quot;)

table(seuratM$dataset)</code></pre>
<pre><code>
     312581_05-5_20230308_Hu_nucseq_SG36_EMB_GEM      312581_06-6_20230308_Hu_nucseq_SG37_EMB_GEM 
                                            4323                                             1123 
   353921_28-28_20240529_Hu_nucseq_Immpath_EMB18    353921_29-29_20240529_Hu_nucseq_Immpath_EMB24 
                                             208                                              105 
   353921_30-30_20240529_Hu_nucseq_Immpath_EMB25    353921_31-31_20240529_Hu_nucseq_Immpath_EMB27 
                                             227                                              579 
   353921_32-32_20240529_Hu_nucseq_Immpath_EMB29    353921_33-33_20240529_Hu_nucseq_Immpath_EMB30 
                                             335                                              730 
   353921_34-34_20240529_Hu_nucseq_Immpath_EMB35    353921_35-35_20240529_Hu_nucseq_Immpath_EMB37 
                                            1217                                              604 
359861_09-11_20240723_Hu_nucseq_Ber_Pat5_EM979_1 359861_10-13_20240723_Hu_nucseq_Ber_Pat5_EM979_2 
                                             771                                             1476 
 359861_11-1_20240730_Hu_nucseq_Ber_Pat1_EMB1537  359861_12-2_20240730_Hu_nucseq_Ber_Pat2_EMB1041 
                                             316                                              608 
    359861_16-8_20240730_Hu_nucseq_Immpath_EMB47     359861_17-9_20240730_Hu_nucseq_Immpath_EMB48 
                                             513                                              426 
     o27533_1_12-12_20220203_Hu_nucseq_EMB31_GEM        o27936_1_7-7_20220309_Hu_nucseq_EMB32_GEM 
                                             236                                             1192 
 o28576_1_01-1_20220525_Hu_nucseq_Graz_1_EMB_GEM  o28576_1_02-2_20220525_Hu_nucseq_Graz_2_EMB_GEM 
                                            2740                                             1684 
 o28576_1_03-3_20220525_Hu_nucseq_Graz_3_EMB_GEM  o28576_1_04-4_20220525_Hu_nucseq_Graz_4_EMB_GEM 
                                            2396                                              545 
 o28576_1_05-5_20220525_Hu_nucseq_Graz_5_EMB_GEM  o28576_1_06-6_20220525_Hu_nucseq_Graz_6_EMB_GEM 
                                             781                                              491 
 o28576_1_07-7_20220525_Hu_nucseq_Graz_7_EMB_GEM   o28576_1_08-8_20220525_Hu_nucseq_Graz_8_HH_GEM 
                                             653                                             3921 
o28576_1_12-12_20220525_Hu_nucseq_Graz_12_HH_GEM      o28576_1_13-13_20220525_Hu_nucseq_EMB32_GEM 
                                            3818                                             1428 
    o292731_1-1_20220818_Hu_nucseq_Graz_9_HH_GEM    o292731_2-2_20220818_Hu_nucseq_Graz_13_HH_GEM 
                                            4908                                             9882 
    o292731_3-3_20220818_Hu_nucseq_SG_33_EMB_GEM     o292731_4-4_20220818_Hu_nucseq_SG_34_EMB_GEM 
                                            6286                                              620 
    o292731_5-5_20220818_Hu_nucseq_SG_35_EMB_GEM   o294781_01-1_20220912_Hu_nucseq_Graz_21_HH_GEM 
                                            2363                                             1442 
  o294781_02-2_20220912_Hu_nucseq_Graz_22_HH_GEM   o294781_03-3_20220912_Hu_nucseq_Graz_23_HH_GEM 
                                            1998                                              841 
  o294781_04-4_20220912_Hu_nucseq_Graz_24_HH_GEM  o294781_05-5_20220912_Hu_nucseq_Graz_14_EMB_GEM 
                                            1480                                             1268 
 o294781_06-6_20220912_Hu_nucseq_Graz_15_EMB_GEM  o294781_09-9_20220912_Hu_nucseq_Graz_18_EMB_GEM 
                                            4439                                             2280 
o294781_10-10_20220912_Hu_nucseq_Graz_19_EMB_GEM o294781_11-11_20220912_Hu_nucseq_Graz_20_EMB_GEM 
                                             111                                             2706 </code></pre>
<pre class="r"><code>table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11    12    13    14 
17699 13639 12754  8222  5475  4836  3723  2365  2156   905   882   694   380   249    91 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
74070 </code></pre>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$RNA_snn_res.0.25
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/Plot%20umap-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>unique(seuratM$clusterName)</code></pre>
<pre><code> [1] PeriFb1       Fb1           VSMC          BEC1          Cardiomyocyte Fb3          
 [7] TC            MCP1          BEC2          INT1          MCP2          NC           
[13] Fb2           AdipoC        PeriFb2      
Levels: Cardiomyocyte Fb1 Fb2 Fb3 INT1 PeriFb1 PeriFb2 VSMC NC BEC1 BEC2 AdipoC MCP1 MCP2 TC</code></pre>
<pre class="r"><code>#### Rename clusters

seuratM$clusterName &lt;- &quot;clusterName&quot;
Idents(seuratM) &lt;- seuratM$clusterName
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;0&quot; )] &lt;- &quot;Fb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;1&quot; )] &lt;- &quot;BEC1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;2&quot; )] &lt;- &quot;PeriFb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;3&quot; )] &lt;- &quot;Cardiomyocyte&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;4&quot; )] &lt;- &quot;MCP1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;5&quot; )] &lt;- &quot;TC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;6&quot; )] &lt;- &quot;MCP2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;7&quot; )] &lt;- &quot;BEC2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;8&quot; )] &lt;- &quot;VSMC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;9&quot; )] &lt;- &quot;NC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;10&quot; )] &lt;- &quot;INT1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;11&quot; )] &lt;- &quot;Fb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;12&quot; )] &lt;- &quot;Fb3&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;13&quot; )] &lt;- &quot;PeriFb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;14&quot; )] &lt;- &quot;AdipoC&quot;


colclusterName &lt;- c(&quot;#D53E4F&quot;, &quot;#f4a582&quot;, &quot;#FEE08B&quot;,&quot;#feb24c&quot;,&quot;#67001f&quot;, &quot;#01665e&quot;,&quot;#66C2A5&quot;,&quot;#c7eae5&quot;,&quot;#BEAED4&quot;, &quot;#355C7D&quot;,&quot;#3288BD&quot;,&quot;#8c510a&quot; ,&quot;#fde0dd&quot;,&quot;#B45B5C&quot;,&quot;#dd1c77&quot;)

                    
names(colclusterName) &lt;- c(&quot;Cardiomyocyte&quot;,&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;,&quot;NC&quot;, &quot;BEC1&quot;,&quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;,&quot;MCP2&quot;,&quot;TC&quot;)

colpat_sub&lt;- c(&quot;#dfc27d&quot;,&quot;#BE3144&quot;,&quot;#355C7D&quot;, &quot;#779d8d&quot;)

seuratM$clusterName &lt;- factor(seuratM$clusterName, levels=c(&quot;Cardiomyocyte&quot;, &quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;INT1&quot;,&quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;, &quot;NC&quot;, &quot;BEC1&quot;, &quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;, &quot;MCP2&quot;, &quot;TC&quot;))


Idents(seuratM) &lt;- seuratM$clusterName
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE, cols = colclusterName)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>Idents(seuratM) &lt;-seuratM$pat_sub

colpat_sub&lt;- c(&quot;#779d8d&quot;,&quot;#BE3144&quot;,&quot;#355C7D&quot;,&quot;#dfc27d&quot;)
names(colpat_sub) &lt;- c(&quot;Healthy&quot;, &quot;AM&quot;, &quot;CMP&quot;, &quot;Sarcoidosis&quot;)

DimPlot(seuratM, reduction = &quot;umap&quot;, cols = colpat_sub)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>table(seuratM$pat_sub)</code></pre>
<pre><code>
    Healthy          AM         CMP Sarcoidosis 
      28290       29076       11635        5069 </code></pre>
<pre class="r"><code>table(seuratM$patient)</code></pre>
<pre><code>
    Healthy 01     Healthy 02     Healthy 03     Healthy 04     Healthy 05     Healthy 06 
          3921           3818           4908           9882           1442           1998 
    Healthy 07     Healthy 08          AM 01          AM 02          AM 03          AM 04 
           841           1480           6286           4323           1123            208 
         AM 05          AM 06          AM 07          AM 08          AM 09          AM 10 
           579            335           1217            604            513            426 
         AM 11          AM 12          AM 13          AM 14          AM 15          AM 16 
           236           1192           2740           1684           4439           2247 
         AM 17          AM 18         CMP 01         CMP 02         CMP 03         CMP 04 
           316            608            105            227            730           1428 
        CMP 05         CMP 06         CMP 07         CMP 08         CMP 09         CMP 10 
           620            653           1268           2280           2396            545 
        CMP 11         CMP 12         CMP 13 Sarcoidosis 01 Sarcoidosis 02 
           781            491            111           2363           2706 </code></pre>
<pre class="r"><code># DE Genes for conditions: AM, CMP, Sarcoidosis, Healthy

DEGenesCond &lt;- FindAllMarkers (seuratM, only.pos = TRUE, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.05)</code></pre>
<pre><code>Calculating cluster Healthy</code></pre>
<pre><code>Calculating cluster AM</code></pre>
<pre><code>Calculating cluster CMP</code></pre>
<pre><code>Calculating cluster Sarcoidosis</code></pre>
<pre class="r"><code>#write.table(DEGenesCond, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenes_condition pos. and neg. AM, Healthy, CMP, Sarcoidosis&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)


##adjust table
DEGenesCond_adj &lt;- DEGenesCond %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

##GSEA for AM (acute Myocarditis) condition
DEGenesAM &lt;- DEGenesCond_adj %&gt;% filter(cluster == &quot;AM&quot;)
#head(DEGenesAM)

egoAM &lt;- enrichGO(gene = unique(DEGenesAM$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM &lt;- setReadable(egoAM, OrgDb = org.Hs.eg.db)
dotplot(egoAM, showCategory=20, title = &quot;Acute Myocarditis&quot;, font.size = 10)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>barplot(egoAM, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Acute Myocarditis&quot;,
        font.size = 10)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>#View(DEGenesAM)
#View(egoAM@result)
#write.table(egoAM, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/ego AM 20.10.2024&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

# PTPRC/B2M/ATM/IRF1/HLA-DRA/NCKAP1L/DOCK10/IKZF1/CD2/DOCK2/IL7R/JAK3/ITGA4/PTK2B/BCL11B/IKZF3/RUNX3/PTPN22/ITK/CD74/FGL2/LAG3/LCK/SPN/ZAP70/IL2RG/IRF8/SLAMF8/ZBTB16/IL4I1/LILRB4/CD3G/KLRC1/CD3E/RHOH/LILRB2/INPP5D/CAMK4/HLA-DRB1/PTPN6/TNFSF13B/BTK/CYLD/JAK1/PIK3CD/PTPRJ/EZH2/CD8A/CD3D/THEMIS/ACTB/TCIRG1/SLAMF1/IL12RB1/CD4/CD1D/PREX1/SASH3/ZNF683/SPI1/TBX21/LY9/FANCA/SOS1/ADAM8/ZFP36L2/STAT4/DOCK11/SLAMF6/KMT2A/CARD11/PTPN2/VAV1/ASCL2/IL15/HLA-DOA/MS4A1/TESPA1/TGFB1/FCER1G/BTN2A2/LGALS9/NOTCH2/CCR2/TOP2B/SOCS1/STAT6/MR1/CTLA4/TMEM131L/GATA3/SYK/NFAM1/NFKBID/EP300/CD27/EOMES/BLNK/SEMA4A/BCL2/SMARCA2/TOX/NFATC3/BATF/CD86/LFNG/IL6R/PBRM1/MALT1/LEF1/CD83/TNFRSF9/PRKDC/DCLRE1C/IL18R1/STAT5B/FANCD2/AP3B1/SMARCC1/SP3/RUNX2/PRDM1/ZBTB1/RASGRP1/CCR7/CDK6/PPP3CB/IRF4/IL18/ZBTB7A/PTGER4/RC3H2/LIG4/ARID1A/AP3D1/RIPK3/PLCG2/PPP2R3C/PLCL2/RORA/SOS2/KLF6/DROSHA/BAX/HDAC4/ZC3H12A/RC3H1/DUSP10/RSAD2/RUNX1/STK11/CBFB/YY1/NDFIP1/IL15RA/NCAPH2/TNFSF8/SMARCD2/RHOA/CD28/LIPA/PHF10/ZBTB7B/RORC/TP53/RIPK2/METTL3/TPD52/CMTM7/VSIR/PIK3R6/MSH2/PRELID1/PCID2/RNF41/FZD5/HHEX/GPR89B/SART1/BAK1/GPR183/MDK/KAT5/MEN1/PDP2/GPR89B/SRF/XBP1/MFNG/PNP/PATZ1/TCF7/HDAC9/NLRP3/CD79B/MIR17HG/DNAJB9/ZFPM1/ARMC5</code></pre>
<pre class="r"><code>## DEGenes for CMP (Cardiomyopathy) condition
DEGenesCMP &lt;- DEGenesCond_adj %&gt;% filter(cluster == &quot;CMP&quot;)
head(DEGenesCMP)</code></pre>
<pre><code>                                   p_val avg_log2FC pct.1 pct.2     p_val_adj cluster   gene
ENSG00000175206.NPPA       5.092720e-188  1.9501280 0.054 0.013 1.991966e-183     CMP   NPPA
ENSG00000090006.LTBP41     1.385007e-175  0.8901326 0.369 0.253 5.417315e-171     CMP  LTBP4
ENSG00000140465.CYP1A1     1.073674e-146  6.2442944 0.013 0.000 4.199568e-142     CMP CYP1A1
ENSG00000273448.AC006480.2 2.105039e-146  1.0284916 0.155 0.080 8.233648e-142     CMP      2
ENSG00000124440.HIF3A1     8.812941e-139  0.7602927 0.343 0.239 3.447094e-134     CMP  HIF3A
ENSG00000251322.SHANK3     4.498541e-134  1.0214967 0.267 0.176 1.759559e-129     CMP SHANK3
                                                 Gene           EnsID
ENSG00000175206.NPPA             ENSG00000175206.NPPA ENSG00000175206
ENSG00000090006.LTBP41          ENSG00000090006.LTBP4 ENSG00000090006
ENSG00000140465.CYP1A1         ENSG00000140465.CYP1A1 ENSG00000140465
ENSG00000273448.AC006480.2 ENSG00000273448.AC006480.2 ENSG00000273448
ENSG00000124440.HIF3A1          ENSG00000124440.HIF3A ENSG00000124440
ENSG00000251322.SHANK3         ENSG00000251322.SHANK3 ENSG00000251322</code></pre>
<pre class="r"><code>egoCMP &lt;- enrichGO(gene = unique(DEGenesCMP$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP &lt;- setReadable(egoCMP, OrgDb = org.Hs.eg.db)
dotplot(egoCMP, showCategory=20, title = &quot;Chronic Cardiomyopathy&quot;, font.size = 10)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>barplot(egoCMP, 
        drop = TRUE, 
        showCategory = 7, 
        title = &quot;Chronic Cardiomyopathy&quot;,
        font.size = 10)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>#View(DEGenesCMP)
#View(egoCMP@result)
#write.table(egoCMP, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/ego CMP 20.10.2024&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

# top DE genes: PTPRC/STAT3/JUNB/NCKAP1L/CD74/PBX1/IL7R/CD2/HLA-DRA/RUNX1/DOCK2/JAK3/SOX4/B2M/DOCK10/LILRB2/IRF1/ITGA4/RUNX3/PTK2B/IRF8/HLA-DRB1/ZBTB16/LILRB4/BCL11B/LAG3/IKZF1/SOCS3/MAFB/SLAMF8/FOSL2/SMAD7/SOX13/PTPN22/IL4I1/PTPN2/BTK/FGL2/BCL3/HMGB1/SOCS1/KLRC1/NRARP/PTPRJ/RBPJ/LCK/ITK/IKZF3/SYK/ITM2A/IL2RA/GAS6/KAT2A/JAG2/VAV1/TYRO3/RELB/IL2RG/DLL4/HLA-DOA/CD1D/PREX1/CR1/BMP4/CD4/PTPN6/BCL6/SPI1/INPP5D/RHOH/PIK3CD/TNFSF13B/IL18/TSC1/BLNK/TMEM98/CAMK4/CD8A/KLF6/AP3B1/ZEB1/ZMIZ1/FOXP1/SLAMF1/FCER1G/STAT4/LEPR/EGR3/TOP2B/CD3E/BRD7/NDFIP1/ZAP70/EZH2/KIT/LMBR1L/LGALS9/BTN2A2/IRF2BP2/STK11/IL12RB1/CD3G/HDAC5/NFKBID/EGR1/SOS2/ERBB2/TNFSF8/THEMIS/FANCA/SASH3/NLRP3/CD86/VCAM1/CD3D/CD28/IL15/CARD11/SMARCD3/GLI2/LY9/ZC3H12A/DCAF1/ZFPM1/PRDM1/SLAMF6/SPN/TBX21/MR1/ADAM8/BMI1/ATG5/LFNG/NFAM1/SEMA4A/FZD7/CTSL/DLL1/IL4R/PHF10/MMP14/CYLD/FANCD2/ID2/SMARCD2/NHEJ1/EP300/FCGR2B/LRRC8A/ST3GAL1</code></pre>
<pre class="r"><code>## DEGenes for Sarcoidosis condition
DEGenesSarco &lt;- DEGenesCond_adj %&gt;% filter(cluster == &quot;Sarcoidosis&quot;)
head(DEGenesSarco)</code></pre>
<pre><code>                                    p_val avg_log2FC pct.1 pct.2     p_val_adj     cluster
ENSG00000134954.ETS1         0.000000e+00   1.188578 0.476 0.238  0.000000e+00 Sarcoidosis
ENSG00000172965.MIR4435-2HG  0.000000e+00   1.144275 0.438 0.210  0.000000e+00 Sarcoidosis
ENSG00000006327.TNFRSF12A    0.000000e+00   2.227723 0.124 0.026  0.000000e+00 Sarcoidosis
ENSG00000250337.PURPL       3.438701e-299   3.175320 0.058 0.006 1.345013e-294 Sarcoidosis
ENSG00000272789.AC010976.21 4.395357e-290   1.870517 0.161 0.044 1.719200e-285 Sarcoidosis
ENSG00000180229.HERC2P3     3.950254e-281   1.907049 0.164 0.046 1.545102e-276 Sarcoidosis
                                   gene                        Gene           EnsID
ENSG00000134954.ETS1               ETS1        ENSG00000134954.ETS1 ENSG00000134954
ENSG00000172965.MIR4435-2HG MIR4435-2HG ENSG00000172965.MIR4435-2HG ENSG00000172965
ENSG00000006327.TNFRSF12A     TNFRSF12A   ENSG00000006327.TNFRSF12A ENSG00000006327
ENSG00000250337.PURPL             PURPL       ENSG00000250337.PURPL ENSG00000250337
ENSG00000272789.AC010976.21           2  ENSG00000272789.AC010976.2 ENSG00000272789
ENSG00000180229.HERC2P3         HERC2P3     ENSG00000180229.HERC2P3 ENSG00000180229</code></pre>
<pre class="r"><code>egoSarco &lt;- enrichGO(gene = unique(DEGenesSarco$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco &lt;- setReadable(egoSarco, OrgDb = org.Hs.eg.db)
dotplot(egoSarco, showCategory=20, title = &quot;Cardiac Sarcoidosis&quot;, font.size = 10)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>barplot(egoSarco, 
        drop = TRUE, 
        showCategory = 7, 
        title = &quot;Cardiac Sarcoidosis&quot;,
        font.size = 10)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>#View(DEGenesSarco)
#View(egoSarco@result)
#write.table(egoSarco, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/ego Sarcoidosis 20.10.2024&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)


# ETS1, TNFRSF12A.2,  CD34/THBS1/PDGFB/JAG1/MICALL2/VWF/ANXA2/PTPRA/ITGA1/CCN1/CCN2/KDR/ADAM15/FN1/CDH13/TNXB/LIMS2/COL8A1/ADAMTS9/ITGA3/FLNA/ITGA6/BCAM/LAMA5/SERPINE1/DAPK3/PEAK1/TEK/DOCK1/PTK2/ITGB1/FUT1/ACTG1/SMAD6/RIN2/P4HB/RHOA/PPM1F/NOTCH1/ACTN4/MAP4K4/PXN/CTNNB1/ACTN1/PKP2/SIGLEC1/CTTN/NRP1/FERMT2/S100A10/PIK3R1/TRIOBP/ARHGEF7/EFNA1/ITGB3/CDC42/HAS2/ITGA5/MYADM/ILK/TBCD/TAOK2/SPRY4/ACVRL1/CRK/PKD1/ANGPT2/CALR/ITGB1BP1/RAC1/ITGA10/SNED1/FER/PARVG/JAM3/PARVA/ROCK2/LIMCH1/FZD4/SORBS3/ITGA11/CD63/CORO1C/ITGAX/THY1/ITGAL/C1QBP/ITGB5/COL1A1/CDK6/HSD17B12/LYVE1/ITGB4/CX3CL1/MELTF/CORO2B/ITGB2/NID1/EMILIN1/POSTN/VIT/CDH11/THSD1/BCR/CCDC80/CORO1A/PTK2B/CSF1/GFUS/TRIP6/RRAS/BCL6/GCNT2/AJUBA/RSU1/PRKCE/VWC2/GBP1/EPHA3/FMN1/PLAU/JAK2/ACTN2/ITGA8/NDNF/TESK1/ITGAM/ITGA9/EDIL3/ABI3BP/COL17A1/EPHB3/ITGA7/PPFIA2/PTPRJ/EGFLAM/ITGA4/DAG1/PARVB

# Receptor for TNFSF12/TWEAK. Weak inducer of apoptosis in some cell types. Promotes angiogenesis and the proliferation of endothelial cells. May modulate cellular adhesion to matrix proteins</code></pre>
<pre class="r"><code>## DEGenes for Healthy condition

DEGenesHealthy &lt;- DEGenesCond_adj %&gt;% filter(cluster == &quot;Healthy&quot;)
head(DEGenesHealthy)</code></pre>
<pre><code>                        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster    gene
ENSG00000210082.MT-RNR2     0  0.9350349 0.687 0.491         0 Healthy MT-RNR2
ENSG00000155657.TTN         0  0.2171542 0.753 0.591         0 Healthy     TTN
ENSG00000198899.MT-ATP6     0  1.1665477 0.322 0.181         0 Healthy MT-ATP6
ENSG00000092054.MYH7        0  0.2515285 0.452 0.315         0 Healthy    MYH7
ENSG00000198712.MT-CO2      0  1.2251589 0.285 0.156         0 Healthy  MT-CO2
ENSG00000198727.MT-CYB      0  1.3541934 0.262 0.140         0 Healthy  MT-CYB
                                           Gene           EnsID
ENSG00000210082.MT-RNR2 ENSG00000210082.MT-RNR2 ENSG00000210082
ENSG00000155657.TTN         ENSG00000155657.TTN ENSG00000155657
ENSG00000198899.MT-ATP6 ENSG00000198899.MT-ATP6 ENSG00000198899
ENSG00000092054.MYH7       ENSG00000092054.MYH7 ENSG00000092054
ENSG00000198712.MT-CO2   ENSG00000198712.MT-CO2 ENSG00000198712
ENSG00000198727.MT-CYB   ENSG00000198727.MT-CYB ENSG00000198727</code></pre>
<pre class="r"><code>egoHealthy &lt;- enrichGO(gene = unique(DEGenesHealthy$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy &lt;- setReadable(egoHealthy, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy, showCategory=20, title = &quot;Healthy&quot;, font.size = 10)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>barplot(egoHealthy, 
        drop = TRUE, 
        showCategory = 7, 
        title = &quot;Healthy&quot;,
        font.size = 10)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>#View(DEGenesHealthy)
#View(egoHealthy@result)
#write.table(egoHealthy, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/ego Healthy 20.10.2024&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)</code></pre>
<pre class="r"><code>## Volcano Plotes with DE Genes for all conditions: 
# DE Genes: Pos. and negative markers (only.pos = FALSE)
DEGenesCond &lt;- FindAllMarkers (seuratM, only.pos = FALSE, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.05)</code></pre>
<pre><code>Calculating cluster Healthy</code></pre>
<pre><code>Calculating cluster AM</code></pre>
<pre><code>Calculating cluster CMP</code></pre>
<pre><code>Calculating cluster Sarcoidosis</code></pre>
<pre class="r"><code>##adjust table
DEGenesCond_adj &lt;- DEGenesCond %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

##GSEA for AM 
DEGenesAM &lt;- DEGenesCond_adj %&gt;% filter(cluster == &quot;AM&quot;)
head(DEGenesAM)</code></pre>
<pre><code>                         p_val avg_log2FC pct.1 pct.2 p_val_adj cluster   gene
ENSG00000229807.XIST.1       0   1.796598 0.434 0.121         0      AM   XIST
ENSG00000234745.HLA-B.1      0   1.399345 0.494 0.280         0      AM  HLA-B
ENSG00000206503.HLA-A.1      0   1.362101 0.424 0.223         0      AM  HLA-A
ENSG00000081237.PTPRC.1      0   2.054675 0.242 0.069         0      AM  PTPRC
ENSG00000154451.GBP5.1       0   5.018198 0.175 0.010         0      AM   GBP5
ENSG00000101596.SMCHD1.1     0   1.017625 0.415 0.252         0      AM SMCHD1
                                           Gene           EnsID
ENSG00000229807.XIST.1     ENSG00000229807.XIST ENSG00000229807
ENSG00000234745.HLA-B.1   ENSG00000234745.HLA-B ENSG00000234745
ENSG00000206503.HLA-A.1   ENSG00000206503.HLA-A ENSG00000206503
ENSG00000081237.PTPRC.1   ENSG00000081237.PTPRC ENSG00000081237
ENSG00000154451.GBP5.1     ENSG00000154451.GBP5 ENSG00000154451
ENSG00000101596.SMCHD1.1 ENSG00000101596.SMCHD1 ENSG00000101596</code></pre>
<pre class="r"><code># pos. and neg. DE genes for AM
res_AM &lt;- as.data.frame(DEGenesAM)

EnhancedVolcano(res_AM, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Acute Myocarditis Condition&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>top_genes_AM &lt;- DEGenesAM %&gt;% 
  filter(p_val_adj &lt; 0.05, abs(avg_log2FC) &gt; 1) %&gt;% 
  arrange(p_val_adj) %&gt;% 
  slice_head(n = 30) %&gt;% 
  pull(gene)</code></pre>
<pre class="r"><code># DE genes for CMP
DEGenesCMP &lt;- DEGenesCond_adj %&gt;% filter(cluster == &quot;CMP&quot;)
head(DEGenesCMP)</code></pre>
<pre><code>                                  p_val avg_log2FC pct.1 pct.2     p_val_adj cluster    gene
ENSG00000251562.MALAT1.2   0.000000e+00 -0.3393961 1.000 1.000  0.000000e+00     CMP  MALAT1
ENSG00000210082.MT-RNR2.2 3.015611e-278 -0.7068360 0.427 0.592 1.179526e-273     CMP MT-RNR2
ENSG00000062716.VMP1.1    1.481298e-275 -1.7169826 0.146 0.293 5.793951e-271     CMP    VMP1
ENSG00000175206.NPPA.2    5.092720e-188  1.9501280 0.054 0.013 1.991966e-183     CMP    NPPA
ENSG00000090006.LTBP4.2   1.385007e-175  0.8901326 0.369 0.253 5.417315e-171     CMP   LTBP4
ENSG00000154451.GBP5.2    3.770002e-175 -3.8545611 0.012 0.087 1.474599e-170     CMP    GBP5
                                             Gene           EnsID
ENSG00000251562.MALAT1.2   ENSG00000251562.MALAT1 ENSG00000251562
ENSG00000210082.MT-RNR2.2 ENSG00000210082.MT-RNR2 ENSG00000210082
ENSG00000062716.VMP1.1       ENSG00000062716.VMP1 ENSG00000062716
ENSG00000175206.NPPA.2       ENSG00000175206.NPPA ENSG00000175206
ENSG00000090006.LTBP4.2     ENSG00000090006.LTBP4 ENSG00000090006
ENSG00000154451.GBP5.2       ENSG00000154451.GBP5 ENSG00000154451</code></pre>
<pre class="r"><code>res_CMP &lt;- as.data.frame(DEGenesCMP)

EnhancedVolcano(res_CMP, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes in Cardiomyopathy Condition&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>top_genes_CMP &lt;- DEGenesCMP %&gt;% 
  filter(p_val_adj &lt; 0.05, abs(avg_log2FC) &gt; 1) %&gt;% 
  arrange(p_val_adj) %&gt;% 
  slice_head(n = 30) %&gt;% 
  pull(gene)</code></pre>
<pre class="r"><code># DE genes for Sarcoidosis
DEGenesSarco &lt;- DEGenesCond_adj %&gt;% filter(cluster == &quot;Sarcoidosis&quot;)
head(DEGenesHealthy)</code></pre>
<pre><code>                        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster    gene
ENSG00000210082.MT-RNR2     0  0.9350349 0.687 0.491         0 Healthy MT-RNR2
ENSG00000155657.TTN         0  0.2171542 0.753 0.591         0 Healthy     TTN
ENSG00000198899.MT-ATP6     0  1.1665477 0.322 0.181         0 Healthy MT-ATP6
ENSG00000092054.MYH7        0  0.2515285 0.452 0.315         0 Healthy    MYH7
ENSG00000198712.MT-CO2      0  1.2251589 0.285 0.156         0 Healthy  MT-CO2
ENSG00000198727.MT-CYB      0  1.3541934 0.262 0.140         0 Healthy  MT-CYB
                                           Gene           EnsID
ENSG00000210082.MT-RNR2 ENSG00000210082.MT-RNR2 ENSG00000210082
ENSG00000155657.TTN         ENSG00000155657.TTN ENSG00000155657
ENSG00000198899.MT-ATP6 ENSG00000198899.MT-ATP6 ENSG00000198899
ENSG00000092054.MYH7       ENSG00000092054.MYH7 ENSG00000092054
ENSG00000198712.MT-CO2   ENSG00000198712.MT-CO2 ENSG00000198712
ENSG00000198727.MT-CYB   ENSG00000198727.MT-CYB ENSG00000198727</code></pre>
<pre class="r"><code>res_Sarco &lt;- as.data.frame(DEGenesSarco)

EnhancedVolcano(res_Sarco, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Sarco$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes in Sarcoidosis Condition&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>top_genes_Sarco &lt;- DEGenesSarco %&gt;% 
  filter(p_val_adj &lt; 0.05, abs(avg_log2FC) &gt; 1) %&gt;% 
  arrange(p_val_adj) %&gt;% 
  slice_head(n = 30) %&gt;% 
  pull(gene)</code></pre>
<pre class="r"><code># DE Genes Healthy condition

DEGenesHealthy &lt;- DEGenesCond_adj %&gt;% filter(cluster == &quot;Healthy&quot;)
head(DEGenesHealthy)</code></pre>
<pre><code>                        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster    gene
ENSG00000234745.HLA-B       0 -1.4197742 0.233 0.445         0 Healthy   HLA-B
ENSG00000210082.MT-RNR2     0  0.9350349 0.687 0.491         0 Healthy MT-RNR2
ENSG00000206503.HLA-A       0 -1.3787395 0.181 0.377         0 Healthy   HLA-A
ENSG00000005810.MYCBP2      0 -0.8737209 0.270 0.459         0 Healthy  MYCBP2
ENSG00000229807.XIST        0 -1.3355026 0.128 0.315         0 Healthy    XIST
ENSG00000101596.SMCHD1      0 -1.1298672 0.204 0.386         0 Healthy  SMCHD1
                                           Gene           EnsID
ENSG00000234745.HLA-B     ENSG00000234745.HLA-B ENSG00000234745
ENSG00000210082.MT-RNR2 ENSG00000210082.MT-RNR2 ENSG00000210082
ENSG00000206503.HLA-A     ENSG00000206503.HLA-A ENSG00000206503
ENSG00000005810.MYCBP2   ENSG00000005810.MYCBP2 ENSG00000005810
ENSG00000229807.XIST       ENSG00000229807.XIST ENSG00000229807
ENSG00000101596.SMCHD1   ENSG00000101596.SMCHD1 ENSG00000101596</code></pre>
<pre class="r"><code>egoHealthy &lt;- enrichGO(gene = unique(DEGenesHealthy$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy &lt;- setReadable(egoHealthy, OrgDb = org.Hs.eg.db)

# DE genes for Healthy condition
res_Healthy &lt;- as.data.frame(DEGenesHealthy)

EnhancedVolcano(res_Healthy, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Healthy$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Healthy Controls&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<img src="figure/DE%20Genes%20by%20Disease%20Condition.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>top_genes_Healthy &lt;- DEGenesHealthy %&gt;% 
  filter(p_val_adj &lt; 0.05, abs(avg_log2FC) &gt; 1) %&gt;% 
  arrange(p_val_adj) %&gt;% 
  slice_head(n = 30) %&gt;% 
  pull(gene)</code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Sun Oct 27 21:12:53 2024&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pathview_1.44.0             wordcloud_2.6               textshaping_0.4.0          
 [4] EnhancedVolcano_1.22.0      ggrepel_0.9.6               RColorBrewer_1.1-3         
 [7] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
[10] ggupset_0.4.0               gridExtra_2.3               DOSE_3.30.5                
[13] enrichplot_1.24.4           msigdbr_7.5.1               org.Hs.eg.db_3.19.1        
[16] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[19] metap_1.11                  scater_1.32.1               scuttle_1.14.0             
[22] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[25] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.3            
[28] forcats_1.0.0               stringr_1.5.1               purrr_1.0.2                
[31] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[34] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[37] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.1       
[40] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[43] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.4.1          
[46] pheatmap_1.0.12             ggpubr_0.6.0                ggplot2_3.5.1              
[49] Seurat_5.1.0                SeuratObject_5.0.2          sp_2.1-4                   
[52] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            EnvStats_3.0.0            aplot_0.2.3              
[112] scatterplot3d_0.3-44      ape_5.8                   spatstat.sparse_3.1-0    
[115] xtable_1.8-4              car_3.1-3                 highr_0.11               
[118] plyr_1.8.9                httr_1.4.7                rbibutils_2.3            
[121] tools_4.4.1               globals_0.16.3            beeswarm_0.4.0           
[124] broom_1.0.7               nlme_3.1-166              lambda.r_1.2.4           
[127] lme4_1.1-35.5             digest_0.6.37             numDeriv_2016.8-1.1      
[130] Matrix_1.7-0              farver_2.1.2              tzdb_0.4.0               
[133] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.1.7        
[136] glue_1.8.0                cachem_1.1.0              polyclip_1.10-7          
[139] generics_0.1.3            Biostrings_2.72.1         mvtnorm_1.3-1            
[142] presto_1.0.0              parallelly_1.38.0         mnormt_2.1.1             
[145] statmod_1.5.0             RcppHNSW_0.6.0            ScaledMatrix_1.12.0      
[148] carData_3.0-5             minqa_1.2.8               pbapply_1.7-2            
[151] httr2_1.0.5               spam_2.11-0               gson_0.1.0               
[154] utf8_1.2.4                graphlayouts_1.2.0        gtools_3.9.5             
[157] ggsignif_0.6.4            RcppEigen_0.3.4.0.2       shiny_1.9.1              
[160] GenomeInfoDbData_1.2.12   glmmTMB_1.1.10            R.utils_2.12.3           
[163] RCurl_1.98-1.16           memoise_2.0.1             rmarkdown_2.28           
[166] scales_1.3.0              R.methodsS3_1.8.2         future_1.34.0            
[169] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-2      
[172] rstudioapi_0.16.0         cluster_2.1.6             mutoss_0.1-13            
[175] spatstat.utils_3.1-0      hms_1.1.3                 fitdistrplus_1.2-1       
[178] munsell_0.5.1             cowplot_1.1.3             colorspace_2.1-1         
[181] rlang_1.1.4               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[184] xts_0.14.0                dotCall64_1.2             shinydashboard_0.7.2     
[187] ggforce_0.4.2             laeken_0.5.3              mgcv_1.9-1               
[190] xfun_0.48                 e1071_1.7-16              TH.data_1.1-2            
[193] iterators_1.0.14          abind_1.4-8               GOSemSim_2.30.2          
[196] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-9             
[199] Rdpack_2.6.1              promises_1.3.0            scatterpie_0.2.4         
[202] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-1           
[205] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[208] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[211] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[214] Rcpp_1.0.13               edgeR_4.2.1               workflowr_1.7.1          
[217] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-61              
[220] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[223] spatstat.random_3.3-2     R6_2.5.1                  fastmap_1.2.0            
[226] multcomp_1.4-26           fastmatch_1.1-4           rstatix_0.7.2            
[229] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[232] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[235] nnet_7.3-19               gtable_0.3.5              KernSmooth_2.23-24       
[238] miniUI_0.1.1.1            deldir_2.0-4              htmltools_0.5.8.1        
[241] ggthemes_5.1.0            bit64_4.5.2               spatstat.explore_3.3-2   
[244] lifecycle_1.0.4           blme_1.0-6                nloptr_2.1.1             
[247] sass_0.4.9                vctrs_0.6.5               robustbase_0.99-4-1      
[250] spatstat.geom_3.3-3       sn_2.1.1                  ggfun_0.1.6              
[253] future.apply_1.11.2       bslib_0.8.0               pillar_1.9.0             
[256] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.10          
[259] jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
<pre><code>





&lt;br&gt;
&lt;p&gt;
&lt;button type=&quot;button&quot; class=&quot;btn btn-default btn-workflowr btn-workflowr-sessioninfo&quot;
  data-toggle=&quot;collapse&quot; data-target=&quot;#workflowr-sessioninfo&quot;
  style = &quot;display: block;&quot;&gt;
  &lt;span class=&quot;glyphicon glyphicon-wrench&quot; aria-hidden=&quot;true&quot;&gt;&lt;/span&gt;
  Session information
&lt;/button&gt;
&lt;/p&gt;

&lt;div id=&quot;workflowr-sessioninfo&quot; class=&quot;collapse&quot;&gt;

``` r
sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pathview_1.44.0             wordcloud_2.6               textshaping_0.4.0          
 [4] EnhancedVolcano_1.22.0      ggrepel_0.9.6               RColorBrewer_1.1-3         
 [7] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
[10] ggupset_0.4.0               gridExtra_2.3               DOSE_3.30.5                
[13] enrichplot_1.24.4           msigdbr_7.5.1               org.Hs.eg.db_3.19.1        
[16] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[19] metap_1.11                  scater_1.32.1               scuttle_1.14.0             
[22] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[25] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.3            
[28] forcats_1.0.0               stringr_1.5.1               purrr_1.0.2                
[31] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[34] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[37] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.1       
[40] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[43] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.4.1          
[46] pheatmap_1.0.12             ggpubr_0.6.0                ggplot2_3.5.1              
[49] Seurat_5.1.0                SeuratObject_5.0.2          sp_2.1-4                   
[52] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            EnvStats_3.0.0            aplot_0.2.3              
[112] scatterplot3d_0.3-44      ape_5.8                   spatstat.sparse_3.1-0    
[115] xtable_1.8-4              car_3.1-3                 highr_0.11               
[118] plyr_1.8.9                httr_1.4.7                rbibutils_2.3            
[121] tools_4.4.1               globals_0.16.3            beeswarm_0.4.0           
[124] broom_1.0.7               nlme_3.1-166              lambda.r_1.2.4           
[127] lme4_1.1-35.5             digest_0.6.37             numDeriv_2016.8-1.1      
[130] Matrix_1.7-0              farver_2.1.2              tzdb_0.4.0               
[133] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.1.7        
[136] glue_1.8.0                cachem_1.1.0              polyclip_1.10-7          
[139] generics_0.1.3            Biostrings_2.72.1         mvtnorm_1.3-1            
[142] presto_1.0.0              parallelly_1.38.0         mnormt_2.1.1             
[145] statmod_1.5.0             RcppHNSW_0.6.0            ScaledMatrix_1.12.0      
[148] carData_3.0-5             minqa_1.2.8               pbapply_1.7-2            
[151] httr2_1.0.5               spam_2.11-0               gson_0.1.0               
[154] utf8_1.2.4                graphlayouts_1.2.0        gtools_3.9.5             
[157] ggsignif_0.6.4            RcppEigen_0.3.4.0.2       shiny_1.9.1              
[160] GenomeInfoDbData_1.2.12   glmmTMB_1.1.10            R.utils_2.12.3           
[163] RCurl_1.98-1.16           memoise_2.0.1             rmarkdown_2.28           
[166] scales_1.3.0              R.methodsS3_1.8.2         future_1.34.0            
[169] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-2      
[172] rstudioapi_0.16.0         cluster_2.1.6             mutoss_0.1-13            
[175] spatstat.utils_3.1-0      hms_1.1.3                 fitdistrplus_1.2-1       
[178] munsell_0.5.1             cowplot_1.1.3             colorspace_2.1-1         
[181] rlang_1.1.4               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[184] xts_0.14.0                dotCall64_1.2             shinydashboard_0.7.2     
[187] ggforce_0.4.2             laeken_0.5.3              mgcv_1.9-1               
[190] xfun_0.48                 e1071_1.7-16              TH.data_1.1-2            
[193] iterators_1.0.14          abind_1.4-8               GOSemSim_2.30.2          
[196] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-9             
[199] Rdpack_2.6.1              promises_1.3.0            scatterpie_0.2.4         
[202] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-1           
[205] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[208] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[211] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[214] Rcpp_1.0.13               edgeR_4.2.1               workflowr_1.7.1          
[217] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-61              
[220] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[223] spatstat.random_3.3-2     R6_2.5.1                  fastmap_1.2.0            
[226] multcomp_1.4-26           fastmatch_1.1-4           rstatix_0.7.2            
[229] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[232] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[235] nnet_7.3-19               gtable_0.3.5              KernSmooth_2.23-24       
[238] miniUI_0.1.1.1            deldir_2.0-4              htmltools_0.5.8.1        
[241] ggthemes_5.1.0            bit64_4.5.2               spatstat.explore_3.3-2   
[244] lifecycle_1.0.4           blme_1.0-6                nloptr_2.1.1             
[247] sass_0.4.9                vctrs_0.6.5               robustbase_0.99-4-1      
[250] spatstat.geom_3.3-3       sn_2.1.1                  ggfun_0.1.6              
[253] future.apply_1.11.2       bslib_0.8.0               pillar_1.9.0             
[256] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.10          
[259] jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
