<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Payne" />

<meta name="date" content="2024-10-25" />

<title>Myocarditis Subgroups vs. Healthy</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myocarditis snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Myocarditis Subgroups vs. Healthy</h1>
<h4 class="author">Emily Payne</h4>
<h4 class="date">2024-10-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Myocarditis
snRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat
files/Myocarditis_RDS__withTCcondition_20.10.2024.rds
</td>
<td style="text-align:left;">
../data/01 Merged Seurat
files/Myocarditis_RDS__withTCcondition_20.10.2024.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat
files/seuratM_withAMvsHealthy and TC_con_27.10.2024.rds
</td>
<td style="text-align:left;">
../data/01 Merged Seurat files/seuratM_withAMvsHealthy and
TC_con_27.10.2024.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis
snRNAseq/docs/Markers/DEGenes_AMvsHealthy_pos
</td>
<td style="text-align:left;">
../docs/Markers/DEGenes_AMvsHealthy_pos
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/V5
25.10.2024 AMvsHealthy/egoAM_TChigh
</td>
<td style="text-align:left;">
../docs/Markers/V5 25.10.2024 AMvsHealthy/egoAM_TChigh
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/V5
25.10.2024 AMvsHealthy/egoAM_TCint
</td>
<td style="text-align:left;">
../docs/Markers/V5 25.10.2024 AMvsHealthy/egoAM_TCint
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/V5
25.10.2024 AMvsHealthy/egoAM_TClow
</td>
<td style="text-align:left;">
../docs/Markers/V5 25.10.2024 AMvsHealthy/egoAM_TClow
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/V5
25.10.2024 AMvsHealthy/egoAM_TCvlow
</td>
<td style="text-align:left;">
../docs/Markers/V5 25.10.2024 AMvsHealthy/egoAM_TCvlow
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/V5
25.10.2024 AMvsHealthy/egoHealthy
</td>
<td style="text-align:left;">
../docs/Markers/V5 25.10.2024 AMvsHealthy/egoHealthy
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis
snRNAseq/docs/Markers/DEGenes_AMvsHealthy_all
</td>
<td style="text-align:left;">
../docs/Markers/DEGenes_AMvsHealthy_all
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree4daf12687e1b1561624f77778fa38396be0bb8detargetblank4daf126a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/4daf12687e1b1561624f77778fa38396be0bb8de" target="_blank">4daf126</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree4daf12687e1b1561624f77778fa38396be0bb8detargetblank4daf126a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/4daf12687e1b1561624f77778fa38396be0bb8de" target="_blank">4daf126</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/00 Inbox Documents and Pictures /
    Ignored:    data/01 Merged Seurat files/
    Ignored:    data/AM/
    Ignored:    data/Berlin/
    Ignored:    data/CM/
    Ignored:    data/Healthy/
    Ignored:    data/Sarcoidosis/

Untracked files:
    Untracked:  analysis/Comparison of Sexes Myocarditis.Rmd
    Untracked:  analysis/DE Genes by Disease Condition.Rmd
    Untracked:  analysis/DE genes AMsub vs. Healthy_V2.Rmd
    Untracked:  analysis/DE-genes-AMsub-vs.-Healthy_V2.Rmd
    Untracked:  analysis/Fibroblast subset V2 reclustering.Rmd
    Untracked:  analysis/Install.packages_MFZ.R
    Untracked:  analysis/MCP and TC reembedding.Rmd
    Untracked:  analysis/Macrophage DE subset V2.Rmd
    Untracked:  analysis/Macrophage Reembedding.Rmd
    Untracked:  analysis/Macrophage embedding.Rmd
    Untracked:  analysis/Macrophage heatmap.Rmd
    Untracked:  analysis/Merging/
    Untracked:  analysis/Myocarditis snRNAseq QC and merging_V2_Tchigh vs. low.Rmd
    Untracked:  analysis/T cell abundance DE Kopie.Rmd
    Untracked:  analysis/T cell abundance DE.Rmd
    Untracked:  analysis/T cell condition Differential expression.Rmd
    Untracked:  analysis/TC high vs. low condition.Rmd
    Untracked:  analysis/TC high vs. low.Rmd
    Untracked:  analysis/Tcell embedding.Rmd
    Untracked:  analysis/Tcell high vs. Tcell low.Rmd
    Untracked:  analysis/V1/

Unstaged changes:
    Deleted:    _workflowr.yml
    Modified:   analysis/Myocarditis snRNAseq.Rmd
    Deleted:    analysis/license.Rmd
    Deleted:    data/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(ExploreSCdataSeurat3)
library(runSeurat3)
library(Seurat)
library(ggpubr)
library(pheatmap)
library(SingleCellExperiment)
library(dplyr)
library(tidyverse)
library(viridis)
library(muscat)
library(circlize)
library(destiny)
library(scater)
library(metap)
library(multtest)
library(clusterProfiler)
library(org.Hs.eg.db)
library(msigdbr)
library(enrichplot)
library(DOSE)
library(grid)
library(gridExtra)
library(ggupset)
library(VennDiagram)
library(NCmisc)
library(RColorBrewer)
library(EnhancedVolcano)
library(rmarkdown)
library(workflowr)

library(textshaping)
library(wordcloud)
library(pathview)</code></pre>
<pre class="r"><code>##load merged file 
seuratM &lt;- readRDS(&quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_RDS__withTCcondition_20.10.2024.rds&quot;)

table(seuratM$dataset)</code></pre>
<pre><code>
     312581_05-5_20230308_Hu_nucseq_SG36_EMB_GEM      312581_06-6_20230308_Hu_nucseq_SG37_EMB_GEM 
                                            4323                                             1123 
   353921_28-28_20240529_Hu_nucseq_Immpath_EMB18    353921_29-29_20240529_Hu_nucseq_Immpath_EMB24 
                                             208                                              105 
   353921_30-30_20240529_Hu_nucseq_Immpath_EMB25    353921_31-31_20240529_Hu_nucseq_Immpath_EMB27 
                                             227                                              579 
   353921_32-32_20240529_Hu_nucseq_Immpath_EMB29    353921_33-33_20240529_Hu_nucseq_Immpath_EMB30 
                                             335                                              730 
   353921_34-34_20240529_Hu_nucseq_Immpath_EMB35    353921_35-35_20240529_Hu_nucseq_Immpath_EMB37 
                                            1217                                              604 
359861_09-11_20240723_Hu_nucseq_Ber_Pat5_EM979_1 359861_10-13_20240723_Hu_nucseq_Ber_Pat5_EM979_2 
                                             771                                             1476 
 359861_11-1_20240730_Hu_nucseq_Ber_Pat1_EMB1537  359861_12-2_20240730_Hu_nucseq_Ber_Pat2_EMB1041 
                                             316                                              608 
    359861_16-8_20240730_Hu_nucseq_Immpath_EMB47     359861_17-9_20240730_Hu_nucseq_Immpath_EMB48 
                                             513                                              426 
     o27533_1_12-12_20220203_Hu_nucseq_EMB31_GEM        o27936_1_7-7_20220309_Hu_nucseq_EMB32_GEM 
                                             236                                             1192 
 o28576_1_01-1_20220525_Hu_nucseq_Graz_1_EMB_GEM  o28576_1_02-2_20220525_Hu_nucseq_Graz_2_EMB_GEM 
                                            2740                                             1684 
 o28576_1_03-3_20220525_Hu_nucseq_Graz_3_EMB_GEM  o28576_1_04-4_20220525_Hu_nucseq_Graz_4_EMB_GEM 
                                            2396                                              545 
 o28576_1_05-5_20220525_Hu_nucseq_Graz_5_EMB_GEM  o28576_1_06-6_20220525_Hu_nucseq_Graz_6_EMB_GEM 
                                             781                                              491 
 o28576_1_07-7_20220525_Hu_nucseq_Graz_7_EMB_GEM   o28576_1_08-8_20220525_Hu_nucseq_Graz_8_HH_GEM 
                                             653                                             3921 
o28576_1_12-12_20220525_Hu_nucseq_Graz_12_HH_GEM      o28576_1_13-13_20220525_Hu_nucseq_EMB32_GEM 
                                            3818                                             1428 
    o292731_1-1_20220818_Hu_nucseq_Graz_9_HH_GEM    o292731_2-2_20220818_Hu_nucseq_Graz_13_HH_GEM 
                                            4908                                             9882 
    o292731_3-3_20220818_Hu_nucseq_SG_33_EMB_GEM     o292731_4-4_20220818_Hu_nucseq_SG_34_EMB_GEM 
                                            6286                                              620 
    o292731_5-5_20220818_Hu_nucseq_SG_35_EMB_GEM   o294781_01-1_20220912_Hu_nucseq_Graz_21_HH_GEM 
                                            2363                                             1442 
  o294781_02-2_20220912_Hu_nucseq_Graz_22_HH_GEM   o294781_03-3_20220912_Hu_nucseq_Graz_23_HH_GEM 
                                            1998                                              841 
  o294781_04-4_20220912_Hu_nucseq_Graz_24_HH_GEM  o294781_05-5_20220912_Hu_nucseq_Graz_14_EMB_GEM 
                                            1480                                             1268 
 o294781_06-6_20220912_Hu_nucseq_Graz_15_EMB_GEM  o294781_09-9_20220912_Hu_nucseq_Graz_18_EMB_GEM 
                                            4439                                             2280 
o294781_10-10_20220912_Hu_nucseq_Graz_19_EMB_GEM o294781_11-11_20220912_Hu_nucseq_Graz_20_EMB_GEM 
                                             111                                             2706 </code></pre>
<pre class="r"><code>table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11    12    13    14 
17699 13639 12754  8222  5475  4836  3723  2365  2156   905   882   694   380   249    91 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
74070 </code></pre>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$RNA_snn_res.0.25
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/Plot%20umap-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>unique(seuratM$clusterName)</code></pre>
<pre><code> [1] PeriFb1       Fb1           VSMC          BEC1          Cardiomyocyte Fb3          
 [7] TC            MCP1          BEC2          INT1          MCP2          NC           
[13] Fb2           AdipoC        PeriFb2      
Levels: Cardiomyocyte Fb1 Fb2 Fb3 INT1 PeriFb1 PeriFb2 VSMC NC BEC1 BEC2 AdipoC MCP1 MCP2 TC</code></pre>
<pre class="r"><code>#### Rename clusters

seuratM$clusterName &lt;- &quot;clusterName&quot;
Idents(seuratM) &lt;- seuratM$clusterName
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;0&quot; )] &lt;- &quot;Fb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;1&quot; )] &lt;- &quot;BEC1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;2&quot; )] &lt;- &quot;PeriFb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;3&quot; )] &lt;- &quot;Cardiomyocyte&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;4&quot; )] &lt;- &quot;MCP1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;5&quot; )] &lt;- &quot;TC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;6&quot; )] &lt;- &quot;MCP2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;7&quot; )] &lt;- &quot;BEC2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;8&quot; )] &lt;- &quot;VSMC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;9&quot; )] &lt;- &quot;NC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;10&quot; )] &lt;- &quot;INT1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;11&quot; )] &lt;- &quot;Fb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;12&quot; )] &lt;- &quot;Fb3&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;13&quot; )] &lt;- &quot;PeriFb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;14&quot; )] &lt;- &quot;AdipoC&quot;


colclusterName &lt;- c(&quot;#D53E4F&quot;, &quot;#f4a582&quot;, &quot;#FEE08B&quot;,&quot;#feb24c&quot;,&quot;#67001f&quot;, &quot;#01665e&quot;,&quot;#66C2A5&quot;,&quot;#c7eae5&quot;,&quot;#BEAED4&quot;, &quot;#355C7D&quot;,&quot;#3288BD&quot;,&quot;#8c510a&quot; ,&quot;#fde0dd&quot;,&quot;#B45B5C&quot;,&quot;#dd1c77&quot;)

                    
names(colclusterName) &lt;- c(&quot;Cardiomyocyte&quot;,&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;,&quot;NC&quot;, &quot;BEC1&quot;,&quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;,&quot;MCP2&quot;,&quot;TC&quot;)

colpat_sub&lt;- c(&quot;#dfc27d&quot;,&quot;#BE3144&quot;,&quot;#355C7D&quot;, &quot;#779d8d&quot;)

seuratM$clusterName &lt;- factor(seuratM$clusterName, levels=c(&quot;Cardiomyocyte&quot;, &quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;INT1&quot;,&quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;, &quot;NC&quot;, &quot;BEC1&quot;, &quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;, &quot;MCP2&quot;, &quot;TC&quot;))


Idents(seuratM) &lt;- seuratM$clusterName
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE, cols = colclusterName)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>Idents(seuratM) &lt;-seuratM$pat_sub

colpat_sub&lt;- c(&quot;#779d8d&quot;,&quot;#BE3144&quot;,&quot;#355C7D&quot;,&quot;#dfc27d&quot;)
names(colpat_sub) &lt;- c(&quot;Healthy&quot;, &quot;AM&quot;, &quot;CMP&quot;, &quot;Sarcoidosis&quot;)

DimPlot(seuratM, reduction = &quot;umap&quot;, cols = colpat_sub)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>table(seuratM$pat_sub)</code></pre>
<pre><code>
    Healthy          AM         CMP Sarcoidosis 
      28290       29076       11635        5069 </code></pre>
<pre class="r"><code>table(seuratM$patient)</code></pre>
<pre><code>
    Healthy 01     Healthy 02     Healthy 03     Healthy 04     Healthy 05     Healthy 06 
          3921           3818           4908           9882           1442           1998 
    Healthy 07     Healthy 08          AM 01          AM 02          AM 03          AM 04 
           841           1480           6286           4323           1123            208 
         AM 05          AM 06          AM 07          AM 08          AM 09          AM 10 
           579            335           1217            604            513            426 
         AM 11          AM 12          AM 13          AM 14          AM 15          AM 16 
           236           1192           2740           1684           4439           2247 
         AM 17          AM 18         CMP 01         CMP 02         CMP 03         CMP 04 
           316            608            105            227            730           1428 
        CMP 05         CMP 06         CMP 07         CMP 08         CMP 09         CMP 10 
           620            653           1268           2280           2396            545 
        CMP 11         CMP 12         CMP 13 Sarcoidosis 01 Sarcoidosis 02 
           781            491            111           2363           2706 </code></pre>
<pre class="r"><code>#subset AM group in TC High, Intermediate, Low, Vlow

seuratAM &lt;- subset(seuratM, pat_sub == &quot;AM&quot;)
DimPlot(seuratAM, reduction = &quot;umap&quot;, pt.size = 0.1, raster=FALSE)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>Idents(seuratAM) &lt;- seuratAM$TC

table(seuratAM$TC)</code></pre>
<pre><code>
        Tcell High Tcell Intermediate          Tcell Low     Tcell Very Low 
              3545               2837               8509              14185 </code></pre>
<pre class="r"><code>levels(seuratAM)</code></pre>
<pre><code>[1] &quot;Tcell High&quot;         &quot;Tcell Intermediate&quot; &quot;Tcell Low&quot;          &quot;Tcell Very Low&quot;    </code></pre>
<pre class="r"><code>table(seuratAM$patient)</code></pre>
<pre><code>
    Healthy 01     Healthy 02     Healthy 03     Healthy 04     Healthy 05     Healthy 06 
             0              0              0              0              0              0 
    Healthy 07     Healthy 08          AM 01          AM 02          AM 03          AM 04 
             0              0           6286           4323           1123            208 
         AM 05          AM 06          AM 07          AM 08          AM 09          AM 10 
           579            335           1217            604            513            426 
         AM 11          AM 12          AM 13          AM 14          AM 15          AM 16 
           236           1192           2740           1684           4439           2247 
         AM 17          AM 18         CMP 01         CMP 02         CMP 03         CMP 04 
           316            608              0              0              0              0 
        CMP 05         CMP 06         CMP 07         CMP 08         CMP 09         CMP 10 
             0              0              0              0              0              0 
        CMP 11         CMP 12         CMP 13 Sarcoidosis 01 Sarcoidosis 02 
             0              0              0              0              0 </code></pre>
<pre class="r"><code>DimPlot(seuratAM, reduction = &quot;umap&quot;, group.by = &quot;clusterName&quot;, cols=colclusterName,
        split.by = &quot;TC&quot;, shuffle = T)+
  theme_void()</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-5-2.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>AM_TChigh &lt;- subset(seuratAM, TC == &quot;Tcell High&quot;)
table(AM_TChigh$orig.ident)</code></pre>
<pre><code>
     
3545 </code></pre>
<pre class="r"><code>AM_TCint &lt;- subset(seuratAM, TC == &quot;Tcell Intermediate&quot;)
table(AM_TCint$orig.ident)</code></pre>
<pre><code>
     
2837 </code></pre>
<pre class="r"><code>AM_TClow &lt;- subset(seuratAM, TC == &quot;Tcell Low&quot;)
table(AM_TClow$orig.ident)</code></pre>
<pre><code>
     
8509 </code></pre>
<pre class="r"><code>AM_TCvlow &lt;- subset(seuratAM, TC == &quot;Tcell Very Low&quot;)
table(AM_TCvlow$orig.ident)</code></pre>
<pre><code>
      
14185 </code></pre>
<pre class="r"><code># seuratM subset for AM and Healthy groups
seuratAMvsHealthy &lt;- subset(seuratM, pat_sub %in% c(&quot;AM&quot;, &quot;Healthy&quot;))

# &quot;Healthy&quot; is default
seuratAMvsHealthy$AMvsHealthy &lt;- &quot;Healthy&quot;  

seuratAMvsHealthy$AMvsHealthy[seuratAMvsHealthy$pat_sub == &quot;AM&quot; &amp; seuratAMvsHealthy$TC == &quot;Tcell High&quot;] &lt;- &quot;AM_TChigh&quot;
seuratAMvsHealthy$AMvsHealthy[seuratAMvsHealthy$pat_sub == &quot;AM&quot; &amp; seuratAMvsHealthy$TC == &quot;Tcell Intermediate&quot;] &lt;- &quot;AM_TCint&quot;
seuratAMvsHealthy$AMvsHealthy[seuratAMvsHealthy$pat_sub == &quot;AM&quot; &amp; seuratAMvsHealthy$TC == &quot;Tcell Low&quot;] &lt;- &quot;AM_TClow&quot;
seuratAMvsHealthy$AMvsHealthy[seuratAMvsHealthy$pat_sub == &quot;AM&quot; &amp; seuratAMvsHealthy$TC == &quot;Tcell Very Low&quot;] &lt;- &quot;AM_TCvlow&quot;

Idents(seuratAMvsHealthy) &lt;- seuratAMvsHealthy$AMvsHealthy

levels(seuratAMvsHealthy)</code></pre>
<pre><code>[1] &quot;AM_TCvlow&quot; &quot;AM_TCint&quot;  &quot;AM_TClow&quot;  &quot;AM_TChigh&quot; &quot;Healthy&quot;  </code></pre>
<pre class="r"><code>table(seuratAMvsHealthy$orig.ident)</code></pre>
<pre><code>
      
57366 </code></pre>
<pre class="r"><code>table(seuratAM$orig.ident)</code></pre>
<pre><code>
      
29076 </code></pre>
<pre class="r"><code>table(seuratM$pat_sub)</code></pre>
<pre><code>
    Healthy          AM         CMP Sarcoidosis 
      28290       29076       11635        5069 </code></pre>
<pre class="r"><code>table(seuratAMvsHealthy$AMvsHealthy)</code></pre>
<pre><code>
AM_TChigh  AM_TCint  AM_TClow AM_TCvlow   Healthy 
     3545      2837      8509     14185     28290 </code></pre>
<pre class="r"><code># New Factor within seuratM
seuratM$AMvsHealthy &lt;- NA
seuratM$AMvsHealthy [Cells(seuratAMvsHealthy)] &lt;- seuratAMvsHealthy$AMvsHealthy

table(seuratM$AMvsHealthy)</code></pre>
<pre><code>
AM_TChigh  AM_TCint  AM_TClow AM_TCvlow   Healthy 
     3545      2837      8509     14185     28290 </code></pre>
<pre class="r"><code>table(seuratM$AMvsHealthy, seuratM$patient)</code></pre>
<pre><code>           
            Healthy 01 Healthy 02 Healthy 03 Healthy 04 Healthy 05 Healthy 06 Healthy 07 Healthy 08
  AM_TChigh          0          0          0          0          0          0          0          0
  AM_TCint           0          0          0          0          0          0          0          0
  AM_TClow           0          0          0          0          0          0          0          0
  AM_TCvlow          0          0          0          0          0          0          0          0
  Healthy         3921       3818       4908       9882       1442       1998        841       1480
           
            AM 01 AM 02 AM 03 AM 04 AM 05 AM 06 AM 07 AM 08 AM 09 AM 10 AM 11 AM 12 AM 13 AM 14
  AM_TChigh   769   529   136    25    70    40   150    74    61    54    26   146   334   205
  AM_TCint    614   422   110    20    56    33   119    60    49    42    22   117   267   164
  AM_TClow   1840  1264   330    60   170    98   356   177   149   125    69   349   802   493
  AM_TCvlow  3063  2108   547   103   283   164   592   293   254   205   119   580  1337   822
  Healthy       0     0     0     0     0     0     0     0     0     0     0     0     0     0
           
            AM 15 AM 16 AM 17 AM 18 CMP 01 CMP 02 CMP 03 CMP 04 CMP 05 CMP 06 CMP 07 CMP 08 CMP 09
  AM_TChigh   541   274    37    74      0      0      0      0      0      0      0      0      0
  AM_TCint    433   220    30    59      0      0      0      0      0      0      0      0      0
  AM_TClow   1300   657    92   178      0      0      0      0      0      0      0      0      0
  AM_TCvlow  2165  1096   157   297      0      0      0      0      0      0      0      0      0
  Healthy       0     0     0     0      0      0      0      0      0      0      0      0      0
           
            CMP 10 CMP 11 CMP 12 CMP 13 Sarcoidosis 01 Sarcoidosis 02
  AM_TChigh      0      0      0      0              0              0
  AM_TCint       0      0      0      0              0              0
  AM_TClow       0      0      0      0              0              0
  AM_TCvlow      0      0      0      0              0              0
  Healthy        0      0      0      0              0              0</code></pre>
<pre class="r"><code>table(seuratM$AMvsHealthy, seuratM$TC)</code></pre>
<pre><code>           
            Tcell High Tcell Intermediate Tcell Low Tcell Very Low
  AM_TChigh       3545                  0         0              0
  AM_TCint           0               2837         0              0
  AM_TClow           0                  0      8509              0
  AM_TCvlow          0                  0         0          14185
  Healthy         3450               2760      8280          13800</code></pre>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$AMvsHealthy
levels(seuratM)</code></pre>
<pre><code>[1] &quot;AM_TCvlow&quot; &quot;AM_TCint&quot;  &quot;AM_TClow&quot;  &quot;AM_TChigh&quot; &quot;Healthy&quot;  </code></pre>
<pre class="r"><code>DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, raster=FALSE)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-5-3.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code># NA are the other disease conditions
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, raster=FALSE, split.by = &quot;AMvsHealthy&quot;)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-5-4.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>#saveRDS(seuratM, file = &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/seuratM_withAMvsHealthy and TC_con_27.10.2024.rds&quot;)</code></pre>
<pre class="r"><code># Find DE genes for AM subgroups (filtered by TC condition) vs. Healthy
Idents(seuratM) &lt;-seuratM$AMvsHealthy

DEGenes_AMvsHealthy_pos &lt;- FindAllMarkers(seuratM, only.pos = TRUE, logfc.threshold = 0.2) %&gt;%
  dplyr::filter(p_val_adj &lt; 0.05)

#head(DEGenes_AMvsHealthy_pos)

#write.table(DEGenes_AMvsHealthy_pos, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenes_AMvsHealthy_pos&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

##adjust table
DEGenesAM_sub &lt;- DEGenes_AMvsHealthy_pos %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))</code></pre>
<pre class="r"><code>## DE Genes for AM T cell high
DEGenesAM_TChigh &lt;- DEGenesAM_sub %&gt;% filter(cluster == &quot;AM_TChigh&quot;)
head(DEGenesAM_TChigh)</code></pre>
<pre><code>                               p_val avg_log2FC pct.1 pct.2     p_val_adj   cluster  gene
ENSG00000229807.XIST3  3.054540e-154  0.8636867 0.435 0.234 1.194753e-149 AM_TChigh  XIST
ENSG00000154451.GBP53  4.244298e-118  1.3598308 0.174 0.070 1.660115e-113 AM_TChigh  GBP5
ENSG00000082074.FYB13   4.912684e-85  1.0734455 0.214 0.109  1.921547e-80 AM_TChigh  FYB1
ENSG00000234745.HLA-B3  2.131302e-82  0.7317787 0.496 0.357  8.336375e-78 AM_TChigh HLA-B
ENSG00000081237.PTPRC3  3.624135e-78  0.9901756 0.241 0.132  1.417544e-73 AM_TChigh PTPRC
ENSG00000206503.HLA-A3  2.641748e-77  0.7550193 0.432 0.295  1.033293e-72 AM_TChigh HLA-A
                                        Gene           EnsID
ENSG00000229807.XIST3   ENSG00000229807.XIST ENSG00000229807
ENSG00000154451.GBP53   ENSG00000154451.GBP5 ENSG00000154451
ENSG00000082074.FYB13   ENSG00000082074.FYB1 ENSG00000082074
ENSG00000234745.HLA-B3 ENSG00000234745.HLA-B ENSG00000234745
ENSG00000081237.PTPRC3 ENSG00000081237.PTPRC ENSG00000081237
ENSG00000206503.HLA-A3 ENSG00000206503.HLA-A ENSG00000206503</code></pre>
<pre class="r"><code>egoAM_TChigh &lt;- enrichGO(gene = unique(DEGenesAM_TChigh$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM_TChigh &lt;- setReadable(egoAM_TChigh, OrgDb = org.Hs.eg.db)
dotplot(egoAM_TChigh, showCategory=20, title = &quot;Gene Ontology Acute Myocarditis T cell High&quot;, font.size = 10)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>barplot(egoAM_TChigh, 
        drop = TRUE, 
        showCategory = 7, 
        title = &quot;Gene Ontology Acute Myocarditis T cell High&quot;,
        font.size = 10)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-8-2.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code># head(egoAM_TChigh)
#View(egoAM_TChigh@result)

# FYB1/PTPRC/HLA-A/GBP1/TRBC2/NCKAP1L/TRAC/LCP2/CYBA/BTN3A2/CD247/HLA-DQB1/MYO1G/ZAP70/PRKCB/KLRC1/ITK/LCK/CLEC7A/BTK/PTPN22/PIK3CD/INPP5D/TNFAIP3/THEMIS/TRBC1/PTPN6/TYROBP/HLA-DRB1/LAT2/LAPTM5/CD3G/CD47/LILRB4/SKAP1/PTPRJ/CD3E/MYD88/IGHM/BTN3A1/CD300A/CD3D/KLRD1/FCN1/CARD11/BTN3A3/MNDA/FCHO1/CYLD/GRB2/CD8B/OAS1/PIK3AP1/CSK/CD226/LAT/FCMR/FGR/PRKCD/NMI/MS4A1/KCNN4/NOD2/HLA-DPB1/LAX1/TLR6/FYN/PAK1/SCIMP/VAV1/BCL2/CD38/PRAM1/TXK/TNIP3/CTLA4/FCER1G/TRAT1/NFAM1/NFKB1/TLR2/NFKBID/UBASH3A/KLHL6/LYN/BLNK/IRAK1/CREBBP/CD8A/SH2D1A/ITGAM/TRGC1/BANK1/SOS1/PTPN2/DENND1B/EP300/RC3H2/SYK/IGKC/TESPA1

#write.table(egoAM_TChigh, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/V5 25.10.2024 AMvsHealthy/egoAM_TChigh&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)</code></pre>
<pre class="r"><code>## DE Genes for AM T cell intermediate
DEGenesAM_TCint &lt;- DEGenesAM_sub %&gt;% filter(cluster == &quot;AM_TCint&quot;)
head(DEGenesAM_TCint)</code></pre>
<pre><code>                               p_val avg_log2FC pct.1 pct.2     p_val_adj  cluster  gene
ENSG00000229807.XIST1  1.298826e-126  0.9104069 0.434 0.236 5.080228e-122 AM_TCint  XIST
ENSG00000154451.GBP51   1.196845e-93  1.2687426 0.174 0.071  4.681341e-89 AM_TCint  GBP5
ENSG00000081237.PTPRC1  8.603204e-68  0.9856375 0.245 0.133  3.365057e-63 AM_TCint PTPRC
ENSG00000082074.FYB11   9.995754e-66  1.0382078 0.212 0.110  3.909739e-61 AM_TCint  FYB1
ENSG00000005844.ITGAL1  4.352593e-59  1.1070272 0.155 0.073  1.702473e-54 AM_TCint ITGAL
ENSG00000107099.DOCK81  3.680243e-55  1.0578340 0.198 0.106  1.439490e-50 AM_TCint DOCK8
                                        Gene           EnsID
ENSG00000229807.XIST1   ENSG00000229807.XIST ENSG00000229807
ENSG00000154451.GBP51   ENSG00000154451.GBP5 ENSG00000154451
ENSG00000081237.PTPRC1 ENSG00000081237.PTPRC ENSG00000081237
ENSG00000082074.FYB11   ENSG00000082074.FYB1 ENSG00000082074
ENSG00000005844.ITGAL1 ENSG00000005844.ITGAL ENSG00000005844
ENSG00000107099.DOCK81 ENSG00000107099.DOCK8 ENSG00000107099</code></pre>
<pre class="r"><code>egoAM_TCint &lt;- enrichGO(gene = unique(DEGenesAM_TCint$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM_TCint &lt;- setReadable(egoAM_TCint, OrgDb = org.Hs.eg.db)
dotplot(egoAM_TCint, showCategory=20, title = &quot;Gene Ontology Acute Myocarditis T cell Intermediate&quot;, font.size = 10)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>barplot(egoAM_TCint, 
        drop = TRUE, 
        showCategory = 7, 
        title = &quot;Gene Ontology Acute Myocarditis T cell Intermediate&quot;,
        font.size = 10)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-9-2.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>#head(egoAM_TCint)
#View(egoAM_TCint@result)

# write.table(egoAM_TCint, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/V5 25.10.2024 AMvsHealthy/egoAM_TCint&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

#PTPRC/FYB1/GBP1/HLA-A/BTN3A2/ITK/NCKAP1L/CYBA/LCK/CD3D/TNFAIP3/MYO1G/PTPN22/FGR/MNDA/CD3E/TRBC2/LAPTM5/ZAP70/CD3G/CD247/CLEC7A/LCP2/TRBC1/FCN1/LILRB4/TRAC/KCNN4/SKAP1/KLRC1/PTPN6/FYN/FCMR/BTK/BTN3A1/CD8A/RBCK1/FCER1G/LAT2/SCIMP/TYROBP/PRKCB/PTPRJ/HLA-DRB1/THEMIS/HLA-DQB1/TRAF3/HLA-DPB1/INPP5D/SOS1/C3AR1/CD226/CD38/CTLA4/SLC39A10/CARD11/KLRK1/LAT/MYD88/NMI/BTN3A3/DGKZ/UBASH3A/PRKCD/IGKC/PRAM1/VAV1/TESPA1/LYN/RAB29/PAK1/GRB2/LPXN/EP300/PIK3AP1/CD47/CD8B/DENND1B/CYLD/TLR2/OAS1/PIK3CD/ITGAM</code></pre>
<pre class="r"><code>## DE Genes for AM T cell low
DEGenesAM_TClow &lt;- DEGenesAM_sub %&gt;% filter(cluster == &quot;AM_TClow&quot;)
head(DEGenesAM_TClow)</code></pre>
<pre><code>                               p_val avg_log2FC pct.1 pct.2     p_val_adj  cluster  gene
ENSG00000229807.XIST2   0.000000e+00  0.9704090 0.436 0.219  0.000000e+00 AM_TClow  XIST
ENSG00000154451.GBP52  8.054547e-308  1.5826534 0.175 0.062 3.150456e-303 AM_TClow  GBP5
ENSG00000081237.PTPRC2 8.660812e-207  1.0649660 0.243 0.123 3.387590e-202 AM_TClow PTPRC
ENSG00000234745.HLA-B2 1.397489e-204  0.8085478 0.493 0.347 5.466140e-200 AM_TClow HLA-B
ENSG00000117228.GBP12  1.340811e-177  1.2646290 0.222 0.116 5.244447e-173 AM_TClow  GBP1
ENSG00000082074.FYB12  4.975289e-176  1.1294199 0.204 0.102 1.946035e-171 AM_TClow  FYB1
                                        Gene           EnsID
ENSG00000229807.XIST2   ENSG00000229807.XIST ENSG00000229807
ENSG00000154451.GBP52   ENSG00000154451.GBP5 ENSG00000154451
ENSG00000081237.PTPRC2 ENSG00000081237.PTPRC ENSG00000081237
ENSG00000234745.HLA-B2 ENSG00000234745.HLA-B ENSG00000234745
ENSG00000117228.GBP12   ENSG00000117228.GBP1 ENSG00000117228
ENSG00000082074.FYB12   ENSG00000082074.FYB1 ENSG00000082074</code></pre>
<pre class="r"><code>egoAM_TClow &lt;- enrichGO(gene = unique(DEGenesAM_TClow$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM_TClow &lt;- setReadable(egoAM_TClow, OrgDb = org.Hs.eg.db)
dotplot(egoAM_TClow, showCategory=20, title = &quot;Gene Ontology Acute Myocarditis T cell Low&quot;, font.size = 10)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>barplot(egoAM_TClow, 
        drop = TRUE, 
        showCategory = 7, 
        title = &quot;Gene Ontology Acute Myocarditis T cell Low&quot;,
        font.size = 10)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-10-2.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>#head(egoAM_TClow)
#View(egoAM_TClow@result)

#write.table(egoAM_TClow, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/V5 25.10.2024 AMvsHealthy/egoAM_TClow&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

# PTPRC/GBP1/FYB1/HLA-A/NCKAP1L/LCP2/BTN3A2/PRKCB/CYBA/PTPN22/ITK/TRBC2/ZAP70/MYO1G/BTN3A1/LCK/TRBC1/CLEC7A/CD247/LAPTM5/TNFAIP3/MNDA/BTK/TRAC/INPP5D/FCN1/HLA-DQB1/CYLD/CD300A/CD3G/LAT2/CD8A/BTN3A3/PIK3CD/LILRB4/HLA-DRB1/SKAP1/TYROBP/CD38/OAS1/FCMR/FGR/TLR6/PTPN6/CD3D/SCIMP/KLRC1/PAK1/BTN2A2/PTPN2/CD226/LIMK1/PTPRJ/GRB2/HLA-DPB1/UBASH3A/KCNN4/TLR2/FCER1G/KLRD1/FYN/DENND1B/CD3E/CD47/IGKC/TESPA1/CSK/MS4A1/PIK3AP1/SOS1/NMI/EP300/LAT/SYK/LAX1/CARD11/THEMIS/WNK1/TRAF3/ITGAM/DGKZ/LYN/VAV1/BANK1/TLR4/IGHM/NFAM1/MYD88/GATA3/IRAK1/PRAM1/SH2D1A/CTLA4/SLC39A10/RBCK1/PRKCD/TRGC1/TNIP3/SPG21/ZC3H12A/IRAK4/BCL2/WDFY1/FCHO1/TLR1/BLNK/NOD2/TXK/PLCL2/BAX/CCR7/LPXN/MALT1/SLA2/KLRK1/C3AR1/NFATC2/HCK/KLHL6/CD8B/NR1H3/IFI35/RAB29/PLCG2/NFKBID/STK11/NFKB1/TRGC2/PRKCH/PAK3/VAV3/PRKD2/PDE4B</code></pre>
<pre class="r"><code>##DE Genes for AM T cell very low
DEGenesAM_TCvlow &lt;- DEGenesAM_sub %&gt;% filter(cluster == &quot;AM_TCvlow&quot;)
head(DEGenesAM_TCvlow)</code></pre>
<pre><code>                      p_val avg_log2FC pct.1 pct.2 p_val_adj   cluster  gene                  Gene
ENSG00000229807.XIST      0  1.0927903 0.432 0.199         0 AM_TCvlow  XIST  ENSG00000229807.XIST
ENSG00000234745.HLA-B     0  0.9191855 0.497 0.332         0 AM_TCvlow HLA-B ENSG00000234745.HLA-B
ENSG00000206503.HLA-A     0  0.9120121 0.429 0.272         0 AM_TCvlow HLA-A ENSG00000206503.HLA-A
ENSG00000081237.PTPRC     0  1.1923275 0.240 0.112         0 AM_TCvlow PTPRC ENSG00000081237.PTPRC
ENSG00000154451.GBP5      0  1.9201603 0.175 0.051         0 AM_TCvlow  GBP5  ENSG00000154451.GBP5
ENSG00000082074.FYB1      0  1.3998114 0.209 0.091         0 AM_TCvlow  FYB1  ENSG00000082074.FYB1
                                EnsID
ENSG00000229807.XIST  ENSG00000229807
ENSG00000234745.HLA-B ENSG00000234745
ENSG00000206503.HLA-A ENSG00000206503
ENSG00000081237.PTPRC ENSG00000081237
ENSG00000154451.GBP5  ENSG00000154451
ENSG00000082074.FYB1  ENSG00000082074</code></pre>
<pre class="r"><code>egoAM_TCvlow &lt;- enrichGO(gene = unique(DEGenesAM_TCvlow$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM_TCvlow &lt;- setReadable(egoAM_TCvlow, OrgDb = org.Hs.eg.db)
dotplot(egoAM_TCvlow, showCategory=20, title = &quot;Gene Ontology Acute Myocarditis T cell Very Low&quot;, font.size = 10)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>barplot(egoAM_TCvlow, 
        drop = TRUE, 
        showCategory = 7, 
        title = &quot;Gene Ontology Acute Myocarditis T cell Very Low&quot;,
        font.size = 10)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-11-2.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>#head(egoAM_TCvlow)

#View(egoAM_TCvlow@result)

#write.table(egoAM_TCvlow, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/V5 25.10.2024 AMvsHealthy/egoAM_TCvlow&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

# PTPRC/CD2/IL7R/IKZF1/BCL11B/B2M/IRF1/RUNX3/DOCK10/ITGA4/IKZF3/NCKAP1L/JAK3/DOCK2/HLA-DRA/PTK2B/PTPN22/LAG3/ATM/LCK/FGL2/ITK/CD74/SPN/ZAP70/CD3E/IL2RG/LILRB4/ZBTB16/KLRC1/IRF8/CD3G/SLAMF8/INPP5D/IL4I1/HLA-DRB1/PTPN6/LILRB2/THEMIS/TNFSF13B/CYLD/JAK1/CD8A/PREX1/PTPRJ/RHOH/CAMK4/SLAMF1/EZH2/SASH3/PIK3CD/ACTB/LY9/CD1D/TCIRG1/TBX21/ZNF683/IL12RB1/SPI1/STAT4/SOS1/CD3D/SLAMF6/ADAM8/BTK/CD4/PTPN2/KMT2A/CCR2/MR1/NFKBID/DOCK11/NOTCH2/VAV1/TMEM131L/STAT6/IL15/HLA-DOA/LGALS9/TESPA1/MS4A1/ASCL2/GATA3/FANCA/CARD11/TGFB1/LEF1/ZFP36L2/SOCS1/SEMA4A/LFNG/BLNK/DCLRE1C/CD80/BTN2A2/CD27/TNFRSF9/TOX/CTLA4/PBRM1/NFATC3/MALT1/SMARCA2/BCL2/NFAM1/SYK/FCER1G/PRDM1/TOP2B/BATF/PRKDC/EOMES/ZBTB1/RUNX2/EP300/IRF4/FANCD2/PPP2R3C/STAT5B/RASGRP1/IL6R/IL18R1/CD86/CD83/SMARCC1/RIPK3/RORA/AP3D1/SOS2/PPP3CB/RC3H2/RHOA/NCAPH2/HDAC4/CDK6/CCR7/MAFB/PLCL2/AP3B1/IL15RA/PLCG2/NDFIP1/LIG4/IL18/PTGER4/LIPA/BAX/RSAD2/DROSHA/YY1/SMARCD2/STK11/TNFSF8/ZBTB7B/ZC3H12A/PCID2/PHF10/CD28/MSH2/HHEX/DUSP10/BAK1/KAT5/GPR89B/PNP/RIPK2/RNF41/SART1/MEN1/CMTM7/MDK</code></pre>
<pre class="r"><code>## DE Genes for Healthy Controls
DEGenesHealthy &lt;- DEGenesAM_sub %&gt;% filter(cluster == &quot;Healthy&quot;)
head(DEGenesHealthy)</code></pre>
<pre><code>                        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster    gene
ENSG00000210082.MT-RNR2     0  0.9350349 0.687 0.491         0 Healthy MT-RNR2
ENSG00000155657.TTN         0  0.2171542 0.753 0.591         0 Healthy     TTN
ENSG00000198899.MT-ATP6     0  1.1665477 0.322 0.181         0 Healthy MT-ATP6
ENSG00000092054.MYH7        0  0.2515285 0.452 0.315         0 Healthy    MYH7
ENSG00000198712.MT-CO2      0  1.2251589 0.285 0.156         0 Healthy  MT-CO2
ENSG00000198727.MT-CYB      0  1.3541934 0.262 0.140         0 Healthy  MT-CYB
                                           Gene           EnsID
ENSG00000210082.MT-RNR2 ENSG00000210082.MT-RNR2 ENSG00000210082
ENSG00000155657.TTN         ENSG00000155657.TTN ENSG00000155657
ENSG00000198899.MT-ATP6 ENSG00000198899.MT-ATP6 ENSG00000198899
ENSG00000092054.MYH7       ENSG00000092054.MYH7 ENSG00000092054
ENSG00000198712.MT-CO2   ENSG00000198712.MT-CO2 ENSG00000198712
ENSG00000198727.MT-CYB   ENSG00000198727.MT-CYB ENSG00000198727</code></pre>
<pre class="r"><code>egoHealthy &lt;- enrichGO(gene = unique(DEGenesHealthy$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy &lt;- setReadable(egoHealthy, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy, showCategory=20, title = &quot;Gene Ontology Healthy Controls&quot;, font.size = 10)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>barplot(egoHealthy, 
        drop = TRUE, 
        showCategory = 7, 
        title = &quot;Gene Ontology Healthy Controls&quot;,
        font.size = 10)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-12-2.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>#head(egoHealthy)
#View(egoHealthy@result)

#write.table(egoHealthy, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/V5 25.10.2024 AMvsHealthy/egoHealthy&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

# TTN/MYH7/CALD1/NR4A1/FGF12/MYL2/MYBPC3/TCAP/KCNE4/MYL3/ATP1A2/SORBS2/RGS2/SCN9A/MYOZ2/CSRP3/PKP2/RNF207/CACNA1C/DES/MLIP/ABCC9/TRDN/TNNC1/CTNNA3/CKMT2/AGT/TPM2/AKAP6/STC1/CRYAB/MYL9/KLF15/PDLIM5/ANKRD2/IGFBP5/PLN/CACNB2/DSP/NKX2-5/KCNJ8/MYL4/HOMER1/PGAM2/CASQ2/LMOD2/TNNI3K/P2RX3/SLC8A1/ADRA2B/ADRA1A/PDE4D/HAND2/PARP2/ATP1B1/DTNA/SMPX/KCND3/PPP1R12B/GJC1/SYNM/ATP1A1/SGCD/RYR1/TMEM38B/GPER1/MEIS1/COL14A1</code></pre>
<pre><code>



``` r
#for Volcano Plot, pos. and neg. markers
DEGenes_AMvsHealthy_all &lt;- FindAllMarkers (seuratM, only.pos = FALSE, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.05)

#write.table(DEGenes_AMvsHealthy_all, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenes_AMvsHealthy_all&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

##adjust table
DEGenes_AMvsHealthy_adj &lt;- DEGenes_AMvsHealthy_all %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

##Volcano Plot for AM with TC high
DEGenesAM_high &lt;- DEGenes_AMvsHealthy_adj %&gt;% filter(cluster == &quot;AM_TChigh&quot;)
head(DEGenesAM_high)</code></pre>
<pre><code>                                p_val avg_log2FC pct.1 pct.2     p_val_adj   cluster  gene
ENSG00000229807.XIST.3  3.054540e-154  0.8636867 0.435 0.234 1.194753e-149 AM_TChigh  XIST
ENSG00000154451.GBP5.3  4.244298e-118  1.3598308 0.174 0.070 1.660115e-113 AM_TChigh  GBP5
ENSG00000082074.FYB1.3   4.912684e-85  1.0734455 0.214 0.109  1.921547e-80 AM_TChigh  FYB1
ENSG00000234745.HLA-B.3  2.131302e-82  0.7317787 0.496 0.357  8.336375e-78 AM_TChigh HLA-B
ENSG00000081237.PTPRC.3  3.624135e-78  0.9901756 0.241 0.132  1.417544e-73 AM_TChigh PTPRC
ENSG00000206503.HLA-A.3  2.641748e-77  0.7550193 0.432 0.295  1.033293e-72 AM_TChigh HLA-A
                                         Gene           EnsID
ENSG00000229807.XIST.3   ENSG00000229807.XIST ENSG00000229807
ENSG00000154451.GBP5.3   ENSG00000154451.GBP5 ENSG00000154451
ENSG00000082074.FYB1.3   ENSG00000082074.FYB1 ENSG00000082074
ENSG00000234745.HLA-B.3 ENSG00000234745.HLA-B ENSG00000234745
ENSG00000081237.PTPRC.3 ENSG00000081237.PTPRC ENSG00000081237
ENSG00000206503.HLA-A.3 ENSG00000206503.HLA-A ENSG00000206503</code></pre>
<pre class="r"><code>res_AMhigh &lt;- as.data.frame(DEGenesAM_high)

EnhancedVolcano(res_AMhigh, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AMhigh$gene,
                # xlim =c(-6, 6), 
                title = &quot;DEGenes AM T cell High Condition&quot;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>top_genes_AM &lt;- DEGenesAM_high %&gt;% 
  filter(p_val_adj &lt; 0.05, abs(avg_log2FC) &gt; 1) %&gt;% 
  arrange(p_val_adj) %&gt;% 
  slice_head(n = 30) %&gt;% 
  pull(gene)</code></pre>
<pre class="r"><code>##Volcano Plot for AM with TC intermediate
DEGenesAM_int &lt;- DEGenes_AMvsHealthy_adj %&gt;% filter(cluster == &quot;AM_TCint&quot;)
head(DEGenesAM_int)</code></pre>
<pre><code>                                p_val avg_log2FC pct.1 pct.2     p_val_adj  cluster  gene
ENSG00000229807.XIST.1  1.298826e-126  0.9104069 0.434 0.236 5.080228e-122 AM_TCint  XIST
ENSG00000154451.GBP5.1   1.196845e-93  1.2687426 0.174 0.071  4.681341e-89 AM_TCint  GBP5
ENSG00000081237.PTPRC.1  8.603204e-68  0.9856375 0.245 0.133  3.365057e-63 AM_TCint PTPRC
ENSG00000082074.FYB1.1   9.995754e-66  1.0382078 0.212 0.110  3.909739e-61 AM_TCint  FYB1
ENSG00000005844.ITGAL.1  4.352593e-59  1.1070272 0.155 0.073  1.702473e-54 AM_TCint ITGAL
ENSG00000107099.DOCK8.1  3.680243e-55  1.0578340 0.198 0.106  1.439490e-50 AM_TCint DOCK8
                                         Gene           EnsID
ENSG00000229807.XIST.1   ENSG00000229807.XIST ENSG00000229807
ENSG00000154451.GBP5.1   ENSG00000154451.GBP5 ENSG00000154451
ENSG00000081237.PTPRC.1 ENSG00000081237.PTPRC ENSG00000081237
ENSG00000082074.FYB1.1   ENSG00000082074.FYB1 ENSG00000082074
ENSG00000005844.ITGAL.1 ENSG00000005844.ITGAL ENSG00000005844
ENSG00000107099.DOCK8.1 ENSG00000107099.DOCK8 ENSG00000107099</code></pre>
<pre class="r"><code>res_AMint &lt;- as.data.frame(DEGenesAM_int)

EnhancedVolcano(res_AMint, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AMint$gene,
                # xlim =c(-6, 6), 
                title = &quot;DE Genes T cell Intermediate Condition&quot;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>top_genes_AM &lt;- DEGenesAM_int %&gt;% 
  filter(p_val_adj &lt; 0.05, abs(avg_log2FC) &gt; 1) %&gt;% 
  arrange(p_val_adj) %&gt;% 
  slice_head(n = 30) %&gt;% 
  pull(gene)</code></pre>
<pre class="r"><code>##Volcano Plot for AM with TC low
DEGenesAM_low &lt;- DEGenes_AMvsHealthy_adj %&gt;% filter(cluster == &quot;AM_TClow&quot;)
head(DEGenesAM_low)</code></pre>
<pre><code>                                p_val avg_log2FC pct.1 pct.2     p_val_adj  cluster  gene
ENSG00000229807.XIST.2   0.000000e+00  0.9704090 0.436 0.219  0.000000e+00 AM_TClow  XIST
ENSG00000154451.GBP5.2  8.054547e-308  1.5826534 0.175 0.062 3.150456e-303 AM_TClow  GBP5
ENSG00000081237.PTPRC.2 8.660812e-207  1.0649660 0.243 0.123 3.387590e-202 AM_TClow PTPRC
ENSG00000234745.HLA-B.2 1.397489e-204  0.8085478 0.493 0.347 5.466140e-200 AM_TClow HLA-B
ENSG00000117228.GBP1.2  1.340811e-177  1.2646290 0.222 0.116 5.244447e-173 AM_TClow  GBP1
ENSG00000082074.FYB1.2  4.975289e-176  1.1294199 0.204 0.102 1.946035e-171 AM_TClow  FYB1
                                         Gene           EnsID
ENSG00000229807.XIST.2   ENSG00000229807.XIST ENSG00000229807
ENSG00000154451.GBP5.2   ENSG00000154451.GBP5 ENSG00000154451
ENSG00000081237.PTPRC.2 ENSG00000081237.PTPRC ENSG00000081237
ENSG00000234745.HLA-B.2 ENSG00000234745.HLA-B ENSG00000234745
ENSG00000117228.GBP1.2   ENSG00000117228.GBP1 ENSG00000117228
ENSG00000082074.FYB1.2   ENSG00000082074.FYB1 ENSG00000082074</code></pre>
<pre class="r"><code>res_AMlow &lt;- as.data.frame(DEGenesAM_low)

EnhancedVolcano(res_AMlow, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AMlow$gene,
                # xlim =c(-6, 6), 
                title = &quot;DE Genes AM T cell Low Condition&quot;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>top_genes_AM &lt;- DEGenesAM_low %&gt;% 
  filter(p_val_adj &lt; 0.05, abs(avg_log2FC) &gt; 1) %&gt;% 
  arrange(p_val_adj) %&gt;% 
  slice_head(n = 30) %&gt;% 
  pull(gene)</code></pre>
<pre class="r"><code>##Volcano Plot for AM with TC low
DEGenesAM_vlow &lt;- DEGenes_AMvsHealthy_adj %&gt;% filter(cluster == &quot;AM_TCvlow&quot;)
head(DEGenesAM_vlow)</code></pre>
<pre><code>                      p_val avg_log2FC pct.1 pct.2 p_val_adj   cluster  gene                  Gene
ENSG00000229807.XIST      0  1.0927903 0.432 0.199         0 AM_TCvlow  XIST  ENSG00000229807.XIST
ENSG00000234745.HLA-B     0  0.9191855 0.497 0.332         0 AM_TCvlow HLA-B ENSG00000234745.HLA-B
ENSG00000206503.HLA-A     0  0.9120121 0.429 0.272         0 AM_TCvlow HLA-A ENSG00000206503.HLA-A
ENSG00000081237.PTPRC     0  1.1923275 0.240 0.112         0 AM_TCvlow PTPRC ENSG00000081237.PTPRC
ENSG00000154451.GBP5      0  1.9201603 0.175 0.051         0 AM_TCvlow  GBP5  ENSG00000154451.GBP5
ENSG00000082074.FYB1      0  1.3998114 0.209 0.091         0 AM_TCvlow  FYB1  ENSG00000082074.FYB1
                                EnsID
ENSG00000229807.XIST  ENSG00000229807
ENSG00000234745.HLA-B ENSG00000234745
ENSG00000206503.HLA-A ENSG00000206503
ENSG00000081237.PTPRC ENSG00000081237
ENSG00000154451.GBP5  ENSG00000154451
ENSG00000082074.FYB1  ENSG00000082074</code></pre>
<pre class="r"><code>res_AMvlow &lt;- as.data.frame(DEGenesAM_vlow)

EnhancedVolcano(res_AMvlow, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AMvlow$gene,
                # xlim =c(-6, 6), 
                title = &quot;DEGenes AM T cell Very Low Condition&quot;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>top_genes_AM &lt;- DEGenesAM_vlow %&gt;% 
  filter(p_val_adj &lt; 0.05, abs(avg_log2FC) &gt; 1) %&gt;% 
  arrange(p_val_adj) %&gt;% 
  slice_head(n = 30) %&gt;% 
  pull(gene)</code></pre>
<pre class="r"><code>##Volcano Plot for Healthy Controls
DEGenesHealthy &lt;- DEGenes_AMvsHealthy_adj %&gt;% filter(cluster == &quot;Healthy&quot;)
head(DEGenesHealthy)</code></pre>
<pre><code>                          p_val avg_log2FC pct.1 pct.2 p_val_adj cluster    gene
ENSG00000234745.HLA-B.4       0 -1.4197742 0.233 0.445         0 Healthy   HLA-B
ENSG00000210082.MT-RNR2.4     0  0.9350349 0.687 0.491         0 Healthy MT-RNR2
ENSG00000206503.HLA-A.4       0 -1.3787395 0.181 0.377         0 Healthy   HLA-A
ENSG00000005810.MYCBP2.4      0 -0.8737209 0.270 0.459         0 Healthy  MYCBP2
ENSG00000229807.XIST.4        0 -1.3355026 0.128 0.315         0 Healthy    XIST
ENSG00000101596.SMCHD1.4      0 -1.1298672 0.204 0.386         0 Healthy  SMCHD1
                                             Gene           EnsID
ENSG00000234745.HLA-B.4     ENSG00000234745.HLA-B ENSG00000234745
ENSG00000210082.MT-RNR2.4 ENSG00000210082.MT-RNR2 ENSG00000210082
ENSG00000206503.HLA-A.4     ENSG00000206503.HLA-A ENSG00000206503
ENSG00000005810.MYCBP2.4   ENSG00000005810.MYCBP2 ENSG00000005810
ENSG00000229807.XIST.4       ENSG00000229807.XIST ENSG00000229807
ENSG00000101596.SMCHD1.4   ENSG00000101596.SMCHD1 ENSG00000101596</code></pre>
<pre class="r"><code>res_Healthy &lt;- as.data.frame(DEGenesHealthy)

EnhancedVolcano(res_Healthy, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Healthy$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Healthy Controls&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<img src="figure/DE%20genes%20AMsub%20vs.%20Healthy_V2.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>top_genes_AM &lt;- DEGenesAM_vlow %&gt;% 
  filter(p_val_adj &lt; 0.05, abs(avg_log2FC) &gt; 1) %&gt;% 
  arrange(p_val_adj) %&gt;% 
  slice_head(n = 30) %&gt;% 
  pull(gene)</code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Wed Oct 30 12:31:23 2024&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pathview_1.44.0             wordcloud_2.6               textshaping_0.4.0          
 [4] workflowr_1.7.1             rmarkdown_2.28              EnhancedVolcano_1.22.0     
 [7] ggrepel_0.9.6               RColorBrewer_1.1-3          NCmisc_1.2.0               
[10] VennDiagram_1.7.3           futile.logger_1.4.3         ggupset_0.4.0              
[13] gridExtra_2.3               DOSE_3.30.5                 enrichplot_1.24.4          
[16] msigdbr_7.5.1               org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0       
[19] clusterProfiler_4.12.6      multtest_2.60.0             metap_1.11                 
[22] scater_1.32.1               scuttle_1.14.0              destiny_3.18.0             
[25] circlize_0.4.16             muscat_1.18.0               viridis_0.6.5              
[28] viridisLite_0.4.2           lubridate_1.9.3             forcats_1.0.0              
[31] stringr_1.5.1               purrr_1.0.2                 readr_2.1.5                
[34] tidyr_1.3.1                 tibble_3.2.1                tidyverse_2.0.0            
[37] dplyr_1.1.4                 SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
[40] Biobase_2.64.0              GenomicRanges_1.56.1        GenomeInfoDb_1.40.1        
[43] IRanges_2.38.1              S4Vectors_0.42.1            BiocGenerics_0.50.0        
[46] MatrixGenerics_1.16.0       matrixStats_1.4.1           pheatmap_1.0.12            
[49] ggpubr_0.6.0                ggplot2_3.5.1               Seurat_5.1.0               
[52] SeuratObject_5.0.2          sp_2.1-4                    runSeurat3_0.1.0           
[55] ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             presto_1.0.0              parallelly_1.38.0        
[145] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[148] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[151] pbapply_1.7-2             httr2_1.0.5               spam_2.11-0              
[154] gson_0.1.0                utf8_1.2.4                graphlayouts_1.2.0       
[157] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[160] shiny_1.9.1               GenomeInfoDbData_1.2.12   glmmTMB_1.1.10           
[163] R.utils_2.12.3            RCurl_1.98-1.16           memoise_2.0.1            
[166] scales_1.3.0              R.methodsS3_1.8.2         future_1.34.0            
[169] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-2      
[172] rstudioapi_0.16.0         whisker_0.4.1             cluster_2.1.6            
[175] mutoss_0.1-13             spatstat.utils_3.1-0      hms_1.1.3                
[178] fitdistrplus_1.2-1        munsell_0.5.1             cowplot_1.1.3            
[181] colorspace_2.1-1          rlang_1.1.4               DelayedMatrixStats_1.26.0
[184] sparseMatrixStats_1.16.0  xts_0.14.0                dotCall64_1.2            
[187] shinydashboard_0.7.2      ggforce_0.4.2             laeken_0.5.3             
[190] mgcv_1.9-1                xfun_0.48                 e1071_1.7-16             
[193] TH.data_1.1-2             iterators_1.0.14          abind_1.4-8              
[196] GOSemSim_2.30.2           treeio_1.28.0             ps_1.8.0                 
[199] futile.options_1.0.1      bitops_1.0-9              Rdpack_2.6.1             
[202] promises_1.3.0            scatterpie_0.2.4          RSQLite_2.3.7            
[205] qvalue_2.36.0             sandwich_3.1-1            fgsea_1.30.0             
[208] DelayedArray_0.30.1       proxy_0.4-27              GO.db_3.19.1             
[211] compiler_4.4.1            prettyunits_1.2.0         boot_1.3-31              
[214] beachmat_2.20.0           listenv_0.9.1             Rcpp_1.0.13              
[217] edgeR_4.2.1               BiocSingular_1.20.0       tensor_1.5               
[220] MASS_7.3-61               progress_1.2.3            BiocParallel_1.38.0      
[223] babelgene_22.9            spatstat.random_3.3-2     R6_2.5.1                 
[226] fastmap_1.2.0             multcomp_1.4-26           fastmatch_1.1-4          
[229] rstatix_0.7.2             vipor_0.4.7               TTR_0.24.4               
[232] ROCR_1.0-11               TFisher_0.2.0             rsvd_1.0.5               
[235] vcd_1.4-13                nnet_7.3-19               gtable_0.3.5             
[238] KernSmooth_2.23-24        miniUI_0.1.1.1            deldir_2.0-4             
[241] htmltools_0.5.8.1         ggthemes_5.1.0            bit64_4.5.2              
[244] spatstat.explore_3.3-2    lifecycle_1.0.4           blme_1.0-6               
[247] processx_3.8.4            callr_3.7.6               nloptr_2.1.1             
[250] sass_0.4.9                vctrs_0.6.5               robustbase_0.99-4-1      
[253] spatstat.geom_3.3-3       sn_2.1.1                  ggfun_0.1.6              
[256] future.apply_1.11.2       bslib_0.8.0               pillar_1.9.0             
[259] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.10          
[262] jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pathview_1.44.0             wordcloud_2.6               textshaping_0.4.0          
 [4] workflowr_1.7.1             rmarkdown_2.28              EnhancedVolcano_1.22.0     
 [7] ggrepel_0.9.6               RColorBrewer_1.1-3          NCmisc_1.2.0               
[10] VennDiagram_1.7.3           futile.logger_1.4.3         ggupset_0.4.0              
[13] gridExtra_2.3               DOSE_3.30.5                 enrichplot_1.24.4          
[16] msigdbr_7.5.1               org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0       
[19] clusterProfiler_4.12.6      multtest_2.60.0             metap_1.11                 
[22] scater_1.32.1               scuttle_1.14.0              destiny_3.18.0             
[25] circlize_0.4.16             muscat_1.18.0               viridis_0.6.5              
[28] viridisLite_0.4.2           lubridate_1.9.3             forcats_1.0.0              
[31] stringr_1.5.1               purrr_1.0.2                 readr_2.1.5                
[34] tidyr_1.3.1                 tibble_3.2.1                tidyverse_2.0.0            
[37] dplyr_1.1.4                 SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
[40] Biobase_2.64.0              GenomicRanges_1.56.1        GenomeInfoDb_1.40.1        
[43] IRanges_2.38.1              S4Vectors_0.42.1            BiocGenerics_0.50.0        
[46] MatrixGenerics_1.16.0       matrixStats_1.4.1           pheatmap_1.0.12            
[49] ggpubr_0.6.0                ggplot2_3.5.1               Seurat_5.1.0               
[52] SeuratObject_5.0.2          sp_2.1-4                    runSeurat3_0.1.0           
[55] ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             presto_1.0.0              parallelly_1.38.0        
[145] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[148] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[151] pbapply_1.7-2             httr2_1.0.5               spam_2.11-0              
[154] gson_0.1.0                utf8_1.2.4                graphlayouts_1.2.0       
[157] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[160] shiny_1.9.1               GenomeInfoDbData_1.2.12   glmmTMB_1.1.10           
[163] R.utils_2.12.3            RCurl_1.98-1.16           memoise_2.0.1            
[166] scales_1.3.0              R.methodsS3_1.8.2         future_1.34.0            
[169] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-2      
[172] rstudioapi_0.16.0         whisker_0.4.1             cluster_2.1.6            
[175] mutoss_0.1-13             spatstat.utils_3.1-0      hms_1.1.3                
[178] fitdistrplus_1.2-1        munsell_0.5.1             cowplot_1.1.3            
[181] colorspace_2.1-1          rlang_1.1.4               DelayedMatrixStats_1.26.0
[184] sparseMatrixStats_1.16.0  xts_0.14.0                dotCall64_1.2            
[187] shinydashboard_0.7.2      ggforce_0.4.2             laeken_0.5.3             
[190] mgcv_1.9-1                xfun_0.48                 e1071_1.7-16             
[193] TH.data_1.1-2             iterators_1.0.14          abind_1.4-8              
[196] GOSemSim_2.30.2           treeio_1.28.0             ps_1.8.0                 
[199] futile.options_1.0.1      bitops_1.0-9              Rdpack_2.6.1             
[202] promises_1.3.0            scatterpie_0.2.4          RSQLite_2.3.7            
[205] qvalue_2.36.0             sandwich_3.1-1            fgsea_1.30.0             
[208] DelayedArray_0.30.1       proxy_0.4-27              GO.db_3.19.1             
[211] compiler_4.4.1            prettyunits_1.2.0         boot_1.3-31              
[214] beachmat_2.20.0           listenv_0.9.1             Rcpp_1.0.13              
[217] edgeR_4.2.1               BiocSingular_1.20.0       tensor_1.5               
[220] MASS_7.3-61               progress_1.2.3            BiocParallel_1.38.0      
[223] babelgene_22.9            spatstat.random_3.3-2     R6_2.5.1                 
[226] fastmap_1.2.0             multcomp_1.4-26           fastmatch_1.1-4          
[229] rstatix_0.7.2             vipor_0.4.7               TTR_0.24.4               
[232] ROCR_1.0-11               TFisher_0.2.0             rsvd_1.0.5               
[235] vcd_1.4-13                nnet_7.3-19               gtable_0.3.5             
[238] KernSmooth_2.23-24        miniUI_0.1.1.1            deldir_2.0-4             
[241] htmltools_0.5.8.1         ggthemes_5.1.0            bit64_4.5.2              
[244] spatstat.explore_3.3-2    lifecycle_1.0.4           blme_1.0-6               
[247] processx_3.8.4            callr_3.7.6               nloptr_2.1.1             
[250] sass_0.4.9                vctrs_0.6.5               robustbase_0.99-4-1      
[253] spatstat.geom_3.3-3       sn_2.1.1                  ggfun_0.1.6              
[256] future.apply_1.11.2       bslib_0.8.0               pillar_1.9.0             
[259] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.10          
[262] jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
