<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Payne" />

<meta name="date" content="2024-11-01" />

<title>Stratification: different types of Macrophages, activated Fibroblasts</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myocarditis snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Stratification: different types of
Macrophages, activated Fibroblasts</h1>
<h4 class="author">Emily Payne</h4>
<h4 class="date">2024-11-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-11-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Myocarditis
snRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat
files/seuratM_all_metadata_withAMvsHealthy_ventricle_sex_TC
con_27.10.2024.rds
</td>
<td style="text-align:left;">
../data/01 Merged Seurat
files/seuratM_all_metadata_withAMvsHealthy_ventricle_sex_TC
con_27.10.2024.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree4daf12687e1b1561624f77778fa38396be0bb8detargetblank4daf126a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/4daf12687e1b1561624f77778fa38396be0bb8de" target="_blank">4daf126</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree4daf12687e1b1561624f77778fa38396be0bb8detargetblank4daf126a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/4daf12687e1b1561624f77778fa38396be0bb8de" target="_blank">4daf126</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/00 Inbox Documents and Pictures /
    Ignored:    data/01 Merged Seurat files/
    Ignored:    data/AM/
    Ignored:    data/Berlin/
    Ignored:    data/CM/
    Ignored:    data/Healthy/
    Ignored:    data/Sarcoidosis/

Untracked files:
    Untracked:  T cell abundance DE Kopie.Rmd
    Untracked:  analysis/Comparison AM by sex.Rmd
    Untracked:  analysis/DE Genes by Disease Condition.Rmd
    Untracked:  analysis/DE genes AMsub vs. Healthy_V2.Rmd
    Untracked:  analysis/Exploratory stratification.Rmd
    Untracked:  analysis/Exploratory-stratification.Rmd
    Untracked:  analysis/Fibroblast subset V2 reclustering.Rmd
    Untracked:  analysis/Install.packages_MFZ.R
    Untracked:  analysis/MCP and TC reembedding.Rmd
    Untracked:  analysis/Macrophage DE subset V2.Rmd
    Untracked:  analysis/Macrophage GSEA.Rmd
    Untracked:  analysis/Macrophage Reembedding.Rmd
    Untracked:  analysis/Macrophage embedding.Rmd
    Untracked:  analysis/Macrophage heatmap.Rmd
    Untracked:  analysis/Merging/
    Untracked:  analysis/Myocarditis snRNAseq QC and merging_V2_Tchigh vs. low.Rmd
    Untracked:  analysis/Relative Abundances and heatmap.Rmd
    Untracked:  analysis/Relative abundances.Rmd
    Untracked:  analysis/T cell abundance DE.Rmd
    Untracked:  analysis/T cell condition Differential expression.Rmd
    Untracked:  analysis/TC high vs. low condition.Rmd
    Untracked:  analysis/Tcell embedding and GSEA.Rmd
    Untracked:  analysis/V1/

Unstaged changes:
    Deleted:    _workflowr.yml
    Modified:   analysis/Myocarditis snRNAseq.Rmd
    Deleted:    analysis/license.Rmd
    Deleted:    data/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>suppressPackageStartupMessages({
  library(ExploreSCdataSeurat3)
  library(runSeurat3)
  library(Seurat)
  library(ggpubr)
  library(pheatmap)
  library(SingleCellExperiment)
  library(dplyr)
  library(tidyverse)
  library(viridis)
  library(muscat)
  library(circlize)
  library(destiny)
  library(scater)
  library(metap)
  library(multtest)
  library(clusterProfiler)
  library(org.Hs.eg.db)
  library(msigdbr)
  library(enrichplot)
  library(DOSE)
  library(grid)
  library(gridExtra)
  library(ggupset)
  library(VennDiagram)
  library(NCmisc)
  library(RColorBrewer)
  library(EnhancedVolcano)
  library(textshaping)
  library(wordcloud)
  library(pathview)
  library(workflowr)
})</code></pre>
<pre class="r"><code>##load merged file 
seuratM &lt;- readRDS(&quot;~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/seuratM_all_metadata_withAMvsHealthy_ventricle_sex_TC con_27.10.2024.rds&quot;)

table(seuratM$dataset)</code></pre>
<pre><code>
     312581_05-5_20230308_Hu_nucseq_SG36_EMB_GEM      312581_06-6_20230308_Hu_nucseq_SG37_EMB_GEM 
                                            4323                                             1123 
   353921_28-28_20240529_Hu_nucseq_Immpath_EMB18    353921_29-29_20240529_Hu_nucseq_Immpath_EMB24 
                                             208                                              105 
   353921_30-30_20240529_Hu_nucseq_Immpath_EMB25    353921_31-31_20240529_Hu_nucseq_Immpath_EMB27 
                                             227                                              579 
   353921_32-32_20240529_Hu_nucseq_Immpath_EMB29    353921_33-33_20240529_Hu_nucseq_Immpath_EMB30 
                                             335                                              730 
   353921_34-34_20240529_Hu_nucseq_Immpath_EMB35    353921_35-35_20240529_Hu_nucseq_Immpath_EMB37 
                                            1217                                              604 
359861_09-11_20240723_Hu_nucseq_Ber_Pat5_EM979_1 359861_10-13_20240723_Hu_nucseq_Ber_Pat5_EM979_2 
                                             771                                             1476 
 359861_11-1_20240730_Hu_nucseq_Ber_Pat1_EMB1537  359861_12-2_20240730_Hu_nucseq_Ber_Pat2_EMB1041 
                                             316                                              608 
    359861_16-8_20240730_Hu_nucseq_Immpath_EMB47     359861_17-9_20240730_Hu_nucseq_Immpath_EMB48 
                                             513                                              426 
     o27533_1_12-12_20220203_Hu_nucseq_EMB31_GEM        o27936_1_7-7_20220309_Hu_nucseq_EMB32_GEM 
                                             236                                             1192 
 o28576_1_01-1_20220525_Hu_nucseq_Graz_1_EMB_GEM  o28576_1_02-2_20220525_Hu_nucseq_Graz_2_EMB_GEM 
                                            2740                                             1684 
 o28576_1_03-3_20220525_Hu_nucseq_Graz_3_EMB_GEM  o28576_1_04-4_20220525_Hu_nucseq_Graz_4_EMB_GEM 
                                            2396                                              545 
 o28576_1_05-5_20220525_Hu_nucseq_Graz_5_EMB_GEM  o28576_1_06-6_20220525_Hu_nucseq_Graz_6_EMB_GEM 
                                             781                                              491 
 o28576_1_07-7_20220525_Hu_nucseq_Graz_7_EMB_GEM   o28576_1_08-8_20220525_Hu_nucseq_Graz_8_HH_GEM 
                                             653                                             3921 
o28576_1_12-12_20220525_Hu_nucseq_Graz_12_HH_GEM      o28576_1_13-13_20220525_Hu_nucseq_EMB32_GEM 
                                            3818                                             1428 
    o292731_1-1_20220818_Hu_nucseq_Graz_9_HH_GEM    o292731_2-2_20220818_Hu_nucseq_Graz_13_HH_GEM 
                                            4908                                             9882 
    o292731_3-3_20220818_Hu_nucseq_SG_33_EMB_GEM     o292731_4-4_20220818_Hu_nucseq_SG_34_EMB_GEM 
                                            6286                                              620 
    o292731_5-5_20220818_Hu_nucseq_SG_35_EMB_GEM   o294781_01-1_20220912_Hu_nucseq_Graz_21_HH_GEM 
                                            2363                                             1442 
  o294781_02-2_20220912_Hu_nucseq_Graz_22_HH_GEM   o294781_03-3_20220912_Hu_nucseq_Graz_23_HH_GEM 
                                            1998                                              841 
  o294781_04-4_20220912_Hu_nucseq_Graz_24_HH_GEM  o294781_05-5_20220912_Hu_nucseq_Graz_14_EMB_GEM 
                                            1480                                             1268 
 o294781_06-6_20220912_Hu_nucseq_Graz_15_EMB_GEM  o294781_09-9_20220912_Hu_nucseq_Graz_18_EMB_GEM 
                                            4439                                             2280 
o294781_10-10_20220912_Hu_nucseq_Graz_19_EMB_GEM o294781_11-11_20220912_Hu_nucseq_Graz_20_EMB_GEM 
                                             111                                             2706 </code></pre>
<pre class="r"><code>table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11    12    13    14 
17699 13639 12754  8222  5475  4836  3723  2365  2156   905   882   694   380   249    91 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
74070 </code></pre>
<pre class="r"><code># Stratification by resident macrophages MCP1

datList &lt;- NULL
for(con in unique(seuratM$patient)){
  seuratSub &lt;- subset(seuratM, patient==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 39114  4323
[1] 39114  1123
[1] 39114   208
[1] 39114   579
[1] 39114   335
[1] 39114  1217
[1] 39114   604
[1] 39114   513
[1] 39114   426
[1] 39114   236
[1] 39114  1192
[1] 39114  2740
[1] 39114  1684
[1] 39114  6286
[1] 39114  4439
[1] 39114  2247
[1] 39114   316
[1] 39114   608
[1] 39114   105
[1] 39114   227
[1] 39114   730
[1] 39114  2396
[1] 39114   545
[1] 39114   781
[1] 39114   491
[1] 39114   653
[1] 39114  1428
[1] 39114   620
[1] 39114  1268
[1] 39114  2280
[1] 39114   111
[1] 39114  3921
[1] 39114  3818
[1] 39114  4908
[1] 39114  9882
[1] 39114  1442
[1] 39114  1998
[1] 39114   841
[1] 39114  1480
[1] 39114  2363
[1] 39114  2706</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)


head(dat_all)</code></pre>
<pre><code>                 Var1 Freq     percent patient
AM 02.1 Cardiomyocyte  801 0.185287994   AM 02
AM 02.2           Fb1 1484 0.343280130   AM 02
AM 02.3           Fb2   15 0.003469813   AM 02
AM 02.4           Fb3   31 0.007170946   AM 02
AM 02.5          INT1   70 0.016192459   AM 02
AM 02.6       PeriFb1  645 0.149201943   AM 02</code></pre>
<pre class="r"><code># Filter MCP1
dat_all &lt;- filter(dat_all, Var1 == (&quot;MCP1&quot;))

dat_all$MCP1grp &lt;- NA

dat_all$MCP1grp [dat_all$percent &gt; 0.12] &lt;- &quot;MCP1 High &gt; 12%&quot;
dat_all$MCP1grp [dat_all$percent &lt;= 0.12 &amp; dat_all$percent &gt;=0.06] &lt;- &quot;MCP1 Intermediate 6-12%&quot;
dat_all$MCP1grp [dat_all$percent &lt; 0.06] &lt;- &quot;MCP1 Low &lt;6%&quot;

seuratM$MCP1grp &lt;- paste0(dat_all$MCP1grp, &#39;_&#39;, dat_all$patient)

table(dat_all$MCP1grp)</code></pre>
<pre><code>
        MCP1 High &gt; 12% MCP1 Intermediate 6-12%            MCP1 Low &lt;6% 
                      9                      19                      13 </code></pre>
<pre class="r"><code>#View(dat_all)

# Create the plot
ggplot(dat_all, aes(x = MCP1grp, y = percent, fill = MCP1grp)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.8) +  # Box plot without outliers
  geom_jitter(size = 0.5, width = 0.2, aes(color = MCP1grp)) +  # Jittered points
  theme_classic() +
  ggtitle(&quot;Frequencies of the Macrophages type 1 - resident macrophages&quot;) +
  xlab(&quot;MCP1 groups&quot;) +
  ylab(&quot;MCP1 Frequency&quot;) +
  stat_compare_means(method = &quot;anova&quot;, label = &quot;p.signif&quot;, hide.ns = FALSE) +
  stat_compare_means(label = &quot;p.format&quot;, comparisons = list(c(&quot;MCP1 High &gt; 12%&quot;, &quot;MCP1 Intermediate 6-12%&quot;, &quot;MCP1 Low &lt;6%&quot;), method = &quot;kruskal&quot;))</code></pre>
<img src="figure/Exploratory%20stratification.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code># Stratification by inflammatory Macrophages MCP2

datList &lt;- NULL
for(con in unique(seuratM$patient)){
  seuratSub &lt;- subset(seuratM, patient==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 39114  4323
[1] 39114  1123
[1] 39114   208
[1] 39114   579
[1] 39114   335
[1] 39114  1217
[1] 39114   604
[1] 39114   513
[1] 39114   426
[1] 39114   236
[1] 39114  1192
[1] 39114  2740
[1] 39114  1684
[1] 39114  6286
[1] 39114  4439
[1] 39114  2247
[1] 39114   316
[1] 39114   608
[1] 39114   105
[1] 39114   227
[1] 39114   730
[1] 39114  2396
[1] 39114   545
[1] 39114   781
[1] 39114   491
[1] 39114   653
[1] 39114  1428
[1] 39114   620
[1] 39114  1268
[1] 39114  2280
[1] 39114   111
[1] 39114  3921
[1] 39114  3818
[1] 39114  4908
[1] 39114  9882
[1] 39114  1442
[1] 39114  1998
[1] 39114   841
[1] 39114  1480
[1] 39114  2363
[1] 39114  2706</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)


#View(dat_all)


# Filter MCP2 cluster and subgroup into high, intermediate and low

dat_all &lt;- filter (dat_all, Var1 == c(&quot;MCP2&quot;))

dat_all$MCP2grp &lt;- NA

dat_all$MCP2grp[dat_all$percent &gt; 0.05] &lt;- &quot;MCP2 High &gt; 5%&quot;
dat_all$MCP2grp[dat_all$percent &lt;= 0.05 &amp; dat_all$percent &gt;= 0.01] &lt;- &quot;MCP2 Intermediate 1-5%&quot;
dat_all$MCP2grp[dat_all$percent &lt; 0.01] &lt;- &quot;MCP2 Low &lt;1%&quot;

seuratM$MCP2grp_patient &lt;- paste0(dat_all$MCP2grp, &#39;_&#39;, dat_all$patient)

table(dat_all$MCP2grp)</code></pre>
<pre><code>
        MCP2 High &gt; 5% MCP2 Intermediate 1-5%           MCP2 Low &lt;1% 
                     2                      8                     31 </code></pre>
<pre class="r"><code># MCP2 --&gt; 2 high, 6 intermediate, 35 low

head(dat_all)</code></pre>
<pre><code>         Var1 Freq     percent patient                MCP2grp
AM 02.14 MCP2   14 0.003238492   AM 02           MCP2 Low &lt;1%
AM 03.14 MCP2    2 0.001780944   AM 03           MCP2 Low &lt;1%
AM 04.14 MCP2    4 0.019230769   AM 04 MCP2 Intermediate 1-5%
AM 05.14 MCP2    7 0.012089810   AM 05 MCP2 Intermediate 1-5%
AM 06.14 MCP2    1 0.002985075   AM 06           MCP2 Low &lt;1%
AM 07.14 MCP2    7 0.005751849   AM 07           MCP2 Low &lt;1%</code></pre>
<pre class="r"><code># Create the plot
ggplot(dat_all, aes(x = MCP2grp, y = percent, fill = MCP2grp)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.8) +  # Box plot without outliers
  geom_jitter(size = 0.5, width = 0.2, aes(color = MCP2grp)) +  # Jittered points
  theme_classic() +
  ggtitle(&quot;Frequencies of the Macrophages type 2 - inflammatory subtype&quot;) +
  xlab(&quot;MCP2 groups&quot;) +
  ylab(&quot;MCP2 Frequency&quot;) +
  stat_compare_means(method = &quot;anova&quot;, label = &quot;p.signif&quot;, hide.ns = FALSE) +
  stat_compare_means(label = &quot;p.format&quot;, comparisons = list(c(&quot;MCP2 High &gt; 5%&quot;, &quot;MCP2 Intermediate 1-5%&quot;, &quot;MCP2 Low &lt;1%&quot;), method = &quot;kruskal&quot;))</code></pre>
<img src="figure/Exploratory%20stratification.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code># Stratification by Fibroblasts

datList &lt;- NULL
for(con in unique(seuratM$patient)){
  seuratSub &lt;- subset(seuratM, patient==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 39114  4323
[1] 39114  1123
[1] 39114   208
[1] 39114   579
[1] 39114   335
[1] 39114  1217
[1] 39114   604
[1] 39114   513
[1] 39114   426
[1] 39114   236
[1] 39114  1192
[1] 39114  2740
[1] 39114  1684
[1] 39114  6286
[1] 39114  4439
[1] 39114  2247
[1] 39114   316
[1] 39114   608
[1] 39114   105
[1] 39114   227
[1] 39114   730
[1] 39114  2396
[1] 39114   545
[1] 39114   781
[1] 39114   491
[1] 39114   653
[1] 39114  1428
[1] 39114   620
[1] 39114  1268
[1] 39114  2280
[1] 39114   111
[1] 39114  3921
[1] 39114  3818
[1] 39114  4908
[1] 39114  9882
[1] 39114  1442
[1] 39114  1998
[1] 39114   841
[1] 39114  1480
[1] 39114  2363
[1] 39114  2706</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)


head(dat_all)</code></pre>
<pre><code>                 Var1 Freq     percent patient
AM 02.1 Cardiomyocyte  801 0.185287994   AM 02
AM 02.2           Fb1 1484 0.343280130   AM 02
AM 02.3           Fb2   15 0.003469813   AM 02
AM 02.4           Fb3   31 0.007170946   AM 02
AM 02.5          INT1   70 0.016192459   AM 02
AM 02.6       PeriFb1  645 0.149201943   AM 02</code></pre>
<pre class="r"><code># Filter Fb2
dat_all &lt;- filter(dat_all, Var1 == (&quot;Fb2&quot;))

#View(dat_all) 
dat_all$Fb2grp &lt;- NA

dat_all$Fb2grp  [dat_all$percent &gt;= 0.01] &lt;- &quot;Fb2 High &gt;= 1%&quot;
dat_all$Fb2grp  [dat_all$percent &lt; 0.01] &lt;- &quot;Fb2 Low &lt; 1%&quot;

seuratM$Fb2grp &lt;- paste0(dat_all$Fb2grp, &#39;_&#39;, dat_all$patient)

table(dat_all$Fb2grp)</code></pre>
<pre><code>
Fb2 High &gt;= 1%   Fb2 Low &lt; 1% 
             3             38 </code></pre>
<pre class="r"><code>head(dat_all)</code></pre>
<pre><code>        Var1 Freq     percent patient         Fb2grp
AM 02.3  Fb2   15 0.003469813   AM 02   Fb2 Low &lt; 1%
AM 03.3  Fb2    0 0.000000000   AM 03   Fb2 Low &lt; 1%
AM 04.3  Fb2    1 0.004807692   AM 04   Fb2 Low &lt; 1%
AM 05.3  Fb2    0 0.000000000   AM 05   Fb2 Low &lt; 1%
AM 06.3  Fb2    0 0.000000000   AM 06   Fb2 Low &lt; 1%
AM 07.3  Fb2   28 0.023007395   AM 07 Fb2 High &gt;= 1%</code></pre>
<pre class="r"><code># Create the plot
ggplot(dat_all, aes(x = Fb2grp, y = percent, fill = Fb2grp)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.8) +  # Box plot without outliers
  geom_jitter(size = 2, width = 0.2, aes(color = Fb2grp)) +  # Jittered points
  theme_classic() +
  ggtitle(&quot;Frequencies of Fibroblasts type 2 - proinflammatory subtype&quot;) +
  xlab(&quot;Fb2 groups&quot;) +
  ylab(&quot;Fb2 Frequency&quot;) +
  stat_compare_means(method = &quot;anova&quot;, label = &quot;p.signif&quot;, hide.ns = FALSE) +
  stat_compare_means(label = &quot;p.format&quot;, comparisons = list(c(&quot;Fb2 High &gt;= 1%&quot;, &quot;Fb2 Low &lt;1%&quot;), method = &quot;kruskal&quot;))</code></pre>
<img src="figure/Exploratory%20stratification.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Sun Nov  3 17:29:47 2024&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.7.1             pathview_1.44.0             wordcloud_2.6              
 [4] textshaping_0.4.0           EnhancedVolcano_1.22.0      ggrepel_0.9.6              
 [7] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
[10] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
[13] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_7.5.1              
[16] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[19] multtest_2.60.0             metap_1.11                  scater_1.32.1              
[22] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[25] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[28] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[31] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[34] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[37] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[40] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[43] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[46] matrixStats_1.4.1           pheatmap_1.0.12             ggpubr_0.6.0               
[49] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[52] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             parallelly_1.38.0         mnormt_2.1.1             
[145] statmod_1.5.0             RcppHNSW_0.6.0            ScaledMatrix_1.12.0      
[148] carData_3.0-5             minqa_1.2.8               pbapply_1.7-2            
[151] httr2_1.0.5               spam_2.11-0               gson_0.1.0               
[154] utf8_1.2.4                graphlayouts_1.2.0        gtools_3.9.5             
[157] ggsignif_0.6.4            RcppEigen_0.3.4.0.2       shiny_1.9.1              
[160] GenomeInfoDbData_1.2.12   glmmTMB_1.1.10            R.utils_2.12.3           
[163] RCurl_1.98-1.16           memoise_2.0.1             rmarkdown_2.28           
[166] scales_1.3.0              R.methodsS3_1.8.2         future_1.34.0            
[169] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-2      
[172] rstudioapi_0.16.0         whisker_0.4.1             cluster_2.1.6            
[175] mutoss_0.1-13             spatstat.utils_3.1-0      hms_1.1.3                
[178] fitdistrplus_1.2-1        munsell_0.5.1             cowplot_1.1.3            
[181] colorspace_2.1-1          rlang_1.1.4               DelayedMatrixStats_1.26.0
[184] sparseMatrixStats_1.16.0  xts_0.14.0                dotCall64_1.2            
[187] shinydashboard_0.7.2      ggforce_0.4.2             laeken_0.5.3             
[190] mgcv_1.9-1                xfun_0.48                 e1071_1.7-16             
[193] TH.data_1.1-2             iterators_1.0.14          abind_1.4-8              
[196] GOSemSim_2.30.2           treeio_1.28.0             ps_1.8.0                 
[199] futile.options_1.0.1      bitops_1.0-9              Rdpack_2.6.1             
[202] promises_1.3.0            scatterpie_0.2.4          RSQLite_2.3.7            
[205] qvalue_2.36.0             sandwich_3.1-1            fgsea_1.30.0             
[208] DelayedArray_0.30.1       proxy_0.4-27              GO.db_3.19.1             
[211] compiler_4.4.1            prettyunits_1.2.0         boot_1.3-31              
[214] beachmat_2.20.0           listenv_0.9.1             Rcpp_1.0.13              
[217] edgeR_4.2.1               BiocSingular_1.20.0       tensor_1.5               
[220] MASS_7.3-61               progress_1.2.3            BiocParallel_1.38.0      
[223] babelgene_22.9            spatstat.random_3.3-2     R6_2.5.1                 
[226] fastmap_1.2.0             multcomp_1.4-26           fastmatch_1.1-4          
[229] rstatix_0.7.2             vipor_0.4.7               TTR_0.24.4               
[232] ROCR_1.0-11               TFisher_0.2.0             rsvd_1.0.5               
[235] vcd_1.4-13                nnet_7.3-19               gtable_0.3.5             
[238] KernSmooth_2.23-24        miniUI_0.1.1.1            deldir_2.0-4             
[241] htmltools_0.5.8.1         ggthemes_5.1.0            bit64_4.5.2              
[244] spatstat.explore_3.3-2    lifecycle_1.0.4           blme_1.0-6               
[247] processx_3.8.4            callr_3.7.6               nloptr_2.1.1             
[250] sass_0.4.9                vctrs_0.6.5               robustbase_0.99-4-1      
[253] spatstat.geom_3.3-3       sn_2.1.1                  ggfun_0.1.6              
[256] future.apply_1.11.2       bslib_0.8.0               pillar_1.9.0             
[259] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.10          
[262] jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.7.1             pathview_1.44.0             wordcloud_2.6              
 [4] textshaping_0.4.0           EnhancedVolcano_1.22.0      ggrepel_0.9.6              
 [7] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
[10] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
[13] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_7.5.1              
[16] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[19] multtest_2.60.0             metap_1.11                  scater_1.32.1              
[22] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[25] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[28] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[31] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[34] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[37] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[40] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[43] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[46] matrixStats_1.4.1           pheatmap_1.0.12             ggpubr_0.6.0               
[49] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[52] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             parallelly_1.38.0         mnormt_2.1.1             
[145] statmod_1.5.0             RcppHNSW_0.6.0            ScaledMatrix_1.12.0      
[148] carData_3.0-5             minqa_1.2.8               pbapply_1.7-2            
[151] httr2_1.0.5               spam_2.11-0               gson_0.1.0               
[154] utf8_1.2.4                graphlayouts_1.2.0        gtools_3.9.5             
[157] ggsignif_0.6.4            RcppEigen_0.3.4.0.2       shiny_1.9.1              
[160] GenomeInfoDbData_1.2.12   glmmTMB_1.1.10            R.utils_2.12.3           
[163] RCurl_1.98-1.16           memoise_2.0.1             rmarkdown_2.28           
[166] scales_1.3.0              R.methodsS3_1.8.2         future_1.34.0            
[169] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-2      
[172] rstudioapi_0.16.0         whisker_0.4.1             cluster_2.1.6            
[175] mutoss_0.1-13             spatstat.utils_3.1-0      hms_1.1.3                
[178] fitdistrplus_1.2-1        munsell_0.5.1             cowplot_1.1.3            
[181] colorspace_2.1-1          rlang_1.1.4               DelayedMatrixStats_1.26.0
[184] sparseMatrixStats_1.16.0  xts_0.14.0                dotCall64_1.2            
[187] shinydashboard_0.7.2      ggforce_0.4.2             laeken_0.5.3             
[190] mgcv_1.9-1                xfun_0.48                 e1071_1.7-16             
[193] TH.data_1.1-2             iterators_1.0.14          abind_1.4-8              
[196] GOSemSim_2.30.2           treeio_1.28.0             ps_1.8.0                 
[199] futile.options_1.0.1      bitops_1.0-9              Rdpack_2.6.1             
[202] promises_1.3.0            scatterpie_0.2.4          RSQLite_2.3.7            
[205] qvalue_2.36.0             sandwich_3.1-1            fgsea_1.30.0             
[208] DelayedArray_0.30.1       proxy_0.4-27              GO.db_3.19.1             
[211] compiler_4.4.1            prettyunits_1.2.0         boot_1.3-31              
[214] beachmat_2.20.0           listenv_0.9.1             Rcpp_1.0.13              
[217] edgeR_4.2.1               BiocSingular_1.20.0       tensor_1.5               
[220] MASS_7.3-61               progress_1.2.3            BiocParallel_1.38.0      
[223] babelgene_22.9            spatstat.random_3.3-2     R6_2.5.1                 
[226] fastmap_1.2.0             multcomp_1.4-26           fastmatch_1.1-4          
[229] rstatix_0.7.2             vipor_0.4.7               TTR_0.24.4               
[232] ROCR_1.0-11               TFisher_0.2.0             rsvd_1.0.5               
[235] vcd_1.4-13                nnet_7.3-19               gtable_0.3.5             
[238] KernSmooth_2.23-24        miniUI_0.1.1.1            deldir_2.0-4             
[241] htmltools_0.5.8.1         ggthemes_5.1.0            bit64_4.5.2              
[244] spatstat.explore_3.3-2    lifecycle_1.0.4           blme_1.0-6               
[247] processx_3.8.4            callr_3.7.6               nloptr_2.1.1             
[250] sass_0.4.9                vctrs_0.6.5               robustbase_0.99-4-1      
[253] spatstat.geom_3.3-3       sn_2.1.1                  ggfun_0.1.6              
[256] future.apply_1.11.2       bslib_0.8.0               pillar_1.9.0             
[259] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.10          
[262] jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
