<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Payne" />

<meta name="date" content="2024-11-04" />

<title>DE genes per patient</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myocarditis snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DE genes per patient</h1>
<h4 class="author">Emily Payne</h4>
<h4 class="date">2024-11-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-11-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Myocarditis
snRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis
snRNAseq/docs/Markers/DEGenes_patient positive markers_05.11.2024
</td>
<td style="text-align:left;">
../docs/Markers/DEGenes_patient positive markers_05.11.2024
</td>
</tr>
<tr>
<td style="text-align:left;">
~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat
files/seuratM_all_metadata_withAMvsHealthy_ventricle_sex_TC
con_27.10.2024.rds
</td>
<td style="text-align:left;">
../data/01 Merged Seurat
files/seuratM_all_metadata_withAMvsHealthy_ventricle_sex_TC
con_27.10.2024.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtreed254b0c164669a09465401a608ae7e7021a28867targetblankd254b0ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/d254b0c164669a09465401a608ae7e7021a28867" target="_blank">d254b0c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtreed254b0c164669a09465401a608ae7e7021a28867targetblankd254b0ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/d254b0c164669a09465401a608ae7e7021a28867" target="_blank">d254b0c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/01 Merged Seurat files/
    Ignored:    data/AM/
    Ignored:    data/Berlin/
    Ignored:    data/CM/
    Ignored:    data/Healthy/
    Ignored:    data/Sarcoidosis/

Untracked files:
    Untracked:  analysis/DE_Genes_Disease_Condition.Rmd
    Untracked:  analysis/DEgenes_patient_stratificationV3.Rmd
    Untracked:  analysis/Fibroblast subset V2 reclustering.Rmd
    Untracked:  analysis/Immune_cells.Rmd
    Untracked:  analysis/Myocarditis_snRNAseq_QC_merging_V3.Rmd
    Untracked:  analysis/Relative_AbundancesV2.Rmd
    Untracked:  analysis/SingleCellBrowserSeurat3_plotly.R
    Untracked:  analysis/Stratification_Tcell_condition.Rmd
    Untracked:  analysis/Stratification_sex.Rmd

Unstaged changes:
    Deleted:    analysis/DE Genes by Disease Condition.Rmd
    Deleted:    analysis/Myocarditis snRNAseq QC and merging_V2.Rmd
    Deleted:    analysis/Myocarditis snRNAseq.Rmd
    Deleted:    analysis/Relative AbundancesV2.Rmd
    Deleted:    analysis/Stratification AM by sex.Rmd
    Deleted:    analysis/Stratification by T cell condition and DE genes.Rmd
    Deleted:    analysis/license.Rmd
    Deleted:    data/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>##load merged file 
seuratM &lt;- readRDS(&quot;~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/seuratM_all_metadata_withAMvsHealthy_ventricle_sex_TC con_27.10.2024.rds&quot;)

table(seuratM$dataset)</code></pre>
<pre><code>
     312581_05-5_20230308_Hu_nucseq_SG36_EMB_GEM      312581_06-6_20230308_Hu_nucseq_SG37_EMB_GEM 
                                            4323                                             1123 
   353921_28-28_20240529_Hu_nucseq_Immpath_EMB18    353921_29-29_20240529_Hu_nucseq_Immpath_EMB24 
                                             208                                              105 
   353921_30-30_20240529_Hu_nucseq_Immpath_EMB25    353921_31-31_20240529_Hu_nucseq_Immpath_EMB27 
                                             227                                              579 
   353921_32-32_20240529_Hu_nucseq_Immpath_EMB29    353921_33-33_20240529_Hu_nucseq_Immpath_EMB30 
                                             335                                              730 
   353921_34-34_20240529_Hu_nucseq_Immpath_EMB35    353921_35-35_20240529_Hu_nucseq_Immpath_EMB37 
                                            1217                                              604 
359861_09-11_20240723_Hu_nucseq_Ber_Pat5_EM979_1 359861_10-13_20240723_Hu_nucseq_Ber_Pat5_EM979_2 
                                             771                                             1476 
 359861_11-1_20240730_Hu_nucseq_Ber_Pat1_EMB1537  359861_12-2_20240730_Hu_nucseq_Ber_Pat2_EMB1041 
                                             316                                              608 
    359861_16-8_20240730_Hu_nucseq_Immpath_EMB47     359861_17-9_20240730_Hu_nucseq_Immpath_EMB48 
                                             513                                              426 
     o27533_1_12-12_20220203_Hu_nucseq_EMB31_GEM        o27936_1_7-7_20220309_Hu_nucseq_EMB32_GEM 
                                             236                                             1192 
 o28576_1_01-1_20220525_Hu_nucseq_Graz_1_EMB_GEM  o28576_1_02-2_20220525_Hu_nucseq_Graz_2_EMB_GEM 
                                            2740                                             1684 
 o28576_1_03-3_20220525_Hu_nucseq_Graz_3_EMB_GEM  o28576_1_04-4_20220525_Hu_nucseq_Graz_4_EMB_GEM 
                                            2396                                              545 
 o28576_1_05-5_20220525_Hu_nucseq_Graz_5_EMB_GEM  o28576_1_06-6_20220525_Hu_nucseq_Graz_6_EMB_GEM 
                                             781                                              491 
 o28576_1_07-7_20220525_Hu_nucseq_Graz_7_EMB_GEM   o28576_1_08-8_20220525_Hu_nucseq_Graz_8_HH_GEM 
                                             653                                             3921 
o28576_1_12-12_20220525_Hu_nucseq_Graz_12_HH_GEM      o28576_1_13-13_20220525_Hu_nucseq_EMB32_GEM 
                                            3818                                             1428 
    o292731_1-1_20220818_Hu_nucseq_Graz_9_HH_GEM    o292731_2-2_20220818_Hu_nucseq_Graz_13_HH_GEM 
                                            4908                                             9882 
    o292731_3-3_20220818_Hu_nucseq_SG_33_EMB_GEM     o292731_4-4_20220818_Hu_nucseq_SG_34_EMB_GEM 
                                            6286                                              620 
    o292731_5-5_20220818_Hu_nucseq_SG_35_EMB_GEM   o294781_01-1_20220912_Hu_nucseq_Graz_21_HH_GEM 
                                            2363                                             1442 
  o294781_02-2_20220912_Hu_nucseq_Graz_22_HH_GEM   o294781_03-3_20220912_Hu_nucseq_Graz_23_HH_GEM 
                                            1998                                              841 
  o294781_04-4_20220912_Hu_nucseq_Graz_24_HH_GEM  o294781_05-5_20220912_Hu_nucseq_Graz_14_EMB_GEM 
                                            1480                                             1268 
 o294781_06-6_20220912_Hu_nucseq_Graz_15_EMB_GEM  o294781_09-9_20220912_Hu_nucseq_Graz_18_EMB_GEM 
                                            4439                                             2280 
o294781_10-10_20220912_Hu_nucseq_Graz_19_EMB_GEM o294781_11-11_20220912_Hu_nucseq_Graz_20_EMB_GEM 
                                             111                                             2706 </code></pre>
<pre class="r"><code>table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11    12    13    14 
17699 13639 12754  8222  5475  4836  3723  2365  2156   905   882   694   380   249    91 </code></pre>
<pre class="r"><code>table(seuratM$patient, seuratM$sex)</code></pre>
<pre><code>                
                 female male unknown
  AM 01            6286    0       0
  AM 02               0 4323       0
  AM 03               0 1123       0
  AM 04             208    0       0
  AM 05               0  579       0
  AM 06             335    0       0
  AM 07            1217    0       0
  AM 08               0  604       0
  AM 09             513    0       0
  AM 10             426    0       0
  AM 11               0  236       0
  AM 12               0 1192       0
  AM 13               0 2740       0
  AM 14            1684    0       0
  AM 15               0 4439       0
  AM 16            2247    0       0
  AM 17               0  316       0
  AM 18             608    0       0
  CMP 01              0  105       0
  CMP 02            227    0       0
  CMP 03              0  730       0
  CMP 04           1428    0       0
  CMP 05              0  620       0
  CMP 06              0  653       0
  CMP 07              0 1268       0
  CMP 08              0 2280       0
  CMP 09              0 2396       0
  CMP 10              0  545       0
  CMP 11              0  781       0
  CMP 12              0  491       0
  CMP 13            111    0       0
  Healthy 01          0    0    3921
  Healthy 02          0    0    3818
  Healthy 03          0    0    4908
  Healthy 04          0    0    9882
  Healthy 05          0    0    1442
  Healthy 06          0    0    1998
  Healthy 07          0    0     841
  Healthy 08          0    0    1480
  Sarcoidosis 01      0 2363       0
  Sarcoidosis 02      0 2706       0</code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
74070 </code></pre>
<pre class="r"><code>seuratM$clusterName &lt;- &quot;clusterName&quot;
Idents(seuratM) &lt;- seuratM$clusterName
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;0&quot; )] &lt;- &quot;Fb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;1&quot; )] &lt;- &quot;BEC1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;2&quot; )] &lt;- &quot;PeriFb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;3&quot; )] &lt;- &quot;Cardiomyocyte&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;4&quot; )] &lt;- &quot;MCP1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;5&quot; )] &lt;- &quot;TC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;6&quot; )] &lt;- &quot;MCP2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;7&quot; )] &lt;- &quot;BEC2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;8&quot; )] &lt;- &quot;VSMC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;9&quot; )] &lt;- &quot;NC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;10&quot; )] &lt;- &quot;INT1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;11&quot; )] &lt;- &quot;Fb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;12&quot; )] &lt;- &quot;Fb3&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;13&quot; )] &lt;- &quot;PeriFb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;14&quot; )] &lt;- &quot;AdipoC&quot;


colclusterName &lt;- c(&quot;#D53E4F&quot;, &quot;#f4a582&quot;, &quot;#FEE08B&quot;,&quot;#feb24c&quot;,&quot;#67001f&quot;, &quot;#01665e&quot;,&quot;#66C2A5&quot;,&quot;#c7eae5&quot;,&quot;#BEAED4&quot;, &quot;#355C7D&quot;,&quot;#3288BD&quot;,&quot;#8c510a&quot; ,&quot;#fde0dd&quot;,&quot;#B45B5C&quot;,&quot;#dd1c77&quot;)

                    
names(colclusterName) &lt;- c(&quot;Cardiomyocyte&quot;,&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;,&quot;NC&quot;, &quot;BEC1&quot;,&quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;,&quot;MCP2&quot;,&quot;TC&quot;)

colpat_sub&lt;- c(&quot;#dfc27d&quot;,&quot;#BE3144&quot;,&quot;#355C7D&quot;, &quot;#779d8d&quot;)

seuratM$clusterName &lt;- factor(seuratM$clusterName, levels=c(&quot;Cardiomyocyte&quot;, &quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;INT1&quot;,&quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;, &quot;NC&quot;, &quot;BEC1&quot;, &quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;, &quot;MCP2&quot;, &quot;TC&quot;))


Idents(seuratM) &lt;- seuratM$clusterName
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE, cols = colclusterName)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$patient

table(seuratM$patient, seuratM$clusterName)</code></pre>
<pre><code>                
                 Cardiomyocyte  Fb1  Fb2  Fb3 INT1 PeriFb1 PeriFb2 VSMC   NC BEC1 BEC2 AdipoC MCP1
  AM 01                    206   91  303    0    0       7     106   20   20  168   35      6   17
  AM 02                    801 1484   15   31   70     645       1  222   35  726   74      3  131
  AM 03                    109  397    0    4    2     158       1   32    4  156  148      0   91
  AM 04                     76   30    1    2    5      25       0    3    1   42    2      0   14
  AM 05                     64  155    0    7   10     158       0    3    4   92   11      0   58
  AM 06                     62   72    0    2    5      12       0    8    6   51   63      1   30
  AM 07                    233  488   28    3    4     182       0   19   12   37  132      1   66
  AM 08                     84  196    0   17   21     120       0   17   12   58   11      0   51
  AM 09                    192  132    0    4    3      51       0    4    8   38   10      0   58
  AM 10                     62  183    1    5    5      48       0    1    4   65   18      1   29
  AM 11                     47   41    0    1    0      36       0   20    0   48   29      0   14
  AM 12                    178  350    2    1    5     187       0   40   22  204   51      0  122
  AM 13                    208  758    3   19   21     489       1  197   59  315  146      0  294
  AM 14                    109  641    0   18   33     259       0   44   30  318  110      1   82
  AM 15                    116 1062   29   28  111    1356       0  141   80 1034    2      0  313
  AM 16                     26   20  228    0    1       4     124   12    7  374    1      0    2
  AM 17                     10   57    2    1    3      22       0   14    4   55    3      0   58
  AM 18                     30  111    2    2    6      83       2   15    3  213    0      0   48
  CMP 01                     9   40    0    0    0       6       0    2    1   11   15      0   14
  CMP 02                    18   95    0    8    4      30       0    2    4   28   16      0   17
  CMP 03                   107  266    0   12   16      77       0   16   10   74    7      3   99
  CMP 04                    40  309    1    7   25     247       0   62   28  547   33      5   65
  CMP 05                   179  112    3    3    4     129       0   11    5   86   13      0   49
  CMP 06                    88  236    0    0    4      64       0   59    0  123    4     54   15
  CMP 07                    50  314    1    2   11     363       0   76   10  314    3      0   86
  CMP 08                   332  613    0    3   19     386       0  128   28  482  166      0   86
  CMP 09                   240  824    7   17   35     425       0   57   35  391  208      3  118
  CMP 10                    45  140    2    1    3      64       0   20    7  120   62      0   48
  CMP 11                    39  260    2    1   10     116       0   37   27  214   27      0   31
  CMP 12                    12  153    0    0    4      53       0   15    2  115   41      0   62
  CMP 13                     1   17    0    0    0      28       0    4    1   27    3      0   15
  Healthy 01               714 1135    3   15   31     581       0  153   64  752   28      4  362
  Healthy 02               115 1111    0   17   75    1000       1   71   22 1130   18      0  216
  Healthy 03               656 1496   40   14   16     815       2   94   33  594  433      5  647
  Healthy 04              2224 2191    3   91  210    1466       2  239  207 2039  212      3  910
  Healthy 05                58   93    0    4   12     766       0   37   15  249    0      0  185
  Healthy 06               159  308    1    6   14     680       2   44   30  264    1      0  444
  Healthy 07                54   65    0    3    6     321       1   17    7  207   13      0  131
  Healthy 08               118  259    5    4   12     604       0   60   10  271   10      1  101
  Sarcoidosis 01           305  758   10   24   40     328       5   33   28  539   30      0  172
  Sarcoidosis 02            46  636    2    3   26     363       1  107   20 1068  176      0  124
                
                 MCP2   TC
  AM 01          2483 2824
  AM 02            14   71
  AM 03             2   19
  AM 04             4    3
  AM 05             7   10
  AM 06             1   22
  AM 07             7    5
  AM 08             2   15
  AM 09             0   13
  AM 10             0    4
  AM 11             0    0
  AM 12             7   23
  AM 13            18  212
  AM 14             4   35
  AM 15            23  144
  AM 16           982  466
  AM 17             5   82
  AM 18             9   84
  CMP 01            0    7
  CMP 02            1    4
  CMP 03           16   27
  CMP 04            3   56
  CMP 05            2   24
  CMP 06            1    5
  CMP 07            6   32
  CMP 08            8   29
  CMP 09            3   33
  CMP 10            7   26
  CMP 11            0   17
  CMP 12            2   32
  CMP 13            2   13
  Healthy 01        6   73
  Healthy 02        8   34
  Healthy 03       21   42
  Healthy 04        7   78
  Healthy 05        1   22
  Healthy 06        6   39
  Healthy 07        1   15
  Healthy 08        6   19
  Sarcoidosis 01   35   56
  Sarcoidosis 02   13  121</code></pre>
<pre class="r"><code>DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DEGenesall_patient &lt;- FindAllMarkers(seuratM, only.pos = FALSE, logfc.threshold = 0.2) %&gt;%
  dplyr::filter(p_val_adj &lt; 0.01)

#write.table(DEGenes_patient, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenes_patient positive markers_05.11.2024&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

#View(DEGenesall_patient)

##adjust table
DEGenesall_patient_adj &lt;- DEGenesall_patient %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))</code></pre>
<pre class="r"><code>## Heatmap patient clustering with common marker genes

Idents(seuratM) &lt;- seuratM$patient
seuratM$patient &lt;- factor(seuratM$patient, levels = c(&quot;Healthy 01&quot;,&quot;Healthy 02&quot;,&quot;Healthy 03&quot;,&quot;Healthy 04&quot;, &quot;Healthy 05&quot;, &quot;Healthy 06&quot;, &quot;Healthy 07&quot;, &quot;Healthy 08&quot;,&quot;AM 01&quot;, &quot;AM 02&quot;,&quot;AM 03&quot;, &quot;AM 04&quot;, &quot;AM 05&quot;,&quot;AM 06&quot;,&quot;AM 07&quot;, &quot;AM 08&quot;, &quot;AM 09&quot;, &quot;AM 10&quot;, &quot;AM 11&quot;, &quot;AM 12&quot;, &quot;AM 13&quot;, &quot;AM 14&quot;, &quot;AM 15&quot;, &quot;AM 16&quot;, &quot;AM 17&quot;, &quot;AM 18&quot;, &quot;CMP 01&quot;, &quot;CMP 02&quot;, &quot;CMP 03&quot;, &quot;CMP 04&quot;, &quot;CMP 05&quot;, &quot;CMP 06&quot;, &quot;CMP 07&quot;, &quot;CMP 08&quot;, &quot;CMP 09&quot;, &quot;CMP 10&quot;, &quot;CMP 11&quot;, &quot;CMP 12&quot;, &quot;CMP 13&quot;, &quot;Sarcoidosis 01&quot;, &quot;Sarcoidosis 02&quot;))

genes &lt;- data.frame(gene=rownames(seuratM)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID=rev(c(&quot;MYBPC3&quot;, &quot;RYR2&quot;, &quot;NEBL&quot;, &quot;TNNT2&quot;, &quot;CMYA5&quot;, &quot;COL6A3&quot;, &quot;DCN&quot;,  &quot;FBN1&quot;, &quot;C7&quot;, &quot;PDGFRA&quot;, &quot;CDH19&quot;, &quot;PDGFRB&quot;, &quot;BMP4&quot;, &quot;GREM1&quot;, &quot;ITGA7&quot;,&quot;RGS5&quot;, &quot;NOTCH3&quot;, &quot;MYH11&quot;, &quot;ACTA2&quot;,&quot;PECAM1&quot;, &quot;VWF&quot;, &quot;EGFL7&quot;, &quot;POSTN&quot;, &quot;ITGA10&quot;, &quot;CDH11&quot;,&quot;CCL21&quot;, &quot;FLT4&quot;, &quot;NRXN1&quot;, &quot;ANK3&quot;, &quot;PTPRZ1&quot;, &quot;PLIN1&quot;, &quot;GPAM&quot;, &quot;CD163&quot;, &quot;LYVE1&quot;, &quot;MRC1&quot;, &quot;SIGLEC1&quot;, &quot;STAB1&quot;, &quot;CSF1R&quot;,&quot;MERTK&quot;, &quot;HLA-DQB1&quot;,&quot;CCR2&quot;, &quot;CD79B&quot;,&quot;JCHAIN&quot;, &quot;IL7R&quot;, &quot;CD3E&quot;, &quot;PTPRC&quot;, &quot;CD2&quot;))) %&gt;% left_join(., genes, by=&quot;geneID&quot;) %&gt;% filter(gene != &quot;ENSG00000232995.RGS5&quot;) 


avgHeatmap &lt;- function(seuratM, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL, cc=FALSE,
                       cr=FALSE, condCol=FALSE) {
  
  selGenes &lt;- selGenes$gene
  

  clusterAssigned &lt;- as.data.frame(Idents(seuratM)) %&gt;%
    dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seuratM)
  

  genes &lt;- data.frame(gene=rownames(seuratM)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes)

 
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% 
    dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(., clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;%
    dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;, &quot;&quot;, rownames(logNormExpresMa))
  
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]
  
  logNormExpresMa &lt;- logNormExpresMa[selGenes,]
  if (is.null(ordVec)) {
    ordVec &lt;- levels(seuratM$patient)
  }
  logNormExpresMa &lt;- logNormExpresMa[, ordVec, drop=FALSE]
  
  pheatmap(logNormExpresMa, scale = &quot;row&quot;, treeheight_row = 0, cluster_rows = cr, 
           cluster_cols = cc, color = colorRampPalette(c(&quot;#2166AC&quot;, &quot;#F7F7F7&quot;, &quot;#B2182B&quot;))(50),
           annotation_col = NULL, cellwidth = 15, cellheight = 10,
           annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}

genesPlot &lt;- data.frame(gene=c(&quot;MYBPC3&quot;, &quot;RYR2&quot;, &quot;NEBL&quot;, &quot;TNNT2&quot;, &quot;CMYA5&quot;, &quot;COL6A3&quot;, &quot;DCN&quot;,  &quot;FBN1&quot;, &quot;C7&quot;, &quot;PDGFRA&quot;, &quot;CDH19&quot;, &quot;PDGFRB&quot;, &quot;BMP4&quot;, &quot;GREM1&quot;, &quot;ITGA7&quot;,&quot;RGS5&quot;, &quot;NOTCH3&quot;, &quot;MYH11&quot;, &quot;ACTA2&quot;,&quot;PECAM1&quot;, &quot;VWF&quot;, &quot;EGFL7&quot;, &quot;POSTN&quot;, &quot;ITGA10&quot;, &quot;CDH11&quot;,&quot;CCL21&quot;, &quot;FLT4&quot;, &quot;NRXN1&quot;, &quot;ANK3&quot;, &quot;PTPRZ1&quot;, &quot;PLIN1&quot;, &quot;GPAM&quot;, &quot;CD163&quot;, &quot;LYVE1&quot;, &quot;MRC1&quot;, &quot;SIGLEC1&quot;, &quot;STAB1&quot;, &quot;CSF1R&quot;,&quot;MERTK&quot;, &quot;HLA-DQB1&quot;, &quot;CCR2&quot;, &quot;CD79B&quot;,&quot;JCHAIN&quot;, &quot;IL7R&quot;,&quot;CD3E&quot;, &quot;PTPRC&quot;, &quot;CD2&quot;))

levels(seuratM)</code></pre>
<pre><code> [1] &quot;AM 02&quot;          &quot;AM 03&quot;          &quot;AM 04&quot;          &quot;AM 05&quot;          &quot;AM 06&quot;         
 [6] &quot;AM 07&quot;          &quot;AM 08&quot;          &quot;AM 09&quot;          &quot;AM 10&quot;          &quot;AM 11&quot;         
[11] &quot;AM 12&quot;          &quot;AM 13&quot;          &quot;AM 14&quot;          &quot;AM 01&quot;          &quot;AM 15&quot;         
[16] &quot;AM 16&quot;          &quot;AM 17&quot;          &quot;AM 18&quot;          &quot;CMP 01&quot;         &quot;CMP 02&quot;        
[21] &quot;CMP 03&quot;         &quot;CMP 09&quot;         &quot;CMP 10&quot;         &quot;CMP 11&quot;         &quot;CMP 12&quot;        
[26] &quot;CMP 06&quot;         &quot;CMP 04&quot;         &quot;CMP 05&quot;         &quot;CMP 07&quot;         &quot;CMP 08&quot;        
[31] &quot;CMP 13&quot;         &quot;Healthy 01&quot;     &quot;Healthy 02&quot;     &quot;Healthy 03&quot;     &quot;Healthy 04&quot;    
[36] &quot;Healthy 05&quot;     &quot;Healthy 06&quot;     &quot;Healthy 07&quot;     &quot;Healthy 08&quot;     &quot;Sarcoidosis 01&quot;
[41] &quot;Sarcoidosis 02&quot;</code></pre>
<pre class="r"><code>avgHeatmap(seuratM, selGenes = genesPlot, colVecIdent = colVec)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/Heatmap%20for%20patient%20clustering%20with%20marker%20genes-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Heatmap for patients with DE genes

Idents(seuratM) &lt;- seuratM$patient
seuratM$patient &lt;- factor(seuratM$patient, levels = c(&quot;Healthy 01&quot;,&quot;Healthy 02&quot;,&quot;Healthy 03&quot;,&quot;Healthy 04&quot;, &quot;Healthy 05&quot;, &quot;Healthy 06&quot;, &quot;Healthy 07&quot;, &quot;Healthy 08&quot;,&quot;AM 01&quot;, &quot;AM 02&quot;,&quot;AM 03&quot;, &quot;AM 04&quot;, &quot;AM 05&quot;,&quot;AM 06&quot;,&quot;AM 07&quot;, &quot;AM 08&quot;, &quot;AM 09&quot;, &quot;AM 10&quot;, &quot;AM 11&quot;, &quot;AM 12&quot;, &quot;AM 13&quot;, &quot;AM 14&quot;, &quot;AM 15&quot;, &quot;AM 16&quot;, &quot;AM 17&quot;, &quot;AM 18&quot;, &quot;CMP 01&quot;, &quot;CMP 02&quot;, &quot;CMP 03&quot;, &quot;CMP 04&quot;, &quot;CMP 05&quot;, &quot;CMP 06&quot;, &quot;CMP 07&quot;, &quot;CMP 08&quot;, &quot;CMP 09&quot;, &quot;CMP 10&quot;, &quot;CMP 11&quot;, &quot;CMP 12&quot;, &quot;CMP 13&quot;, &quot;Sarcoidosis 01&quot;, &quot;Sarcoidosis 02&quot;))

genes &lt;- data.frame(gene=rownames(seuratM)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID=rev(c(&quot;XIST&quot;, &quot;RYR2&quot;,&quot;TNNT2&quot;,&quot;MALAT1&quot;, &quot;COL8A1&quot;, &quot;LAMA2&quot;, &quot;SOX4&quot;, &quot;NPPA&quot;, &quot;GREM1&quot;, &quot;GREM2&quot;, &quot;BMP4&quot;, &quot;BMPER&quot;, &quot;TGFBR3&quot;, &quot;ADAMTS5&quot;, &quot;HIF3A&quot;,&quot;THBS1&quot;, &quot;FOXO3&quot;, &quot;EGR1&quot;, &quot;HGF&quot;, &quot;SERPINE1&quot;, &quot;TIMP1&quot;, &quot;FBN1&quot;, &quot;IRS2&quot;, &quot;GCGR&quot;, &quot;SLC6A1&quot;, &quot;POU3F3&quot;, &quot;ZBTB16&quot;, &quot;ACTA2&quot;,&quot;VWF&quot;, &quot;POSTN&quot;, &quot;CCL21&quot;, &quot;LYVE1&quot;, &quot;TNFAIP3&quot;,&quot;CXCL9&quot;, &quot;HLA-B&quot;, &quot;CCR2&quot;,&quot;JAK2&quot;, &quot;STAT1&quot;, &quot;TLR4&quot;, &quot;IRF1&quot;, &quot;GBP5&quot;, &quot;IGHG3&quot;, &quot;IL7R&quot;, &quot;CD3E&quot;, &quot;PTPRC&quot;, &quot;CD2&quot;, &quot;CD74&quot;, &quot;IL17RA&quot;))) %&gt;% left_join(., genes, by=&quot;geneID&quot;) %&gt;% filter(gene != &quot;ENSG00000232995.RGS5&quot;) 


avgHeatmap &lt;- function(seuratM, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=TRUE, gapVecR=NULL, gapVecC=NULL, cc=FALSE,
                       cr=FALSE, condCol=FALSE) {
  
  selGenes &lt;- selGenes$gene
  

  clusterAssigned &lt;- as.data.frame(Idents(seuratM)) %&gt;%
    dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seuratM)
  

  genes &lt;- data.frame(gene=rownames(seuratM)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes)

 
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% 
    dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(., clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;%
    dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;, &quot;&quot;, rownames(logNormExpresMa))
  
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]
  
  logNormExpresMa &lt;- logNormExpresMa[selGenes,]
  if (is.null(ordVec)) {
    ordVec &lt;- levels(seuratM$patient)
  }
  logNormExpresMa &lt;- logNormExpresMa[, ordVec, drop=FALSE]
  
  pheatmap(logNormExpresMa, scale = &quot;row&quot;, treeheight_row = 0, cluster_rows = cr, 
           cluster_cols = cc, color = colorRampPalette(c(&quot;#2166AC&quot;, &quot;#F7F7F7&quot;, &quot;#B2182B&quot;))(50),
           annotation_col = NULL, cellwidth = 15, cellheight = 10,
           annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}

genesPlot &lt;- data.frame(gene=c(&quot;XIST&quot;, &quot;RYR2&quot;,&quot;TNNT2&quot;,&quot;MALAT1&quot;, &quot;COL8A1&quot;, &quot;LAMA2&quot;, &quot;SOX4&quot;, &quot;NPPA&quot;,&quot;GREM1&quot;, &quot;GREM2&quot;, &quot;BMP4&quot;, &quot;BMPER&quot;, &quot;TGFBR3&quot;, &quot;ADAMTS5&quot;, &quot;HIF3A&quot;,&quot;FOXO3&quot;, &quot;EGR1&quot;, &quot;HGF&quot;, &quot;SERPINE1&quot;, &quot;TIMP1&quot;, &quot;IRS2&quot;, &quot;GCGR&quot;, &quot;SLC6A1&quot;, &quot;POU3F3&quot;, &quot;ZBTB16&quot;, &quot;ACTA2&quot;,&quot;VWF&quot;, &quot;POSTN&quot;, &quot;CCL21&quot;, &quot;LYVE1&quot;, &quot;TNFAIP3&quot;,&quot;CXCL9&quot;, &quot;HLA-B&quot;, &quot;CCR2&quot;,&quot;JAK2&quot;, &quot;STAT1&quot;, &quot;TLR4&quot;, &quot;IRF1&quot;, &quot;GBP5&quot;, &quot;IGHG3&quot;, &quot;IL7R&quot;, &quot;CD3E&quot;, &quot;PTPRC&quot;, &quot;CD2&quot;, &quot;CD74&quot;, &quot;IL17RA&quot;))

levels(seuratM)</code></pre>
<pre><code> [1] &quot;Healthy 01&quot;     &quot;Healthy 02&quot;     &quot;Healthy 03&quot;     &quot;Healthy 04&quot;     &quot;Healthy 05&quot;    
 [6] &quot;Healthy 06&quot;     &quot;Healthy 07&quot;     &quot;Healthy 08&quot;     &quot;AM 01&quot;          &quot;AM 02&quot;         
[11] &quot;AM 03&quot;          &quot;AM 04&quot;          &quot;AM 05&quot;          &quot;AM 06&quot;          &quot;AM 07&quot;         
[16] &quot;AM 08&quot;          &quot;AM 09&quot;          &quot;AM 10&quot;          &quot;AM 11&quot;          &quot;AM 12&quot;         
[21] &quot;AM 13&quot;          &quot;AM 14&quot;          &quot;AM 15&quot;          &quot;AM 16&quot;          &quot;AM 17&quot;         
[26] &quot;AM 18&quot;          &quot;CMP 01&quot;         &quot;CMP 02&quot;         &quot;CMP 03&quot;         &quot;CMP 04&quot;        
[31] &quot;CMP 05&quot;         &quot;CMP 06&quot;         &quot;CMP 07&quot;         &quot;CMP 08&quot;         &quot;CMP 09&quot;        
[36] &quot;CMP 10&quot;         &quot;CMP 11&quot;         &quot;CMP 12&quot;         &quot;CMP 13&quot;         &quot;Sarcoidosis 01&quot;
[41] &quot;Sarcoidosis 02&quot;</code></pre>
<pre class="r"><code>avgHeatmap(seuratM, selGenes = genesPlot, colVecIdent = colVec)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># AM 16-18 + CMP 13 = added = NA

# &quot;XIST&quot;, &quot;RYR2&quot;,&quot;TNNT2&quot;,&quot;MALAT1&quot;, &quot;COL8A1&quot;, &quot;LAMA2&quot;, &quot;SOX4&quot;, &quot;NPPA&quot;, &quot;BMP4&quot;, &quot;BMPER&quot;, &quot;TGFBR3&quot;, &quot;ADAMTS5&quot;, &quot;HIF3A&quot;,&quot;THBS1&quot;, &quot;FOXO3&quot;, &quot;EGR1&quot;, &quot;HGF&quot;, &quot;SERPINE1&quot;, &quot;TIMP1&quot;, &quot;FBN1&quot;, &quot;IRS2&quot;, &quot;GCGR&quot;, &quot;SLC6A1&quot;, &quot;POU3F3&quot;, &quot;ZBTB16&quot;, &quot;ACTA2&quot;,&quot;VWF&quot;, &quot;POSTN&quot;, &quot;CCL21&quot;, &quot;LYVE1&quot;, &quot;TNFAIP3&quot;,&quot;CXCL9&quot;,&quot;CXCL10&quot;, &quot;HLA-B&quot;, &quot;CCR2&quot;,&quot;JAK2&quot;, &quot;STAT1&quot;, &quot;TLR4&quot;, &quot;IRF1&quot;, &quot;GBP5&quot;, &quot;IGHM&quot;, &quot;IGHG3&quot;, &quot;IL7R&quot;, &quot;CD3E&quot;, &quot;PTPRC&quot;, &quot;CD2&quot;, &quot;CD74&quot;, &quot;IL17RA&quot;</code></pre>
<pre><code></code></pre>
<pre><code>







``` r
## Volcano Plot with pos. and neg. DE Genes for all conditions: 

##adjust table
DEGenesall_patient_adj &lt;- DEGenesall_patient %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

DEGenesallAM1 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 01&quot;)

res_AM1 &lt;- as.data.frame(DEGenesallAM1)

EnhancedVolcano(res_AM1, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM1$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 01&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2001-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with pos. and neg. DE Genes for Patient AM 02 

DEGenesallAM2 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 02&quot;)

res_AM2 &lt;- as.data.frame(DEGenesallAM2)

EnhancedVolcano(res_AM2, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM2$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 02&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2002-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## single numbers as gene names relate to pseudogenes or RNA</code></pre>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 03 


DEGenesallAM3 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 03&quot;)

res_AM3 &lt;- as.data.frame(DEGenesallAM3)

EnhancedVolcano(res_AM3, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM3$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 03&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2003-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 04 

DEGenesallAM4 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 04&quot;)

res_AM4 &lt;- as.data.frame(DEGenesallAM4)

EnhancedVolcano(res_AM4, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM4$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 04&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2004-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 05 


DEGenesallAM5 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 05&quot;)

res_AM5 &lt;- as.data.frame(DEGenesallAM5)

EnhancedVolcano(res_AM5, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM5$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 05&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2005-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 06

DEGenesallAM6 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 06&quot;)

res_AM6 &lt;- as.data.frame(DEGenesallAM6)

EnhancedVolcano(res_AM6, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM6$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 06&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2006-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 07
DEGenesallAM7 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 07&quot;)

res_AM7 &lt;- as.data.frame(DEGenesallAM7)

EnhancedVolcano(res_AM7, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM7$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 07&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2007-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 08
DEGenesallAM8 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 08&quot;)

res_AM8 &lt;- as.data.frame(DEGenesallAM8)

EnhancedVolcano(res_AM8, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM8$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 08&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2008-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 09
DEGenesallAM9 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 09&quot;)

res_AM9 &lt;- as.data.frame(DEGenesallAM9)

EnhancedVolcano(res_AM9, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM9$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 09&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2009-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 10
DEGenesallAM10 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 10&quot;)

res_AM10 &lt;- as.data.frame(DEGenesallAM10)

EnhancedVolcano(res_AM10, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM10$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 10&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2010-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 11
DEGenesallAM11 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 11&quot;)

res_AM11 &lt;- as.data.frame(DEGenesallAM11)

EnhancedVolcano(res_AM11, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM11$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 11&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2011-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 12

DEGenesallAM12 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 12&quot;)

# pos. and neg. DE genes 
res_AM12 &lt;- as.data.frame(DEGenesallAM12)

EnhancedVolcano(res_AM12, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM12$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 12&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2012-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 13
DEGenesallAM13 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 13&quot;)

# pos. and neg. DE genes
res_AM13 &lt;- as.data.frame(DEGenesallAM13)

EnhancedVolcano(res_AM13, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM13$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 13&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2013-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plots with DE Genes for Patient AM 14 

DEGenesallAM14 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 14&quot;)

# pos. and neg. DE genes
res_AM14 &lt;- as.data.frame(DEGenesallAM14)

EnhancedVolcano(res_AM14, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM14$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 14&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2014-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## single numbers as gene names refer to pseudogenes or RNA</code></pre>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 15
DEGenesallAM15 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 15&quot;)

# pos. and neg. DE genes
res_AM15 &lt;- as.data.frame(DEGenesallAM15)

EnhancedVolcano(res_AM15, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM15$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 15&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2015-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 16 

DEGenesallAM16 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 16&quot;)

# pos. and neg. DE genes
res_AM16 &lt;- as.data.frame(DEGenesallAM16)

EnhancedVolcano(res_AM16, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM16$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 16&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2016-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 17 

DEGenesallAM17 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 17&quot;)

res_AM17 &lt;- as.data.frame(DEGenesallAM17)

EnhancedVolcano(res_AM17, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM17$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 17&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2017-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient AM 18 

DEGenesallAM18 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;AM 18&quot;)

res_AM18 &lt;- as.data.frame(DEGenesallAM18)

EnhancedVolcano(res_AM18, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_AM18$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient AM 18&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20AM%2018-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 01

DEGenesallCMP1 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 01&quot;)

# pos. and neg. DE genes for AM
res_CMP1 &lt;- as.data.frame(DEGenesallCMP1)

EnhancedVolcano(res_CMP1, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP1$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 01&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP1-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 02

DEGenesallCMP2 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 02&quot;)

res_CMP2 &lt;- as.data.frame(DEGenesallCMP2)

EnhancedVolcano(res_CMP2, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP2$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 02&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP2-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 03


DEGenesallCMP3 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 03&quot;)

res_CMP3 &lt;- as.data.frame(DEGenesallCMP3)

EnhancedVolcano(res_CMP3, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP3$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 03&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP3-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 04


DEGenesallCMP4 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 04&quot;)


res_CMP4 &lt;- as.data.frame(DEGenesallCMP4)

EnhancedVolcano(res_CMP4, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP4$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 04&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP4-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 05


DEGenesallCMP5 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 05&quot;)

res_CMP5 &lt;- as.data.frame(DEGenesallCMP5)

EnhancedVolcano(res_CMP5, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP5$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 05&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP5-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 06


DEGenesallCMP6 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 06&quot;)

res_CMP6 &lt;- as.data.frame(DEGenesallCMP6)

EnhancedVolcano(res_CMP6, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP6$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 06&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP6-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 07


DEGenesallCMP7 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 07&quot;)

res_CMP7 &lt;- as.data.frame(DEGenesallCMP7)

EnhancedVolcano(res_CMP7, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP7$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 07&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP7-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 08


DEGenesallCMP8 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 08&quot;)

res_CMP8 &lt;- as.data.frame(DEGenesallCMP8)

EnhancedVolcano(res_CMP8, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP8$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 08&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP8-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 09


DEGenesallCMP9 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 09&quot;)

res_CMP9 &lt;- as.data.frame(DEGenesallCMP9)

EnhancedVolcano(res_CMP9, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP9$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 09&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP9-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 10


DEGenesallCMP10 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 10&quot;)

res_CMP10 &lt;- as.data.frame(DEGenesallCMP10)

EnhancedVolcano(res_CMP10, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP10$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 10&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP10-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 11


DEGenesallCMP11 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 11&quot;)

res_CMP11 &lt;- as.data.frame(DEGenesallCMP11)

EnhancedVolcano(res_CMP11, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP11$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 11&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP11-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 12


DEGenesallCMP12 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 12&quot;)

res_CMP12 &lt;- as.data.frame(DEGenesallCMP12)

EnhancedVolcano(res_CMP12, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP12$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 12&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP12-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient CMP 13


DEGenesallCMP13 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;CMP 13&quot;)

res_CMP13 &lt;- as.data.frame(DEGenesallCMP13)

EnhancedVolcano(res_CMP13, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_CMP13$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient CMP 13&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20CMP13-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient Sarcoidosis 01

DEGenesallSarco1 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;Sarcoidosis 01&quot;)

res_Sarco1 &lt;- as.data.frame(DEGenesallSarco1)

EnhancedVolcano(res_Sarco1, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Sarco1$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Sarcoidosis 01&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.5,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20Sarcoidosis%2001-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient Sarcoidosis 02

DEGenesallSarco2 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;Sarcoidosis 02&quot;)

res_Sarco2 &lt;- as.data.frame(DEGenesallSarco2)

EnhancedVolcano(res_Sarco2, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Sarco2$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Sarcoidosis 02&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.5,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20Sarcoidosis%2002-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient Healthy 01

DEGenesallHealthy1 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;Healthy 01&quot;)

res_Healthy1 &lt;- as.data.frame(DEGenesallHealthy1)

EnhancedVolcano(res_Healthy1, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Healthy1$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Healthy 01&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20Healthy%2001-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient Healthy 02

DEGenesallHealthy2 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;Healthy 02&quot;)

res_Healthy2 &lt;- as.data.frame(DEGenesallHealthy2)

EnhancedVolcano(res_Healthy2, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Healthy2$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Healthy 02&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20Healthy%2002-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient Healthy 03

DEGenesallHealthy3 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;Healthy 03&quot;)

res_Healthy3 &lt;- as.data.frame(DEGenesallHealthy3)

EnhancedVolcano(res_Healthy3, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Healthy3$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient Healthy 03&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20Healthy%2003-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient Healthy 04

DEGenesallHealthy4 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;Healthy 04&quot;)

res_Healthy4 &lt;- as.data.frame(DEGenesallHealthy4)

EnhancedVolcano(res_Healthy4, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Healthy4$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient Healthy 04&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20Healthy%2004-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient Healthy 05

DEGenesallHealthy5 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;Healthy 05&quot;)

res_Healthy5 &lt;- as.data.frame(DEGenesallHealthy5)

EnhancedVolcano(res_Healthy5, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Healthy5$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient Healthy 05&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20Healthy%2005-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient Healthy 06

DEGenesallHealthy6 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;Healthy 06&quot;)

res_Healthy6 &lt;- as.data.frame(DEGenesallHealthy6)

EnhancedVolcano(res_Healthy6, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Healthy6$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient Healthy 06&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20Healthy%2006-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient Healthy 07

DEGenesallHealthy7 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;Healthy 07&quot;)

res_Healthy7 &lt;- as.data.frame(DEGenesallHealthy7)

EnhancedVolcano(res_Healthy7, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Healthy7$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient Healthy 07&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20Healthy%2007-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot with DE Genes for Patient Healthy 08
## &quot;Healthy 08&quot; shows inflammatory signature

DEGenesallHealthy8 &lt;- DEGenesall_patient_adj %&gt;% filter(cluster == &quot;Healthy 08&quot;)

res_Healthy8 &lt;- as.data.frame(DEGenesallHealthy8)

EnhancedVolcano(res_Healthy8, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Healthy8$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Patient Healthy 08&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/DE%20Genes%20Volcano%20Plot%20Healthy%2008-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Gene Ontology and GSEA for all Patients, positive markers

DEGenes_patient &lt;- FindAllMarkers(seuratM, only.pos = TRUE, logfc.threshold = 0.2) %&gt;%
  dplyr::filter(p_val_adj &lt; 0.01)

#head(DEGenes_patient)

#write.table(DEGenes_patient, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenes_patient positive markers_05.11.2024&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

##adjust table
DEGenes_patient &lt;- DEGenes_patient %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

#View(DEGenes_patient)</code></pre>
<pre class="r"><code># Patient Gene Ontology AM 01
DEGenesAM1 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 01&quot;)

egoAM1 &lt;- enrichGO(gene = unique(DEGenesAM1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM1 &lt;- setReadable(egoAM1, OrgDb = org.Hs.eg.db)

#View(egoAM1)

dotplot(egoAM1, showCategory=20, title = &quot;Patient AM 01 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-4-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM1, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 01 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-4-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM1 &lt;- enrichGO(gene = unique(DEGenesAM1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM1 &lt;- setReadable(egoAM1, OrgDb = org.Hs.eg.db)
dotplot(egoAM1, showCategory=20, title = &quot;Patient AM 01 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-4-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM1 &lt;- enrichGO(gene = unique(DEGenesAM1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM1 &lt;- setReadable(egoAM1, OrgDb = org.Hs.eg.db)
dotplot(egoAM1, showCategory=20, title = &quot;Patient AM 01 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-4-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Gene patterns and signature sorted by function found in Patient AM 01


# Genes associated with T cell differentiation: PTPRC/B2M/IRF1/CD74/NCKAP1L/IL7R/CD2/DOCK2/JAK3/HLA-DRA/BCL11B/HLA-DRB1/IKZF3/TCIRG1/RUNX3/SPN/PREX1/PTPN22/LILRB2/ZAP70/ITK/PIK3CD/LCK/FGL2/LAG3/LILRB4/IL2RG/CD3G/STAT4/KLRC1/VAV1/CD3E/RHOH/CAMK4/ADAM8/CD1D/ZNF683/SASH3/THEMIS/IL12RB1/LY9/IL4I1/CARD11/CD3D/CD8A/CD4/SLAMF6/FANCA/CTLA4/CYLD/LGALS9/TBX21/TESPA1/SPI1/PTPN2/BTN2A2/NFKBID/CCR7/CD27/TOX/LEF1/TNFRSF9/SYK/GATA3/RASGRP1/HLA-DOA/PRDM1/TMEM131L/CD28/TNFSF8/MAFB/SEMA4A/ZFP36L2/EOMES/IFNG/TGFB1/RUNX1/JAK1/BATF/CD83/STAT6/FANCD2/ACTB/CCR2/MR1/AP3B1/ZFP36L1/RUNX2/IL15/FCER1G/NFATC3/IRF4/ASCL2/RIPK3/IL4R/CD80/TCF7/CD86/NLRP3/WNT10B/JUNB/BCL2/ZC3H12A/PIK3R6/IL18/ZBTB7B/SMARCC1/SOS1/PRKDC/GPR183/PTGER4/RSAD2/BRD2/ARID1A/LFNG/CRTAM/DUSP10/IL6R/VSIR/NCAPH2/PBRM1/RELB/KMT2A/AP3D1/LOXL3/IL18R1/IL2RA/CLEC4E/STAT5B/SOCS1/MALT1/DROSHA/RORC/GPR18/ANXA1/ATG5/SMARCA4/PRKCZ/RIPK2/TP53/PNP/RC3H2

# TCR signaling: GBP1/PTPN22/LCK/PTPN6/LILRB4/LAPTM5/CD226/CD300A/KCNN4/CARD11/UBASH3A/PTPRJ/CYLD/TESPA1/PTPN2/BTN2A2/CCR7/TRAT1/SH2D1A/DGKZ/RAB29/SLA2/PRKD2/MALT1/IKBKG

# Th 17 response: JAK2/CLEC7A/IL12RB1/LY9/PRKCQ/SLAMF6/TBX21/IL17RA/CARD9/IL27RA/NFKBID/JAK1/BATF/TYK2/IRF4/ASCL2/JUNB/ZC3H12A/ZBTB7B/BRD2/IL6R/ARID5A/LOXL3/MALT1/RORC/RC3H2

# type II interferon response STAT1/GBP5/IRF1/GBP1/NLRC5/JAK2/SP100/GBP2/GBP4/HLA-DPA1/WAS/IRF8/CCL5/CCL4/IL12RB1/CCL3/TNF/AIF1/TLR2/LGALS9/PARP9/PTPN2/PARP14/CASP1/TXK/HCK/GBP3/FASLG/NR1H3/IFNG/CCL4L2/ACTR3/JAK1/TYK2/CD47/DAPK1/IFNGR2/STX4/ZYX/KIF5B/CCL8/ACTR2/STXBP3/TLR4/SIRPA/SLC26A6/TP53/RPL13A/SYNCRIP

# B cell activation: PTPRC/CD74/NCKAP1L/IL7R/DOCK10/JAK3/ITGA4/ATM/TBC1D10C/INPP5D/PTK2B/IKZF3/TCIRG1/CASP8/ZAP70/PIK3CD/TNFAIP3/PRKCB/EZH2/PTPN6/SAMSN1/BTK/IRF8/IL2RG/LAPTM5/MNDA/CD300A/SASH3/IL4I1/CARD11/MS4A1/THEMIS2/TNFSF13B/PTPRJ/LAT2/CTLA4/TYROBP/CYLD/TBX21/SPI1/PTPN2/SLAMF8/BANK1/IL27RA/LAX1/PRKCD/CD27/LEF1/NOD2/SYK/CD38/RASGRP1/CD40LG/CD28/SKAP2/ZFP36L2/TGFB1/TFRC/FLT3/CD70/BATF/STAT6/PLCG2/BAX/FCRL1/BLNK/MZB1/ID2/NFAM1/ZFP36L1/DOCK11/NFATC2/LYN/RBPJ/TNFRSF13C/CDKN1A/CD86/GAPT/BCL2/DCLRE1C/CD22/PKN1/PRKDC/GPR183/SH3KBP1/AKAP17A/CD180/VCAM1/LFNG/NSD2/FCRL3/NOTCH2/SLC39A10/ATAD5/MFNG/ZBTB7A/STAT5B/SUPT6H/CD79B/TLR4/BST2/VAV3/KLF6/RIF1/TICAM1/TPD52/PARP3/CMTM7/MALT1/DCAF1/TNFRSF4/EP300/LYL1/PPP2R3C/HHEX/ITFG2/AKIRIN2/NBN/TP53

# Innate immune responses: GBP5/IRF1/TRAF3IP3/CTSS/CYBA/USP15/GBP2/BIRC3/NAGK/ARRB2/PTPN22/NLRC3/PYHIN1/TNFAIP3/FCN1/CLEC7A/ZBP1/OAS1/BTK/HAVCR2/KLRC1/MNDA/CD300A/TASL/AIM2/IRF7/IFI16/TNIP3/TNF/UNC93B1/TLR2/TYROBP/CYLD/OASL/GFI1/CASP1/KLRD1/DDX60/MYD88/PIK3AP1/CD300LF/TRAF3/TLR6/IFIH1/NOD2/SCIMP/PAK1/IRAK1/HCK/SYK/NMI/CGAS/KLRC2/IRAK4/DDX3X/NR1H3/KLRK1/NLRP1/TLR10/AURKB/CARD8/NLRC4/PLCG2/OAS3/AP3B1/TLR8/IRF4/TLR1/IKBKB/FYN/MEFV/DHX58/ZNFX1/LYN/TREM2/RNF135/IRF3/ALPK1/NLRP3/SLC15A3/CD274/IFI35/TLR7/PARP1/PAK2/PRKDC/RSAD2/RIGI/XRCC5/SRC/KLRC4/EIF2AK2/PYCARD/FCRL3/CD14/TAX1BP1/LRCH4/ERBIN/TRIM11/CLEC4E/HSP90AA1/WDFY1/NAIP/TRIM25/GPR108/AKT1/TLR4/LILRA2/RBM14/RIOK3/IRAK2/SLC15A2/TICAM1/TKFC/RPS6KA3/PDPK1/SIN3A/UFD1/EP300/ITCH/ZDHHC18/RIPK2/LACC1

# NK activation HLA-E/JAK2/PTPN22/HAVCR2/FGR/KLRC1/ZNF683/TYROBP/KLRD1/TOX/RASGRP1/CLNK/PRDM1/KLRC2/FCGR3A/TYK2/IL15/IL18/STAT5B/DCAF15/TICAM1/IL15RA

# phagocytosis: PTPRC/ITGAL/NCKAP1L/DOCK2/ITGB2/CORO1A/CYBA/BIN2/ABCA7/LYST/MYO1G/P2RY6/FCN1/CLEC7A/P2RX7/ARHGAP25/BTK/IRF8/IL2RG/UNC13D/FGR/VAV1/SH3BP1/CD300A/IL2RB/ICAM3/SIRPB1/SLAMF1/SIRPG/PTPRJ/MSR1/AIF1/NCF4/TLR2/TYROBP/CDC42SE2/ITGAM/CSK/FCGR1A/PRKCD/NCF2/MYD88/CD300LF/NOD2/PAK1/C2/HCK/SYK/LIMK1/MYO7A/NR1H3/IFNG/PLCG2/CCR2/SPG11/IL15/FCER1G/CD47/FYN/CNN2/LYN/TREM2/RAB27A/CDC42SE1/DNM2/DYSF/ATG3/BLTP1/SLC11A1/ELMO1/RAB31/CFP/SRC/TMEM175/PYCARD/CD14/CEACAM4/SCARB1/RUBCN/CAMK1D/ITGA2/EIF2AK1/CLN3/TLR4/VAV3/SPHK1/RAB39A/FCGR2A/SIRPA/ELMO2/ELMO3/MYH9/LYAR/IL15RA/ANXA1/ATG5/PLA2G6

# IL6 reaaction: HLA-B/CD74/NCKAP1L/CYBA/POU2F2/INPP5D/CD84/ARRB2/PTPN22/NLRC3/LILRB2/TNFAIP3/CLEC7A/P2RX7/PTPN6/LILRB4/BTK/HAVCR2/SLAMF1/TNF/UNC93B1/AIF1/TLR2/TYROBP/LGALS9/CSK/MBP/IL17RA/BANK1/CARD9/IL27RA/SIGLEC16/MYD88/TLR6/IFIH1/NOD2/SCIMP/SYK/IFNG/TGFB1/TMEM106A/IL16/PLCG2/TLR8/TLR1/CD47/PTAFR/TREM2/TLR7/ZC3H12A/AGER/ARHGEF2/RIGI/PYCARD/IL6R/ELF4/ARID5A/ARRB1/TLR4/LILRA2/CD200R1/MAPK13/GHRL/SIRPA/TICAM1/DHX9/TRPV4/AKIRIN2/RIPK2

# response to bacterial origin: GBP5/B2M/IL10RA/JAK2/GBP2/LILRB1/CASP8/CD6/PTPN22/CYRIB/LILRB2/TAP2/TNFAIP3/P2RX7/CD96/BTK/IRF8/HAVCR2/CXCL10/CXCL9/TNIP3/TNF/TLR2/LGALS9/ACP5/GFI1/IRF5/CARD9/CCR7/CASP1/MYD88/TLR6/NOD2/SCIMP/IRAK1/C2/HCK/TNFRSF1B/GCH1/GBP3/PDE4B/FASLG/IRAK4/KMO/NR1H3/KLRK1/CSF2RB/SIGIRR/NFKB2/TLR10/SBNO2/CARD8/LTA/PLCG2/CCR5/LY86/TLR1/PTGER2/LYN/PTAFR/TREM2/CD80/IRF3/CD86/NLRP3/CD274/IDO1/PDCD4/ZC3H12A/IL18BP/FOS/SLC11A1/PTGER4/TRIM41/CD180/SRC/VCAM1/PYCARD/DUSP10/IL12RB2/CD14/SCARB1/ARID5A/ERBIN/NUGGC/CARD16/CDK19/HPGD/TAB2/AKT1/TLR4/LILRA2/ZFP36/CNR2/MAP2K3/PCK2/IRAK2/SIRPA/TICAM1/CMPK2/MALT1/RPS6KA3/OTUD5/CD68/SLC7A5/NFKB1/AKIRIN2/RIPK2/RPL13A/HMGB2/AKAP8

# viral genome replication: MX1/RESF1/GRK2/APOBEC3G/OAS1/CCL5/IFI16/APOBEC3H/ATG16L2/ZC3HAV1/TNF/OASL/OAS2/ISG20/FAM111A/IFIH1/TOP2A/PABPC1/CD28/DDX3X/TRIM38/ADAR/OAS3/ZNFX1/APOBEC3D/SHFL/ATG16L1/CXCR6/DEK/APOBEC3C/BCL2/ZC3H12A/ILF3/ISG15/RSAD2/EIF2AK2/CTBP1/APOBEC3F/CCL8/EEF1A1/SRPK1/FMR1/SETDB1/BST2/MPHOSPH8/LARP1/PKN2/ZBED1/SMC6/ATG5/TASOR</code></pre>
<pre class="r"><code># Patient AM 02
DEGenesAM2 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 02&quot;)


egoAM2 &lt;- enrichGO(gene = unique(DEGenesAM2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

egoAM2 &lt;- setReadable(egoAM2, OrgDb = org.Hs.eg.db)

#View(egoAM2)

dotplot(egoAM2, showCategory=20, title = &quot;Patient AM 02 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-5-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM2, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 02&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-5-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM2 &lt;- enrichGO(gene = unique(DEGenesAM2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

egoAM2 &lt;- setReadable(egoAM2, OrgDb = org.Hs.eg.db)
dotplot(egoAM2, showCategory=20, title = &quot;Patient AM 02 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-5-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM2 &lt;- enrichGO(gene = unique(DEGenesAM2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

egoAM2 &lt;- setReadable(egoAM2, OrgDb = org.Hs.eg.db)
dotplot(egoAM2, showCategory=20, title = &quot;Patient AM 02 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-5-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># heart contraction: TTN/MYBPC3/TPM1/MYH7/RYR2/TNNT2/DMD/SCN7A/MYH6/ATP2A2/TNNI3/PKP2/MYL7/GLP1R/GSN/CYP2J2/AKAP9/SCN5A/CHRM2/MYL3/ANK2/KCNH2/SLC4A3/SNTA1/MYL2/CASQ2/CACNA1C/POPDC2/ACTC1/ATP1A2/TNNC1/RNF207/TBX5/TBX2/DSP/ADRB1/GJA1/THRA/FGF12/JUP/CTNNA3/TNNI3K/TRDN/CACNB2/TMEM38A/TRPC1/DES/PPP1R13L/PLN/SPTBN4/DMPK/SCN2B/GATA6/TMEM65/CAMK2D/FLNA/AKAP13/KCND3/ACE/CORIN/SLC8A1/BVES/SPX

# action potential: RYR2/DMD/SCN7A/ATP2A2/PKP2/AKAP9/SCN5A/SLMAP/ANK2/KCNH2/SLC4A3/SNTA1/CACNA1C/ATP1A2/RNF207/DSP/GJA1/FGF12/JUP/CTNNA3/ANK3/CACNB2/SCN2B/CAMK2D/FLNA/KCND3/SLC8A1

# muscle tissue</code></pre>
<pre class="r"><code># Patient AM 03
DEGenesAM3 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 03&quot;)


egoAM3 &lt;- enrichGO(gene = unique(DEGenesAM3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM3 &lt;- setReadable(egoAM3, OrgDb = org.Hs.eg.db)

#View(egoAM3)
dotplot(egoAM3, showCategory=20, title = &quot;Patient AM 03 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-6-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM3, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 03 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-6-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM3 &lt;- enrichGO(gene = unique(DEGenesAM3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM3 &lt;- setReadable(egoAM3, OrgDb = org.Hs.eg.db)
dotplot(egoAM3, showCategory=20, title = &quot;Patient AM 03 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-6-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM3 &lt;- enrichGO(gene = unique(DEGenesAM3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM3 &lt;- setReadable(egoAM3, OrgDb = org.Hs.eg.db)
dotplot(egoAM3, showCategory=20, title = &quot;Patient AM 03 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-6-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Genes associated with different pathways:
# BMP signaling pathway: BMPER/PCSK6/USP9Y/GPC3/TMEM100/SMAD5/NFIA/BMP4/ENG/HTRA3/TGFBR3/GATA6/PDCD4/SMAD5-AS1/CTDSPL2/GATA4/PPM1A/SMAD6/SMAD2/TBX20/SFRP1/SMAD7/BMPR1A/BMPR2/HTRA1/TOB1/SMAD9/ACVR1/ACVR2A/NEO1/NBL1/FST/HFE

# renal axis and development: BMPER/ALDH1A2/POU3F3/FOXC1/MAGI2/COL4A4/ITGA8/COL4A3/PBX1/TBX18/PDGFD/GFRA1/GPC3/FGFR1/SMAD5/NFIA/PDGFRA/BMP4/WWTR1/CALB1/CC2D2A/ROBO2/DYNC2H1/TCF21/FGF10/BICC1/CDKN1C/FLCN/RGN/PCSK5/PPP3CA/EFNB2/WDPCP/IFT20/ACAT1/SMAD6/SMAD2/SFRP1/SALL1/SMAD7/PODXL/EDNRB/RET/OSR2/RARB/MME/C1GALT1/WNT9B/LGR4/KLHL3/CD24/IFT25/PECAM1/ACTA2/SOX9/PYGO1/FGF2/AKR1B1/TRAF3IP2/ERBB4/SULF2/GDNF/TRAF3IP1/IRX3/WNT2B/ANKS6/AQP11

# endocardial cushion development/aortic vavle morphogenesis: ADAMTS5/TMEM100/BMP4/ENG/ROBO2/GATA4/NOS3/SMAD2/TBX20/BMPR1A/BMPR2/APC/SOX9/ACVR1/SNAI2/CRELD1/ROBO1/APLNR/TWIST1

# determination left/right symmetry PCSK6/HAND2/ENG/C2CD3/MMP21/BBS5/CC2D2A/DYNC2H1/IFT172/FGF10/BICC1/MICAL2/GATA4/DYNC2LI1/MIB1/SMAD2/TBX20/ALG5/CCDC39/GAS8/ACVR1/APLNR/GJA1/TMED2/ACVR2A/IFT57/DISP1/BBS7/PSKH1/SOX18/ANKS6

# septum development ZFPM2/HEG1/MATR3/BMP4/ENG/TGFBR3/GATA6/ROBO2/SLIT3/PARVA/GATA4/SMAD6/NOS3/ANK2/TBX20/SALL1/SMAD7/ZFPM1/BMPR1A/BMPR2/EGLN1/ACVR1/CRELD1/ROBO1/APLNR</code></pre>
<pre class="r"><code># Patient AM 04
DEGenesAM4 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 04&quot;)


egoAM4 &lt;- enrichGO(gene = unique(DEGenesAM4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM4 &lt;- setReadable(egoAM4, OrgDb = org.Hs.eg.db)
dotplot(egoAM4, showCategory=20, title = &quot;Patient AM 04 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-7-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM4, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 04 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-7-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM4 &lt;- enrichGO(gene = unique(DEGenesAM4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM4 &lt;- setReadable(egoAM4, OrgDb = org.Hs.eg.db)
dotplot(egoAM4, showCategory=20, title = &quot;Patient AM 04 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-7-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM4 &lt;- enrichGO(gene = unique(DEGenesAM4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM4 &lt;- setReadable(egoAM4, OrgDb = org.Hs.eg.db)
dotplot(egoAM4, showCategory=20, title = &quot;Patient AM 04 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-7-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># genes --&gt; cardiac muscle</code></pre>
<pre class="r"><code># Patient AM 05
DEGenesAM5 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 05&quot;)


egoAM5 &lt;- enrichGO(gene = unique(DEGenesAM5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM5 &lt;- setReadable(egoAM5, OrgDb = org.Hs.eg.db)
dotplot(egoAM5, showCategory=20, title = &quot;Patient AM 05 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-8-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM5, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 05&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-8-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM5 &lt;- enrichGO(gene = unique(DEGenesAM5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM5 &lt;- setReadable(egoAM5, OrgDb = org.Hs.eg.db)
dotplot(egoAM5, showCategory=20, title = &quot;Patient AM 05 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-8-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM5 &lt;- enrichGO(gene = unique(DEGenesAM5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM5 &lt;- setReadable(egoAM5, OrgDb = org.Hs.eg.db)
dotplot(egoAM5, showCategory=20, title = &quot;Patient AM 05 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-8-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ECM, adhesion, WNT</code></pre>
<pre class="r"><code># Patient AM 06
DEGenesAM6 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 06&quot;)


egoAM6 &lt;- enrichGO(gene = unique(DEGenesAM6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM6 &lt;- setReadable(egoAM6, OrgDb = org.Hs.eg.db)

#View(egoAM6)

dotplot(egoAM6, showCategory=20, title = &quot;Patient AM 06 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-9-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM6, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 06 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-9-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM6 &lt;- enrichGO(gene = unique(DEGenesAM6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM6 &lt;- setReadable(egoAM6, OrgDb = org.Hs.eg.db)
dotplot(egoAM6, showCategory=20, title = &quot;Patient AM 06 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-9-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM6 &lt;- enrichGO(gene = unique(DEGenesAM6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM6 &lt;- setReadable(egoAM6, OrgDb = org.Hs.eg.db)
dotplot(egoAM6, showCategory=20, title = &quot;Patient AM 06 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-9-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># rRNA: MRTO4/RPLP0/RPL7/RPS4X/EIF1AX/RPS17/RPS5/MDN1/RPL26/RPS27/PDCD11/RPL10A/RPS14/PRKDC/RPL10/RPS3A/RPL35/RPL23A/HEATR1/RPS24/RPS16/RPL27/RPL11/WDR75/RPS6/RPS12/DDX10/RPS28/RPS13/WDR12/RPS19/RPL38/CUL4B/GTPBP10/WDR36/RPS8/RPS23/KRI1/RPS15/MTG2/RPL7A/NHP2/RPS7/NOL9/NPM1/NOP58/RPL35A/RPS11/TSR1/CUL4A/RPS9/RPS27L/PA2G4/DIMT1/GLUL/DDX3X/DKC1/C1QBP/GNL3L/RPS21/RPL5/UTP20/EIF2A/NUP88/SBDS/FTSJ3/PWP2/FCF1/URB1/RAN/NOP56/RPSA/NOL10/GTPBP4/DDX52/WDR46/EXOSC2/BUD23/WDR18/METTL5/FASTKD2/LAS1L/MRM1/RIOX2/SNU13/TSC1/MYBBP1A/ABCE1/RRP12/WDR3/LTV1/NOP53/EMG1/MRPS2/NOL6/FRG1/RPS25/MPHOSPH6/RSL1D1/AFG2A/NAT10/MTERF3/BRIX1/SLX9/RPP25/EXOSC9/RRP36/PIH1D1/ERI3/EXOSC7

# purine and nucleotide synthesis: PAPSS2/ND4/ND1/ND5/ND2/ND3/ND6/GCDH/ATP6/ATP8/NNMT/ENO1/NDUFA13/AFMID/PFAS/ACACA/PDK4/AK5/DMAC2L/IMPDH2/PTGIS/HSD17B12/NAMPT/AK3/ADSL/ACAT1/ATIC/AK2/ND4L/PID1/MPC2/NDUFAB1/NDUFS4/TMSB4X/ATP5F1D/PRPS1/ACSS2/VCP/ATP5F1A/ALDOA/SDHA/ELOVL5/ATP5MJ/NDUFS8/NMRK2/PRPSAP1/ATP5PF/PAICS/PDK1/NDUFB8/NDUFB3/NADK/ADSS2/NOS1/ACSL1/NME7/ATP5ME/ACLY/ATP6V1A/AK1/PINK1/NDUFB7/ATP5F1C/NDUFS3/APRT/SDHB/ADCY4/ATP5MK/NMNAT3/ATP5MC1/NPR1/NDUFV1/NDUFA1

# ATP metabolic process: MYH6/ND4/ND1/ND5/ND2/ND3/ND6/ATP6/ATP8/ENO1/NDUFA13/ENO3/HSPA8/AK5/DMAC2L/ALDOC/MLST8/PRKAA1/GAPDH/AK2/GPI/HIF1A/ND4L/PID1/PGK1/NDUFAB1/NDUFS4/HDAC4/TMSB4X/ATP5F1D/VCP/ATP5F1A/ALDOA/SDHA/ATP5MJ/NDUFS8/PRKAA2/TPI1/ATP5PF/FKRP/NDUFB8/PRKAG1/CLPX/NDUFB3/NADK/APP/LDHA/OGT/ATP1B1/ATP5ME/ARL2/ATP6V1A/AK1/PINK1/MYH4/NDUFB7/ATP5F1C/MYH7/NDUFS3/SDHB/ATP5IF1/ATP5MK/ATP5MC1/IER3/NDUFV1/NDUFA1/RPTOR</code></pre>
<pre class="r"><code># Patient AM 07
DEGenesAM7 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 07&quot;)


egoAM7 &lt;- enrichGO(gene = unique(DEGenesAM7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM7 &lt;- setReadable(egoAM7, OrgDb = org.Hs.eg.db)

#View(egoAM7)
dotplot(egoAM7, showCategory=20, title = &quot;Patient AM 07 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-10-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM7, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 07 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-10-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM7 &lt;- enrichGO(gene = unique(DEGenesAM7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM7 &lt;- setReadable(egoAM7, OrgDb = org.Hs.eg.db)
dotplot(egoAM7, showCategory=20, title = &quot;Patient AM 07 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-10-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM7 &lt;- enrichGO(gene = unique(DEGenesAM7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM7 &lt;- setReadable(egoAM7, OrgDb = org.Hs.eg.db)
dotplot(egoAM7, showCategory=20, title = &quot;Patient AM 07 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-10-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Response to hypoxia: VEGFA/THBS1/EGLN3/EGR1/ANGPTL4/ADM/ENO1/HK2/STC2/FOSL2/PDK1/P4HB/PLOD2/LOXL2/HSP90B1/HMOX1/LONP1/ERO1A/HILPDA/SLC2A1/SMAD3/STC1/BNIP3/HIF3A/ND4/PGK1/LIF/AK4/NDRG1/PTGS2/PLOD1/HYOU1/ACVRL1/NF1/HIPK2/ND5/LMNA/ND2/PLEKHN1/MMP14/RORA/BNIP3L/FOXO3/HIF1A/MYC/COX2/ATF4/ND1/DDR2/DDAH1/TGFBR3/MDM4/PTGIS/SCFD1/TMBIM6/FOS/SRF/TBL2/ADAM17/HSPG2/VHL/COX1/SLC9A1/NOL3/PRKCE/SIRT1/ADSL/LIMD1/ALKBH5/CYP1A1/RYR2/PPARA/COMT/POSTN/GATA6/MMP2/TGFB3/SOD3/ARNT/TP53/FAM162A/RWDD3/MYOCD/PLK3/DDIT4/TGFB1/ERCC2/PMAIP1/AGTRAP/ANGPT2/NOP53/LTA/PLAU/VASN/TFAM/NR4A2

# Wnt signaling: EGR1/SMAD3/COL1A1/JADE1/NFATC4/PLEKHA4/IGFBP4/CCN4/EDNRB/NID1/LRP1/VCP/BICC1/UBR5/PLPP3/PKD1/COL6A1/FOXO3/G3BP1/UBE2B/CSNK2A1/RBMS3/ALPK2/PRKAA2/SKI/DDIT3/PPP2CA/ABL1/SNAI2/MAGI2/MDFI/DDB1/TLE2/STRN/DDX3X/SMARCA4/FOXO1/PTK7/RYK/ZNRF3/LMBR1L/ADNP/DAPK3/YAP1/FZD8/NKD2/WNK1/PYGO1/CDK14/EGF/ZRANB1/XIAP/ARHGEF19/KLF15/AMOTL1/NFKB1/PRICKLE1/LIMD1/CDH2/EXT1/SNAI1/CUL3/TGFB1I1/LATS2/SMURF2/PPM1A/PPP2R1A/CSNK1G2/ASPM/PIAS4/PTPRU/CCAR2/TTC21B/GID8/ITGA3/EMD/ROR2/APOE/BCL9L/SMURF1/RPS12/ZNF703/NPHP4/TBL1X/NLK/BMAL1/PRICKLE2/GLI3/DACT3/LGR6/TGFB1/NOTUM/NKD1/ATP6AP2/RAB5A/RNF138/LGR4/TAX1BP3/FGF9/AMFR/DVL2/PPM1B/AXIN2/TBX18/IGFBP2/LRRK1/AMOTL2/SOX9/CELSR1/FZD3/RUVBL1/FZD1/GSKIP/RUVBL2/HIC1/NR4A2

# TGFbeta:THBS1/LOX/GDF15/HSPA5/COL3A1/SMAD3/FBN1/COL1A2/ADAM9/ACVRL1/HIPK2/FLCN/SKIL/FMOD/LTBP1/SKI/SPRY1/HSP90AB1/SNX25/STAT3/TGFBR3/USP9X/ZEB1/ADAMTSL2/SPRED3/FOS/JUN/TGFBR2/SMAD2/ADAM17/OGT/FKBP1A/ZBTB7A/HDAC2/ZNF451/SMAD5/SIRT1/LEMD3/SPRY2/FBN2/TGFBR1/TGFB1I1/LATS2/SMURF2/PPM1A/VEPH1/PPARA/ITGB5/TGFB3/ZYX/ITGA3/TP53/BCL9L/SMURF1/APPL1/ZNF703/STRAP/NLK/MYOCD/PDPK1/TGFB1/SINHCAF/SDCBP/LTBP2/ITGB8/CILP/ACVR1/ING1/VASN/BRMS1</code></pre>
<pre class="r"><code># Patient AM 08
DEGenesAM8 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 08&quot;)


egoAM8 &lt;- enrichGO(gene = unique(DEGenesAM8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM8 &lt;- setReadable(egoAM8, OrgDb = org.Hs.eg.db)
dotplot(egoAM8, showCategory=20, title = &quot;Patient AM 08 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-11-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM8, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 08 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-11-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM8 &lt;- enrichGO(gene = unique(DEGenesAM8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM8 &lt;- setReadable(egoAM8, OrgDb = org.Hs.eg.db)
dotplot(egoAM8, showCategory=20, title = &quot;Patient AM 08 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-11-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM8 &lt;- enrichGO(gene = unique(DEGenesAM8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM8 &lt;- setReadable(egoAM8, OrgDb = org.Hs.eg.db)
dotplot(egoAM8, showCategory=20, title = &quot;Patient AM 08 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-11-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient AM 09
DEGenesAM9 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 09&quot;)


egoAM9 &lt;- enrichGO(gene = unique(DEGenesAM9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM9 &lt;- setReadable(egoAM9, OrgDb = org.Hs.eg.db)
dotplot(egoAM9, showCategory=20, title = &quot;Patient AM 09 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-12-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM9, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 09 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-12-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM9 &lt;- enrichGO(gene = unique(DEGenesAM9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM9 &lt;- setReadable(egoAM9, OrgDb = org.Hs.eg.db)
dotplot(egoAM9, showCategory=20, title = &quot;Patient AM 09 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-12-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM9 &lt;- enrichGO(gene = unique(DEGenesAM9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM9 &lt;- setReadable(egoAM9, OrgDb = org.Hs.eg.db)
dotplot(egoAM9, showCategory=20, title = &quot;Patient AM 09 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-12-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient AM 10
DEGenesAM10 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 10&quot;)


egoAM10 &lt;- enrichGO(gene = unique(DEGenesAM10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM10 &lt;- setReadable(egoAM10, OrgDb = org.Hs.eg.db)

#View(egoAM10@result)

dotplot(egoAM10, showCategory=20, title = &quot;Patient AM 10 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-13-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM10, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 10 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-13-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM10 &lt;- enrichGO(gene = unique(DEGenesAM10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM10 &lt;- setReadable(egoAM10, OrgDb = org.Hs.eg.db)
dotplot(egoAM10, showCategory=20, title = &quot;Patient AM 10 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-13-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM10 &lt;- enrichGO(gene = unique(DEGenesAM10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM10 &lt;- setReadable(egoAM10, OrgDb = org.Hs.eg.db)
dotplot(egoAM10, showCategory=20, title = &quot;Patient AM 10 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-13-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># wound healing: SERPINE1/TIMP1/TNFRSF12A/FN1/THBS1/ITGAV/COL3A1/F2RL1/ANXA2/ANXA5/CCN1/TFPI/PPL/SMAD3/DST/AHNAK/CD59/SERPINE2/VCL/GNA12/CAV1/AXL/CSRP1/IGF1/MYH9/ITGA5/CHMP1A/DCBLD2/FLNA/ITGB5/SYT7/ACTB/PDIA3/COL5A1/FKBP10/PDGFRA/EMILIN1/F2R/HSPB1/SVEP1/TGFB1/MYOF/ANO6/EXT1/MRTFA/YAP1/PIK3CA/XBP1/ARFGEF1/NF1/STXBP1/FGFR1OP2/PAPSS2/DDR2/ACTG1/LMAN1/NFE2L2/CLASP2/SDC4/FBLN1/KANK1/DGKI/NOTCH2/PLPP3/PLEC/VAV3/CD34/NFATC1/IL6ST/DAG1/SYTL4

# integrin mediated pathway: TIMP1/LAMA2/NID1/LAMC1/CD63/LIMS1/LAMA1/FLNA/LAMB1/PRKD1

# cell adhesion: SERPINE1/FN1/THBS1/COL8A1/ITGAV/CCN2/MICALL2/COL3A1/ANXA2/ITGBL1/ABI3BP/FBLN5/COL1A1/CCN1/NID1/SMAD3/LAMC1/CALR/SPOCK1/VCL/CD63/LIMS1/AXL/SLK/ITGA5/SNED1/ITGA11/FLNA/ACTN1/ITGB5/LAMB1/P4HB/EMILIN1/CCDC80/CORO1C/VIT/SVEP1/ROCK2/BRAF/HAS2/ANGPT1/BVES/NF1/COL16A1/MYADM/POSTN/ACTG1/ADAM9/MMP14/FER/CLASP2/SDC4/TRIOBP/EPHA3/FBLN1/KANK1/MKLN1/DAPK3/MELTF/PXN/PEAK1/BCL6/ABL1/TNXB/CD34/DAG1

# ECM organisation: OL8A1/LAMA2/CYP1B1/COL1A2/COL4A1/COL3A1/COL4A2/ANXA2/ABI3BP/HMCN1/FBLN5/COL1A1/ADAMTS2/COL12A1/CCN1/ADAMTS5/NID1/COL5A2/ELN/SMAD3/LAMC1/PXDN/COL18A1/TGFBR3/CAV1/PHLDB1/LOXL1/LAMA1/ADAMTS19/LAMB1/FURIN/COL5A1/LRP1/FKBP10/MMP2/PDGFRA/EMILIN1/SOX9/CCDC80/VIT/TGFB1/EXT1/SERPINH1/HAS2/NF1/MMP16/COL16A1/COL6A1/DDR2/POSTN/MMP14/AEBP1/CLASP2/FBLN1/ADAMTS10/MELTF/QSOX1/TLL1/ABL1/SH3PXD2B/TNXB/PRDM5/DAG1</code></pre>
<pre class="r"><code># Patient AM 11
DEGenesAM11 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 11&quot;)


egoAM11 &lt;- enrichGO(gene = unique(DEGenesAM11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM11 &lt;- setReadable(egoAM11, OrgDb = org.Hs.eg.db)

#View(egoAM11@result)

dotplot(egoAM11, showCategory=20, title = &quot;Patient AM 11 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-14-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM11, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 11 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-14-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM11 &lt;- enrichGO(gene = unique(DEGenesAM11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM11 &lt;- setReadable(egoAM11, OrgDb = org.Hs.eg.db)
dotplot(egoAM11, showCategory=20, title = &quot;Patient AM 11 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-14-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM11 &lt;- enrichGO(gene = unique(DEGenesAM11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM11 &lt;- setReadable(egoAM11, OrgDb = org.Hs.eg.db)
dotplot(egoAM11, showCategory=20, title = &quot;Patient AM 11 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-14-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># angiogenesis/vasculature development TNN/SFRP1/TSPAN12/BMPER/GAB1/TIE1/ADAM10/ANGPTL4/PLK2/TJP1/GTF2I/PTPRM/FOXC1/PRKD1/CXCR2/HSPB1/TEK/ATP2B4/ECSCR/TSPAN18/CCBE1/CYSLTR1/GREM1/CD34/SASH1/RAMP2/SPINK5/FGF18/RECK

# heart growth AKAP6/FOXC1/PDLIM5/PRKG1/HEG1/COL14A1/HEY2/ACACB/FOXP1/SORBS2/ADPRHL1/MEF2C/VGLL4

# action potential: PTPN3/KCNJ2/SLMAP/CACNB2/YWHAH/SCN11A/CACNA2D1

# lymph vessel development: BMPR2/TIE1/FOXC1/HEG1/EFNB2/CCBE1

# muscle apoptosis: GF1R/KIFAP3/BMP7/SIRT5/PDE1A/CXCR2/MYOCD/MAP3K5/HAND2/LRP6/HEY2</code></pre>
<pre class="r"><code># Patient AM 12
DEGenesAM12 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 12&quot;)


egoAM12 &lt;- enrichGO(gene = unique(DEGenesAM12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM12 &lt;- setReadable(egoAM12, OrgDb = org.Hs.eg.db)
#View(egoAM12@result)


dotplot(egoAM12, showCategory=20, title = &quot;Patient AM 12 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-15-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM12, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 12 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-15-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM12 &lt;- enrichGO(gene = unique(DEGenesAM12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM12 &lt;- setReadable(egoAM12, OrgDb = org.Hs.eg.db)
dotplot(egoAM12, showCategory=8, title = &quot;Patient AM 12 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-15-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM12 &lt;- enrichGO(gene = unique(DEGenesAM12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM12 &lt;- setReadable(egoAM12, OrgDb = org.Hs.eg.db)
dotplot(egoAM12, showCategory=20, title = &quot;Patient AM 12 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-15-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># cardiac muscle
# sympathetic nerve development: SOX4/NTRK1/HAND2/NRP1/EDNRA/SOX11</code></pre>
<pre class="r"><code># Patient AM 13
DEGenesAM13 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 13&quot;)


egoAM13 &lt;- enrichGO(gene = unique(DEGenesAM13$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM13 &lt;- setReadable(egoAM13, OrgDb = org.Hs.eg.db)
#View(egoAM13@result)

dotplot(egoAM13, showCategory=30, title = &quot;Patient AM 13 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-16-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM13, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 13 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-16-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM13 &lt;- enrichGO(gene = unique(DEGenesAM13$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM13 &lt;- setReadable(egoAM13, OrgDb = org.Hs.eg.db)
dotplot(egoAM13, showCategory=20, title = &quot;Patient AM 13 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-16-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># WNT; adhesion
# Response to Glucocorticosteroid: FOXO3/EGFR/KLF9/DDIT4/FIBIN/PIK3CA/CFLAR/FBXO32/CDKN1A/PDCD7/FOSL2/BCL2/SERPINF1/PTGDS/FAM107A/PAPPA/IGF1R/UCN

# response to hypoxia: HIF3A/MALAT1/FOXO3/MDM2/KCNMA1/DDIT4/TGFBR3/CFLAR/CAT/AQP3/BNIP3L/PDK3/HIPK2/VEGFD/SMAD3/NFE2L2/ANGPTL4/RGCC/FOSL2/BCL2/NDRG1/SMAD4/SIRT1/BNIP3/CRYAB/ABAT/ANG/SOD3/BMP2

# Insulin R signaling: PDK4/TNS2/PID1/RHOQ/IRS2/PIK3CA/PIK3R1/PDK2/SORBS1/FOXO1/PIP4K2A/KANK1/SIRT1/GKAP1/NCL/IGF1R/GRB10/BAIAP2/FOXC2/SOS2/STXBP4

# renal system: ZBTB16/NID1/TNS2/NPNT/ARID5B/EDNRB/ACTA2/COL4A1/CFLAR/CAT/IL6R/KLF15/PKD2/ITGA8/FMN1/SMAD3/BCL2/BMPER/SERPINF1/ODC1/WNT9B/IRX3/MTSS1/SMAD4/FGF10/GPC3/PTCH1/AGTR1/ERBB4/GFRA1/GDF11/JMJD6/FOXC2/PDGFA/MPST/CALB1/BMP2/PPP3CA

# WNT: ADE1/FOXO3/SHISA3/NID1/PTEN/EDNRB/EGFR/CALCOCO1/NKD1/IGFBP6/RBMS3/TRABD2B/NDRG2/KLF15/HBP1/PPM1B/PKD2/SIAH2/DAAM2/FOXO1/TLE1/SMAD3/EMD/OTULIN/LGR6/WNT9B/UBE2B/SMAD4/FGF10/APOE/KANK1/FRAT1/GPC3/BMAL1/FBXW4/TCF7/WNK1/SKIC8/LDB1/BAMBI/STK4/DAAM1/GRB10/DKK3/DIXDC1/BRD7/BMP2</code></pre>
<pre class="r"><code># Patient AM 14
DEGenesAM14 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 14&quot;)


egoAM14 &lt;- enrichGO(gene = unique(DEGenesAM14$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM14 &lt;- setReadable(egoAM14, OrgDb = org.Hs.eg.db)
dotplot(egoAM14, showCategory=30, title = &quot;Patient AM 14 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-17-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM14, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 14 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-17-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM14 &lt;- enrichGO(gene = unique(DEGenesAM14$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM14 &lt;- setReadable(egoAM14, OrgDb = org.Hs.eg.db)
dotplot(egoAM14, showCategory=8, title = &quot;Patient AM 14 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-17-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM14 &lt;- enrichGO(gene = unique(DEGenesAM14$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM14 &lt;- setReadable(egoAM14, OrgDb = org.Hs.eg.db)
dotplot(egoAM14, showCategory=20, title = &quot;Patient AM 14 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-17-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient AM 15
DEGenesAM15 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 15&quot;)


egoAM15 &lt;- enrichGO(gene = unique(DEGenesAM15$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM15 &lt;- setReadable(egoAM15, OrgDb = org.Hs.eg.db)
dotplot(egoAM15, showCategory=20, title = &quot;Patient AM 15 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-18-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM15, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 15 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-18-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM15 &lt;- enrichGO(gene = unique(DEGenesAM15$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM15 &lt;- setReadable(egoAM15, OrgDb = org.Hs.eg.db)
dotplot(egoAM15, showCategory=8, title = &quot;Patient AM 15 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-18-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM15 &lt;- enrichGO(gene = unique(DEGenesAM15$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM15 &lt;- setReadable(egoAM15, OrgDb = org.Hs.eg.db)
dotplot(egoAM15, showCategory=20, title = &quot;Patient AM 15 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-18-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient AM 16
DEGenesAM16 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 16&quot;)


egoAM16 &lt;- enrichGO(gene = unique(DEGenesAM16$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM16 &lt;- setReadable(egoAM16, OrgDb = org.Hs.eg.db)

#View(egoAM16@result)
dotplot(egoAM16, showCategory=20, title = &quot;Patient AM 16 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-19-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM16, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 16 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-19-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM16 &lt;- enrichGO(gene = unique(DEGenesAM16$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM16 &lt;- setReadable(egoAM16, OrgDb = org.Hs.eg.db)
dotplot(egoAM16, showCategory=8, title = &quot;Patient AM 16 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-19-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM16 &lt;- enrichGO(gene = unique(DEGenesAM16$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM16 &lt;- setReadable(egoAM16, OrgDb = org.Hs.eg.db)
dotplot(egoAM16, showCategory=20, title = &quot;Patient AM 16 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-19-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># T lymphocyte differentiation: CD74/IRF1/HLA-DRB1/HLA-DRA/B2M/PTPRC/ACTB/FGL2/LILRB2/NCKAP1L/TCIRG1/JAK3/RUNX1/DOCK2/SOCS1/LGALS9/BTN2A2/IL15/FCER1G/CD2/RSAD2/LILRB4/SYK/SPI1/RUNX3/IL4I1/LAG3/CD4/PTPN22/IL2RG/HLA-DOA/ASCL2/CD86/CLEC4E/CCR2/CD80/PTPN2/CD8A/RHOH/IL12RB1/IKZF3/BATF/PREX1/BCL11B/MR1/IL18/CD3D/SASH3/TBX21/ITK/VAV1/SPN/FANCA/TGFB1/SEMA4A/CYLD/IL27/IRF4/PIK3CD/THEMIS/CD3G/CD1D/CDK6/LCK/EOMES/MAFB/CD3E/RHOA/PRELID1/ZC3H12A/IL4R/RELB/PRDM1/RIPK3/IL1B/NFKBID/ADAM8/CD83/KLRC1/RIPK2/ZAP70/SOCS3/IFNG/MYB/IL7R/PTGER4/STAT6/RASGRP1/PNP/TMEM131L/CTLA4/CLEC4D/ANXA1/LIPA/RC3H2/SLAMF6/SOS1/JUNB/BCL3/GATA3/CAMK4/STAT4/BRD2/FOSL2/SP3/MALT1/ARID1A/IL6R/CTNNB1/CD27/JAK1/VSIR/PBRM1/FANCD2/RUNX2/ZBTB7B/PRKDC/ZBTB1/CTSL/TGFBR2/PIK3R6/NFKBIZ/RPL22/CBFB/TNFRSF9/DUSP10/FOXP3/ADAM17/LOXL3/BCL2/IL18R1/CRTAM/BRD7/AP3B1/NFATC3/LIG4/NLRP3/XBP1/RC3H1/TP53/KMT2A/ZFP36L2/FZD5/MDK/PKNOX1/CARD11/SMAD7/TESPA1/METTL3/SLC4A2/CCL19/ATG5/DROSHA/GPR89B/GPR89B/EGR3/EGR1/PATZ1/SRF/MEN1/LY9/TOX/LEF1/FADD/GPR183/TNFSF8/CD28/IL7/CCR7/PIK3R3/ACTL6A/ZNF683/IL2RA/PDP2/GPR18/WNT10B

# TCR activation: RF1/HLA-A/HLA-DRB1/HLA-DRA/PTPRC/HLA-E/JAK2/NCKAP1L/TCIRG1/JAK3/LILRB1/RUNX1/DOCK2/SOCS1/CD274/LGALS9/IL15/RSAD2/LILRB4/SYK/RUNX3/IL2RG/CD300A/NKG7/ASCL2/CD86/CCR2/CD80/EBI3/IL12RB1/BATF/TRBC2/BCL11B/IL18/CD3D/SASH3/TBX21/HSPH1/ITK/SPN/SEMA4A/RASAL3/WDFY4/IL27/IRF4/TRAC/CD3G/CD247/EOMES/CD3E/ELF4/RHOA/ZC3H12A/IL4R/TNFRSF14/RELB/PRDM1/NFKBID/CD83/RIPK2/ZAP70/SOCS3/IFNG/MYB/PTGER4/STAT6/PNP/ANXA1/RC3H2/SLAMF6/JUNB/BCL3/GATA3/STAT4/BRD2/CD81/FOSL2/MALT1/IL6R/JAK1/VSIR/ZBTB7B/CTSL/TGFBR2/NFKBIZ/PRKCQ/RPL22/CBFB/FOXP3/LOXL3/BCL2/IL18R1/CRTAM/TYK2/AP3B1/NLRP3/RC3H1/KMT2A/NDFIP1/AGER/CARD11/SMAD7/CBLB/SLC4A2/CCL19/CLEC4A/TRBC1/IL12A/MEN1/LY9/TOX/LEF1/GPR183/TNFSF8/CD55/CD28/ZNF683/IL2RA/CD160/PDP2/GPR18

# B lymphocyte activation: CD74/PTPRC/CD38/TNFAIP3/NCKAP1L/TCIRG1/JAK3/DOCK10/TYROBP/LAPTM5/TLR4/TNFSF13B/ITGA4/SLAMF8/PTK2B/SAMSN1/SKAP2/LYN/CASP8/IRF8/PTPRJ/PRKCB/LAT2/BTK/PTPN6/SYK/SPI1/MNDA/IL4I1/IL2RG/CD300A/CD86/NFAM1/PTPN2/EZH2/IKZF3/BATF/CD40/BST2/CDKN1A/BLNK/THEMIS2/SASH3/TBX21/PRKCD/INPP5D/NOD2/TGFB1/MZB1/TBC1D10C/NBN/CYLD/VCAM1/PIK3CD/KLF6/BAX/DOCK11/PLCG2/IL27RA/ZAP70/IL7R/NOTCH2/STAT6/RASGRP1/CTLA4/DCLRE1C/AKIRIN2/ATM/EXO1/BCL3/TNIP2/RBPJ/LAX1/CD81/TNFRSF4/TXLNA/FOSL2/HSPD1/SP3/HHEX/CMTM7/AHR/NSD2/CASP3/MALT1/FLT3/TFRC/CD27/CD70/PPP2R3C/ATAD5/LGALS1/PRKDC/ZBTB1/NFKBIZ/LYL1/SLC15A4/ID2/NFATC2/ZBTB7A/BAK1/FOXP3/ADAM17/BCL2/EXOSC6/PKN1/LIG4/VAV3/IL10/XBP1/RC3H1/PLCL2/SLC25A5/TP53/TICAM1/MIF/ZFP36L2/SHLD2/NDFIP1/BST1/CARD11/TOP2B/FCGR2B/BANK1/SYVN1/MIR17HG/HDAC4/MLH1/PHB2/TNFSF13/CD180/MSH2/HMGB3/DCAF1/MFNG/TCF3/IGBP1/SHLD1/CTPS1/EPHB2/FCRL3/POU2AF1/SH3KBP1/TPD52/PARP3/LEF1/PAXIP1/GPR183/PHB1/CD28/IL7/DNAJB9/SHLD3/GAPT/CD79B/PIK3R3/TNFRSF13C/MAD2L2/TIRAP/CD40LG/MS4A1

# Innate: GBP5/IRF1/GBP2/CTSS/IFI16/BIRC3/TNFAIP3/CYBA/DDX60/HSPA8/ALPK1/TYROBP/IFIH1/OAS1/OAS3/TLR4/CLEC7A/CASP1/TLR2/IRF7/CD274/CGAS/LYN/IFI35/RSAD2/BTK/MYD88/HAVCR2/UNC93B1/NMI/SYK/SCIMP/MNDA/S100A9/TLR1/PTPN22/PIK3AP1/CD300A/S100A8/AIM2/CD300LF/TLR8/CLEC4E/SLC15A3/HCK/PAK1/TRAF3/TIFA/TNF/IRAK1/ZBP1/TASL/TRAF3IP3/USP15/ACOD1/TLR6/MEFV/KLRD1/FFAR2/PYCARD/WDFY1/NOD2/PYHIN1/LACC1/CYLD/IRF4/ZNFX1/ARRB2/HSP90AA1/DDX3X/NAGK/RIGI/GFI1/CARD8/TRIM25/DHX58/OASL/NFKBIA/AURKB/PLCG2/NINJ1/IRAK4/OTUD4/KLRC1/RIPK2/EIF2AK2/LY96/FCN1/TLR7/TBK1/ERBIN/RTN4/SRC/IRAK3/NLRC4/OTULIN/TNIP3/NR1H3/TNIP2/BIRC2/NLRC3/IKBKB/RNF135/ZDHHC5/HSPD1/NAIP/XIAP/TRIM5/SLC15A2/UFD1/PPP2CA/EPG5/BECN1/PAK2/RPS6KA3/PRKDC/NFKBIZ/PUM2/ZCCHC3/PDPK1/RELA/SLC15A4/UBQLN1/TRIM11/NOD1/PARP1/BCL10/ZDHHC12/MAP3K7/CD14/RNF34/TNIP1/TREM2/TICAM2/AP3B1/NLRP3/ZDHHC18/PSPC1/AKT1/LILRA2/TICAM1/KLRK1/TLR10/MAPKAPK2/TRAF6/SIN3A/LSM14A/KLRC3/TLR3/BANF1/IRAK2/FYN/SARM1/PHB2/NOP53/RAB11FIP2/HSPA1A/MFHAS1/RNF125/CHUK/FCRL3/C1QBP/TOMM70/HSPA1B/MAPKAPK3/PHB1/KLRC4/PPP6C/BRCC3/FCN3/LRCH4/TKFC/LRRC14/KLRC2/TIRAP

# NFKAPPB: STAT1/CD74/HLA-DRB1/TMSB4X/TRIM22/TNFSF10/APOL3/BIRC3/TNFAIP3/SLC20A1/ALPK1/TLR4/CLEC7A/CASP1/TLR2/CASP8/LGALS9/PRDX1/PRKCB/LILRB4/MYD88/IKBKE/CD4/SLC39A8/SECTM1/CARD16/TLR8/UBD/RHOH/CD40/TRAF3/BST2/TIFA/TNF/IRAK1/RASSF2/TMEM106A/TLR6/TANK/PYCARD/LITAF/TRIM38/NOD2/TGFB1/RBCK1/ARRB2/SIRPA/CARD9/CARD8/TRIM25/RHOA/NFKBIA/ZC3H12A/CASP10/RELB/NUP62/IL1B/NFKB1/PLCG2/ADAM8/HMOX1/IRAK4/RIPK2/NFKB2/TLR7/TBK1/TRIM21/FASLG/FKBP1A/ECT2/GAPDH/IFIT5/TRAF2/CLEC4D/S100A4/TNIP3/BCL3/TNIP2/BIRC2/REL/TAB3/TFG/NLRC3/IKBKB/LTB/XIAP/TNFRSF1A/CX3CL1/TRIM5/LIMS1/MALT1/TRAF1/MTDH/TFRC/NEK6/RIPK1/CUL1/ATP2C1/USP10/LGALS1/TRIM13/TRADD/PDPK1/RELA/OPTN/PPM1B/NFAT5/TNFRSF10B/TAB2/NOD1/NKIRAS2/PARP1/BCL10/DHX15/MIER1/MAP3K7/ZNF675/TNIP1/PIM2/HDAC1/DDX21/TREM2/TRIM59/TICAM2/SHISA5/GOLT1B/ZDHHC13/F2R/AKT1/TICAM1/IKBKG/TRAF5/NDFIP1/CANT1/TRAF6/RBX1/CARD11/ZFAND6/SLC35B2/TLR3/CCL19/IRAK2/FLOT2/GPR89B/UBE2N/GPR89B/NFKBIB/USP20/CHUK/LAMTOR5/CX3CR1/CARD19/UBE2V1/FBXW11/EDN1/TMEM9B/FADD/SUMO4/CCR7/JMJD8/NDFIP2/TIRAP/TMEM101</code></pre>
<pre class="r"><code># Patient AM 17
DEGenesAM17 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 17&quot;)


egoAM17 &lt;- enrichGO(gene = unique(DEGenesAM17$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM17 &lt;- setReadable(egoAM17, OrgDb = org.Hs.eg.db)
dotplot(egoAM17, showCategory= 30, title = &quot;Patient AM 17 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-20-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM17, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 17 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-20-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM17 &lt;- enrichGO(gene = unique(DEGenesAM17$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM17 &lt;- setReadable(egoAM17, OrgDb = org.Hs.eg.db)
dotplot(egoAM17, showCategory=20, title = &quot;Patient AM 17 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-20-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM17 &lt;- enrichGO(gene = unique(DEGenesAM17$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM17 &lt;- setReadable(egoAM17, OrgDb = org.Hs.eg.db)
#View(egoAM17@result)
dotplot(egoAM17, showCategory=20, title = &quot;Patient AM 17 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-20-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># IGHG3/IGKC/IGHG1
# IGHM/IGKC
# presynaptic: GPM6A/LPAR2/KCNJ3/OPRD1
# voltage gated channels: KCNJ3/CNTNAP2/KCNG1/SCNN1A</code></pre>
<pre class="r"><code># Patient AM 18
DEGenesAM18 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;AM 18&quot;)


egoAM18 &lt;- enrichGO(gene = unique(DEGenesAM18$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM18 &lt;- setReadable(egoAM18, OrgDb = org.Hs.eg.db)
dotplot(egoAM18, showCategory=30, title = &quot;Patient AM 18 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-21-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoAM18, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient AM 18 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-21-2.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM18 &lt;- enrichGO(gene = unique(DEGenesAM18$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM18 &lt;- setReadable(egoAM18, OrgDb = org.Hs.eg.db)
dotplot(egoAM18, showCategory=20, title = &quot;Patient AM 18 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-21-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoAM18 &lt;- enrichGO(gene = unique(DEGenesAM18$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM18 &lt;- setReadable(egoAM18, OrgDb = org.Hs.eg.db)
dotplot(egoAM18, showCategory=20, title = &quot;Patient AM 18 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-21-4.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient CMP 01
DEGenesCMP1 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 01&quot;)


egoCMP1 &lt;- enrichGO(gene = unique(DEGenesCMP1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP1 &lt;- setReadable(egoCMP1, OrgDb = org.Hs.eg.db)
dotplot(egoCMP1, showCategory=30, title = &quot;Patient CMP 01 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/CMP%2001-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP1 &lt;- enrichGO(gene = unique(DEGenesCMP1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP1 &lt;- setReadable(egoCMP1, OrgDb = org.Hs.eg.db)
dotplot(egoCMP1, showCategory=20, title = &quot;Patient CMP 01 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/CMP%2001-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP1 &lt;- enrichGO(gene = unique(DEGenesCMP1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP1 &lt;- setReadable(egoCMP1, OrgDb = org.Hs.eg.db)
dotplot(egoCMP1, showCategory=20, title = &quot;Patient CMP 01 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/CMP%2001-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient CMP 02
DEGenesCMP2 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 02&quot;)


egoCMP2 &lt;- enrichGO(gene = unique(DEGenesCMP2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP2 &lt;- setReadable(egoCMP2, OrgDb = org.Hs.eg.db)
dotplot(egoCMP2, showCategory=30, title = &quot;Patient CMP 02 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-22-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP2, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 02 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-22-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP2 &lt;- enrichGO(gene = unique(DEGenesCMP2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP2 &lt;- setReadable(egoCMP2, OrgDb = org.Hs.eg.db)
dotplot(egoCMP2, showCategory=20, title = &quot;Patient CMP 02 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-22-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP2 &lt;- enrichGO(gene = unique(DEGenesCMP2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP2 &lt;- setReadable(egoCMP2, OrgDb = org.Hs.eg.db)
dotplot(egoCMP2, showCategory=20, title = &quot;Patient CMP 02 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-22-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># transmembrane receptor signaling, TGFbeta and BMP pathway</code></pre>
<pre class="r"><code># Patient CMP 03
DEGenesCMP3 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 03&quot;)


egoCMP3 &lt;- enrichGO(gene = unique(DEGenesCMP3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP3 &lt;- setReadable(egoCMP3, OrgDb = org.Hs.eg.db)
#View(egoCMP3@result)

dotplot(egoCMP3, showCategory=30, title = &quot;Patient CMP 03 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-23-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP3, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 03 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-23-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP3 &lt;- enrichGO(gene = unique(DEGenesCMP3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP3 &lt;- setReadable(egoCMP3, OrgDb = org.Hs.eg.db)
dotplot(egoCMP3, showCategory=20, title = &quot;Patient CMP 03 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-23-3.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP3 &lt;- enrichGO(gene = unique(DEGenesCMP3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP3 &lt;- setReadable(egoCMP3, OrgDb = org.Hs.eg.db)
dotplot(egoCMP3, showCategory=20, title = &quot;Patient CMP 03 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-23-4.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># response to peptide hormone: GCGR/TGFBR3/IRS2/CYTB/TNS2/CDO1/ADM/DDR2/INSR/FBN1/SORL1/COL3A1/LPIN1/TRIM72/STK11/AGTRAP/TBC1D4/PID1/OGT/NCOA1/EIF4EBP2/SESN3/AGTR1/EDNRB/HMGCS2/ARID1B/CFLAR/IRS1/PXN/IGF1R/COL6A1/MAPK14/ZBTB7B/MAPK3/STAT6/CTSD/FOXO1/C2CD5/SLC39A14/PDK4/PIP4K2A/CTSK/COL1A1/SLC27A1/STAT5B/PDK2/SCAP/PHIP/HDAC5/BCAR1/PIK3CA/PIK3R2/GHR/MAPK1/SOS2/CAMK2A/EREG/RANGAP1/CSK/JAK3/KLF15/TLR2/RHOQ/TSC1/LPIN3/DENND4C

# response to ROS: POD/HBA2/HBA1/HBB/FBLN5/MMP9/ND5/LRRK2/DDR2/MAP1LC3A/FOXO3/MMP2/BNIP3/PDGFRA/BCL2/SESN3/ANKZF1/PTPRK/PXN/TOP2B/COL6A1/KDM6B/MAPK3/STAT6/TXNRD2/ZNF580/FOXP1/COL1A1/TXNIP/PDK2/IL18RAP/EGFR/FXN/MAPK1/CCS/MAP3K5/ZNF277/MAP2K4/SETX/OSER1

# insulin pathway: IRS2/TNS2/ADM/INSR/SORL1/LPIN1/TRIM72/TBC1D4/PID1/OGT/NCOA1/EIF4EBP2/SESN3/HMGCS2/CFLAR/IRS1/IGF1R/COL6A1/MAPK14/ZBTB7B/MAPK3/CTSD/FOXO1/C2CD5/SLC39A14/PDK4/PIP4K2A/CTSK/COL1A1/SLC27A1/PDK2/SCAP/PHIP/HDAC5/BCAR1/PIK3CA/PIK3R2/GHR/MAPK1/SOS2/KLF15/TLR2/RHOQ/TSC1/LPIN3/DENND4C

# glucose homeostasis:  GCGR/FFAR2/SERPINF1/IRS2/NOX4/INSR/FOXO3/ZBTB20/STK11/OGT/PIM3/MLXIPL/FOSL2/SESN3/PLA2G6/NEUROD1/IRS1/NADK/IGF1R/CRTC2/FOXO1/LRRC8A/PFKM/SLC39A14/PDK4/COL1A1/PDK2/SLC12A6/RAB11B/ASPSCR1/PIK3CA/PIK3R2/DHPS/ALOX5/PIH1D1/KLF15

# Fatty acid: ACSM3/CYP4F3/CES1/AKR1C2/ALDH3A2/ECHDC2/IRS2/ACSL1/GGT5/C3/CYP4B1/NR1H3/GPX4/ACADL/ABHD5/LPIN1/ALOX5AP/AKR1C1/PTGR3/MLXIPL/BDH2/PLIN5/ACSS1/ABCD2/ACADVL/ACADSB/ADH5/IRS1/CPT1B/GCDH/PTGIS/PLA2G5/MAPK14/PTGDS/CYP4Z1/ACACB/PDK4/ABCB11/SLC27A1/PDK2/SCAP/FADS3/PLA2G15/ETFDH/ABHD2/SOX9/ABCD1/FMO4/ALOX5/AACS/GPAT4/PTGR1/PRKAG2/ABCD3/LPIN3

# organic acid metabolism: CYP4F3/ALDH1L1/ECHDC2/IRS2/AASS/CDO1/HAL/MPST/ACADL/GLUL/LPIN1/BDH2/PLIN5/ABCD2/ALDH6A1/ACADVL/ACADSB/BCKDHB/IRS1/BCAT2/ENOSF1/CPT1B/GCDH/PGD/DCXR/ACACB/ABCB11/ARG1/CBS/DDO/PLA2G15/ETFDH/ABHD2/GPT2/ABCD1/DBT/ABCD3/LPIN3/PPM1K</code></pre>
<pre class="r"><code># Patient CMP 04
DEGenesCMP4 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 04&quot;)


egoCMP4 &lt;- enrichGO(gene = unique(DEGenesCMP4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP4 &lt;- setReadable(egoCMP4, OrgDb = org.Hs.eg.db)
dotplot(egoCMP4, showCategory=30, title = &quot;Patient CMP 04 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-24-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP4, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 04 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-24-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP4 &lt;- enrichGO(gene = unique(DEGenesCMP4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP4 &lt;- setReadable(egoCMP4, OrgDb = org.Hs.eg.db)
dotplot(egoCMP4, showCategory=20, title = &quot;Patient CMP 04 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-24-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP4 &lt;- enrichGO(gene = unique(DEGenesCMP4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP4 &lt;- setReadable(egoCMP4, OrgDb = org.Hs.eg.db)
dotplot(egoCMP4, showCategory=20, title = &quot;Patient CMP 04 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-24-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># angiogenesis, lymph vessel development</code></pre>
<pre class="r"><code># Patient CMP 05
DEGenesCMP5 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 05&quot;)


egoCMP5 &lt;- enrichGO(gene = unique(DEGenesCMP5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP5 &lt;- setReadable(egoCMP5, OrgDb = org.Hs.eg.db)
dotplot(egoCMP5, showCategory=30, title = &quot;Patient CMP 05 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-25-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP5, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 05 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-25-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP5 &lt;- enrichGO(gene = unique(DEGenesCMP5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP5 &lt;- setReadable(egoCMP5, OrgDb = org.Hs.eg.db)
dotplot(egoCMP5, showCategory=20, title = &quot;Patient CMP 05 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-25-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP5 &lt;- enrichGO(gene = unique(DEGenesCMP5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP5 &lt;- setReadable(egoCMP5, OrgDb = org.Hs.eg.db)
dotplot(egoCMP5, showCategory=20, title = &quot;Patient CMP 05 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-25-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient CMP 06
DEGenesCMP6 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 06&quot;)


egoCMP6 &lt;- enrichGO(gene = unique(DEGenesCMP6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP6 &lt;- setReadable(egoCMP6, OrgDb = org.Hs.eg.db)
dotplot(egoCMP6, showCategory=30, title = &quot;Patient CMP 06 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-26-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP6, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 06 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-26-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP6 &lt;- enrichGO(gene = unique(DEGenesCMP6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP6 &lt;- setReadable(egoCMP6, OrgDb = org.Hs.eg.db)
dotplot(egoCMP6, showCategory=20, title = &quot;Patient CMP 06 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-26-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#response to peptide, RNA splicing, insulin, circadian rhythm, </code></pre>
<pre class="r"><code>egoCMP6 &lt;- enrichGO(gene = unique(DEGenesCMP6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP6 &lt;- setReadable(egoCMP6, OrgDb = org.Hs.eg.db)
dotplot(egoCMP6, showCategory=20, title = &quot;Patient CMP 06 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient CMP 07
DEGenesCMP7 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 07&quot;)


egoCMP7 &lt;- enrichGO(gene = unique(DEGenesCMP7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP7 &lt;- setReadable(egoCMP7, OrgDb = org.Hs.eg.db)
dotplot(egoCMP7, showCategory=30, title = &quot;Patient CMP 07 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-28-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP7, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 07 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-28-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP7 &lt;- enrichGO(gene = unique(DEGenesCMP7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP7 &lt;- setReadable(egoCMP7, OrgDb = org.Hs.eg.db)
dotplot(egoCMP7, showCategory=20, title = &quot;Patient CMP 07 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-28-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP7 &lt;- enrichGO(gene = unique(DEGenesCMP7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP7 &lt;- setReadable(egoCMP7, OrgDb = org.Hs.eg.db)
dotplot(egoCMP7, showCategory=20, title = &quot;Patient CMP 07 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-28-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient CMP 08
DEGenesCMP8 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 08&quot;)


egoCMP8 &lt;- enrichGO(gene = unique(DEGenesCMP8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP8 &lt;- setReadable(egoCMP8, OrgDb = org.Hs.eg.db)
dotplot(egoCMP8, showCategory=30, title = &quot;Patient CMP 08 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-29-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP8, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 08 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-29-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP8 &lt;- enrichGO(gene = unique(DEGenesCMP8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP8 &lt;- setReadable(egoCMP8, OrgDb = org.Hs.eg.db)
dotplot(egoCMP8, showCategory=20, title = &quot;Patient CMP 08 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-29-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP8 &lt;- enrichGO(gene = unique(DEGenesCMP8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP8 &lt;- setReadable(egoCMP8, OrgDb = org.Hs.eg.db)
dotplot(egoCMP8, showCategory=20, title = &quot;Patient CMP 08 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-29-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient CMP 09
DEGenesCMP9 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 09&quot;)


egoCMP9 &lt;- enrichGO(gene = unique(DEGenesCMP9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP9 &lt;- setReadable(egoCMP9, OrgDb = org.Hs.eg.db)
dotplot(egoCMP9, showCategory=30, title = &quot;Patient CMP 09 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-30-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP9, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 09 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-30-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP9 &lt;- enrichGO(gene = unique(DEGenesCMP9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP9 &lt;- setReadable(egoCMP9, OrgDb = org.Hs.eg.db)
dotplot(egoCMP9, showCategory=20, title = &quot;Patient CMP 09 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-30-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP9 &lt;- enrichGO(gene = unique(DEGenesCMP9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP9 &lt;- setReadable(egoCMP9, OrgDb = org.Hs.eg.db)
dotplot(egoCMP9, showCategory=20, title = &quot;Patient CMP 09 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/Patient%20CMP%2009-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#View(egoCMP9@result)

# platelet alpha granule: CFD/SPARC/IGF1/CD9/MMRN1/CLU/APP/APLP2/MAGED2/ISLR/PCDH7/THBS2/SYTL4/PROS1/SELP/HGF/PHACTR2/VEGFC/CD109

# postsynaptic membrane: LRRC4C/GRID2/PRRT2/SSPN/ITGA8/FAIM2/NECTIN3/SLC6A1/CNKSR2/ADGRB3/EFNB2/CHRM2/DLG2/FXYD6/NRP2/KCNC4/IQSEC3/TIAM1/LRFN5/ATP6AP2/GABRE/KCTD12/GRM3/RNF10/SLC6A4/PCDH17/IL1RAPL1/ERBB4/P2RY1/ANK2/TRAPPC4/NLGN4X/HIP1/LRRC7/LRRTM3/CHRNB1/TRPV1/ABHD17B/COL13A1/PTPRT/SHISA6</code></pre>
<pre class="r"><code># Patient CMP 10
DEGenesCMP10 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 10&quot;)


egoCMP10 &lt;- enrichGO(gene = unique(DEGenesCMP10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP10 &lt;- setReadable(egoCMP10, OrgDb = org.Hs.eg.db)
dotplot(egoCMP10, showCategory=30, title = &quot;Patient CMP 10 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP10, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 10 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-31-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># BMP signaling pathway!</code></pre>
<pre class="r"><code>egoCMP10 &lt;- enrichGO(gene = unique(DEGenesCMP10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP10 &lt;- setReadable(egoCMP10, OrgDb = org.Hs.eg.db)
dotplot(egoCMP10, showCategory=20, title = &quot;Patient CMP 10 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP10 &lt;- enrichGO(gene = unique(DEGenesCMP10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP10 &lt;- setReadable(egoCMP10, OrgDb = org.Hs.eg.db)
dotplot(egoCMP10, showCategory=20, title = &quot;Patient CMP 10 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-32-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient CMP 11
DEGenesCMP11 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 11&quot;)


egoCMP11 &lt;- enrichGO(gene = unique(DEGenesCMP11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP11 &lt;- setReadable(egoCMP11, OrgDb = org.Hs.eg.db)
dotplot(egoCMP11, showCategory=30, title = &quot;Patient CMP 11 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-33-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP11, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 11 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-33-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP11 &lt;- enrichGO(gene = unique(DEGenesCMP11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP11 &lt;- setReadable(egoCMP11, OrgDb = org.Hs.eg.db)
dotplot(egoCMP11, showCategory=20, title = &quot;Patient CMP 11 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-33-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP11 &lt;- enrichGO(gene = unique(DEGenesCMP11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP11 &lt;- setReadable(egoCMP11, OrgDb = org.Hs.eg.db)
dotplot(egoCMP11, showCategory=20, title = &quot;Patient CMP 11 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-33-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient CMP 12
DEGenesCMP12 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;CMP 12&quot;)


egoCMP12 &lt;- enrichGO(gene = unique(DEGenesCMP12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP12 &lt;- setReadable(egoCMP12, OrgDb = org.Hs.eg.db)
dotplot(egoCMP12, showCategory=30, title = &quot;Patient CMP 12 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-34-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoCMP12, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Patient CMP 12 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-34-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#View(egoCMP12@result)

egoCMP12 &lt;- enrichGO(gene = unique(DEGenesCMP12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP12 &lt;- setReadable(egoCMP12, OrgDb = org.Hs.eg.db)
dotplot(egoCMP12, showCategory=20, title = &quot;Patient CMP 12 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-34-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoCMP12 &lt;- enrichGO(gene = unique(DEGenesCMP12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP12 &lt;- setReadable(egoCMP12, OrgDb = org.Hs.eg.db)
dotplot(egoCMP12, showCategory=20, title = &quot;Patient CMP 12 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-34-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient CMP 13
# p adjusted not significant</code></pre>
<pre class="r"><code># Patient Sarcoidosis 01
DEGenesSarco1 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;Sarcoidosis 01&quot;)


egoSarco1 &lt;- enrichGO(gene = unique(DEGenesSarco1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco1 &lt;- setReadable(egoSarco1, OrgDb = org.Hs.eg.db)
dotplot(egoSarco1, showCategory=20, title = &quot;Patient Sarcoidosis 01 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-35-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoSarco1, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Patient Sarcoidosis 01 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-35-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoSarco1 &lt;- enrichGO(gene = unique(DEGenesSarco1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco1 &lt;- setReadable(egoSarco1, OrgDb = org.Hs.eg.db)
dotplot(egoSarco1, showCategory=20, title = &quot;Patient Sarcoidosis 01 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-35-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoSarco1 &lt;- enrichGO(gene = unique(DEGenesSarco1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco1 &lt;- setReadable(egoSarco1, OrgDb = org.Hs.eg.db)
dotplot(egoSarco1, showCategory=20, title = &quot;Patient Sarcoidosis 01 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-35-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># rRNA processing</code></pre>
<pre class="r"><code># Patient Sarcoidosis 02
DEGenesSarco2 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;Sarcoidosis 02&quot;)


egoSarco2 &lt;- enrichGO(gene = unique(DEGenesSarco2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco2 &lt;- setReadable(egoSarco2, OrgDb = org.Hs.eg.db)
dotplot(egoSarco2, showCategory=20, title = &quot;Patient Sarcoidosis 02 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-36-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoSarco2, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Patient Sarcoidosis 02 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-36-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoSarco2 &lt;- enrichGO(gene = unique(DEGenesSarco2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco2 &lt;- setReadable(egoSarco2, OrgDb = org.Hs.eg.db)
dotplot(egoSarco2, showCategory=20, title = &quot;Patient Sarcoidosis 02 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-36-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoSarco2 &lt;- enrichGO(gene = unique(DEGenesSarco2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco2 &lt;- setReadable(egoSarco2, OrgDb = org.Hs.eg.db)
dotplot(egoSarco2, showCategory=20, title = &quot;Patient Sarcoidosis 02 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-36-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># angiogenesis, adhesion, TGFbeta, SMAD</code></pre>
<pre class="r"><code># Patient Healthy 01
DEGenesHealthy1 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;Healthy 01&quot;)


egoHealthy1 &lt;- enrichGO(gene = unique(DEGenesHealthy1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy1 &lt;- setReadable(egoHealthy1, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy1, showCategory=20, title = &quot;Patient Healthy 01 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-37-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoHealthy1, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Patient Healthy 02 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-37-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy1 &lt;- enrichGO(gene = unique(DEGenesHealthy1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy1 &lt;- setReadable(egoHealthy1, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy1, showCategory=20, title = &quot;Patient Healthy 01 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-37-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy1 &lt;- enrichGO(gene = unique(DEGenesHealthy1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy1 &lt;- setReadable(egoHealthy1, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy1, showCategory=20, title = &quot;Patient Healthy 01 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-37-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient Healthy 02
DEGenesHealthy2 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;Healthy 02&quot;)


egoHealthy2 &lt;- enrichGO(gene = unique(DEGenesHealthy2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy2 &lt;- setReadable(egoHealthy2, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy2, showCategory=20, title = &quot;Patient Healthy 02 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-38-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoHealthy2, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Patient Healthy 02 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-38-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy2 &lt;- enrichGO(gene = unique(DEGenesHealthy2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy2 &lt;- setReadable(egoHealthy2, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy2, showCategory=20, title = &quot;Patient Healthy 02 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-38-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy2 &lt;- enrichGO(gene = unique(DEGenesHealthy2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy2 &lt;- setReadable(egoHealthy2, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy2, showCategory=20, title = &quot;Patient Healthy 02 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-38-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient Healthy 03
DEGenesHealthy3 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;Healthy 03&quot;)


egoHealthy3 &lt;- enrichGO(gene = unique(DEGenesHealthy3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy3 &lt;- setReadable(egoHealthy3, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy3, showCategory=20, title = &quot;Patient Healthy 03 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-39-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoHealthy3, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Patient Healthy 03 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-39-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy3 &lt;- enrichGO(gene = unique(DEGenesHealthy3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy3 &lt;- setReadable(egoHealthy3, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy3, showCategory=20, title = &quot;Patient Healthy 03 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-39-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy3 &lt;- enrichGO(gene = unique(DEGenesHealthy3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy3 &lt;- setReadable(egoHealthy3, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy3, showCategory=20, title = &quot;Patient Healthy 03 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-39-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient Healthy 04
DEGenesHealthy4 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;Healthy 04&quot;)


egoHealthy4 &lt;- enrichGO(gene = unique(DEGenesHealthy4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy4 &lt;- setReadable(egoHealthy4, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy4, showCategory=20, title = &quot;Patient Healthy 04 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-40-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoHealthy4, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Patient Healthy 04 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-40-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy4 &lt;- enrichGO(gene = unique(DEGenesHealthy4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy4 &lt;- setReadable(egoHealthy4, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy4, showCategory=20, title = &quot;Patient Healthy 04 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-40-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy4 &lt;- enrichGO(gene = unique(DEGenesHealthy4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy4 &lt;- setReadable(egoHealthy4, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy4, showCategory=20, title = &quot;Patient Healthy 04 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-40-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient Healthy 05
DEGenesHealthy5 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;Healthy 05&quot;)


egoHealthy5 &lt;- enrichGO(gene = unique(DEGenesHealthy5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy5 &lt;- setReadable(egoHealthy5, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy5, showCategory=20, title = &quot;Patient Healthy 05 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-41-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoHealthy5, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Patient Healthy 05 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-41-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy5 &lt;- enrichGO(gene = unique(DEGenesHealthy5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy5 &lt;- setReadable(egoHealthy5, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy5, showCategory=10, title = &quot;Patient Healthy 05 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-41-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy5 &lt;- enrichGO(gene = unique(DEGenesHealthy5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy5 &lt;- setReadable(egoHealthy5, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy5, showCategory=10, title = &quot;Patient Healthy 05 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-41-4.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient Healthy 06
DEGenesHealthy6 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;Healthy 06&quot;)


egoHealthy6 &lt;- enrichGO(gene = unique(DEGenesHealthy6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy6 &lt;- setReadable(egoHealthy6, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy6, showCategory=20, title = &quot;Patient Healthy 06 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-42-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoHealthy6, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Patient Healthy 06 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-42-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy6 &lt;- enrichGO(gene = unique(DEGenesHealthy6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy6 &lt;- setReadable(egoHealthy6, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy6, showCategory=20, title = &quot;Patient Healthy 06 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-42-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy6 &lt;- enrichGO(gene = unique(DEGenesHealthy6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy6 &lt;- setReadable(egoHealthy6, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy6, showCategory=20, title = &quot;Patient Healthy 06 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-42-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient Healthy 07
DEGenesHealthy7 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;Healthy 07&quot;)


egoHealthy7 &lt;- enrichGO(gene = unique(DEGenesHealthy7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy7 &lt;- setReadable(egoHealthy7, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy7, showCategory=20, title = &quot;Patient Healthy 07 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-43-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoHealthy7, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Patient Healthy 07 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-43-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy3 &lt;- enrichGO(gene = unique(DEGenesHealthy3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy7 &lt;- setReadable(egoHealthy7, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy7, showCategory=20, title = &quot;Patient Healthy 07 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-43-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy7 &lt;- enrichGO(gene = unique(DEGenesHealthy7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy7 &lt;- setReadable(egoHealthy7, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy7, showCategory=20, title = &quot;Patient Healthy 07 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-43-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Patient Healthy 08
DEGenesHealthy8 &lt;- DEGenes_patient %&gt;% filter(cluster == &quot;Healthy 08&quot;)


egoHealthy8 &lt;- enrichGO(gene = unique(DEGenesHealthy8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy8 &lt;- setReadable(egoHealthy8, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy8, showCategory=10, title = &quot;Patient Healthy 08 Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-44-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoHealthy8, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Patient Healthy 08 Biological Process&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-44-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy8 &lt;- enrichGO(gene = unique(DEGenesHealthy8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy8 &lt;- setReadable(egoHealthy8, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy8, showCategory=20, title = &quot;Patient Healthy 08 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-44-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoHealthy8 &lt;- enrichGO(gene = unique(DEGenesHealthy8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy8 &lt;- setReadable(egoHealthy8, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy8, showCategory=20, title = &quot;Patient Healthy 08 Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/DEgenes_patient_stratificationV3.Rmd/unnamed-chunk-44-4.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Healthy 08 is apparently not healthy, but has inflammatory signature (response LPS, innate, hypoxia)</code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Thu Nov 14 22:14:06 2024&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.7.1             pathview_1.44.0             wordcloud_2.6              
 [4] textshaping_0.4.0           EnhancedVolcano_1.22.0      ggrepel_0.9.6              
 [7] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
[10] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
[13] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_7.5.1              
[16] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[19] multtest_2.60.0             metap_1.11                  scater_1.32.1              
[22] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[25] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[28] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[31] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[34] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[37] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[40] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[43] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[46] matrixStats_1.4.1           pheatmap_1.0.12             ggpubr_0.6.0               
[49] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[52] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             presto_1.0.0              parallelly_1.38.0        
[145] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[148] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[151] pbapply_1.7-2             httr2_1.0.5               spam_2.11-0              
[154] gson_0.1.0                utf8_1.2.4                graphlayouts_1.2.0       
[157] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[160] shiny_1.9.1               GenomeInfoDbData_1.2.12   glmmTMB_1.1.10           
[163] R.utils_2.12.3            RCurl_1.98-1.16           memoise_2.0.1            
[166] rmarkdown_2.28            scales_1.3.0              R.methodsS3_1.8.2        
[169] future_1.34.0             RANN_2.6.2                Cairo_1.6-2              
[172] spatstat.data_3.1-2       rstudioapi_0.16.0         whisker_0.4.1            
[175] cluster_2.1.6             mutoss_0.1-13             spatstat.utils_3.1-0     
[178] hms_1.1.3                 fitdistrplus_1.2-1        munsell_0.5.1            
[181] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.4              
[184] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.0               
[187] dotCall64_1.2             shinydashboard_0.7.2      ggforce_0.4.2            
[190] laeken_0.5.3              mgcv_1.9-1                xfun_0.48                
[193] e1071_1.7-16              TH.data_1.1-2             iterators_1.0.14         
[196] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[199] ps_1.8.0                  futile.options_1.0.1      bitops_1.0-9             
[202] Rdpack_2.6.1              promises_1.3.0            scatterpie_0.2.4         
[205] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-1           
[208] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[211] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[214] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[217] Rcpp_1.0.13               edgeR_4.2.1               BiocSingular_1.20.0      
[220] tensor_1.5                MASS_7.3-61               progress_1.2.3           
[223] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.3-2    
[226] R6_2.5.1                  fastmap_1.2.0             multcomp_1.4-26          
[229] fastmatch_1.1-4           rstatix_0.7.2             vipor_0.4.7              
[232] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[235] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-19              
[238] gtable_0.3.5              KernSmooth_2.23-24        miniUI_0.1.1.1           
[241] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[244] bit64_4.5.2               spatstat.explore_3.3-2    lifecycle_1.0.4          
[247] blme_1.0-6                processx_3.8.4            callr_3.7.6              
[250] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[253] robustbase_0.99-4-1       spatstat.geom_3.3-3       sn_2.1.1                 
[256] ggfun_0.1.6               future.apply_1.11.2       bslib_0.8.0              
[259] pillar_1.9.0              gplots_3.2.0              pcaMethods_1.96.0        
[262] locfit_1.5-9.10           jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.7.1             pathview_1.44.0             wordcloud_2.6              
 [4] textshaping_0.4.0           EnhancedVolcano_1.22.0      ggrepel_0.9.6              
 [7] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
[10] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
[13] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_7.5.1              
[16] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[19] multtest_2.60.0             metap_1.11                  scater_1.32.1              
[22] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[25] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[28] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[31] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[34] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[37] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[40] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[43] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[46] matrixStats_1.4.1           pheatmap_1.0.12             ggpubr_0.6.0               
[49] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[52] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             presto_1.0.0              parallelly_1.38.0        
[145] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[148] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[151] pbapply_1.7-2             httr2_1.0.5               spam_2.11-0              
[154] gson_0.1.0                utf8_1.2.4                graphlayouts_1.2.0       
[157] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[160] shiny_1.9.1               GenomeInfoDbData_1.2.12   glmmTMB_1.1.10           
[163] R.utils_2.12.3            RCurl_1.98-1.16           memoise_2.0.1            
[166] rmarkdown_2.28            scales_1.3.0              R.methodsS3_1.8.2        
[169] future_1.34.0             RANN_2.6.2                Cairo_1.6-2              
[172] spatstat.data_3.1-2       rstudioapi_0.16.0         whisker_0.4.1            
[175] cluster_2.1.6             mutoss_0.1-13             spatstat.utils_3.1-0     
[178] hms_1.1.3                 fitdistrplus_1.2-1        munsell_0.5.1            
[181] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.4              
[184] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.0               
[187] dotCall64_1.2             shinydashboard_0.7.2      ggforce_0.4.2            
[190] laeken_0.5.3              mgcv_1.9-1                xfun_0.48                
[193] e1071_1.7-16              TH.data_1.1-2             iterators_1.0.14         
[196] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[199] ps_1.8.0                  futile.options_1.0.1      bitops_1.0-9             
[202] Rdpack_2.6.1              promises_1.3.0            scatterpie_0.2.4         
[205] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-1           
[208] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[211] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[214] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[217] Rcpp_1.0.13               edgeR_4.2.1               BiocSingular_1.20.0      
[220] tensor_1.5                MASS_7.3-61               progress_1.2.3           
[223] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.3-2    
[226] R6_2.5.1                  fastmap_1.2.0             multcomp_1.4-26          
[229] fastmatch_1.1-4           rstatix_0.7.2             vipor_0.4.7              
[232] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[235] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-19              
[238] gtable_0.3.5              KernSmooth_2.23-24        miniUI_0.1.1.1           
[241] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[244] bit64_4.5.2               spatstat.explore_3.3-2    lifecycle_1.0.4          
[247] blme_1.0-6                processx_3.8.4            callr_3.7.6              
[250] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[253] robustbase_0.99-4-1       spatstat.geom_3.3-3       sn_2.1.1                 
[256] ggfun_0.1.6               future.apply_1.11.2       bslib_0.8.0              
[259] pillar_1.9.0              gplots_3.2.0              pcaMethods_1.96.0        
[262] locfit_1.5-9.10           jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
