<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-09-19" />

<title>Fibroblasts Subgroups Preliminary</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myocarditis snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Fibroblasts Subgroups Preliminary</h1>
<h4 class="date">2024-09-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-11-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Myocarditis
snRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat
files/Myocarditis_allfiltered_12.11.24.rds
</td>
<td style="text-align:left;">
../data/01 Merged Seurat files/Myocarditis_allfiltered_12.11.24.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis
snRNAseq/docs/Markers/DEGenes_condition pos. and neg. AM
</td>
<td style="text-align:left;">
../docs/Markers/DEGenes_condition pos. and neg. AM
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenesFb
</td>
<td style="text-align:left;">
../docs/Markers/DEGenesFb
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Fb markers after
HH2_HH3 removal
</td>
<td style="text-align:left;">
../docs/Fb markers after HH2_HH3 removal
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis
snRNAseq/docs/Markers_dat_all_clusters_frequency_relative abundance_V2
</td>
<td style="text-align:left;">
../docs/Markers_dat_all_clusters_frequency_relative abundance_V2
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis
snRNAseq/docs/allFb_PeriFb_vsmc_int markers
</td>
<td style="text-align:left;">
../docs/allFb_PeriFb_vsmc_int markers
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree0583f93f70a80d38bd1f2a60b01fc6f80cf675d8targetblank0583f93a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/0583f93f70a80d38bd1f2a60b01fc6f80cf675d8" target="_blank">0583f93</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree0583f93f70a80d38bd1f2a60b01fc6f80cf675d8targetblank0583f93a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/0583f93f70a80d38bd1f2a60b01fc6f80cf675d8" target="_blank">0583f93</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    data/

Untracked files:
    Untracked:  analysis/Fibroblast_subset_V2.Rmd

Unstaged changes:
    Modified:   analysis/DE_Genes_Disease_Condition.Rmd
    Deleted:    analysis/Fibroblast subset V2 reclustering.Rmd
    Deleted:    analysis/Fibroblast-subset-V2-reclustering.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>##load merged file 
seuratM &lt;- readRDS(&quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_allfiltered_12.11.24.rds&quot;)


table(seuratM$dataset)</code></pre>
<pre><code>
     312581_05-5_20230308_Hu_nucseq_SG36_EMB_GEM      312581_06-6_20230308_Hu_nucseq_SG37_EMB_GEM 
                                            4323                                             1123 
   353921_28-28_20240529_Hu_nucseq_Immpath_EMB18    353921_29-29_20240529_Hu_nucseq_Immpath_EMB24 
                                             208                                              105 
   353921_30-30_20240529_Hu_nucseq_Immpath_EMB25    353921_31-31_20240529_Hu_nucseq_Immpath_EMB27 
                                             227                                              579 
   353921_32-32_20240529_Hu_nucseq_Immpath_EMB29    353921_33-33_20240529_Hu_nucseq_Immpath_EMB30 
                                             335                                              730 
   353921_34-34_20240529_Hu_nucseq_Immpath_EMB35    353921_35-35_20240529_Hu_nucseq_Immpath_EMB37 
                                            1217                                              604 
359861_09-11_20240723_Hu_nucseq_Ber_Pat5_EM979_1 359861_10-13_20240723_Hu_nucseq_Ber_Pat5_EM979_2 
                                             771                                             1476 
 359861_11-1_20240730_Hu_nucseq_Ber_Pat1_EMB1537  359861_12-2_20240730_Hu_nucseq_Ber_Pat2_EMB1041 
                                             316                                              608 
    359861_16-8_20240730_Hu_nucseq_Immpath_EMB47     359861_17-9_20240730_Hu_nucseq_Immpath_EMB48 
                                             513                                              426 
     o27533_1_12-12_20220203_Hu_nucseq_EMB31_GEM        o27936_1_7-7_20220309_Hu_nucseq_EMB32_GEM 
                                             236                                             1192 
 o28576_1_01-1_20220525_Hu_nucseq_Graz_1_EMB_GEM  o28576_1_02-2_20220525_Hu_nucseq_Graz_2_EMB_GEM 
                                            2740                                             1684 
 o28576_1_03-3_20220525_Hu_nucseq_Graz_3_EMB_GEM  o28576_1_04-4_20220525_Hu_nucseq_Graz_4_EMB_GEM 
                                            2396                                              545 
 o28576_1_05-5_20220525_Hu_nucseq_Graz_5_EMB_GEM  o28576_1_06-6_20220525_Hu_nucseq_Graz_6_EMB_GEM 
                                             781                                              491 
 o28576_1_07-7_20220525_Hu_nucseq_Graz_7_EMB_GEM   o28576_1_08-8_20220525_Hu_nucseq_Graz_8_HH_GEM 
                                             653                                             3921 
o28576_1_12-12_20220525_Hu_nucseq_Graz_12_HH_GEM      o28576_1_13-13_20220525_Hu_nucseq_EMB32_GEM 
                                            3818                                             1428 
    o292731_1-1_20220818_Hu_nucseq_Graz_9_HH_GEM    o292731_2-2_20220818_Hu_nucseq_Graz_13_HH_GEM 
                                            4908                                             9882 
    o292731_3-3_20220818_Hu_nucseq_SG_33_EMB_GEM     o292731_4-4_20220818_Hu_nucseq_SG_34_EMB_GEM 
                                            6286                                              620 
    o292731_5-5_20220818_Hu_nucseq_SG_35_EMB_GEM   o294781_01-1_20220912_Hu_nucseq_Graz_21_HH_GEM 
                                            2363                                             1442 
  o294781_02-2_20220912_Hu_nucseq_Graz_22_HH_GEM   o294781_03-3_20220912_Hu_nucseq_Graz_23_HH_GEM 
                                            1998                                              841 
  o294781_04-4_20220912_Hu_nucseq_Graz_24_HH_GEM  o294781_05-5_20220912_Hu_nucseq_Graz_14_EMB_GEM 
                                            1480                                             1268 
 o294781_06-6_20220912_Hu_nucseq_Graz_15_EMB_GEM  o294781_09-9_20220912_Hu_nucseq_Graz_18_EMB_GEM 
                                            4439                                             2280 
o294781_10-10_20220912_Hu_nucseq_Graz_19_EMB_GEM o294781_11-11_20220912_Hu_nucseq_Graz_20_EMB_GEM 
                                             111                                             2706 </code></pre>
<pre class="r"><code>table(seuratM$dataset)</code></pre>
<pre><code>
     312581_05-5_20230308_Hu_nucseq_SG36_EMB_GEM      312581_06-6_20230308_Hu_nucseq_SG37_EMB_GEM 
                                            4323                                             1123 
   353921_28-28_20240529_Hu_nucseq_Immpath_EMB18    353921_29-29_20240529_Hu_nucseq_Immpath_EMB24 
                                             208                                              105 
   353921_30-30_20240529_Hu_nucseq_Immpath_EMB25    353921_31-31_20240529_Hu_nucseq_Immpath_EMB27 
                                             227                                              579 
   353921_32-32_20240529_Hu_nucseq_Immpath_EMB29    353921_33-33_20240529_Hu_nucseq_Immpath_EMB30 
                                             335                                              730 
   353921_34-34_20240529_Hu_nucseq_Immpath_EMB35    353921_35-35_20240529_Hu_nucseq_Immpath_EMB37 
                                            1217                                              604 
359861_09-11_20240723_Hu_nucseq_Ber_Pat5_EM979_1 359861_10-13_20240723_Hu_nucseq_Ber_Pat5_EM979_2 
                                             771                                             1476 
 359861_11-1_20240730_Hu_nucseq_Ber_Pat1_EMB1537  359861_12-2_20240730_Hu_nucseq_Ber_Pat2_EMB1041 
                                             316                                              608 
    359861_16-8_20240730_Hu_nucseq_Immpath_EMB47     359861_17-9_20240730_Hu_nucseq_Immpath_EMB48 
                                             513                                              426 
     o27533_1_12-12_20220203_Hu_nucseq_EMB31_GEM        o27936_1_7-7_20220309_Hu_nucseq_EMB32_GEM 
                                             236                                             1192 
 o28576_1_01-1_20220525_Hu_nucseq_Graz_1_EMB_GEM  o28576_1_02-2_20220525_Hu_nucseq_Graz_2_EMB_GEM 
                                            2740                                             1684 
 o28576_1_03-3_20220525_Hu_nucseq_Graz_3_EMB_GEM  o28576_1_04-4_20220525_Hu_nucseq_Graz_4_EMB_GEM 
                                            2396                                              545 
 o28576_1_05-5_20220525_Hu_nucseq_Graz_5_EMB_GEM  o28576_1_06-6_20220525_Hu_nucseq_Graz_6_EMB_GEM 
                                             781                                              491 
 o28576_1_07-7_20220525_Hu_nucseq_Graz_7_EMB_GEM   o28576_1_08-8_20220525_Hu_nucseq_Graz_8_HH_GEM 
                                             653                                             3921 
o28576_1_12-12_20220525_Hu_nucseq_Graz_12_HH_GEM      o28576_1_13-13_20220525_Hu_nucseq_EMB32_GEM 
                                            3818                                             1428 
    o292731_1-1_20220818_Hu_nucseq_Graz_9_HH_GEM    o292731_2-2_20220818_Hu_nucseq_Graz_13_HH_GEM 
                                            4908                                             9882 
    o292731_3-3_20220818_Hu_nucseq_SG_33_EMB_GEM     o292731_4-4_20220818_Hu_nucseq_SG_34_EMB_GEM 
                                            6286                                              620 
    o292731_5-5_20220818_Hu_nucseq_SG_35_EMB_GEM   o294781_01-1_20220912_Hu_nucseq_Graz_21_HH_GEM 
                                            2363                                             1442 
  o294781_02-2_20220912_Hu_nucseq_Graz_22_HH_GEM   o294781_03-3_20220912_Hu_nucseq_Graz_23_HH_GEM 
                                            1998                                              841 
  o294781_04-4_20220912_Hu_nucseq_Graz_24_HH_GEM  o294781_05-5_20220912_Hu_nucseq_Graz_14_EMB_GEM 
                                            1480                                             1268 
 o294781_06-6_20220912_Hu_nucseq_Graz_15_EMB_GEM  o294781_09-9_20220912_Hu_nucseq_Graz_18_EMB_GEM 
                                            4439                                             2280 
o294781_10-10_20220912_Hu_nucseq_Graz_19_EMB_GEM o294781_11-11_20220912_Hu_nucseq_Graz_20_EMB_GEM 
                                             111                                             2706 </code></pre>
<pre class="r"><code>table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11    12    13    14 
17699 13639 12754  8222  5475  4836  3723  2365  2156   905   882   694   380   249    91 </code></pre>
<pre class="r"><code>table(seuratM$patient)</code></pre>
<pre><code>
    Healthy 01     Healthy 02     Healthy 03     Healthy 04     Healthy 05     Healthy 06 
          3921           3818           4908           9882           1442           1998 
    Healthy 07     Healthy 08          AM 01          AM 02          AM 03          AM 04 
           841           1480           6286           4323           1123            208 
         AM 05          AM 06          AM 07          AM 08          AM 09          AM 10 
           579            335           1217            604            513            426 
         AM 11          AM 12          AM 13          AM 14          AM 15          AM 16 
           236           1192           2740           1684           4439           2247 
         AM 17          AM 18         CMP 01         CMP 02         CMP 03         CMP 04 
           316            608            105            227            730           1428 
        CMP 05         CMP 06         CMP 07         CMP 08         CMP 09         CMP 10 
           620            653           1268           2280           2396            545 
        CMP 11         CMP 12         CMP 13 Sarcoidosis 01 Sarcoidosis 02 
           781            491            111           2363           2706 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
74070 </code></pre>
<pre class="r"><code>#### Rename clusters

seuratM$clusterName &lt;- &quot;clusterName&quot;
Idents(seuratM) &lt;- seuratM$clusterName
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;0&quot; )] &lt;- &quot;Fb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;1&quot; )] &lt;- &quot;BEC1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;2&quot; )] &lt;- &quot;PeriFb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;3&quot; )] &lt;- &quot;Cardiomyocyte&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;4&quot; )] &lt;- &quot;MCP1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;5&quot; )] &lt;- &quot;TC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;6&quot; )] &lt;- &quot;MCP2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;7&quot; )] &lt;- &quot;BEC2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;8&quot; )] &lt;- &quot;VSMC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;9&quot; )] &lt;- &quot;NC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;10&quot; )] &lt;- &quot;INT1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;11&quot; )] &lt;- &quot;Fb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;12&quot; )] &lt;- &quot;Fb3&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;13&quot; )] &lt;- &quot;PeriFb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;14&quot; )] &lt;- &quot;AdipoC&quot;


colclusterName &lt;- c(&quot;#D53E4F&quot;, &quot;#f4a582&quot;, &quot;#FEE08B&quot;,&quot;#feb24c&quot;,&quot;#67001f&quot;, &quot;#01665e&quot;,&quot;#66C2A5&quot;,&quot;#c7eae5&quot;,&quot;#BEAED4&quot;, &quot;#355C7D&quot;,&quot;#3288BD&quot;,&quot;#8c510a&quot; ,&quot;#fde0dd&quot;,&quot;#B45B5C&quot;,&quot;#dd1c77&quot;)

                    
names(colclusterName) &lt;- c(&quot;Cardiomyocyte&quot;,&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;,&quot;NC&quot;, &quot;BEC1&quot;,&quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;,&quot;MCP2&quot;,&quot;TC&quot;)

colpat_sub&lt;- c(&quot;#dfc27d&quot;,&quot;#BE3144&quot;,&quot;#355C7D&quot;, &quot;#779d8d&quot;)

seuratM$clusterName &lt;- factor(seuratM$clusterName, levels=c(&quot;Cardiomyocyte&quot;, &quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;INT1&quot;,&quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;, &quot;NC&quot;, &quot;BEC1&quot;, &quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;, &quot;MCP2&quot;, &quot;TC&quot;))


Idents(seuratM) &lt;- seuratM$clusterName
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE, cols = colclusterName)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># DE Genes 

DEGenesCluster &lt;- FindAllMarkers (seuratM, only.pos = TRUE, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.01)


#write.table(DEGenesCluster, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenes_condition pos. and neg. AM, Healthy, CMP, Sarcoidosis&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)


##adjust table
DEGenesCluster_adj &lt;- DEGenesCluster %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

#View(DEGenesCluster)</code></pre>
<pre class="r"><code>##GSEA for Fb2
DEGenesFb2 &lt;- DEGenesCluster_adj %&gt;% filter(cluster == &quot;Fb2&quot;)
#head(DEGenesFb2)

#write.table(DEGenesFb2, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenesFb&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

egoFb2 &lt;- enrichGO(gene = unique(DEGenesFb2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoFb2 &lt;- setReadable(egoFb2, OrgDb = org.Hs.eg.db)
dotplot(egoFb2, showCategory=20, title = &quot;Fibroblast 2 Cluster Biological Process&quot;, font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/Fibroblast%20Cluster%202-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Barplot Fb2
barplot(egoFb2, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Fibroblast 2 Cluster&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoFb2 &lt;- enrichGO(gene = unique(DEGenesFb2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;CC&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoFb2 &lt;- setReadable(egoFb2, OrgDb = org.Hs.eg.db)


dotplot(egoFb2, showCategory=20, title = &quot;Fibroblast 2 Cluster Cellular Component&quot;, font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-4-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>egoFb2 &lt;- enrichGO(gene = unique(DEGenesFb2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;MF&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoFb2 &lt;- setReadable(egoFb2, OrgDb = org.Hs.eg.db)
dotplot(egoFb2, showCategory=20, title = &quot;Fibroblast 2 Metabolic Function&quot;, font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-4-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##GSEA for Fb1
DEGenesFb1 &lt;- DEGenesCluster_adj %&gt;% filter(cluster == &quot;Fb1&quot;)
#head(DEGenesFb2)

egoFb1 &lt;- enrichGO(gene = unique(DEGenesFb1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoFb1 &lt;- setReadable(egoFb1, OrgDb = org.Hs.eg.db)
dotplot(egoFb1, showCategory=30, title = &quot;Fibroblast 1 Cluster&quot;, font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Barplot Fb1
barplot(egoFb1, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Fibroblast 1 Cluster&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##GSEA for Fb3
DEGenesFb3 &lt;- DEGenesCluster_adj %&gt;% filter(cluster == &quot;Fb3&quot;)
#head(DEGenesFb3)

egoFb3 &lt;- enrichGO(gene = unique(DEGenesFb3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoFb3 &lt;- setReadable(egoFb3, OrgDb = org.Hs.eg.db)
dotplot(egoFb3, showCategory=10, title = &quot;Fibroblast 3 Cluster&quot;, font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-7-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Barplot Fb3
barplot(egoFb3, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Fibroblast 3 Cluster&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-8-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##GSEA for Int non significant
##GSEA for PeriFb1
DEGenesPeriFb1 &lt;- DEGenesCluster_adj %&gt;% filter(cluster == &quot;PeriFb1&quot;)
#head(DEGenesPeriFb1)

egoPeriFb1 &lt;- enrichGO(gene = unique(DEGenesPeriFb1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoPeriFb1 &lt;- setReadable(egoPeriFb1, OrgDb = org.Hs.eg.db)
dotplot(egoPeriFb1, showCategory=10, title = &quot;Perivascular Fb 1&quot;, font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoPeriFb1, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Perivascular Fb 1&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-9-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##GSEA for PeriFb2
DEGenesPeriFb2 &lt;- DEGenesCluster_adj %&gt;% filter(cluster == &quot;PeriFb2&quot;)
#head(DEGenesPeriFb1)

egoPeriFb2 &lt;- enrichGO(gene = unique(DEGenesPeriFb2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoPeriFb2 &lt;- setReadable(egoPeriFb1, OrgDb = org.Hs.eg.db)
dotplot(egoPeriFb2, showCategory=10, title = &quot;Perivascular Fb 2&quot;, font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoPeriFb2, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;Perivascular Fb 2&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-10-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##GSEA for PeriFb2
DEGenesVsmc &lt;- DEGenesCluster_adj %&gt;% filter(cluster == &quot;VSMC&quot;)
#head(DEGenesVsmc)

egoVsmc &lt;- enrichGO(gene = unique(DEGenesVsmc$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoPeriVsmc &lt;- setReadable(egoVsmc, OrgDb = org.Hs.eg.db)
dotplot(egoPeriVsmc, showCategory=10, title = &quot;VSMC&quot;, font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(egoPeriVsmc, 
        drop = TRUE, 
        showCategory = 20, 
        title = &quot;VSMC&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-11-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DEGenesClusterall &lt;- FindAllMarkers(seuratM, only.pos = FALSE, logfc.threshold = 0.2) %&gt;%
  dplyr::filter(p_val_adj &lt; 0.01)


##adjust table
DEGenesClusterall_adj &lt;- DEGenesClusterall %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))</code></pre>
<pre class="r"><code>## Volcano Plot with pos. and neg. DE Genes Fb2

DEGenesFb2all &lt;- DEGenesClusterall_adj %&gt;% filter(cluster == &quot;Fb2&quot;)

head(DEGenesFb2all)</code></pre>
<pre><code>                          p_val avg_log2FC pct.1 pct.2 p_val_adj cluster    gene
ENSG00000154175.ABI3BP.2      0   3.305118 0.742 0.137         0     Fb2  ABI3BP
ENSG00000166147.FBN1.2        0   2.588820 0.774 0.180         0     Fb2    FBN1
ENSG00000117228.GBP1.2        0   2.877544 0.682 0.123         0     Fb2    GBP1
ENSG00000115414.FN1.2         0   3.175310 0.811 0.256         0     Fb2     FN1
ENSG00000102265.TIMP1.2       0   3.314412 0.599 0.116         0     Fb2   TIMP1
ENSG00000164877.MICALL2.2     0   2.357303 0.549 0.085         0     Fb2 MICALL2
                                             Gene           EnsID
ENSG00000154175.ABI3BP.2   ENSG00000154175.ABI3BP ENSG00000154175
ENSG00000166147.FBN1.2       ENSG00000166147.FBN1 ENSG00000166147
ENSG00000117228.GBP1.2       ENSG00000117228.GBP1 ENSG00000117228
ENSG00000115414.FN1.2         ENSG00000115414.FN1 ENSG00000115414
ENSG00000102265.TIMP1.2     ENSG00000102265.TIMP1 ENSG00000102265
ENSG00000164877.MICALL2.2 ENSG00000164877.MICALL2 ENSG00000164877</code></pre>
<pre class="r"><code>#View(DEGenesFb2all)

res_Fb2 &lt;- as.data.frame(DEGenesFb2all)

EnhancedVolcano(res_Fb2, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Fb2$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Fibroblast 2 Cluster&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 1.0,
    labSize = 5.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/Volcano%20Plot-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>EnhancedVolcano(res_Fb2, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_Fb2$gene,
                selectLab = c(&quot;CXCL6&quot;,&quot;STAT1&quot;, &quot;VCAN&quot;,  &quot;GBP1P1&quot;, &quot;GBP2&quot;, &quot;SRPX2&quot;, &quot;SLIT2&quot;, &quot;CCL19&quot;,&quot;CLSPN&quot;, &quot;SLIT2&quot;,&quot;GBP2&quot;, &quot;VDR&quot;, &quot;CCL11&quot;,&quot;CCL2&quot;, 
                             &quot;CXCL9&quot;, &quot;CXCL10&quot;, &quot;PDGFA&quot;, &quot;CXCL1&quot;, &quot;FN1&quot;,&quot;ABI3BP&quot;,&quot;FBN1&quot;,&quot;TIMP&quot;, &quot;GBP1&quot;, &quot;C1GALT1&quot;, &quot;EFEMP1&quot;, &quot;APOL1&quot;),
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Fibroblast 2 Cluster&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 1.0,
    labSize = 5.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-13-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Select only Fibroblast 2 Cluster
seuratFb2 &lt;- subset(seuratM, clusterName %in% &quot;Fb2&quot;)

table(seuratFb2$clusterName)</code></pre>
<pre><code>
Cardiomyocyte           Fb1           Fb2           Fb3          INT1       PeriFb1       PeriFb2 
            0             0           694             0             0             0             0 
         VSMC            NC          BEC1          BEC2        AdipoC          MCP1          MCP2 
            0             0             0             0             0             0             0 
           TC 
            0 </code></pre>
<pre class="r"><code>table(seuratFb2$patient)</code></pre>
<pre><code>
    Healthy 01     Healthy 02     Healthy 03     Healthy 04     Healthy 05     Healthy 06 
             3              0             40              3              0              1 
    Healthy 07     Healthy 08          AM 01          AM 02          AM 03          AM 04 
             0              5            303             15              0              1 
         AM 05          AM 06          AM 07          AM 08          AM 09          AM 10 
             0              0             28              0              0              1 
         AM 11          AM 12          AM 13          AM 14          AM 15          AM 16 
             0              2              3              0             29            228 
         AM 17          AM 18         CMP 01         CMP 02         CMP 03         CMP 04 
             2              2              0              0              0              1 
        CMP 05         CMP 06         CMP 07         CMP 08         CMP 09         CMP 10 
             3              0              1              0              7              2 
        CMP 11         CMP 12         CMP 13 Sarcoidosis 01 Sarcoidosis 02 
             2              0              0             10              2 </code></pre>
<pre class="r"><code>table(seuratFb2$orig.ident)</code></pre>
<pre><code>
    
694 </code></pre>
<pre class="r"><code># nuclei originate from AM 01 and AM 16 primarily</code></pre>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$clusterName

seuratFb &lt;- subset(seuratM, clusterName %in% c(&quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;))

table(seuratFb$patient)</code></pre>
<pre><code>
    Healthy 01     Healthy 02     Healthy 03     Healthy 04     Healthy 05     Healthy 06 
          1918           2275           2477           4202            912           1055 
    Healthy 07     Healthy 08          AM 01          AM 02          AM 03          AM 04 
           413            944            527           2468            594             66 
         AM 05          AM 06          AM 07          AM 08          AM 09          AM 10 
           333             99            724            371            194            243 
         AM 11          AM 12          AM 13          AM 14          AM 15          AM 16 
            98            585           1488            995           2727            389 
         AM 17          AM 18         CMP 01         CMP 02         CMP 03         CMP 04 
            99            221             48            139            387            651 
        CMP 05         CMP 06         CMP 07         CMP 08         CMP 09         CMP 10 
           262            363            767           1149           1365            230 
        CMP 11         CMP 12         CMP 13 Sarcoidosis 01 Sarcoidosis 02 
           426            225             49           1198           1138 </code></pre>
<pre class="r"><code>levels(seuratFb)</code></pre>
<pre><code>[1] &quot;Fb1&quot;     &quot;Fb2&quot;     &quot;Fb3&quot;     &quot;INT1&quot;    &quot;PeriFb1&quot; &quot;PeriFb2&quot; &quot;VSMC&quot;   </code></pre>
<pre class="r"><code>DimPlot(seuratFb, reduction = &quot;umap&quot;, cols = colclusterName, pt.size = 0.3)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$pat_sub
DimPlot(seuratFb, reduction = &quot;umap&quot;, cols = colpat_sub, pt.size = 0.3)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Fb_markers_patsub &lt;- FindAllMarkers(seuratFb, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01)

#View(Fb_markers_patsub)</code></pre>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$clusterName

Fbmarkers &lt;- FindAllMarkers(seuratFb, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01)</code></pre>
<pre class="r"><code>## 

Idents(seuratFb) &lt;- seuratFb$clusterName
seuratFb$clusterName &lt;- factor(seuratFb$clusterName, levels =c(&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;))</code></pre>
<pre class="r"><code>## Create Heatmap

seuratFb$clusterName &lt;- factor(seuratFb$clusterName, levels =c(&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;))

genes &lt;- data.frame(gene=rownames(seuratFb)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))


selGenesFb &lt;- data.frame(geneID=rev(c(&quot;XIST&quot;, &quot;ABI3BP&quot;, &quot;NOX4&quot;, &quot;FBN1&quot;, &quot;GBP1&quot;, &quot;FN1&quot;, &quot;TIMP1&quot;, &quot;MICALL2&quot;, &quot;CCL2&quot;, &quot;SRPX2&quot;, &quot;CXCL6&quot;, &quot;CXCL9&quot;, &quot;CXCL10&quot;, &quot;CXCL11&quot;, &quot;SPHK1&quot;,  &quot;HIF3A&quot;,  &quot;LAMA2&quot;, &quot;STAT1&quot;, &quot;PDGFRA&quot;, &quot;MEG3&quot;, &quot;ABCA8&quot;, &quot;LAMB1&quot;, &quot;C7&quot;, &quot;BMP4&quot;, &quot;GREM1&quot;, &quot;CARMN&quot;, &quot;COL5A3&quot;, &quot;PDGFRB&quot;,&quot;ACTA2&quot;, &quot;NOTCH3&quot;,&quot;ADAP1&quot;, &quot;ADAMTS4&quot;, &quot;RGS5&quot;, &quot;HLA-B&quot;)))  %&gt;% left_join(., genes, by=&quot;geneID&quot;) %&gt;% distinct(geneID, .keep_all = TRUE)


DotPlot(seuratFb, features = selGenesFb, group.by= &quot;clusterName&quot;) + RotatedAxis() + scale_color_viridis(option=&quot;T&quot;) + coord_flip()</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/Dotplot%20Fb-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>avgHeatmap &lt;- function(seuratFb, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  selGenes &lt;- selGenes$gene
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seuratFb)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seuratFb)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seuratFb)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]
  
  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;(^[0-9]_?)&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      celltype=colVec)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes,]
  if(is.null(ordVec)){
  ordVec &lt;- levels(seuratFb)
  ordVec &lt;- c(c(&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;))
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0,cluster_rows = cr, 
         cluster_cols = cc, color = colorRampPalette(c(&quot;#2166AC&quot;, &quot;#F7F7F7&quot;, &quot;#B2182B&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}

genesPlot &lt;- data.frame(gene=c(&quot;XIST&quot;, &quot;ABI3BP&quot;, &quot;NOX4&quot;, &quot;FBN1&quot;, &quot;GBP1&quot;, &quot;FN1&quot;, &quot;TIMP1&quot;, &quot;MICALL2&quot;, &quot;CCL2&quot;, &quot;SRPX2&quot;, &quot;CXCL6&quot;, &quot;CXCL9&quot;, &quot;CXCL10&quot;, &quot;CXCL11&quot;, &quot;SPHK1&quot;,  &quot;HIF3A&quot;,  &quot;LAMA2&quot;, &quot;STAT1&quot;, &quot;PDGFRA&quot;, &quot;MEG3&quot;, &quot;ABCA8&quot;, &quot;LAMB1&quot;, &quot;C7&quot;, &quot;BMP4&quot;, &quot;GREM1&quot;, &quot;CARMN&quot;, &quot;COL5A3&quot;, &quot;PDGFRB&quot;, &quot;NOTCH3&quot;,&quot;ADAP1&quot;, &quot;ADAMTS4&quot;, &quot;RGS5&quot;, &quot;HLA-B&quot;))

levels(seuratFb)</code></pre>
<pre><code>[1] &quot;Fb1&quot;     &quot;Fb2&quot;     &quot;Fb3&quot;     &quot;INT1&quot;    &quot;PeriFb1&quot; &quot;PeriFb2&quot; &quot;VSMC&quot;   </code></pre>
<pre class="r"><code>colVec &lt;- colclusterName

# colVec &lt;- c(colPal, colPal, colPal)
# colVec &lt;- c(&quot;blue&quot;, &quot;red&quot;)

avgHeatmap(seuratFb, selGenes = genesPlot, colVecIdent = colVec)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/Heatmap%20Fb%20cell%20clusters-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Filter for clusters Fb1, Fb2, Fb3, Int1, PeriFb1,2,3

seuratFb &lt;- subset(seuratM, clusterName %in% c(&quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;))
# cell count N = 21641

DimPlot(seuratFb, reduction = &quot;umap&quot;, cols = colclusterName, pt.size = 0.1)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table(seuratFb$orig.ident)</code></pre>
<pre><code>
      
34814 </code></pre>
<pre class="r"><code>table(seuratFb$patient)</code></pre>
<pre><code>
    Healthy 01     Healthy 02     Healthy 03     Healthy 04     Healthy 05     Healthy 06 
          1918           2275           2477           4202            912           1055 
    Healthy 07     Healthy 08          AM 01          AM 02          AM 03          AM 04 
           413            944            527           2468            594             66 
         AM 05          AM 06          AM 07          AM 08          AM 09          AM 10 
           333             99            724            371            194            243 
         AM 11          AM 12          AM 13          AM 14          AM 15          AM 16 
            98            585           1488            995           2727            389 
         AM 17          AM 18         CMP 01         CMP 02         CMP 03         CMP 04 
            99            221             48            139            387            651 
        CMP 05         CMP 06         CMP 07         CMP 08         CMP 09         CMP 10 
           262            363            767           1149           1365            230 
        CMP 11         CMP 12         CMP 13 Sarcoidosis 01 Sarcoidosis 02 
           426            225             49           1198           1138 </code></pre>
<pre class="r"><code>Fb_markers &lt;- FindAllMarkers(seuratFb, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01)
#
#View(Fb_markers)
#write.table(Fb_markers, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Fb markers after HH2_HH3 removal&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)</code></pre>
<pre class="r"><code>datList &lt;- NULL
for(con in unique(seuratFb$patient)){
  seuratSub &lt;- subset(seuratFb, patient==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 39114  2468
[1] 39114   594
[1] 39114    66
[1] 39114   333
[1] 39114    99
[1] 39114   724
[1] 39114   371
[1] 39114   194
[1] 39114   243
[1] 39114    98
[1] 39114   585
[1] 39114  1488
[1] 39114   995
[1] 39114   527
[1] 39114  2727
[1] 39114   389
[1] 39114    99
[1] 39114   221
[1] 39114    48
[1] 39114   139
[1] 39114   387
[1] 39114  1365
[1] 39114   230
[1] 39114   426
[1] 39114   225
[1] 39114   363
[1] 39114   651
[1] 39114   262
[1] 39114   767
[1] 39114  1149
[1] 39114    49
[1] 39114  1918
[1] 39114  2275
[1] 39114  2477
[1] 39114  4202
[1] 39114   912
[1] 39114  1055
[1] 39114   413
[1] 39114   944
[1] 39114  1198
[1] 39114  1138</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

# Order of patient levels 
patient_order &lt;- c(&quot;Healthy 01&quot;, &quot;Healthy 02&quot;, &quot;Healthy 03&quot;, &quot;Healthy 04&quot;, &quot;Healthy 05&quot;, 
                   &quot;Healthy 06&quot;, &quot;Healthy 07&quot;, &quot;Healthy 08&quot;, &quot;AM 01&quot;, &quot;AM 02&quot;, 
                   &quot;AM 03&quot;, &quot;AM 04&quot;, &quot;AM 05&quot;, &quot;AM 06&quot;, &quot;AM 07&quot;, 
                   &quot;AM 08&quot;, &quot;AM 09&quot;, &quot;AM 10&quot;, &quot;AM 11&quot;, &quot;AM 12&quot;, &quot;AM 13&quot;, &quot;AM 14&quot;, 
                   &quot;AM 15&quot;, &quot;AM 16&quot;, &quot;AM 17&quot;, &quot;AM 18&quot;, &quot;CMP 01&quot;, &quot;CMP 02&quot;, &quot;CMP 03&quot;, 
                   &quot;CMP 04&quot;, &quot;CMP 05&quot;, &quot;CMP 06&quot;, &quot;CMP 07&quot;, &quot;CMP 08&quot;, &quot;CMP 09&quot;, 
                   &quot;CMP 10&quot;, &quot;CMP 11&quot;, &quot;CMP 12&quot;, &quot;CMP 13&quot;, &quot;Sarcoidosis 01&quot;, &quot;Sarcoidosis 02&quot;)

View(dat_all)



# Convert the patient column to a factor with the desired order
dat_all$patient &lt;- factor(dat_all$patient, levels = patient_order)

## plot abundance
ggbarplot(dat_all, x= &quot;patient&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;, palette = colclusterName )  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>datList &lt;- NULL
for(con in unique(seuratFb$pat_sub)){
  seuratSub &lt;- subset(seuratFb, pat_sub==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(diseaseCond=con)
  datList[[con]] &lt;- dat_con
  
}</code></pre>
<pre><code>[1] 39114 12221
[1] 39114  6061
[1] 39114 14196
[1] 39114  2336</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

order_patsub &lt;- c(&quot;Healthy&quot;, &quot;AM&quot;, &quot;CMP&quot;, &quot;Sarcoidosis&quot;)

ggbarplot(dat_all, x = &quot;diseaseCond&quot;, y = &quot;percent&quot;, fill = &quot;Var1&quot;, palette = colclusterName, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, xlab = &quot;condition&quot;, ylab = &quot;frequency&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_x_discrete(limits = order_patsub)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/Relative%20abundance%20dis%20con-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$clusterName
seuratFb &lt;- subset(seuratM, clusterName %in% c(&quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;))
levels(seuratFb)</code></pre>
<pre><code>[1] &quot;Fb1&quot;     &quot;Fb2&quot;     &quot;Fb3&quot;     &quot;INT1&quot;    &quot;PeriFb1&quot; &quot;PeriFb2&quot; &quot;VSMC&quot;   </code></pre>
<pre class="r"><code>DimPlot(seuratFb, reduction = &quot;umap&quot;, cols = colclusterName, pt.size = 0.1, label = TRUE)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>seuratFb &lt;- NormalizeData (object = seuratFb)
seuratFb &lt;- FindVariableFeatures(object = seuratFb)
seuratFb &lt;- ScaleData(object = seuratFb, verbose = TRUE)
seuratFb &lt;- RunPCA(object=seuratFb, npcs = 30, verbose = FALSE)
seuratFb &lt;- RunTSNE(object=seuratFb, reduction=&quot;pca&quot;, dims = 1:20)
seuratFb &lt;- RunUMAP(object=seuratFb, reduction=&quot;pca&quot;, dims = 1:20)
seuratFb &lt;- FindNeighbors(object = seuratFb, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4)
for (i in 1:length(res)) {
  seuratFb &lt;- FindClusters(object = seuratFb, resolution = res[i], random.seed = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 34814
Number of edges: 1034395

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9237
Number of communities: 11
Elapsed time: 4 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 34814
Number of edges: 1034395

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8770
Number of communities: 21
Elapsed time: 5 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 34814
Number of edges: 1034395

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8620
Number of communities: 22
Elapsed time: 4 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 34814
Number of edges: 1034395

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8989
Number of communities: 16
Elapsed time: 4 seconds</code></pre>
<pre class="r"><code>table(seuratFb$orig.ident)</code></pre>
<pre><code>
      
34814 </code></pre>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$RNA_snn_res.0.25
DimPlot(seuratFb, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$pat_sub
DimPlot(seuratFb, reduction = &quot;umap&quot;, pt.size = 0.1, repel = TRUE, cols = colpat_sub)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-20-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>seuratFb$clusterName &lt;- &quot;clusterName&quot;
Idents(seuratFb) &lt;- seuratFb$clusterName
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;0&quot; )] &lt;- &quot;PeriFb1&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;1&quot; )] &lt;- &quot;Fb1&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;2&quot; )] &lt;- &quot;Fb3&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;3&quot; )] &lt;- &quot;VSMC&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;4&quot; )] &lt;- &quot;Fb4&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;5&quot; )] &lt;- &quot;Fb5&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;6&quot; )] &lt;- &quot;Fb8&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;7&quot; )] &lt;- &quot;Fb6&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;8&quot; )] &lt;- &quot;Fb2&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;9&quot; )] &lt;- &quot;PeriFb2&quot;
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% &quot;10&quot; )] &lt;- &quot;Fb7&quot;


seuratFb$clusterName &lt;- factor(seuratFb$clusterName, levels =c(&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;,&quot;Fb4&quot;, &quot;Fb5&quot;, &quot;Fb6&quot;, &quot;Fb7&quot;, &quot;Fb8&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;))

table(seuratFb$patient,seuratFb$clusterName)</code></pre>
<pre><code>                
                  Fb1  Fb2  Fb3  Fb4  Fb5  Fb6  Fb7  Fb8 PeriFb1 PeriFb2 VSMC
  Healthy 01      123    6 1012    4    0    1   12   37     568       0  155
  Healthy 02      110    3   18    6    1  969   10   84     984       2   88
  Healthy 03      280   61 1171   24    1    0    5   40     786       5  104
  Healthy 04      278   16 1880   14    0    2   74  244    1430       4  260
  Healthy 05       33    1   46    4   10    0    3   12     731       0   72
  Healthy 06      113    1  173   15    0    8    4   14     659       1   67
  Healthy 07       35    0   26    1    0    3    1    9     310       2   26
  Healthy 08       48    3  199    8    0    2    0   26     596       2   60
  AM 01             0  376   10    6    2    0    0    2       3     112   16
  AM 02          1439   24   33    6    3    0   22   78     623       2  238
  AM 03           396    1    1    1    1    0    3    2     156       1   32
  AM 04            25    1    1    2    1    1    1    7      23       0    4
  AM 05           152    2    1    1    2    0    3   11     156       2    3
  AM 06             4    1    0   11   56    1    0    7      12       0    7
  AM 07            16   30   16  425   24    2    0    9     182       1   19
  AM 08           166    3   17   10    2    2   14   24     116       0   17
  AM 09            85    0    1    5   40    2    2    3      49       0    7
  AM 10            20    1    0   74   91    0    0    7      48       0    2
  AM 11            37    1    1    2    0    0    1    2      32       0   22
  AM 12           325    6   10   11    0    0    1    5     179       1   47
  AM 13            37    3   12   11  704    2    2   29     480       8  200
  AM 14           624    3    3    1    0   15   11   41     248       1   48
  AM 15          1022   46   37    1    0    0   21  120    1309       2  169
  AM 16             1  243    1    4    0    0    0    2       2     133    3
  AM 17            37    5    6    2    1    0    0    4      27       4   13
  AM 18            98    4    0    5    3    0    1   12      78       6   14
  CMP 01           19    1    1    0   19    0    0    1       6       0    1
  CMP 02           33    1    0    0   61    6    1    5      29       1    2
  CMP 03           22   14    2    0  234    3    1   21      74       3   13
  CMP 04          290    1    6    1    2   11    3   31     235       4   67
  CMP 05           52    3   17   36    6    0    2    5     125       1   15
  CMP 06          232    0    3    0    1    0    0    6      64       0   57
  CMP 07          296    1   13    1    3    1    1   13     348       0   90
  CMP 08          584    1   13    1   10    0    2   25     379       1  133
  CMP 09          803   11   11    7    0    3   12   39     399       0   80
  CMP 10          130    3    1    7    0    0    1    4      61       0   23
  CMP 11          256    4    1    1    0    0    1   13     113       0   37
  CMP 12          144    0    2    5    1    1    0    4      50       0   18
  CMP 13           17    0    0    0    0    0    0    0      28       0    4
  Sarcoidosis 01   54   15   39  674    0    0    3   48     306       6   53
  Sarcoidosis 02  611    2    7    3    7    6    3   32     344       0  123</code></pre>
<pre class="r"><code>table(seuratFb$clusterName)</code></pre>
<pre><code>
    Fb1     Fb2     Fb3     Fb4     Fb5     Fb6     Fb7     Fb8 PeriFb1 PeriFb2    VSMC 
   9047     898    4791    1390    1286    1041     221    1078   12348     305    2409 </code></pre>
<pre class="r"><code>print(levels(seuratFb$clusterName))</code></pre>
<pre><code> [1] &quot;Fb1&quot;     &quot;Fb2&quot;     &quot;Fb3&quot;     &quot;Fb4&quot;     &quot;Fb5&quot;     &quot;Fb6&quot;     &quot;Fb7&quot;     &quot;Fb8&quot;     &quot;PeriFb1&quot;
[10] &quot;PeriFb2&quot; &quot;VSMC&quot;   </code></pre>
<pre class="r"><code>colFb &lt;- c(&quot;#f4a582&quot;, &quot;#FEE08B&quot;,&quot;#feb24c&quot;, &quot;#9ebcda&quot;, &quot;#2166AC&quot; , &quot;#756bb1&quot;, &quot;#3182bd&quot;, &quot;#c994c7&quot;, &quot;#01665e&quot;,&quot;#66C2A5&quot;,&quot;#c7eae5&quot;)

names(colFb) &lt;- c(&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;,&quot;Fb4&quot;, &quot;Fb5&quot;, &quot;Fb6&quot;, &quot;Fb7&quot;, &quot;Fb8&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;)

Idents(seuratFb) &lt;- seuratFb$clusterName
DimPlot(seuratFb, reduction = &quot;umap&quot;, pt.size = 0.1, label = FALSE, cols =colFb)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>datList &lt;- NULL
for(con in unique(seuratFb$patient)){
  seuratSub &lt;- subset(seuratFb, patient==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 39114  2468
[1] 39114   594
[1] 39114    66
[1] 39114   333
[1] 39114    99
[1] 39114   724
[1] 39114   371
[1] 39114   194
[1] 39114   243
[1] 39114    98
[1] 39114   585
[1] 39114  1488
[1] 39114   995
[1] 39114   527
[1] 39114  2727
[1] 39114   389
[1] 39114    99
[1] 39114   221
[1] 39114    48
[1] 39114   139
[1] 39114   387
[1] 39114  1365
[1] 39114   230
[1] 39114   426
[1] 39114   225
[1] 39114   363
[1] 39114   651
[1] 39114   262
[1] 39114   767
[1] 39114  1149
[1] 39114    49
[1] 39114  1918
[1] 39114  2275
[1] 39114  2477
[1] 39114  4202
[1] 39114   912
[1] 39114  1055
[1] 39114   413
[1] 39114   944
[1] 39114  1198
[1] 39114  1138</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

# Order 
patient_order &lt;- c(&quot;Healthy 01&quot;, &quot;Healthy 02&quot;, &quot;Healthy 03&quot;, &quot;Healthy 04&quot;, &quot;Healthy 05&quot;, 
                   &quot;Healthy 06&quot;, &quot;Healthy 07&quot;, &quot;Healthy 08&quot;, &quot;Healthy 09&quot;, &quot;Healthy 10&quot;, 
                   &quot;AM 01&quot;, &quot;AM 02&quot;, &quot;AM 03&quot;, &quot;AM 04&quot;, &quot;AM 05&quot;, &quot;AM 06&quot;, &quot;AM 07&quot;, 
                   &quot;AM 08&quot;, &quot;AM 09&quot;, &quot;AM 10&quot;, &quot;AM 11&quot;, &quot;AM 12&quot;, &quot;AM 13&quot;, &quot;AM 14&quot;, 
                   &quot;AM 15&quot;, &quot;AM 16&quot;, &quot;AM 17&quot;, &quot;AM 18&quot;, &quot;CMP 01&quot;, &quot;CMP 02&quot;, &quot;CMP 03&quot;, 
                   &quot;CMP 04&quot;, &quot;CMP 05&quot;, &quot;CMP 06&quot;, &quot;CMP 07&quot;, &quot;CMP 08&quot;, &quot;CMP 09&quot;, 
                   &quot;CMP 10&quot;, &quot;CMP 11&quot;, &quot;CMP 12&quot;, &quot;CMP 13&quot;, &quot;Sarcoidosis 01&quot;, &quot;Sarcoidosis 02&quot;)

#View(dat_all)

#write.table(dat_all, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers_dat_all_clusters_frequency_relative abundance_V2&quot;, sep = &quot;\t&quot;,quote=F,row.names=T,col.names=T)


dat_all$patient &lt;- factor(dat_all$patient, levels = patient_order)

## plot abundance
ggbarplot(dat_all, x= &quot;patient&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;, palette = colFb )  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/Abundance%20per%20patient%20Fibroblast%20subgroup%20analysis-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>datList &lt;- NULL
for(con in unique(seuratFb$pat_sub)){
  seuratSub &lt;- subset(seuratFb, pat_sub==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(diseaseCond=con)
  datList[[con]] &lt;- dat_con
  
}</code></pre>
<pre><code>[1] 39114 12221
[1] 39114  6061
[1] 39114 14196
[1] 39114  2336</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

order_patsub &lt;- c(&quot;Healthy&quot;, &quot;AM&quot;, &quot;CMP&quot;, &quot;Sarcoidosis&quot;)

ggbarplot(dat_all, x = &quot;diseaseCond&quot;, y = &quot;percent&quot;, fill = &quot;Var1&quot;, palette = colFb, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, xlab = &quot;condition&quot;, ylab = &quot;frequency&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_x_discrete(limits = order_patsub)</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/Relative%20abundance%20per%20disease%20condition-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##cluster markers
Fbmarkers &lt;- FindAllMarkers(seuratFb, only.pos=F, logfc.threshold = 0.2) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01)

#View(Fbmarkers)

# write.table(Fbmarkers, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/allFb_PeriFb_vsmc_int markers&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)</code></pre>
<pre class="r"><code>Idents(seuratFb) &lt;- seuratFb$clusterName

genes &lt;- data.frame(gene = rownames (seuratFb)) %&gt;%
  mutate(geneID =gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

# Selected genes for DotPlot
selGenes &lt;- data.frame(geneID=rev(c(&quot;MEG3&quot;, &quot;ABCA6&quot;, &quot;ABCA10&quot;, &quot;COL6A3&quot;,&quot;C7&quot;, &quot;PDGFRA&quot;, &quot;ACTA2&quot;,&quot;PECAM1&quot;, &quot;VWF&quot;, &quot;BMP4&quot;, &quot;XIST&quot;, &quot;GBP1&quot;, &quot;GBP5&quot;, &quot;CXCL9&quot;, &quot;IRF8&quot;, &quot;VMP1&quot;, &quot;HLA-DMB&quot;, &quot;APOL2&quot;, &quot;IL7R&quot;, &quot;CXCR2P1&quot;, &quot;JAK2&quot;, &quot;SLAMF8&quot;, &quot;CD2&quot;, &quot;CCL8&quot;, &quot;CXCL6&quot;, &quot;IL2RG&quot;, &quot;COL1A2&quot;, &quot;TNFSF10&quot;, &quot;COl1A2&quot;, &quot;SNX10&quot;, &quot;XAF1&quot;,&quot;ZBTB16&quot;, &quot;APOD&quot;, &quot;ADAMTS5&quot;, &quot;ADAMTS1&quot;, &quot;RGS2&quot;, &quot;GRAMD2B&quot;, &quot;WNK3&quot;, &quot;SLC7A8&quot;, &quot;PLEKHG4&quot;, &quot;S100A8&quot;)))  %&gt;% left_join(., genes, by=&quot;geneID&quot;) %&gt;% filter(gene != &quot;ENSG00000232995.RGS5&quot;) 

DotPlot(seuratFb, features = selGenes, group.by= &quot;clusterName&quot;) + RotatedAxis() + scale_color_viridis(option=&quot;T&quot;) + coord_flip()</code></pre>
<p><img src="figure/Fibroblast_subset_V2.Rmd/find%20markers-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## overlap with mcp genes
####selGenes &lt;- data.frame(geneID=rev(c(&quot;MEG3&quot;, &quot;ABCA6&quot;, &quot;ABCA10&quot;, &quot;COL6A3&quot;, &quot;DCN&quot;,  &quot;FBN1&quot;, &quot;C7&quot;, &quot;PDGFRA&quot;, &quot;BMP4&quot;, &quot;GREM1&quot;,&quot;GREM2&quot;, &quot;HLA-B&quot;, &quot;GBP1&quot;, &quot;STAT1&quot;, &quot;CDH19&quot;, &quot;PDGFRB&quot;,&quot;ITGA7&quot;,&quot;RGS5&quot;, &quot;NOTCH3&quot;, &quot;MYH11&quot;,&quot;CARMN&quot;, &quot;CD36&quot;, &quot;ACTA2&quot;,&quot;PECAM1&quot;, &quot;VWF&quot;, &quot;SMOC1&quot;, &quot;EGFL7&quot;, &quot;ITGA10&quot;, &quot;CXCL9&quot;, &quot;CXCL10&quot;, &quot;CDH11&quot;,&quot;CCL21&quot;, &quot;PROX1&quot;, &quot;FLT4&quot;, &quot;EGFL71&quot;, &quot;NOTCH41&quot;,&quot;SHANK31&quot;, &quot;BTLN9&quot;, &quot;CD163&quot;, &quot;LYVE1&quot;, &quot;MRC1&quot;, &quot;SIGLEC1&quot;, &quot;STAB1&quot;,&quot;MS4A6A&quot;, &quot;CSF1R&quot;,&quot;MERTK&quot;, &quot;HLA-DQB1&quot;, &quot;CCR2&quot;, &quot;IL7R&quot;, &quot;CD2&quot;, &quot;CD79B&quot;, &quot;NRXN1&quot;, &quot;TTN&quot;))) %&gt;% left_join(., genes, by=&quot;geneID&quot;) %&gt;% filter(gene != &quot;ENSG00000232995.RGS5&quot;) 

## Fb1: ECM &quot;normal&quot; fibroblasts

## Fb2
# HLA-B and HLA-A
# STAT1 signal transducer and activator of transcription, can be activated by inferon, IL-6, involved in immune responses
# CD74 MHCII antigen presentation
# LAP3: Metallopeptidase
# WARS1: catalyzes the aminoacylation of tRNA with tryptophan and is induced by interferon
# NLRC5: cytokine response and antiviral immunity
# CXCL9 and CXCL10

## Fb3
# STAB1 Stabilin, angiogenesis, lymphocyte homing, cell adhesion, binds to gram pos and neg. bacteria, defense against bacterial infections
# CD163 scavenger receptor family expressed in monocytes and macrophages, acute phase regulated receptor, acute phase reaction
# MS4A7 CD20 antigen like 4, high affinity immunoglobulin epsilon receptor beta subunit, associated with cellular function in the monocytic lineage
# MRC1 Mannose receptor type 1, cell-cell recognition, neutroalizuation pathogens, mediates endocytosis by macrophages
# F13A1 coagulation factor XIII, crosslinking between fibrin molecules, stabilizes fibrin clots
# SIGLEC1  Sialic acid binding immunoglobulin like lectin: on subpopulation of macrophages, mediates cell-cell interactions, bacterial and viral infection
# LILRB5 Leukocyte immunoglobulin like receptor subfamily B, innate immune system, antigen processing
# CYTH4 Cytohesin 4: vesicle mediated transport. Golgi and ER
# FGD2 fuanine nucleotide exchange factors, cytoskeleton dependent membrane rearrancgement, B lymphocytes, macrophages, dendritic cells
# LCGR2C: not found
# LYVE1: lymphatic vessel endothelial hyaluronan receptor 1: membrane glycoproteine, lymphatic hyaluronan transport
# NCF4: cytosolic regulatory component of NADPH oxidase; important for host defense

###
## PeriFb1
# CARMN cardiac mesoderm enhancer associated non coding RNA: responsible for contractile phenotype of VSMCs
# CD36: glycoprotein on platelet surfaces, cell adhesion, binds collagen, trhobmospondin, angiogenesis, inflammatroy responses, muscle lipid utilization
# ITGA7: Integrin subunit alpha, mainly expressed in skeletal and cardiac muscles; receptor for Lamini-1, myofiber cytoarchitecture maintenance
# ABCC9: ATP sensitive potassium channel in cardiac skeletal and vascular and non-vascular smooth muscle
# RGS5: GTPasa activity Gi
# DLC1: deleted in liver cancer 1, GTPAse, regulatory processes cytoskeletal changes


##PeriFb3
#GBP1: Interferon induced GTPase: innate immunity; inflammasome assembly Guanlyate binding protein expression is induced by interferon
#GBP4: Interferon induced GTPase
#TAP1: Antigen transporter associated with antigen processing
#CXCL9: binds CXCR3, monokine induced by Interferon-Gamma; immunoregulatory and inflammatory processes, T cell trafficking
#CXCL10: cytokine storm, binds CXCR3, stimulates monocytes, NK, T cells
#ADAP1: GTPase activator activity
#CXCL11: induced chemotactic response in activated T cells, binds to CXCR3
#CCL8: Monocyte chemotactic protein 2: chemotactic activity for monocytes, lymphocytes, basophils and eosinophils, contributes to antiviral state
#HLA-B: antigen presentation

###INT1
# EGFL71: secreted endothelial cell protein that contains two epidermal growth factor-like domains. The encoded protein may play a role in regulating vasculogenesis. EGFL7 Secreted By Human Bone Mesenchymal Stem Cells Promotes Osteoblast Differentiation
# NOTCH41: not found in gene database (only Notch4), epidermal growth factor vascular, renal and hepatic development
# SHANK31: not found in gene database (only shank3synapse formation and dendritic spine maturation)
# BTLN9: Butyrophylin like 9 Predicted to be involved in T cell receptor signaling pathway and regulation of cytokine production
# SLC9A3R2 nf --&gt; SLC9A3 (Solute Carrier Family 9 Member A3) sodium channel
# CR381670.1 ankyrin repeat domain 20 family, member A3 (ANKRD20A3) pseudogene
# PTPRB cell growth, differentiation, mitotic cycle, and oncogenic transformation. 
# SYNE22 nf
# ADGRF52 nf, ADGRF5 glomerular filtration; pharyngeal arch artery morphogenesis; and surfactant homeostasis
# VWF platelet aggregation endothelial cells
# PECAM12: Platelet And Endothelial Cell Adhesion Molecule 1, PECAM1 The protein encoded by this gene is found on the surface of platelets, monocytes, neutrophils, and some types of T-cells, and makes up a large portion of endothelial cell intercellular junction

## PeriFb2 first 7 highly expressed genes identical to INT1:
#EGFL7
#SHANK3
#NOTCH4

## VSMC</code></pre>
<p>`</p>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Sat Nov 16 01:56:52 2024&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.7.1             pathview_1.44.0             wordcloud_2.6              
 [4] textshaping_0.4.0           EnhancedVolcano_1.22.0      ggrepel_0.9.6              
 [7] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
[10] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
[13] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_7.5.1              
[16] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[19] multtest_2.60.0             metap_1.11                  scater_1.32.1              
[22] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[25] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[28] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[31] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[34] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[37] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[40] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[43] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[46] matrixStats_1.4.1           pheatmap_1.0.12             ggpubr_0.6.0               
[49] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[52] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             presto_1.0.0              parallelly_1.38.0        
[145] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[148] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[151] pbapply_1.7-2             httr2_1.0.5               spam_2.11-0              
[154] gson_0.1.0                utf8_1.2.4                graphlayouts_1.2.0       
[157] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[160] shiny_1.9.1               GenomeInfoDbData_1.2.12   glmmTMB_1.1.10           
[163] R.utils_2.12.3            RCurl_1.98-1.16           memoise_2.0.1            
[166] rmarkdown_2.28            scales_1.3.0              R.methodsS3_1.8.2        
[169] future_1.34.0             RANN_2.6.2                Cairo_1.6-2              
[172] spatstat.data_3.1-2       rstudioapi_0.16.0         whisker_0.4.1            
[175] cluster_2.1.6             mutoss_0.1-13             spatstat.utils_3.1-0     
[178] hms_1.1.3                 fitdistrplus_1.2-1        munsell_0.5.1            
[181] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.4              
[184] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.0               
[187] dotCall64_1.2             shinydashboard_0.7.2      ggforce_0.4.2            
[190] laeken_0.5.3              mgcv_1.9-1                xfun_0.48                
[193] e1071_1.7-16              TH.data_1.1-2             iterators_1.0.14         
[196] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[199] ps_1.8.0                  futile.options_1.0.1      bitops_1.0-9             
[202] Rdpack_2.6.1              promises_1.3.0            scatterpie_0.2.4         
[205] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-1           
[208] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[211] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[214] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[217] Rcpp_1.0.13               edgeR_4.2.1               BiocSingular_1.20.0      
[220] tensor_1.5                MASS_7.3-61               progress_1.2.3           
[223] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.3-2    
[226] R6_2.5.1                  fastmap_1.2.0             multcomp_1.4-26          
[229] fastmatch_1.1-4           rstatix_0.7.2             vipor_0.4.7              
[232] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[235] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-19              
[238] gtable_0.3.5              KernSmooth_2.23-24        miniUI_0.1.1.1           
[241] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[244] bit64_4.5.2               spatstat.explore_3.3-2    lifecycle_1.0.4          
[247] blme_1.0-6                processx_3.8.4            callr_3.7.6              
[250] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[253] robustbase_0.99-4-1       spatstat.geom_3.3-3       sn_2.1.1                 
[256] ggfun_0.1.6               future.apply_1.11.2       bslib_0.8.0              
[259] pillar_1.9.0              gplots_3.2.0              pcaMethods_1.96.0        
[262] locfit_1.5-9.10           jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.7.1             pathview_1.44.0             wordcloud_2.6              
 [4] textshaping_0.4.0           EnhancedVolcano_1.22.0      ggrepel_0.9.6              
 [7] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
[10] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
[13] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_7.5.1              
[16] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[19] multtest_2.60.0             metap_1.11                  scater_1.32.1              
[22] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[25] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[28] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[31] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[34] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[37] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[40] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[43] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[46] matrixStats_1.4.1           pheatmap_1.0.12             ggpubr_0.6.0               
[49] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[52] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             presto_1.0.0              parallelly_1.38.0        
[145] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[148] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[151] pbapply_1.7-2             httr2_1.0.5               spam_2.11-0              
[154] gson_0.1.0                utf8_1.2.4                graphlayouts_1.2.0       
[157] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[160] shiny_1.9.1               GenomeInfoDbData_1.2.12   glmmTMB_1.1.10           
[163] R.utils_2.12.3            RCurl_1.98-1.16           memoise_2.0.1            
[166] rmarkdown_2.28            scales_1.3.0              R.methodsS3_1.8.2        
[169] future_1.34.0             RANN_2.6.2                Cairo_1.6-2              
[172] spatstat.data_3.1-2       rstudioapi_0.16.0         whisker_0.4.1            
[175] cluster_2.1.6             mutoss_0.1-13             spatstat.utils_3.1-0     
[178] hms_1.1.3                 fitdistrplus_1.2-1        munsell_0.5.1            
[181] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.4              
[184] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.0               
[187] dotCall64_1.2             shinydashboard_0.7.2      ggforce_0.4.2            
[190] laeken_0.5.3              mgcv_1.9-1                xfun_0.48                
[193] e1071_1.7-16              TH.data_1.1-2             iterators_1.0.14         
[196] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[199] ps_1.8.0                  futile.options_1.0.1      bitops_1.0-9             
[202] Rdpack_2.6.1              promises_1.3.0            scatterpie_0.2.4         
[205] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-1           
[208] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[211] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[214] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[217] Rcpp_1.0.13               edgeR_4.2.1               BiocSingular_1.20.0      
[220] tensor_1.5                MASS_7.3-61               progress_1.2.3           
[223] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.3-2    
[226] R6_2.5.1                  fastmap_1.2.0             multcomp_1.4-26          
[229] fastmatch_1.1-4           rstatix_0.7.2             vipor_0.4.7              
[232] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[235] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-19              
[238] gtable_0.3.5              KernSmooth_2.23-24        miniUI_0.1.1.1           
[241] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[244] bit64_4.5.2               spatstat.explore_3.3-2    lifecycle_1.0.4          
[247] blme_1.0-6                processx_3.8.4            callr_3.7.6              
[250] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[253] robustbase_0.99-4-1       spatstat.geom_3.3-3       sn_2.1.1                 
[256] ggfun_0.1.6               future.apply_1.11.2       bslib_0.8.0              
[259] pillar_1.9.0              gplots_3.2.0              pcaMethods_1.96.0        
[262] locfit_1.5-9.10           jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
