<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Payne" />

<meta name="date" content="2024-11-07" />

<title>Activated clusters and immune cell clusters</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myocarditis snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Activated clusters and immune cell
clusters</h1>
<h4 class="author">Emily Payne</h4>
<h4 class="date">2024-11-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-11-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Myocarditis
snRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis
snRNAseq/docs/Markers/DEGenesimmune
</td>
<td style="text-align:left;">
../docs/Markers/DEGenesimmune
</td>
</tr>
<tr>
<td style="text-align:left;">
~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat
files/Myocarditis_allfiltered_V3_15.11.2024.rds
</td>
<td style="text-align:left;">
../data/01 Merged Seurat files/Myocarditis_allfiltered_V3_15.11.2024.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree05dd5a33149bad1924497530bf1b627bc18d9a60targetblank05dd5a3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/05dd5a33149bad1924497530bf1b627bc18d9a60" target="_blank">05dd5a3</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree05dd5a33149bad1924497530bf1b627bc18d9a60targetblank05dd5a3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/05dd5a33149bad1924497530bf1b627bc18d9a60" target="_blank">05dd5a3</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/

Untracked files:
    Untracked:  analysis/AM vs. Healthy subsets.Rmd
    Untracked:  analysis/DE genes AMsub vs. Healthy_V2.Rmd
    Untracked:  analysis/Immune_cells_V3.Rmd
    Untracked:  analysis/Interactome Analysis.Rmd
    Untracked:  analysis/Interactome Chellchat Analysis.Rmd
    Untracked:  analysis/T cells.Rmd

Unstaged changes:
    Modified:   analysis/Fibroblast_subset_V2.Rmd
    Deleted:    analysis/Immune_cells.Rmd
    Modified:   analysis/Myocarditis_snRNAseq_QC_merging_V3.Rmd
    Modified:   analysis/Stratification_Tcell_conditionV2.Rmd
    Modified:   analysis/Stratification_sex.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>## load merged file 
seuratM &lt;- readRDS(&quot;~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_allfiltered_V3_15.11.2024.rds&quot;)

table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11    12    13    14 
17699 13639 12754  8222  5475  4836  3723  2365  2156   905   882   694   380   249    91 </code></pre>
<pre class="r"><code>table(seuratM$patient)</code></pre>
<pre><code>
    Healthy 01     Healthy 02     Healthy 03     Healthy 04     Healthy 05     Healthy 06 
          3921           3818           4908           9882           1442           1998 
    Healthy 07     Healthy 08          AM 01          AM 02          AM 03          AM 04 
           841           1480           6286           4323           1123            208 
         AM 05          AM 06          AM 07          AM 08          AM 09          AM 10 
           579            335           1217            604            513            426 
         AM 11          AM 12          AM 13          AM 14          AM 15          AM 16 
           236           1192           2740           1684           4439           2247 
         AM 17          AM 18         CMP 01         CMP 02         CMP 03         CMP 04 
           316            608            105            227            730           1428 
        CMP 05         CMP 06         CMP 07         CMP 08         CMP 09         CMP 10 
           620            653           1268           2280           2396            545 
        CMP 11         CMP 12         CMP 13 Sarcoidosis 01 Sarcoidosis 02 
           781            491            111           2363           2706 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
74070 </code></pre>
<pre class="r"><code>## name Clusters
seuratM$clusterName &lt;- &quot;clusterName&quot;
Idents(seuratM) &lt;- seuratM$clusterName
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;0&quot; )] &lt;- &quot;Fb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;1&quot; )] &lt;- &quot;BEC1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;2&quot; )] &lt;- &quot;PeriFb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;3&quot; )] &lt;- &quot;Cardiomyocyte&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;4&quot; )] &lt;- &quot;MCP1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;5&quot; )] &lt;- &quot;TC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;6&quot; )] &lt;- &quot;MCP2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;7&quot; )] &lt;- &quot;BEC2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;8&quot; )] &lt;- &quot;VSMC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;9&quot; )] &lt;- &quot;NC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;10&quot; )] &lt;- &quot;INT1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;11&quot; )] &lt;- &quot;Fb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;12&quot; )] &lt;- &quot;Fb3&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;13&quot; )] &lt;- &quot;PeriFb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;14&quot; )] &lt;- &quot;AdipoC&quot;


colclusterName &lt;- c(&quot;#D53E4F&quot;, &quot;#f4a582&quot;, &quot;#FEE08B&quot;,&quot;#feb24c&quot;,&quot;#67001f&quot;, &quot;#01665e&quot;,&quot;#66C2A5&quot;,&quot;#c7eae5&quot;,&quot;#BEAED4&quot;, &quot;#355C7D&quot;,&quot;#3288BD&quot;,&quot;#8c510a&quot; ,&quot;#fde0dd&quot;,&quot;#B45B5C&quot;,&quot;#dd1c77&quot;)

                    
names(colclusterName) &lt;- c(&quot;Cardiomyocyte&quot;,&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;,&quot;NC&quot;, &quot;BEC1&quot;,&quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;,&quot;MCP2&quot;,&quot;TC&quot;)

colpat_sub&lt;- c(&quot;#dfc27d&quot;,&quot;#BE3144&quot;,&quot;#355C7D&quot;, &quot;#779d8d&quot;)

seuratM$clusterName &lt;- factor(seuratM$clusterName, levels=c(&quot;Cardiomyocyte&quot;, &quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;INT1&quot;,&quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;, &quot;NC&quot;, &quot;BEC1&quot;, &quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;, &quot;MCP2&quot;, &quot;TC&quot;))


Idents(seuratM) &lt;- seuratM$clusterName
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE, cols = colclusterName)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/original%20clusterName-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table(seuratM$patient, seuratM$TC)</code></pre>
<pre><code>                
                 Tcell High Tcell Intermediate Tcell Low Tcell Very Low
  Healthy 01              0                  0         0           3921
  Healthy 02              0                  0         0           3818
  Healthy 03              0                  0         0           4908
  Healthy 04              0                  0         0           9882
  Healthy 05              0                  0         0           1442
  Healthy 06              0                  0         0           1998
  Healthy 07              0                  0         0            841
  Healthy 08              0                  0         0           1480
  AM 01                6286                  0         0              0
  AM 02                   0                  0         0           4323
  AM 03                   0                  0         0           1123
  AM 04                   0                  0         0            208
  AM 05                   0                  0         0            579
  AM 06                   0                335         0              0
  AM 07                   0                  0         0           1217
  AM 08                   0                  0       604              0
  AM 09                   0                  0       513              0
  AM 10                   0                  0         0            426
  AM 11                   0                  0         0            236
  AM 12                   0                  0         0           1192
  AM 13                   0               2740         0              0
  AM 14                   0                  0      1684              0
  AM 15                   0                  0      4439              0
  AM 16                2247                  0         0              0
  AM 17                 316                  0         0              0
  AM 18                 608                  0         0              0
  CMP 01                  0                105         0              0
  CMP 02                  0                  0         0            227
  CMP 03                  0                  0       730              0
  CMP 04                  0                  0      1428              0
  CMP 05                  0                  0       620              0
  CMP 06                  0                  0         0            653
  CMP 07                  0                  0      1268              0
  CMP 08                  0                  0         0           2280
  CMP 09                  0                  0         0           2396
  CMP 10                  0                  0       545              0
  CMP 11                  0                  0       781              0
  CMP 12                  0                491         0              0
  CMP 13                111                  0         0              0
  Sarcoidosis 01          0                  0      2363              0
  Sarcoidosis 02          0                  0      2706              0</code></pre>
<pre class="r"><code>## Subset from AM cohort, inflammatory clusters

seurat_immun &lt;- subset(seuratM, clusterName %in% c(&quot;MCP1&quot;, &quot;MCP2&quot;,&quot;TC&quot;))
table(seurat_immun$orig.ident)</code></pre>
<pre><code>
      
14034 </code></pre>
<pre class="r"><code>table(seurat_immun$patient, seurat_immun$clusterName)</code></pre>
<pre><code>                
                 Cardiomyocyte  Fb1  Fb2  Fb3 INT1 PeriFb1 PeriFb2 VSMC   NC BEC1 BEC2 AdipoC MCP1
  Healthy 01                 0    0    0    0    0       0       0    0    0    0    0      0  362
  Healthy 02                 0    0    0    0    0       0       0    0    0    0    0      0  216
  Healthy 03                 0    0    0    0    0       0       0    0    0    0    0      0  647
  Healthy 04                 0    0    0    0    0       0       0    0    0    0    0      0  910
  Healthy 05                 0    0    0    0    0       0       0    0    0    0    0      0  185
  Healthy 06                 0    0    0    0    0       0       0    0    0    0    0      0  444
  Healthy 07                 0    0    0    0    0       0       0    0    0    0    0      0  131
  Healthy 08                 0    0    0    0    0       0       0    0    0    0    0      0  101
  AM 01                      0    0    0    0    0       0       0    0    0    0    0      0   17
  AM 02                      0    0    0    0    0       0       0    0    0    0    0      0  131
  AM 03                      0    0    0    0    0       0       0    0    0    0    0      0   91
  AM 04                      0    0    0    0    0       0       0    0    0    0    0      0   14
  AM 05                      0    0    0    0    0       0       0    0    0    0    0      0   58
  AM 06                      0    0    0    0    0       0       0    0    0    0    0      0   30
  AM 07                      0    0    0    0    0       0       0    0    0    0    0      0   66
  AM 08                      0    0    0    0    0       0       0    0    0    0    0      0   51
  AM 09                      0    0    0    0    0       0       0    0    0    0    0      0   58
  AM 10                      0    0    0    0    0       0       0    0    0    0    0      0   29
  AM 11                      0    0    0    0    0       0       0    0    0    0    0      0   14
  AM 12                      0    0    0    0    0       0       0    0    0    0    0      0  122
  AM 13                      0    0    0    0    0       0       0    0    0    0    0      0  294
  AM 14                      0    0    0    0    0       0       0    0    0    0    0      0   82
  AM 15                      0    0    0    0    0       0       0    0    0    0    0      0  313
  AM 16                      0    0    0    0    0       0       0    0    0    0    0      0    2
  AM 17                      0    0    0    0    0       0       0    0    0    0    0      0   58
  AM 18                      0    0    0    0    0       0       0    0    0    0    0      0   48
  CMP 01                     0    0    0    0    0       0       0    0    0    0    0      0   14
  CMP 02                     0    0    0    0    0       0       0    0    0    0    0      0   17
  CMP 03                     0    0    0    0    0       0       0    0    0    0    0      0   99
  CMP 04                     0    0    0    0    0       0       0    0    0    0    0      0   65
  CMP 05                     0    0    0    0    0       0       0    0    0    0    0      0   49
  CMP 06                     0    0    0    0    0       0       0    0    0    0    0      0   15
  CMP 07                     0    0    0    0    0       0       0    0    0    0    0      0   86
  CMP 08                     0    0    0    0    0       0       0    0    0    0    0      0   86
  CMP 09                     0    0    0    0    0       0       0    0    0    0    0      0  118
  CMP 10                     0    0    0    0    0       0       0    0    0    0    0      0   48
  CMP 11                     0    0    0    0    0       0       0    0    0    0    0      0   31
  CMP 12                     0    0    0    0    0       0       0    0    0    0    0      0   62
  CMP 13                     0    0    0    0    0       0       0    0    0    0    0      0   15
  Sarcoidosis 01             0    0    0    0    0       0       0    0    0    0    0      0  172
  Sarcoidosis 02             0    0    0    0    0       0       0    0    0    0    0      0  124
                
                 MCP2   TC
  Healthy 01        6   73
  Healthy 02        8   34
  Healthy 03       21   42
  Healthy 04        7   78
  Healthy 05        1   22
  Healthy 06        6   39
  Healthy 07        1   15
  Healthy 08        6   19
  AM 01          2483 2824
  AM 02            14   71
  AM 03             2   19
  AM 04             4    3
  AM 05             7   10
  AM 06             1   22
  AM 07             7    5
  AM 08             2   15
  AM 09             0   13
  AM 10             0    4
  AM 11             0    0
  AM 12             7   23
  AM 13            18  212
  AM 14             4   35
  AM 15            23  144
  AM 16           982  466
  AM 17             5   82
  AM 18             9   84
  CMP 01            0    7
  CMP 02            1    4
  CMP 03           16   27
  CMP 04            3   56
  CMP 05            2   24
  CMP 06            1    5
  CMP 07            6   32
  CMP 08            8   29
  CMP 09            3   33
  CMP 10            7   26
  CMP 11            0   17
  CMP 12            2   32
  CMP 13            2   13
  Sarcoidosis 01   35   56
  Sarcoidosis 02   13  121</code></pre>
<pre class="r"><code>table(seurat_immun$TC, seurat_immun$sex)</code></pre>
<pre><code>                    
                     female male unknown
  Tcell High           6945  145       0
  Tcell Intermediate     53  641       0
  Tcell Low             316 1539       0
  Tcell Very Low        154  867    3374</code></pre>
<pre class="r"><code>Idents(seurat_immun) &lt;- seurat_immun$sex

colsex &lt;- c(&quot;#dd1c77&quot;,&quot;#355C7D&quot;)
names(colsex) &lt;- c(&quot;female&quot;, &quot;male&quot;)
            

colcluster&lt;- c( &quot;#fde0dd&quot;,&quot;#B45B5C&quot;,&quot;#dd1c77&quot;)

                    
names(colcluster) &lt;- c(&quot;MCP1&quot;,&quot;MCP2&quot;,&quot;TC&quot;)

Idents(seurat_immun) &lt;- seurat_immun$clusterName

DimPlot(seurat_immun, reduction = &quot;umap&quot;, cols = colcluster, pt.size = 0.3)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/select%20inflammatory%20cells-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table(seuratM$clusterName)</code></pre>
<pre><code>
Cardiomyocyte           Fb1           Fb2           Fb3          INT1       PeriFb1       PeriFb2 
         8222         17699           694           380           882         12754           249 
         VSMC            NC          BEC1          BEC2        AdipoC          MCP1          MCP2 
         2156           905         13639          2365            91          5475          3723 
           TC 
         4836 </code></pre>
<pre class="r"><code>## GSEA and Normalization
seurat_immun &lt;- NormalizeData (object = seurat_immun)
seurat_immun &lt;- FindVariableFeatures(object = seurat_immun)
seurat_immun &lt;- ScaleData(object = seurat_immun, verbose = TRUE)
seurat_immun &lt;- RunPCA(object=seurat_immun, npcs = 30, verbose = FALSE)
seurat_immun &lt;- RunTSNE(object=seurat_immun, reduction=&quot;pca&quot;, dims = 1:20)
seurat_immun &lt;- RunUMAP(object=seurat_immun, reduction=&quot;pca&quot;, dims = 1:20)
seurat_immun &lt;- FindNeighbors(object = seurat_immun, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4, 0.3)
for (i in 1:length(res)) {
  seurat_immun &lt;- FindClusters(object = seurat_immun, resolution = res[i], random.seed = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 14034
Number of edges: 505982

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9343
Number of communities: 11
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 14034
Number of edges: 505982

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8934
Number of communities: 16
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 14034
Number of edges: 505982

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8741
Number of communities: 18
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 14034
Number of edges: 505982

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9149
Number of communities: 14
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 14034
Number of edges: 505982

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9267
Number of communities: 13
Elapsed time: 1 seconds</code></pre>
<pre class="r"><code>table(seurat_immun$orig.ident)</code></pre>
<pre><code>
      
14034 </code></pre>
<pre class="r"><code>table(seurat_immun$clusterName)</code></pre>
<pre><code>
Cardiomyocyte           Fb1           Fb2           Fb3          INT1       PeriFb1       PeriFb2 
            0             0             0             0             0             0             0 
         VSMC            NC          BEC1          BEC2        AdipoC          MCP1          MCP2 
            0             0             0             0             0          5475          3723 
           TC 
         4836 </code></pre>
<pre class="r"><code>## Different resolutions
Idents (seurat_immun) &lt;- seurat_immun$clusterName

DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Umap%20different%20resolutions-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents (seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.25

DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Umap%20different%20resolutions-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents (seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.3

DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Umap%20different%20resolutions-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents (seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.4

DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Umap%20different%20resolutions-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents (seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.6

DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Umap%20different%20resolutions-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents (seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.8

DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Umap%20different%20resolutions-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Immun cell clusters by patient
Idents (seurat_immun) &lt;- (seurat_immun$patient)
DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.3)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Umap%20by%20patient-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table(seurat_immun$patient, seurat_immun$RNA_snn_res.0.3)</code></pre>
<pre><code>                
                    0    1    2    3    4    5    6    7    8    9   10   11   12
  Healthy 01       52  299   22    0   11    0   26    8    4   14    1    1    3
  Healthy 02       27   62  134    2    6    0    8    7    2    5    0    1    4
  Healthy 03       34  442  129    1   25    0   27   25    0   23    2    0    2
  Healthy 04       46  658   67    0   10    0   89   64    3   52    3    3    0
  Healthy 05        8  131   38    0    1    0    1    5    1   15    0    0    8
  Healthy 06       28  365   56    3    0    1    5    4    0   16    0    0   11
  Healthy 07        3   93   20    1    9    0    0    9    4    4    0    0    4
  Healthy 08       14   65   27    1    3    0    4    6    1    5    0    0    0
  AM 01          2000    2   10 1989  766   81   31   44   39   27  215  119    1
  AM 02            47    7  111    1    0    0   20    3   15    9    0    2    1
  AM 03             8    0   92    0    0    0    2    2    8    0    0    0    0
  AM 04             2    0   13    3    0    0    1    2    0    0    0    0    0
  AM 05             2    1   50    0    1    0    7    3    5    5    0    1    0
  AM 06            16    7   15    0    0    0    5    4    5    0    1    0    0
  AM 07             3   26   22    1   11    0    7    1    0    5    1    0    1
  AM 08             5    6   34    0    2    0    7    0    4    9    0    0    1
  AM 09            10    6   45    0    0    0    4    1    2    3    0    0    0
  AM 10             3    9   14    0    0    0    2    3    0    1    0    0    1
  AM 11             0    3   10    0    0    0    0    0    0    1    0    0    0
  AM 12             9    0  116    0    5    0    3    6   11    1    0    0    1
  AM 13           153  246   14    3   10    1   39   16    6   28    7    1    0
  AM 14            20    4   66    1    1    0    8    5    8    7    1    0    0
  AM 15            89    2  271    1    1    0   16   22   34   29    6    6    3
  AM 16           209    0    8   82  216  770    2   78   17   27   22   19    0
  AM 17            68   23   32    1    1    0    5    5    2    0    8    0    0
  AM 18            65   30   18    3    1    0    7    5    0    8    3    1    0
  CMP 01            6    6    7    0    0    0    0    2    0    0    0    0    0
  CMP 02            1    0   12    1    0    0    3    2    2    0    0    0    1
  CMP 03           12   17   47    0    0    0   40    7    8   10    1    0    0
  CMP 04           26    9   46    0    1    0    6   12   20    4    0    0    0
  CMP 05            6   32   13    1    2    0    2    2   15    2    0    0    0
  CMP 06            4    3   11    0    0    0    1    1    0    1    0    0    0
  CMP 07           13    3   84    1    0    0    5    1   11    1    1    0    4
  CMP 08           25    6   84    0    1    0    1    0    4    0    1    0    1
  CMP 09           15    1  102    0    0    0   13    5   10    6    2    0    0
  CMP 10           10    5   46    1    1    0    0    0   15    2    1    0    0
  CMP 11            6    1   29    0    0    0    2    0    6    1    0    0    3
  CMP 12           14    4   54    1    0    0    2    3   16    0    0    0    2
  CMP 13            7    1   14    1    0    0    0    0    7    0    0    0    0
  Sarcoidosis 01   26   21  138    9    0    0   15   11   23   12    5    3    0
  Sarcoidosis 02   46    3  122    0    1    0    4    6   71    2    3    0    0</code></pre>
<pre class="r"><code>table(seurat_immun$orig.ident)</code></pre>
<pre><code>
      
14034 </code></pre>
<pre class="r"><code>table(seurat_immun$patient)</code></pre>
<pre><code>
    Healthy 01     Healthy 02     Healthy 03     Healthy 04     Healthy 05     Healthy 06 
           441            258            710            995            208            489 
    Healthy 07     Healthy 08          AM 01          AM 02          AM 03          AM 04 
           147            126           5324            216            112             21 
         AM 05          AM 06          AM 07          AM 08          AM 09          AM 10 
            75             53             78             68             71             33 
         AM 11          AM 12          AM 13          AM 14          AM 15          AM 16 
            14            152            524            121            480           1450 
         AM 17          AM 18         CMP 01         CMP 02         CMP 03         CMP 04 
           145            141             21             22            142            124 
        CMP 05         CMP 06         CMP 07         CMP 08         CMP 09         CMP 10 
            75             21            124            123            154             81 
        CMP 11         CMP 12         CMP 13 Sarcoidosis 01 Sarcoidosis 02 
            48             96             30            263            258 </code></pre>
<pre class="r"><code>## TC condition for immun cell clusters
Idents (seurat_immun) &lt;- (seurat_immun$TC)

colTC &lt;- c(&quot;#BE3144&quot;,&quot;#de2d26&quot;,&quot;#fc9272&quot;, &quot;#fee0d2&quot;)
names(colTC) &lt;- c(&quot;Tcell High&quot;, &quot;Tcell Intermediate&quot;, &quot;Tcell Low&quot;, &quot;Tcell Very Low&quot;)

DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.3, cols = colTC)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/TC%20condition%20for%20immun%20cell%20clusters-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.3

DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Res%200.6%20find%20all%20markers%20for%20immune%20cells-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Find all markers within AM cohort, only positive
DEGenesimmun &lt;- FindAllMarkers (seurat_immun, only.pos = TRUE, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.01)


#write.table(DEGenesimmun, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenesimmune&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)


table(seurat_immun$clusterName)</code></pre>
<pre><code>
Cardiomyocyte           Fb1           Fb2           Fb3          INT1       PeriFb1       PeriFb2 
            0             0             0             0             0             0             0 
         VSMC            NC          BEC1          BEC2        AdipoC          MCP1          MCP2 
            0             0             0             0             0          5475          3723 
           TC 
         4836 </code></pre>
<pre class="r"><code>table(seurat_immun$RNA_snn_res.0.3)</code></pre>
<pre><code>
   0    1    2    3    4    5    6    7    8    9   10   11   12 
3138 2599 2243 2108 1086  853  420  380  379  335  284  157   52 </code></pre>
<pre class="r"><code>head(DEGenesimmun)</code></pre>
<pre><code>                        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster                    gene
ENSG00000054654.SYNE2       0   2.570780 0.744 0.235         0       0   ENSG00000054654.SYNE2
ENSG00000168685.IL7R        0   4.242586 0.562 0.076         0       0    ENSG00000168685.IL7R
ENSG00000116824.CD2         0   2.436738 0.497 0.134         0       0     ENSG00000116824.CD2
ENSG00000127152.BCL11B      0   2.932358 0.448 0.095         0       0  ENSG00000127152.BCL11B
ENSG00000072818.ACAP1       0   2.319228 0.484 0.152         0       0   ENSG00000072818.ACAP1
ENSG00000215252.GOLGA8B     0   2.649740 0.462 0.135         0       0 ENSG00000215252.GOLGA8B</code></pre>
<pre class="r"><code>#View(DEGenesimmun)</code></pre>
<pre class="r"><code>## Marker Genes 

get_full_gene_name &lt;- function(gene, obj){
  return(grep(gene,rownames(obj), value =TRUE))
}
get_full_gene_name(&quot;XIST&quot;,seurat_immun) </code></pre>
<pre><code>[1] &quot;ENSG00000229807.XIST&quot;</code></pre>
<pre class="r"><code>FeaturePlot(seuratM, features = get_full_gene_name(&quot;XIST&quot;,seuratM) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# NK</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000081237.PTPRC&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000168685.IL7R&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000010610.CD4&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000172116.CD8B&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CD8A&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000116824.CD2&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000167286.CD3D&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000198851.CD3E&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000178562.CD28&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CD96&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;RIPOR2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000227507.LTB&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Cluster 1 spec</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000069667.RORA&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Cluster 1</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;PATJ&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-15.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#cl 6: CD8+ T cells proliferating ASPM, RRM2, KNL1, FANC; TOP2A, CEP152, DIAPH3, KIF18B, KIF14, CDCA3, TROAP, SAC3D1, MS4A7, ZNF683, GEN1, SAC8D1, SKA3, RAD54L, UWOMT. SPC25, BRACA1, BUB1B, Gen1, FoxM1, SKA3, CCNB1, SAC8D1, FANCI, CIT, SPAG5, ZNF683, E2F2, ZWINT


FeaturePlot(seurat_immun, features = &quot;ENSG00000163599.CTLA4&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;LAG3&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# exhaustion</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-17.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TIGIT&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;PDCD1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-19.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;HAVCR2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;FOXP3&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Tregs</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-21.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000136634.IL10&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-22.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TGFB1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-23.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;STAT5A&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-24.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;STAT5B&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-25.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;RORC&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Th17 (and also IL3)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-26.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IL22&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-27.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;STAT3&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-28.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IL17RA&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# did not find IL17A, only R</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-29.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CCR6&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-30.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000162594.IL23R&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-31.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000138795.LEF1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-32.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000109906.ZBTB16&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))   </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-33.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000165810.BTNL9&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))    # MAIT</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-34.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TBX21&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Th1</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-35.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IFNG&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-36.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IL12RB1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-37.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CXCR3&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-38.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;STAT4&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-39.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;GATA3&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Th2</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-40.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;BCL6&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Tfh</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-41.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IL21&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-42.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000163600.ICOS&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-43.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CCR5&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-44.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000172116.CD8B&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-45.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;KIR2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-46.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;GZMB&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-47.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;PRF1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-48.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000104970.KIR3DX1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-49.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000111537.IFNG&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-50.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TRGV9&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) #yd Tcells</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-51.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TRDV2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-52.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TRGC1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-53.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;KIR3DL1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# NK</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-54.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TXK&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-55.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;NCAM1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) #CD56,NK marker</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-56.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;KLRD1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # CD94</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-57.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;FCGR3A&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) #CD16</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-58.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;GZMB&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-59.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;NCR1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# NK</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-60.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000189430.NCR1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-61.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000004468.CD38&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-62.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;JCHAIN&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# plasma cells</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-63.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000057657.PRDM1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-64.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000211899.IGHM&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))## proliferating cells </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-65.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000156738.MS4A1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # CD20</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-66.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CD19&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-67.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000148773.MKI67&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))## proliferating cells</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-68.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TOP2A&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-69.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;SELL&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # SELL = CD62L# naive</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-70.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;PTPRC&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) #CD45</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-71.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CCR7&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-72.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CXCL12&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-73.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000110448.CD5&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-74.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FeaturePlot(seurat_immun, features = &quot;ENSG00000197471.SPN&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # CD43 (sialoprotein of leukos, regulates T cell traffickinig to LN)

FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CLEC9A&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# DC</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-75.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;XCR1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-76.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;SEPTIN3&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-77.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;WDFY4&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-78.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CCR9&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-79.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;PACSIN1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-80.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000182578.CSF1R&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-81.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;VCAM1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-82.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;MERTK&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-83.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;MS4A4A&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-84.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;FKBP5&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-85.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;ITGAM&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # CD11b</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-86.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;STAT1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-87.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TIMD4&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# TLFpos res.Macrophages</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-88.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000177575.CD163&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-89.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;LYVE1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-90.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;FOLR2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-91.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;GAS6&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-92.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;NINJ1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-93.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;MRC1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-94.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IGF1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-95.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;SELENOP&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-96.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IGFBP4&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-97.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## features of resident mph cluster 2 vs. cluster 1 (LYVE1+): SLC40A1 (iron transport), ITGAX (integrin), FCN1 (PRR), CD300E, ITGA4 (integrin), SPN, UICLM, CD1C, CLEC4F (pDCs), PCNX2, SOX4 Cluster 2 resident mph

FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;SLC40A1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-98.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;ITGAX&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-99.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;FCN1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-100.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CD1C&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-101.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CLEC4F&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-102.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;PCNX2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-103.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;SOX4&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-104.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;APOE&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# MHC-II hi res. MCPs</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-105.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CD14&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-106.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CST3&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-107.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TREM2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-108.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;LILRA5&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-109.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;C5orf1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-110.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;STAB1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-111.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CCR2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# CCR2+ MCP</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-112.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CD52&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-113.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;HLA-DMB&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-114.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;HLA-A&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-115.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;HLA-DQB1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-116.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;S100A6&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-117.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000163191.S100A11&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-118.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CORO1A&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-119.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TMEM119&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-120.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000163508.EOMES&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))    </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-121.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000170458.CD14&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-122.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CXCR3&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-123.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CXCL9&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# inflammatory macrophages</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-124.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CXCL10&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-125.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CXCL11&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-126.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IL23A&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-127.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TREM2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-128.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;MMP9&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-129.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IL1B&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-130.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IL6R&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-131.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000232810.TNF&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-132.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000136244.IL6&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-133.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TLR4&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-134.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;XIST&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-135.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IL32&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-136.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000183160.TMEM119&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # also found on microglia</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-137.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CCL18&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-138.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CPM&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-139.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;ZBTB16&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-140.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;ALOX15B&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-141.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;HCLS1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-142.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;ADAMTS2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-143.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TBC1D16&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-144.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;SLC31A2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # copper transporter</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-145.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;LPAR6&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-146.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;SPRED1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# inhibits growth factor activation, negatively regulates HSC </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-147.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;ABCA6&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # macrophage lipid transport</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-148.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE genes resident macrophages cl 2 and 7
## cl 7 ZBTB16, CPM, PDK4, ALOX15B, CCL18, SLCA3, KCNMA1, ADAMTS2, HCLS1, APOE1, ZNF812P, HCLS1, TBC1D16, SLC31A2, TFBI, 
## cl2: LPAR6, ABCA6, SPRED1          

## ZBTB16 Acts as a transcriptional repressor (PubMed:10688654, 24359566). Transcriptional repression may be mediated through recruitment of histone deacetylases to target promoters (PubMed:10688654). May play a role in myeloid maturation and in the development and/or maintenance of other differentiated tissues. Probable substrate-recognition component of an E3 ubiquitin-protein ligase complex which mediates the ubiquitination and subsequent proteasomal degradation of target proteins


FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CD72&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # activated macrophages, cluster 0</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-149.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;LY86&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-150.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;SIRPA&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-151.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IL10RA&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-152.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000112799.LY86&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-153.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># cluster activated macrophages vs. inflammatory do not express CXCL10 and 9; FAM118A, HLADRB5, HLA-V --&gt; activated macrophages
#0 vs.4: CXCL9, CXCL10, MT2A, CXCL11, CCL8, CD274, MT1G, HLA-DQA2, TIPA, C3, IL31RA, CALHM6, MT1H, MT1G, MT1X, WARS1, GBP1


FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;DAB2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-154.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CD274&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # inflammatory mch, cl4</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-155.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;MT1H&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # glucocroticoid reg.metallothioneins</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-156.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CCL8&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-157.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;WARS1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#stress response, angiogenesis</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-158.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000117228.GBP1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-159.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000125148.MT2A&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-160.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;FAM118A&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # membrane; cl0</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-161.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;HLA-V&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# cl0</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-162.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;HLA-DRB5&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-163.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Eosinophilic markers
FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;EOMES&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#also CD8, MBP1, CD193 not found</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-164.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000172156.CCL11&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-165.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = &quot;ENSG00000105366.SIGLEC8&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-166.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;IL5RA&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-167.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CCL24&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-168.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;CCR3&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-169.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;PRG2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-170.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;GATA1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-171.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Mast cells # KIT/SLC18A2/GATA2/RAB44/ENPP3/BTK/ADGRE2, NDST2

FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;RAB44&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-172.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;BTK&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-173.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;SLC18A2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-174.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;GATA2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-175.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;NDST2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-176.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;ADGRE2&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-177.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;ENPP3&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-178.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;KIT&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # CD117</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-179.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seurat_immun, features = get_full_gene_name(&quot;TPSAB1&quot;,seurat_immun) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Feature%20plots%20with%20markers-180.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## GSEA embedding of immune cells
Idents(seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.3

seurat_immun$clusterName &lt;- &quot;clusterName&quot;
Idents(seurat_immun) &lt;- seurat_immun$clusterName
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;0&quot; )] &lt;- &quot;Tcells&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;1&quot; )] &lt;- &quot;LYVE1pos TRM&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;2&quot; )] &lt;- &quot;LYVE1neg TRM&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;3&quot; )] &lt;- &quot;inflammatory MPH1&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;4&quot; )] &lt;- &quot;prolif Tcells&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;5&quot; )] &lt;- &quot;inflammatory MPH2&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;6&quot; )] &lt;- &quot;Fb&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;7&quot; )] &lt;- &quot;BEC&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;8&quot; )] &lt;- &quot;NK&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;9&quot; )] &lt;- &quot;Mural cells&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;10&quot; )] &lt;- &quot;Bcells&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;11&quot; )] &lt;- &quot;DC&quot;
seurat_immun$clusterName[which(seurat_immun$RNA_snn_res.0.3 %in% &quot;12&quot; )] &lt;- &quot;Mast cells&quot;


colclusterName&lt;- c(&quot;#dd1c77&quot;,&quot;#fa9fb5&quot;, &quot;#fde0dd&quot;, &quot;#66C2A5&quot;, &quot;#c7eae5&quot;, &quot;#2c7fb8&quot;,&quot;#b3cde3&quot;,  &quot;darkblue&quot;, &quot;#D53E4F&quot;, &quot;orange&quot;,&quot;#bcbddc&quot;, &quot;#756bb1&quot;, &quot;#8c510a&quot;)
              
             
names(colclusterName) &lt;- c(&quot;Tcells&quot;,&quot;prolif Tcells&quot;,&quot;NK&quot;, &quot;Bcells&quot;,&quot;DC&quot;, &quot;LYVE1pos TRM&quot;, &quot;LYVE1neg TRM&quot;, &quot;inflammatory MPH1&quot;, &quot;inflammatory MPH2&quot;, &quot;Fb&quot;,&quot;Mural cells&quot;, &quot;BEC&quot;, &quot;Mast cells&quot;)

colpat_sub&lt;- c(&quot;#dfc27d&quot;,&quot;#BE3144&quot;,&quot;#355C7D&quot;, &quot;#779d8d&quot;)

seurat_immun$clusterName &lt;- factor(seurat_immun$clusterName, levels = c(&quot;Tcells&quot;,&quot;prolif Tcells&quot;,&quot;NK&quot;, &quot;Bcells&quot;,&quot;DC&quot;, &quot;LYVE1pos TRM&quot;, &quot;LYVE1neg TRM&quot;, &quot;inflammatory MPH1&quot;, &quot;inflammatory MPH2&quot;, &quot;Fb&quot;,&quot;Mural cells&quot;, &quot;BEC&quot;,&quot;Mast cells&quot;))

Idents(seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.3
DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Gene%20Set%20Enrichtment%20Analysis%20rename%20clusters%20with%20UMAP-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seurat_immun) &lt;- seurat_immun$clusterName
DimPlot(seurat_immun, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Gene%20Set%20Enrichtment%20Analysis%20rename%20clusters%20with%20UMAP-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Create Dotplot/Heatmap

seurat_immun$clusterName &lt;- factor(seurat_immun$clusterName, levels = c(&quot;Tcells&quot;,&quot;prolif Tcells&quot;,&quot;NK&quot;, &quot;Bcells&quot;,&quot;DC&quot;, &quot;LYVE1pos TRM&quot;, &quot;LYVE1neg TRM&quot;, &quot;inflammatory MPH1&quot;, &quot;inflammatory MPH2&quot;, &quot;Fb&quot;,&quot;Mural cells&quot;, &quot;BEC&quot;,&quot;Mast cells&quot;))

genes &lt;- data.frame(gene=rownames(seurat_immun)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID=rev(c(&quot;IL7R&quot;, &quot;CD2&quot;, &quot;CD3E&quot;, &quot;CD4&quot;, &quot;MKI67&quot;, &quot;TOP2A&quot;, &quot;CD8A&quot;,&quot;CD8B&quot;, &quot;GZMB&quot;, &quot;TBX21&quot;, &quot;IFGN&quot;, &quot;STAT4&quot;, &quot;CTLA4&quot;, &quot;PDCD1&quot;, &quot;TIGIT&quot;, &quot;NCAM1&quot;, &quot;KLRG1&quot;, &quot;TXK&quot;, &quot;LAG3&quot;, &quot;RORA&quot;, &quot;FOXP3&quot;, &quot;IKZF2&quot;, &quot;RORC&quot;, &quot;IL17A&quot;, &quot;CCR6&quot;, &quot;TRGV9&quot;, &quot;GATA3&quot;, &quot;SELL&quot;, &quot;CXCR5&quot;, &quot;BCL6&quot;, &quot;IGHM&quot;, &quot;MS4A1&quot;, &quot;CD79A&quot;, &quot;JCHAIN&quot;, &quot;CLEC9A&quot;, &quot;XCR1&quot;, &quot;CD163&quot;, &quot;HLA-DMB&quot;, &quot;HLA-A&quot;, &quot;MRC1&quot;, &quot;STAB1&quot;, &quot;CSF1R&quot;, &quot;MERTK&quot;, &quot;LYVE1&quot;, &quot;IL23A&quot;, &quot;IL6R&quot;, &quot;CD14&quot;, &quot;GBP5&quot;, &quot;IL10RA&quot;, &quot;CXCL9&quot;, &quot;CXCL10&quot;, &quot;EOMES&quot;, &quot;IL5RA&quot;, &quot;CEACAM8&quot;, &quot;ITGB2&quot;, &quot;XIST&quot;, &quot;COL6A3&quot;, &quot;C7&quot;, &quot;PDGFRA&quot;, &quot;PDGFRB&quot;,&quot;ITGA7&quot;,&quot;RGS5&quot;, &quot;CDH19&quot;, &quot;VWF&quot;, &quot;SLC18A2&quot;, &quot;KIT&quot;)))  %&gt;% left_join(., genes, by=&quot;geneID&quot;) %&gt;% distinct(geneID, .keep_all = TRUE)

DotPlot(seurat_immun, features = selGenes, group.by= &quot;clusterName&quot;) + RotatedAxis() + scale_color_viridis(option=&quot;T&quot;) + coord_flip()</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Heatmap-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Relative abundance per disease condition
datList &lt;- NULL
for(con in unique(seurat_immun$pat_sub)){
  seuratSub &lt;- subset(seurat_immun, pat_sub==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(diseaseCond=con)
  datList[[con]] &lt;- dat_con
  
}</code></pre>
<pre><code>[1] 39114  9078
[1] 39114  1061
[1] 39114  3374
[1] 39114   521</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

order_patsub &lt;- c(&quot;Healthy&quot;, &quot;AM&quot;, &quot;CMP&quot;, &quot;Sarcoidosis&quot;)

ggbarplot(dat_all, x = &quot;diseaseCond&quot;, y = &quot;percent&quot;, fill = &quot;Var1&quot;, palette = colclusterName, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, xlab = &quot;condition&quot;, ylab = &quot;frequency&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + 
  scale_x_discrete(limits = order_patsub)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Relative%20abundance-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Relative abundance per patient
datList &lt;- NULL
for(con in unique(seurat_immun$patient)){
  seuratSub &lt;- subset(seurat_immun, patient==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 39114   216
[1] 39114   112
[1] 39114    21
[1] 39114    75
[1] 39114    53
[1] 39114    78
[1] 39114    68
[1] 39114    71
[1] 39114    33
[1] 39114    14
[1] 39114   152
[1] 39114   524
[1] 39114   121
[1] 39114  5324
[1] 39114   480
[1] 39114  1450
[1] 39114   145
[1] 39114   141
[1] 39114    21
[1] 39114    22
[1] 39114   142
[1] 39114   154
[1] 39114    81
[1] 39114    48
[1] 39114    96
[1] 39114    21
[1] 39114   124
[1] 39114    75
[1] 39114   124
[1] 39114   123
[1] 39114    30
[1] 39114   441
[1] 39114   258
[1] 39114   710
[1] 39114   995
[1] 39114   208
[1] 39114   489
[1] 39114   147
[1] 39114   126
[1] 39114   263
[1] 39114   258</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

## Order of patient levels 
patient_order &lt;- c(&quot;Healthy 01&quot;, &quot;Healthy 02&quot;, &quot;Healthy 03&quot;, &quot;Healthy 04&quot;, &quot;Healthy 05&quot;, 
                   &quot;Healthy 06&quot;, &quot;Healthy 07&quot;, &quot;Healthy 08&quot;, &quot;AM 01&quot;, &quot;AM 02&quot;, 
                   &quot;AM 03&quot;, &quot;AM 04&quot;, &quot;AM 05&quot;, &quot;AM 06&quot;, &quot;AM 07&quot;, 
                   &quot;AM 08&quot;, &quot;AM 09&quot;, &quot;AM 10&quot;, &quot;AM 11&quot;, &quot;AM 12&quot;, &quot;AM 13&quot;, &quot;AM 14&quot;, 
                   &quot;AM 15&quot;, &quot;AM 16&quot;, &quot;AM 17&quot;, &quot;AM 18&quot;, &quot;CMP 01&quot;, &quot;CMP 02&quot;, &quot;CMP 03&quot;, 
                   &quot;CMP 04&quot;, &quot;CMP 05&quot;, &quot;CMP 06&quot;, &quot;CMP 07&quot;, &quot;CMP 08&quot;, &quot;CMP 09&quot;, 
                   &quot;CMP 10&quot;, &quot;CMP 11&quot;, &quot;CMP 12&quot;, &quot;CMP 13&quot;, &quot;Sarcoidosis 01&quot;, &quot;Sarcoidosis 02&quot;)

#View(dat_all)



## Convert the patient column to a factor with the desired order
dat_all$patient &lt;- factor(dat_all$patient, levels = patient_order)

## plot abundance
ggbarplot(dat_all, x= &quot;patient&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;, palette = colclusterName )  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/relative%20abundance%20per%20patient-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes immune cells, find all markers, positive and negative
Idents(seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.3

DEGenes_IC &lt;- FindAllMarkers(seurat_immun, only.pos = F, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.01)

DEGenes_IC &lt;- DEGenes_IC %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

#View(DEGenes_IC)</code></pre>
<pre class="r"><code>## Volcano Plot Cluster 0, T cells

cl0 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;0&quot;)

res_0 &lt;- as.data.frame(cl0)

EnhancedVolcano(res_0, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_0$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes T cells&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## IL7R, ETS1, SCML4, DGKA, RIPOR2, MIAT, SYNE2, CD6, SPOCK2, ATP8B2, CAMK4, ITK, TMC8, GOLGA8A, KLRG1, TCF7LCK, STAT4, TRBC2, RBL2, LEF1, CD247, LY9, SYTL2, CASP8, RASGRP2, SAMD3, LTK, CD40LG; ENO2, SIDT1, CASK, GNB5, HELB, KCNA3</code></pre>
<pre class="r"><code>## Volcano Plot Cluster 1, TRM LYVE1 pos.

cl1 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;1&quot;)

res_1 &lt;- as.data.frame(cl1)

EnhancedVolcano(res_1, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_1$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes LYVE1pos tissue resident macrophages&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl1%20lyve1pos.%20TRM-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## STAT1, SCO2, SAMD9L, NR1H3, SCO2, CLEC12A, FMNL2, PSTPIP2, MYOF, EPST1, CD1D, RUFY4, PFKFB3, ADA2, DYSF, LILRB4</code></pre>
<pre class="r"><code>## Volcano Plot Cluster 2: TRM, LYVE1 neg.

cl2 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;2&quot;)

res_2 &lt;- as.data.frame(cl2)

EnhancedVolcano(res_2, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_2$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes LYVE1 neg tissue resident macrophages&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl2%20macrophage%20TRM-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DAB2, FRMD4B, WWP1, NRP1, SLC40A1, LTC4S, HRH1, CD163L1, MS4A4A, IGF1, HRH1, CCDC152, FOLR2, NAV2, PLEKHG5, HPGDS, ARHGAP24, ABCA6, MEGF6, RENBP, KCNJ5, GGTA1, PID1, FAM177B, CNRIP1, KHDRBS2, CRHBP, RENBP, GAS2L3 </code></pre>
<pre class="r"><code>## Volcano Plot Cluster 3: activated inflammatory Macrophages, IL-6 regulation high

cl3 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;3&quot;)

res_3 &lt;- as.data.frame(cl3)

EnhancedVolcano(res_3, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_3$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes activated macrophages (CXCL9 neg)&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl3%20-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># MRC1, MS4A</code></pre>
<pre class="r"><code>## Volcano Plot Cluster 4: proliferating T cells

cl4 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;4&quot;)

res_4 &lt;- as.data.frame(cl4)

EnhancedVolcano(res_4, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_4$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Proliferating T cells&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl4%20-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot: inflammatory macrophages, CXCL9 pos.

cl5 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;5&quot;)

res_5 &lt;- as.data.frame(cl5)

EnhancedVolcano(res_5, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_5$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Inflammatory Macrophages, CXCL9 pos.&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl5%20-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot Fibroblasts: 

cl6 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;6&quot;)

res_6 &lt;- as.data.frame(cl6)

EnhancedVolcano(res_6, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_6$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Fibroblasts&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl6%20-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ABCA9, APCDD1, COL6A11, MEG3, FREM1, DIO3OS, DNM3O&quot;, PRELP, COL4A1, PDGFRB, PXDN, HIF3A, VCAN, NFIB, SVEP1, NFATC4, LAMC2, TIMP3, FOXP2, GFPT2, RGR, FAT1, CNBD1TM6SF2, PLEHHA5, VCAN, SPRY1 LAMC1</code></pre>
<pre class="r"><code>## Volcano Plot Cluster 7: Endothelial cells

cl7 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;7&quot;)

res_7 &lt;- as.data.frame(cl7)

EnhancedVolcano(res_7, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_7$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Endothelial cells&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl7,%20BEC%20-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot Cluster 8: NK cells

cl8 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;8&quot;)

res_8 &lt;- as.data.frame(cl8)

EnhancedVolcano(res_8, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_8$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes NK cells&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl8%20NK-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot Cluster 9: Mural cells: VSMC markers and Pericyte markers

cl9 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;9&quot;)

res_9 &lt;- as.data.frame(cl9)

EnhancedVolcano(res_9, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_9$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Mural cells&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl9,%20mural%20cells-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot Cluster 10: B cells

cl10 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;10&quot;)

res_10 &lt;- as.data.frame(cl10)

EnhancedVolcano(res_10, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_10$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes B cells&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl10%20,%20B%20cells-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot Cluster 11: Dendritic cells

cl11 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;11&quot;)

res_11 &lt;- as.data.frame(cl10)

EnhancedVolcano(res_11, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_11$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Cluster DCs&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl11,%20Dendritic%20cells-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Volcano Plot Cluster 12: Mast cells

cl12 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;12&quot;)

res_12 &lt;- as.data.frame(cl12)

EnhancedVolcano(res_12, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_12$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Mast cells&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20Cl12%20mast%20cells-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes immune cells, find all markers, positive and negative
Idents(seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.3

DEGenes_IC &lt;- FindAllMarkers(seurat_immun, only.pos = F, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.01)

DEGenes_IC &lt;- DEGenes_IC %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

#View(DEGenes_IC)</code></pre>
<pre class="r"><code>## Difference between resident macrophages: Clusters 1 and 2
res_mph &lt;- subset(seurat_immun, RNA_snn_res.0.3 %in% c(&quot;1&quot;, &quot;2&quot;)) 

res_mph &lt;- FindAllMarkers(res_mph, only.pos = F, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.01)

res_mph &lt;- res_mph %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

#View(res_mph)


res_mph2 &lt;- res_mph %&gt;% filter(cluster == &quot;1&quot;)

res_2 &lt;- as.data.frame(res_mph2)

## Plot DE Genes between resident macrophages

EnhancedVolcano(res_2, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_2$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes Resident Macrophages Cluster 1 vs. Cluster 2&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20DEGenes%20resident%20mph-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Cluster 2 SLC40A1, ITGAX, FCN1, CD300E, ITGA4, SPN, UICLM, CD1C, CLEC4F, PCNX2, SOX4
## Cluster 1 is LYVE1 positive, Cluster 2 is not; Cluster 2 has more phagocytotic activitiy</code></pre>
<pre class="r"><code>## Difference between T cells: Clusters 8 and 4
res_tcell &lt;- subset(seurat_immun, RNA_snn_res.0.3 %in% c(&quot;8&quot;, &quot;4&quot;)) 

res_tcell &lt;- FindAllMarkers(res_tcell, only.pos = F, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.01)

res_tcell &lt;- res_tcell %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

#View(res_mph)


res_tcell8 &lt;- res_tcell %&gt;% filter(cluster == &quot;8&quot;)

res_tcell8 &lt;- as.data.frame(res_tcell8)

EnhancedVolcano(res_tcell8, 
                x = &#39;avg_log2FC&#39;,
                y = &#39;p_val_adj&#39;,
                lab = res_tcell8$gene,
                # xlim =c(-6, 6), 
                title = &#39;DE Genes T cells Cluster 8 vs. Cluster 4&#39;,
                xlab = bquote(~Log[2]~ &#39;fold change&#39;),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = &#39;twodash&#39;,
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c(&#39;Not sig.&#39;,&#39;Log (base 2) FC&#39;,&#39;p-value&#39;,
      &#39;p-value &amp; Log (base 2) FC&#39;),
    legendPosition = &#39;right&#39;,
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Volcano%20Plot%20DE%20genes%20between%20resident%20mph-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Cluster 2 SLC40A1, ITGAX, FCN1, CD300E, ITGA4, SPN, UICLM, CD1C, CLEC4F, PCNX2, SOX4
## Cluster 1 is LYVE1 positive, Cluster 2 is not; Cluster 2 has more phagocytotic activitiy</code></pre>
<pre class="r"><code>## DE Genes immune cells, find all markers, positive and negative

Idents(seurat_immun) &lt;- seurat_immun$RNA_snn_res.0.3
DEGenes_ICall &lt;- FindAllMarkers(seurat_immun, only.pos = T, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.01)

DEGenes_IC &lt;- DEGenes_ICall %&gt;% 
  mutate(Gene=gene) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, Gene))

#View(DEGenes_IC)</code></pre>
<pre class="r"><code>## DE Genes T cells Dotplot
DEGenes0 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;0&quot;)

ego0 &lt;- enrichGO(gene = unique(DEGenes0$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego0 &lt;- setReadable(ego0, OrgDb = org.Hs.eg.db)


dotplot(ego0, showCategory=20, title = &quot;Cluster 0&quot;, font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Gene%20Ontology%20and%20GSEA-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes T cells Barplot
barplot(ego0, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Cluster 0&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%200%20T%20cells,%20barplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes resident macrophages, Dotplot
DEGenes1 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;1&quot;)

ego1 &lt;- enrichGO(gene = unique(DEGenes1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego1 &lt;- setReadable(ego1, OrgDb = org.Hs.eg.db)

dotplot(ego1, showCategory=20, title = &quot;Tissue resident macrophages 1: LYVE1 pos.&quot;, font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%201%20resident%20mcp,%20dotplot-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes resident macrophages, barplot
barplot(ego1, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Tissue resident macrophages 1: LYVE1 pos.&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%201%20resident%20mcp,%20barplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## resident macrophages Cluster 2, Dotplot
DEGenes2 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;2&quot;)

ego2 &lt;- enrichGO(gene = unique(DEGenes2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego2 &lt;- setReadable(ego2, OrgDb = org.Hs.eg.db)

dotplot(ego2, showCategory=20, title = &quot;Tissue resident macrophages 2: LYVE1neg&quot;, font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%202%20resident%20mcp%20dotplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## resident macrophages Cluster 2, Dotplot
barplot(ego2, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Tissue resident macrophages 2: LYVE1neg&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%202%20resident%20mcp%20barplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## activated macrophages with Dotplot

DEGenes3 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;3&quot;)

ego3 &lt;- enrichGO(gene = unique(DEGenes3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego3 &lt;- setReadable(ego3, OrgDb = org.Hs.eg.db)

#View(ego3)

dotplot(ego3, showCategory=20, title = &quot;Inflammatory macrophages, regulation of IL-6 production&quot;, font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%203%20activated%20macrophages-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># IL6 pathway associated genes CD74/LILRB2/P2RX7/BTK/CLEC7A/LGALS9/SCIMP/IL17RA/SIGLEC16/IFIH1/CARD9/CYBA/AIF1/TLR2/ARHGEF2/MYD88/UNC93B1/NOD2/POU2F2/IL6R/RIGI/TYROBP/TMEM106A/TLR6/TLR8/PTAFR/TLR1/NOD1/TNF/TLR4/TLR7/STAT3/SYK</code></pre>
<pre class="r"><code>## activated macrophages barplot
barplot(ego3, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;IL-6 producing activated macrophages&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%203%20activated%20macrophages%20Barplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes Proliferating T cells with Dotplot
DEGenes4 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;4&quot;)

ego4 &lt;- enrichGO(gene = unique(DEGenes4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego4 &lt;- setReadable(ego4, OrgDb = org.Hs.eg.db)

dotplot(ego4, showCategory=20, title = &quot;Proliferating T cells&quot;, font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%204,%20proliferating%20inflammatory%20T%20cells,%20dotplot-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes Proliferating T cells with barplot
barplot(ego4, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Proliferating T cells: mixed Th1 and CD8+&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%204,%20barplot%20proliferating%20inflammatory%20T%20cells-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes inflammatory macrophages with Dotplot
DEGenes5 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;5&quot;)

ego5 &lt;- enrichGO(gene = unique(DEGenes5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego5 &lt;- setReadable(ego5, OrgDb = org.Hs.eg.db)

#View(ego5@result)

dotplot(ego5, showCategory=20, title = &quot;CXCL9+ Inflammatory Macrophages&quot;, font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%20inflammatory%20mph%20dotplot-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes inflammatory macrophages
barplot(ego5, 
        drop = TRUE, 
        showCategory = 7, 
        title = &quot;CXCL9+ Inflammatory Macrophages&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%205%20barplot%20inflammatory%20macrophages-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes Fibroblasts

DEGenes6 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;6&quot;)

ego6 &lt;- enrichGO(gene = unique(DEGenes6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego6 &lt;- setReadable(ego6, OrgDb = org.Hs.eg.db)


#dotplot(ego6, showCategory=30, title = &quot;Fibroblasts&quot;, font.size = 10)

barplot(ego6, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Fibroblasts&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%206-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes BECs
DEGenes7 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;7&quot;)

ego7 &lt;- enrichGO(gene = unique(DEGenes7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego7 &lt;- setReadable(ego7, OrgDb = org.Hs.eg.db)

#dotplot(ego7, showCategory=20, title = &quot;Cluster 7&quot;, font.size = 10)

barplot(ego7, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Endothelial Cells&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20cl7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes Natural Killer Cells
DEGenes8 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;8&quot;)

ego8 &lt;- enrichGO(gene = unique(DEGenes8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego8 &lt;- setReadable(ego8, OrgDb = org.Hs.eg.db)

dotplot(ego8, showCategory=20, title = &quot;NKs&quot;, font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%208%20Dotplot%20NK-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(ego8, 
        drop = TRUE, 
        showCategory = 10, 
        title = &quot;Natural Killer cells&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%208%20barplot%20NK-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes Mural Cells (pericytes and VSMC)
DEGenes9 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;9&quot;)

ego9 &lt;- enrichGO(gene = unique(DEGenes9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego9 &lt;- setReadable(ego9, OrgDb = org.Hs.eg.db)

#dotplot(ego9, showCategory=20, title = &quot;Cluster 9&quot;, font.size = 10)

barplot(ego1, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Mural Cells&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%209-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes B cells
DEGenes10 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;10&quot;)

ego10 &lt;- enrichGO(gene = unique(DEGenes10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego10 &lt;- setReadable(ego10, OrgDb = org.Hs.eg.db)

#dotplot(ego1, showCategory=20, title = &quot;Cluster 10&quot;, font.size = 10)

barplot(ego10, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;B cells&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%2010-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes Dendritic Cells
DEGenes11 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;11&quot;)

ego11 &lt;- enrichGO(gene = unique(DEGenes11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego11 &lt;- setReadable(ego11, OrgDb = org.Hs.eg.db)

#dotplot(ego11, showCategory=20, title = &quot;Cluster 11&quot;, font.size = 10)

barplot(ego11, 
        drop = TRUE, 
        showCategory = 5, 
        title = &quot;Dendritic cells&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%2011-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## DE Genes Mast Cells
DEGenes12 &lt;- DEGenes_IC %&gt;% filter(cluster == &quot;12&quot;)

ego12 &lt;- enrichGO(gene = unique(DEGenes12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

ego12 &lt;- setReadable(ego12, OrgDb = org.Hs.eg.db)

dotplot(ego12, showCategory=20, title = &quot;Cluster 12&quot;, font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%2012-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(ego12, 
        drop = T, 
        showCategory = 7, 
        title = &quot;Mast cells&quot;,
        font.size = 10)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/GSEA%20Cluster%2012-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#View(ego12)

# KIT/SLC18A2/GATA2/RAB44/ENPP3/BTK/ADGRE2, NDST2</code></pre>
<pre class="r"><code>## Convert seurat_immun to single cell experiment (sce)
sce &lt;- as.SingleCellExperiment(seurat_immun)
genes &lt;- data.frame(geneID=rownames(sce)) %&gt;% mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;, geneID))
pal = colorRampPalette(c(&quot;#053061&quot;, &quot;#2166ac&quot;, &quot;#f7f7f7&quot;, &quot;#f4a582&quot;,&quot;orange&quot;, &quot;#d95f0e&quot;,&quot;#b2183c&quot;)) </code></pre>
<pre class="r"><code>## Subset healthy and convert seurat_Healthy to sce
seurat_Healthy &lt;- subset(seurat_immun, pat_sub == &quot;Healthy&quot;)
table(seurat_Healthy$pat_sub)</code></pre>
<pre><code>
    Healthy          AM         CMP Sarcoidosis 
       3374           0           0           0 </code></pre>
<pre class="r"><code>sceHealthy &lt;- as.SingleCellExperiment(seurat_Healthy)</code></pre>
<pre class="r"><code>## subset Acute Myocarditis (AM) from all disease conditions and convert seurat object to sce 
seuratAM &lt;- subset(seurat_immun, pat_sub == &quot;AM&quot;)
table(seuratAM$pat_sub)</code></pre>
<pre><code>
    Healthy          AM         CMP Sarcoidosis 
          0        9078           0           0 </code></pre>
<pre class="r"><code>sceAM &lt;- as.SingleCellExperiment(seuratAM)

## subgroup AM into different T cell conditions and convert into sce

seuratAM_TChigh &lt;- subset(seurat_immun, TC == &quot;Tcell High&quot;)
table(seuratAM_TChigh$TC)</code></pre>
<pre><code>
        Tcell High Tcell Intermediate          Tcell Low     Tcell Very Low 
              7090                  0                  0                  0 </code></pre>
<pre class="r"><code>sceAM_TChigh &lt;- as.SingleCellExperiment(seuratAM_TChigh)

seuratAM_TCint &lt;- subset(seurat_immun, TC == &quot;Tcell Intermediate&quot;)
table(seuratAM_TCint$TC)</code></pre>
<pre><code>
        Tcell High Tcell Intermediate          Tcell Low     Tcell Very Low 
                 0                694                  0                  0 </code></pre>
<pre class="r"><code>sceAM_TCint &lt;- as.SingleCellExperiment(seuratAM_TCint)

seuratAM_TClow &lt;- subset(seurat_immun, TC == &quot;Tcell Low&quot;)
table(seuratAM_TClow$TC)</code></pre>
<pre><code>
        Tcell High Tcell Intermediate          Tcell Low     Tcell Very Low 
                 0                  0               1855                  0 </code></pre>
<pre class="r"><code>sceAM_TClow &lt;- as.SingleCellExperiment(seuratAM_TClow)

seuratAM_TCvlow &lt;- subset(seurat_immun, TC == &quot;Tcell Very Low&quot;)
table(seuratAM_TCvlow$TC)</code></pre>
<pre><code>
        Tcell High Tcell Intermediate          Tcell Low     Tcell Very Low 
                 0                  0                  0               4395 </code></pre>
<pre class="r"><code>sceAM_TCvlow &lt;- as.SingleCellExperiment(seuratAM_TCvlow)</code></pre>
<pre class="r"><code>## subset Cardiomyopathy (CMP) and convert seurat object to sce 
seuratCMP &lt;- subset(seurat_immun, pat_sub == &quot;CMP&quot;)
table(seuratCMP$pat_sub)</code></pre>
<pre><code>
    Healthy          AM         CMP Sarcoidosis 
          0           0        1061           0 </code></pre>
<pre class="r"><code>sceCMP &lt;- as.SingleCellExperiment(seuratCMP)</code></pre>
<pre class="r"><code>## subset Sarcoidosis and convert seurat object to sce 
seuratSarco &lt;- subset(seurat_immun, pat_sub == &quot;Sarcoidosis&quot;)
table(seuratSarco$pat_sub)</code></pre>
<pre><code>
    Healthy          AM         CMP Sarcoidosis 
          0           0           0         521 </code></pre>
<pre class="r"><code>sceSarco &lt;- as.SingleCellExperiment(seuratSarco)</code></pre>
<pre class="r"><code>## Filter lymphocyte differentiation T cell GSEA, redundant terms are listed 
ego0 &lt;- dplyr::filter(ego0@result, ego0@result$Description %in% c(
  &quot;T cell differentiation&quot;,
  &quot;lymphocyte differentiation&quot;,
  &quot;T cell receptor signaling pathway&quot;,
  &quot;alpha-beta T cell activation&quot;,
  &quot;T cell differentiation in thymus&quot;,
  &quot;T cell selection&quot;,
  &quot;positive regulation of T cell activation&quot;,
  &quot;alpha-beta T cell differentiation&quot;,
  &quot;positive regulation of lymphocyte activation&quot;,
  &quot;regulation of T cell differentiation&quot;,
  &quot;regulation of lymphocyte differentiation&quot;,
  &quot;positive T cell selection&quot;,
  &quot;lymphocyte proliferation&quot;,
  &quot;thymic T cell selection&quot;,
  &quot;T cell proliferation&quot;,
  &quot;CD4-positive, alpha-beta T cell activation&quot;,
  &quot;lymphocyte activation involved in immune response&quot;,
  &quot;T cell migration&quot;,
  &quot;T cell costimulation&quot;,
  &quot;alpha-beta T cell activation involved in immune response&quot;,
  &quot;alpha-beta T cell differentiation involved in immune response&quot;,
  &quot;CD4-positive, alpha-beta T cell differentiation&quot;,
  &quot;T cell differentiation involved in immune response&quot;,
  &quot;T-helper cell differentiation&quot;,
  &quot;regulation of lymphocyte proliferation&quot;,
  &quot;CD4-positive, alpha-beta T cell differentiation involved in immune response&quot;,
  &quot;positive regulation of T cell differentiation&quot;))

## geneID extraction
g0 &lt;- ego0$geneID
Str &lt;- unlist(strsplit(g0, &quot;/&quot;))
df &lt;- as.data.frame(Str)
colnames(df) &lt;- c(&quot;gene&quot;)

## signature Genes in count matrix
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)
if (nrow(signGenes) == 0) stop(&quot;No signature genes found.&quot;)

## sce object for signature genes
sceSub &lt;- sce[rownames(sce) %in% unique(signGenes$geneID), ]
if (nrow(sceSub) == 0) stop(&quot;No matching genes found in sce.&quot;)

## gene signature matrix
cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts))) / nrow(signGenes)
if (length(cntMat) != ncol(sceSub)) stop(&quot;Mismatch between cntMat and sceSub dimensions.&quot;)

sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign

sceSub$sign2[which(sceSub$sign &gt; 2)] &lt;- 2

## umap of cells affected by T lymphocyte activation, proliferation and differentiation
sc &lt;- scale_colour_gradientn(colours = pal(100), limits = c(0, 2))
plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc </code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20lymphocyte%20differentiation%20all-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Healthy
sceHealthySub &lt;- sceHealthy[which(rownames(sceHealthy) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceHealthySub@assays@data$logcounts)))/nrow(signGenes)
sceHealthySub$sign &lt;- cntMat
sceHealthySub$sign2 &lt;- sceHealthySub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceHealthySub$sign2[which(sceHealthySub$sign &gt; 1)] &lt;- 1
plotUMAP(sceHealthySub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20healthy%20only-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM all, single cell experiment with high expression of lymphocyte activation markers
sceAMSub &lt;- sceAM[which(rownames(sceAM) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSub@assays@data$logcounts)))/nrow(signGenes)
sceAMSub$sign &lt;- cntMat
sceAMSub$sign2 &lt;- sceAMSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSub$sign2[which(sceAMSub$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20only%20lympho%20activation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC high lymphocyte activation
sceAMSubhigh &lt;- sceAM_TChigh[which(rownames(sceAM) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSubhigh@assays@data$logcounts)))/nrow(signGenes)
sceAMSubhigh$sign &lt;- cntMat
sceAMSubhigh$sign2 &lt;- sceAMSubhigh$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSubhigh$sign2[which(sceAMSubhigh$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSubhigh, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20Tc%20high%20lympho%20activation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC int lymphocyte activation
sceAMSubint &lt;- sceAM_TCint[which(rownames(sceAM_TCint) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSubint@assays@data$logcounts)))/nrow(signGenes)
sceAMSubint$sign &lt;- cntMat
sceAMSubint$sign2 &lt;- sceAMSubint$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSubint$sign2[which(sceAMSubint$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSubint, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20Tc%20int%20lympho%20activation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC low lymphocyte activation
sceAMSublow &lt;- sceAM_TClow[which(rownames(sceAM_TClow) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSublow@assays@data$logcounts)))/nrow(signGenes)
sceAMSublow$sign &lt;- cntMat
sceAMSublow$sign2 &lt;- sceAMSublow$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSublow$sign2[which(sceAMSublow$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSublow, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20Tc%20low%20and%20vlow%20lympho%20activation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC vlow lymphocyte activation
sceAMSubvlow &lt;- sceAM_TCvlow[which(rownames(sceAM_TCvlow) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSubvlow@assays@data$logcounts)))/nrow(signGenes)
sceAMSubvlow$sign &lt;- cntMat
sceAMSubvlow$sign2 &lt;- sceAMSubvlow$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSubvlow$sign2[which(sceAMSubvlow$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSubvlow, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20Tc%20low%20and%20vlow%20lympho%20activation-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## CMP lymphocyte activation
sceCMPSub &lt;- sceCMP[which(rownames(sceCMP) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceCMPSub@assays@data$logcounts)))/nrow(signGenes)
sceCMPSub$sign &lt;- cntMat
sceCMPSub$sign2 &lt;- sceCMPSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceCMPSub$sign2[which(sceCMPSub$sign &gt; 1)] &lt;- 1
plotUMAP(sceCMPSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20CMP%20sce%20lymphocyte%20activation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis lymphocyte activation
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20Sarcoidosis%20sce%20lymphocyte%20activation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## response to type II interferon
ego5 &lt;- dplyr::filter(ego5@result,ego5@result$Description==&quot;response to type II interferon&quot;)
g5 &lt;- ego5$geneID
Str &lt;-(g5)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

## Count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign

##  max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 2))
sceSub$sign2[which(sceSub$sign &gt; 2)] &lt;- 2
plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/response%20type%20II%20interferon,%20all%20immuno%20clusters-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## SCE Healthy, response to type II interferon
sceHealthySub &lt;- sceHealthy[which(rownames(sceHealthy) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceHealthySub@assays@data$logcounts)))/nrow(signGenes)
sceHealthySub$sign &lt;- cntMat
sceHealthySub$sign2 &lt;- sceHealthySub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceHealthySub$sign2[which(sceHealthySub$sign &gt; 1)] &lt;- 1
plotUMAP(sceHealthySub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20Healthy,%20response%20to%20type%20II%20interferon-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM all, response type II interferon
sceAMSub &lt;- sceAM[which(rownames(sceAM) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSub@assays@data$logcounts)))/nrow(signGenes)
sceAMSub$sign &lt;- cntMat
sceAMSub$sign2 &lt;- sceAMSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSub$sign2[which(sceAMSub$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20only%20response%20type%20II%20interferon-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC high response type II interferon
sceAMSubhigh &lt;- sceAM_TChigh[which(rownames(sceAM) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSubhigh@assays@data$logcounts)))/nrow(signGenes)
sceAMSubhigh$sign &lt;- cntMat
sceAMSubhigh$sign2 &lt;- sceAMSubhigh$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSubhigh$sign2[which(sceAMSubhigh$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSubhigh, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20Tc%20high%20response%20type%20II%20interferon-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC int response type II interferon
sceAMSubint &lt;- sceAM_TCint[which(rownames(sceAM_TCint) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSubint@assays@data$logcounts)))/nrow(signGenes)
sceAMSubint$sign &lt;- cntMat
sceAMSubint$sign2 &lt;- sceAMSubint$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSubint$sign2[which(sceAMSubint$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSubint, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20TCint%20response%20type%20II%20interferon-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC low response type II interferon
sceAMSublow &lt;- sceAM_TClow[which(rownames(sceAM_TClow) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSublow@assays@data$logcounts)))/nrow(signGenes)
sceAMSublow$sign &lt;- cntMat
sceAMSublow$sign2 &lt;- sceAMSublow$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSublow$sign2[which(sceAMSublow$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSublow, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20Tc%20low%20response%20type%20II%20interferon-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC vlow response type II interferon
sceAMSubvlow &lt;- sceAM_TCvlow[which(rownames(sceAM_TCvlow) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSubvlow@assays@data$logcounts)))/nrow(signGenes)
sceAMSubvlow$sign &lt;- cntMat
sceAMSubvlow$sign2 &lt;- sceAMSubvlow$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSubvlow$sign2[which(sceAMSubvlow$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSubvlow, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20Tc%20vlow%20response%20type%20II%20interferon-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## CMP response type II interferon
sceCMPSub &lt;- sceCMP[which(rownames(sceCMP) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceCMPSub@assays@data$logcounts)))/nrow(signGenes)
sceCMPSub$sign &lt;- cntMat
sceCMPSub$sign2 &lt;- sceCMPSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceCMPSub$sign2[which(sceCMPSub$sign &gt; 1)] &lt;- 1
plotUMAP(sceCMPSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/CMP%20response%20type%20II%20interferon-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis response type II interferon
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Sarco%20response%20type%20II%20interferon-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Filter Regulation of Interleukin-6 production
  

ego3 &lt;- dplyr::filter(ego3@result,ego3@result$Description==&quot;positive regulation of interleukin-6 production&quot;)
g3 &lt;- ego3$geneID
Str &lt;-(g3)
StrSub &lt;- strsplit(Str, &quot;/&quot;)
df &lt;- as.data.frame(StrSub)
colnames(df) &lt;- c(&quot;gene&quot;)

##make a count matrix of signature genes
signGenes &lt;- genes %&gt;% dplyr::filter(gene %in% df$gene)

sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign2 &lt;- sceSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSub$sign2[which(sceSub$sign &gt; 1)] &lt;- 1
plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Interleukin-6%20production%20across%20all%20clusters-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM all, gradient where IL-6 activation is high
sceAMSub &lt;- sceAM[which(rownames(sceAM) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSub@assays@data$logcounts)))/nrow(signGenes)
sceAMSub$sign &lt;- cntMat
sceAMSub$sign2 &lt;- sceAMSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSub$sign2[which(sceAMSub$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/AM%20IL6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC high IL6 activation
sceAMSubhigh &lt;- sceAM_TChigh[which(rownames(sceAM) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSubhigh@assays@data$logcounts)))/nrow(signGenes)
sceAMSubhigh$sign &lt;- cntMat
sceAMSubhigh$sign2 &lt;- sceAMSubhigh$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSubhigh$sign2[which(sceAMSubhigh$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSubhigh, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20Tc%20high%20IL6%20activation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC int IL6 activation
sceAMSubint &lt;- sceAM_TCint[which(rownames(sceAM_TCint) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSubint@assays@data$logcounts)))/nrow(signGenes)
sceAMSubint$sign &lt;- cntMat
sceAMSubint$sign2 &lt;- sceAMSubint$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSubint$sign2[which(sceAMSubint$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSubint, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20TCint%20IL6%20activation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## AM TC low IL6 activation
sceAMSublow &lt;- sceAM_TClow[which(rownames(sceAM_TClow) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSublow@assays@data$logcounts)))/nrow(signGenes)
sceAMSublow$sign &lt;- cntMat
sceAMSublow$sign2 &lt;- sceAMSublow$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSublow$sign2[which(sceAMSublow$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSublow, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20Tc%20int%20IL6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># AM TC vlow lymphocyte activation
sceAMSubvlow &lt;- sceAM_TCvlow[which(rownames(sceAM_TCvlow) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceAMSubvlow@assays@data$logcounts)))/nrow(signGenes)
sceAMSubvlow$sign &lt;- cntMat
sceAMSubvlow$sign2 &lt;- sceAMSubvlow$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceAMSubvlow$sign2[which(sceAMSubvlow$sign &gt; 1)] &lt;- 1
plotUMAP(sceAMSubvlow, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/SCE%20AM%20Tc%20int%20IL6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## CMP IL6 activation
sceCMPSub &lt;- sceCMP[which(rownames(sceCMP) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceCMPSub@assays@data$logcounts)))/nrow(signGenes)
sceCMPSub$sign &lt;- cntMat
sceCMPSub$sign2 &lt;- sceCMPSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceCMPSub$sign2[which(sceCMPSub$sign &gt; 1)] &lt;- 1
plotUMAP(sceCMPSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/CMP%20IL6%20activation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Sarcoidosis IL6 activation
sceSarcoSub &lt;- sceSarco[which(rownames(sceSarco) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSarcoSub@assays@data$logcounts)))/nrow(signGenes)
sceSarcoSub$sign &lt;- cntMat
sceSarcoSub$sign2 &lt;- sceSarcoSub$sign
##check max and min values
##max(sceSub$sign)
sc &lt;- scale_colour_gradientn(colours = pal(100), limits=c(0, 1))
sceSarcoSub$sign2[which(sceSarcoSub$sign &gt; 1)] &lt;- 1
plotUMAP(sceSarcoSub, colour_by = &quot;sign2&quot;, point_size = 0.5) + sc +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/Immune_cells_V3.Rmd/Sarco%20IL6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>`</p>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Mon Nov 25 22:06:57 2024&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.7.1             pathview_1.44.0             wordcloud_2.6              
 [4] textshaping_0.4.0           EnhancedVolcano_1.22.0      ggrepel_0.9.6              
 [7] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
[10] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
[13] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_7.5.1              
[16] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[19] multtest_2.60.0             metap_1.11                  scater_1.32.1              
[22] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[25] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[28] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[31] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[34] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[37] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[40] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[43] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[46] matrixStats_1.4.1           pheatmap_1.0.12             ggpubr_0.6.0               
[49] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[52] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             presto_1.0.0              parallelly_1.38.0        
[145] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[148] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[151] pbapply_1.7-2             httr2_1.0.5               spam_2.11-0              
[154] gson_0.1.0                utf8_1.2.4                graphlayouts_1.2.0       
[157] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[160] shiny_1.9.1               GenomeInfoDbData_1.2.12   glmmTMB_1.1.10           
[163] R.utils_2.12.3            RCurl_1.98-1.16           memoise_2.0.1            
[166] rmarkdown_2.28            scales_1.3.0              R.methodsS3_1.8.2        
[169] future_1.34.0             RANN_2.6.2                Cairo_1.6-2              
[172] spatstat.data_3.1-2       rstudioapi_0.16.0         whisker_0.4.1            
[175] cluster_2.1.6             mutoss_0.1-13             spatstat.utils_3.1-0     
[178] hms_1.1.3                 fitdistrplus_1.2-1        munsell_0.5.1            
[181] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.4              
[184] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.0               
[187] dotCall64_1.2             shinydashboard_0.7.2      ggforce_0.4.2            
[190] laeken_0.5.3              mgcv_1.9-1                xfun_0.48                
[193] e1071_1.7-16              TH.data_1.1-2             iterators_1.0.14         
[196] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[199] ps_1.8.0                  futile.options_1.0.1      bitops_1.0-9             
[202] Rdpack_2.6.1              promises_1.3.0            scatterpie_0.2.4         
[205] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-1           
[208] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[211] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[214] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[217] Rcpp_1.0.13               edgeR_4.2.1               BiocSingular_1.20.0      
[220] tensor_1.5                MASS_7.3-61               progress_1.2.3           
[223] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.3-2    
[226] R6_2.5.1                  fastmap_1.2.0             multcomp_1.4-26          
[229] fastmatch_1.1-4           rstatix_0.7.2             vipor_0.4.7              
[232] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[235] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-19              
[238] gtable_0.3.5              KernSmooth_2.23-24        miniUI_0.1.1.1           
[241] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[244] bit64_4.5.2               spatstat.explore_3.3-2    lifecycle_1.0.4          
[247] blme_1.0-6                processx_3.8.4            callr_3.7.6              
[250] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[253] robustbase_0.99-4-1       spatstat.geom_3.3-3       sn_2.1.1                 
[256] ggfun_0.1.6               future.apply_1.11.2       bslib_0.8.0              
[259] pillar_1.9.0              gplots_3.2.0              pcaMethods_1.96.0        
[262] locfit_1.5-9.10           jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.7.1             pathview_1.44.0             wordcloud_2.6              
 [4] textshaping_0.4.0           EnhancedVolcano_1.22.0      ggrepel_0.9.6              
 [7] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
[10] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
[13] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_7.5.1              
[16] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[19] multtest_2.60.0             metap_1.11                  scater_1.32.1              
[22] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[25] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[28] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[31] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[34] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[37] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[40] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[43] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[46] matrixStats_1.4.1           pheatmap_1.0.12             ggpubr_0.6.0               
[49] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[52] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             presto_1.0.0              parallelly_1.38.0        
[145] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[148] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[151] pbapply_1.7-2             httr2_1.0.5               spam_2.11-0              
[154] gson_0.1.0                utf8_1.2.4                graphlayouts_1.2.0       
[157] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[160] shiny_1.9.1               GenomeInfoDbData_1.2.12   glmmTMB_1.1.10           
[163] R.utils_2.12.3            RCurl_1.98-1.16           memoise_2.0.1            
[166] rmarkdown_2.28            scales_1.3.0              R.methodsS3_1.8.2        
[169] future_1.34.0             RANN_2.6.2                Cairo_1.6-2              
[172] spatstat.data_3.1-2       rstudioapi_0.16.0         whisker_0.4.1            
[175] cluster_2.1.6             mutoss_0.1-13             spatstat.utils_3.1-0     
[178] hms_1.1.3                 fitdistrplus_1.2-1        munsell_0.5.1            
[181] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.4              
[184] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.0               
[187] dotCall64_1.2             shinydashboard_0.7.2      ggforce_0.4.2            
[190] laeken_0.5.3              mgcv_1.9-1                xfun_0.48                
[193] e1071_1.7-16              TH.data_1.1-2             iterators_1.0.14         
[196] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[199] ps_1.8.0                  futile.options_1.0.1      bitops_1.0-9             
[202] Rdpack_2.6.1              promises_1.3.0            scatterpie_0.2.4         
[205] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-1           
[208] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[211] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[214] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[217] Rcpp_1.0.13               edgeR_4.2.1               BiocSingular_1.20.0      
[220] tensor_1.5                MASS_7.3-61               progress_1.2.3           
[223] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.3-2    
[226] R6_2.5.1                  fastmap_1.2.0             multcomp_1.4-26          
[229] fastmatch_1.1-4           rstatix_0.7.2             vipor_0.4.7              
[232] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[235] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-19              
[238] gtable_0.3.5              KernSmooth_2.23-24        miniUI_0.1.1.1           
[241] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[244] bit64_4.5.2               spatstat.explore_3.3-2    lifecycle_1.0.4          
[247] blme_1.0-6                processx_3.8.4            callr_3.7.6              
[250] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[253] robustbase_0.99-4-1       spatstat.geom_3.3-3       sn_2.1.1                 
[256] ggfun_0.1.6               future.apply_1.11.2       bslib_0.8.0              
[259] pillar_1.9.0              gplots_3.2.0              pcaMethods_1.96.0        
[262] locfit_1.5-9.10           jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
