<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Payne" />

<meta name="date" content="2024-11-25" />

<title>T cells preliminary GSEA</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myocarditis snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">T cells preliminary GSEA</h1>
<h4 class="author">Emily Payne</h4>
<h4 class="date">2024-11-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-11-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Myocarditis
snRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Emily/Myocarditis
snRNAseq/docs/Markers/DEGenesTcell
</td>
<td style="text-align:left;">
../docs/Markers/DEGenesTcell
</td>
</tr>
<tr>
<td style="text-align:left;">
~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat
files/Myocarditis_allfiltered_V3_15.11.2024.rds
</td>
<td style="text-align:left;">
../data/01 Merged Seurat files/Myocarditis_allfiltered_V3_15.11.2024.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree05dd5a33149bad1924497530bf1b627bc18d9a60targetblank05dd5a3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/05dd5a33149bad1924497530bf1b627bc18d9a60" target="_blank">05dd5a3</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomEpaynehubMyocarditissnRNAseqtree05dd5a33149bad1924497530bf1b627bc18d9a60targetblank05dd5a3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Epayne-hub/Myocarditis-snRNAseq/tree/05dd5a33149bad1924497530bf1b627bc18d9a60" target="_blank">05dd5a3</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    data/

Untracked files:
    Untracked:  analysis/AM vs. Healthy subsets.Rmd
    Untracked:  analysis/DE genes AMsub vs. Healthy_V2.Rmd
    Untracked:  analysis/Immune_cells_V3.Rmd
    Untracked:  analysis/Interactome Analysis.Rmd
    Untracked:  analysis/Interactome Chellchat Analysis.Rmd
    Untracked:  analysis/T-cells.Rmd
    Untracked:  analysis/T_cells.Rmd

Unstaged changes:
    Modified:   analysis/Fibroblast_subset_V2.Rmd
    Deleted:    analysis/Immune_cells.Rmd
    Modified:   analysis/Myocarditis_snRNAseq_QC_merging_V3.Rmd
    Modified:   analysis/Stratification_Tcell_conditionV2.Rmd
    Modified:   analysis/Stratification_sex.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>suppressPackageStartupMessages({
  library(ExploreSCdataSeurat3)
  library(runSeurat3)
  library(Seurat)
  library(ggpubr)
  library(pheatmap)
  library(SingleCellExperiment)
  library(dplyr)
  library(tidyverse)
  library(viridis)
  library(muscat)
  library(circlize)
  library(destiny)
  library(scater)
  library(metap)
  library(multtest)
  library(clusterProfiler)
  library(org.Hs.eg.db)
  library(msigdbr)
  library(enrichplot)
  library(DOSE)
  library(grid)
  library(gridExtra)
  library(ggupset)
  library(VennDiagram)
  library(NCmisc)
  library(RColorBrewer)
  library(EnhancedVolcano)
  library(textshaping)
  library(wordcloud)
  library(pathview)
  library(workflowr)
})</code></pre>
<pre class="r"><code>##load merged file 
seuratM &lt;- readRDS(&quot;~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_allfiltered_V3_15.11.2024.rds&quot;)

table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11    12    13    14 
17699 13639 12754  8222  5475  4836  3723  2365  2156   905   882   694   380   249    91 </code></pre>
<pre class="r"><code>table(seuratM$patient)</code></pre>
<pre><code>
    Healthy 01     Healthy 02     Healthy 03     Healthy 04     Healthy 05     Healthy 06 
          3921           3818           4908           9882           1442           1998 
    Healthy 07     Healthy 08          AM 01          AM 02          AM 03          AM 04 
           841           1480           6286           4323           1123            208 
         AM 05          AM 06          AM 07          AM 08          AM 09          AM 10 
           579            335           1217            604            513            426 
         AM 11          AM 12          AM 13          AM 14          AM 15          AM 16 
           236           1192           2740           1684           4439           2247 
         AM 17          AM 18         CMP 01         CMP 02         CMP 03         CMP 04 
           316            608            105            227            730           1428 
        CMP 05         CMP 06         CMP 07         CMP 08         CMP 09         CMP 10 
           620            653           1268           2280           2396            545 
        CMP 11         CMP 12         CMP 13 Sarcoidosis 01 Sarcoidosis 02 
           781            491            111           2363           2706 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
74070 </code></pre>
<pre class="r"><code>seuratM$clusterName &lt;- &quot;clusterName&quot;
Idents(seuratM) &lt;- seuratM$clusterName
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;0&quot; )] &lt;- &quot;Fb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;1&quot; )] &lt;- &quot;BEC1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;2&quot; )] &lt;- &quot;PeriFb1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;3&quot; )] &lt;- &quot;Cardiomyocyte&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;4&quot; )] &lt;- &quot;MCP1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;5&quot; )] &lt;- &quot;TC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;6&quot; )] &lt;- &quot;MCP2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;7&quot; )] &lt;- &quot;BEC2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;8&quot; )] &lt;- &quot;VSMC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;9&quot; )] &lt;- &quot;NC&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;10&quot; )] &lt;- &quot;INT1&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;11&quot; )] &lt;- &quot;Fb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;12&quot; )] &lt;- &quot;Fb3&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;13&quot; )] &lt;- &quot;PeriFb2&quot;
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% &quot;14&quot; )] &lt;- &quot;AdipoC&quot;


colclusterName &lt;- c(&quot;#D53E4F&quot;, &quot;#f4a582&quot;, &quot;#FEE08B&quot;,&quot;#feb24c&quot;,&quot;#67001f&quot;, &quot;#01665e&quot;,&quot;#66C2A5&quot;,&quot;#c7eae5&quot;,&quot;#BEAED4&quot;, &quot;#355C7D&quot;,&quot;#3288BD&quot;,&quot;#8c510a&quot; ,&quot;#fde0dd&quot;,&quot;#B45B5C&quot;,&quot;#dd1c77&quot;)

                    
names(colclusterName) &lt;- c(&quot;Cardiomyocyte&quot;,&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;INT1&quot;, &quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;,&quot;NC&quot;, &quot;BEC1&quot;,&quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;,&quot;MCP2&quot;,&quot;TC&quot;)

colpat_sub&lt;- c(&quot;#dfc27d&quot;,&quot;#BE3144&quot;,&quot;#355C7D&quot;, &quot;#779d8d&quot;)

seuratM$clusterName &lt;- factor(seuratM$clusterName, levels=c(&quot;Cardiomyocyte&quot;, &quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;INT1&quot;,&quot;PeriFb1&quot;, &quot;PeriFb2&quot;, &quot;VSMC&quot;, &quot;NC&quot;, &quot;BEC1&quot;, &quot;BEC2&quot;, &quot;AdipoC&quot;, &quot;MCP1&quot;, &quot;MCP2&quot;, &quot;TC&quot;))


Idents(seuratM) &lt;- seuratM$clusterName
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, label = TRUE, cols = colclusterName)</code></pre>
<p><img src="figure/T_cells.Rmd/clusterName-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table(seuratM$patient, seuratM$TC)</code></pre>
<pre><code>                
                 Tcell High Tcell Intermediate Tcell Low Tcell Very Low
  Healthy 01              0                  0         0           3921
  Healthy 02              0                  0         0           3818
  Healthy 03              0                  0         0           4908
  Healthy 04              0                  0         0           9882
  Healthy 05              0                  0         0           1442
  Healthy 06              0                  0         0           1998
  Healthy 07              0                  0         0            841
  Healthy 08              0                  0         0           1480
  AM 01                6286                  0         0              0
  AM 02                   0                  0         0           4323
  AM 03                   0                  0         0           1123
  AM 04                   0                  0         0            208
  AM 05                   0                  0         0            579
  AM 06                   0                335         0              0
  AM 07                   0                  0         0           1217
  AM 08                   0                  0       604              0
  AM 09                   0                  0       513              0
  AM 10                   0                  0         0            426
  AM 11                   0                  0         0            236
  AM 12                   0                  0         0           1192
  AM 13                   0               2740         0              0
  AM 14                   0                  0      1684              0
  AM 15                   0                  0      4439              0
  AM 16                2247                  0         0              0
  AM 17                 316                  0         0              0
  AM 18                 608                  0         0              0
  CMP 01                  0                105         0              0
  CMP 02                  0                  0         0            227
  CMP 03                  0                  0       730              0
  CMP 04                  0                  0      1428              0
  CMP 05                  0                  0       620              0
  CMP 06                  0                  0         0            653
  CMP 07                  0                  0      1268              0
  CMP 08                  0                  0         0           2280
  CMP 09                  0                  0         0           2396
  CMP 10                  0                  0       545              0
  CMP 11                  0                  0       781              0
  CMP 12                  0                491         0              0
  CMP 13                111                  0         0              0
  Sarcoidosis 01          0                  0      2363              0
  Sarcoidosis 02          0                  0      2706              0</code></pre>
<pre class="r"><code>## Subset from AM cohort, inflammatory clusters

seuratTcell &lt;- subset(seuratM, clusterName %in% &quot;TC&quot;)
table(seuratTcell$orig.ident)</code></pre>
<pre><code>
     
4836 </code></pre>
<pre class="r"><code>table(seuratTcell$patient, seuratTcell$clusterName)</code></pre>
<pre><code>                
                 Cardiomyocyte  Fb1  Fb2  Fb3 INT1 PeriFb1 PeriFb2 VSMC   NC BEC1 BEC2 AdipoC MCP1
  Healthy 01                 0    0    0    0    0       0       0    0    0    0    0      0    0
  Healthy 02                 0    0    0    0    0       0       0    0    0    0    0      0    0
  Healthy 03                 0    0    0    0    0       0       0    0    0    0    0      0    0
  Healthy 04                 0    0    0    0    0       0       0    0    0    0    0      0    0
  Healthy 05                 0    0    0    0    0       0       0    0    0    0    0      0    0
  Healthy 06                 0    0    0    0    0       0       0    0    0    0    0      0    0
  Healthy 07                 0    0    0    0    0       0       0    0    0    0    0      0    0
  Healthy 08                 0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 01                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 02                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 03                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 04                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 05                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 06                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 07                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 08                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 09                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 10                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 11                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 12                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 13                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 14                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 15                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 16                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 17                      0    0    0    0    0       0       0    0    0    0    0      0    0
  AM 18                      0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 01                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 02                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 03                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 04                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 05                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 06                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 07                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 08                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 09                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 10                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 11                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 12                     0    0    0    0    0       0       0    0    0    0    0      0    0
  CMP 13                     0    0    0    0    0       0       0    0    0    0    0      0    0
  Sarcoidosis 01             0    0    0    0    0       0       0    0    0    0    0      0    0
  Sarcoidosis 02             0    0    0    0    0       0       0    0    0    0    0      0    0
                
                 MCP2   TC
  Healthy 01        0   73
  Healthy 02        0   34
  Healthy 03        0   42
  Healthy 04        0   78
  Healthy 05        0   22
  Healthy 06        0   39
  Healthy 07        0   15
  Healthy 08        0   19
  AM 01             0 2824
  AM 02             0   71
  AM 03             0   19
  AM 04             0    3
  AM 05             0   10
  AM 06             0   22
  AM 07             0    5
  AM 08             0   15
  AM 09             0   13
  AM 10             0    4
  AM 11             0    0
  AM 12             0   23
  AM 13             0  212
  AM 14             0   35
  AM 15             0  144
  AM 16             0  466
  AM 17             0   82
  AM 18             0   84
  CMP 01            0    7
  CMP 02            0    4
  CMP 03            0   27
  CMP 04            0   56
  CMP 05            0   24
  CMP 06            0    5
  CMP 07            0   32
  CMP 08            0   29
  CMP 09            0   33
  CMP 10            0   26
  CMP 11            0   17
  CMP 12            0   32
  CMP 13            0   13
  Sarcoidosis 01    0   56
  Sarcoidosis 02    0  121</code></pre>
<pre class="r"><code>table(seuratTcell$patient, seuratTcell$sex)</code></pre>
<pre><code>                
                 female male unknown
  Healthy 01          0    0      73
  Healthy 02          0    0      34
  Healthy 03          0    0      42
  Healthy 04          0    0      78
  Healthy 05          0    0      22
  Healthy 06          0    0      39
  Healthy 07          0    0      15
  Healthy 08          0    0      19
  AM 01            2824    0       0
  AM 02               0   71       0
  AM 03               0   19       0
  AM 04               3    0       0
  AM 05               0   10       0
  AM 06              22    0       0
  AM 07               5    0       0
  AM 08               0   15       0
  AM 09              13    0       0
  AM 10               4    0       0
  AM 11               0    0       0
  AM 12               0   23       0
  AM 13               0  212       0
  AM 14              35    0       0
  AM 15               0  144       0
  AM 16             466    0       0
  AM 17               0   82       0
  AM 18              84    0       0
  CMP 01              0    7       0
  CMP 02              4    0       0
  CMP 03              0   27       0
  CMP 04             56    0       0
  CMP 05              0   24       0
  CMP 06              0    5       0
  CMP 07              0   32       0
  CMP 08              0   29       0
  CMP 09              0   33       0
  CMP 10              0   26       0
  CMP 11              0   17       0
  CMP 12              0   32       0
  CMP 13             13    0       0
  Sarcoidosis 01      0   56       0
  Sarcoidosis 02      0  121       0</code></pre>
<pre class="r"><code>## Plot single Tcell cluster
Idents(seuratTcell) &lt;- seuratTcell$clusterName
DimPlot(seuratTcell, reduction = &quot;umap&quot;, cols = colclusterName, pt.size = 0.3)</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Plot Tcell sorted by sex
colsex &lt;- c(&quot;#dd1c77&quot;,&quot;#355C7D&quot;)
names(colsex) &lt;- c(&quot;female&quot;, &quot;male&quot;)

Idents(seuratTcell) &lt;- seuratTcell$sex
DimPlot(seuratTcell, reduction = &quot;umap&quot;, pt.size = 0.3, cols = colsex)</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Plot Tcell sorted by Tcell
Idents(seuratTcell) &lt;- (seuratTcell$TC)

colTC &lt;- c(&quot;#BE3144&quot;,&quot;#de2d26&quot;,&quot;#fc9272&quot;, &quot;#fee0d2&quot;)
names(colTC) &lt;- c(&quot;Tcell High&quot;, &quot;Tcell Intermediate&quot;, &quot;Tcell Low&quot;, &quot;Tcell Very Low&quot;)

DimPlot(seuratTcell, reduction = &quot;umap&quot;, pt.size = 0.3, cols = colTC)</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratTcell) &lt;- seuratTcell$clusterName

table(seuratM$clusterName)</code></pre>
<pre><code>
Cardiomyocyte           Fb1           Fb2           Fb3          INT1       PeriFb1       PeriFb2 
         8222         17699           694           380           882         12754           249 
         VSMC            NC          BEC1          BEC2        AdipoC          MCP1          MCP2 
         2156           905         13639          2365            91          5475          3723 
           TC 
         4836 </code></pre>
<pre class="r"><code>seuratTcell &lt;- NormalizeData (object = seuratTcell)
seuratTcell &lt;- FindVariableFeatures(object = seuratTcell)
seuratTcell &lt;- ScaleData(object = seuratTcell, verbose = TRUE)
seuratTcell &lt;- RunPCA(object = seuratTcell, npcs = 30, verbose = FALSE)
seuratTcell &lt;- RunTSNE(object = seuratTcell, reduction=&quot;pca&quot;, dims = 1:20)
seuratTcell &lt;- RunUMAP(object = seuratTcell, reduction=&quot;pca&quot;, dims = 1:20)
seuratTcell &lt;- FindNeighbors(object = seuratTcell, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4, 0.3)
for (i in 1:length(res)) {
  seuratTcell &lt;- FindClusters(object = seuratTcell, resolution = res[i], random.seed = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 4836
Number of edges: 175920

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8926
Number of communities: 9
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 4836
Number of edges: 175920

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8314
Number of communities: 12
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 4836
Number of edges: 175920

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7997
Number of communities: 12
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 4836
Number of edges: 175920

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8648
Number of communities: 11
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 4836
Number of edges: 175920

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8829
Number of communities: 10
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>table(seuratTcell$orig.ident)</code></pre>
<pre><code>
     
4836 </code></pre>
<pre class="r"><code>table(seuratTcell$clusterName)</code></pre>
<pre><code>
Cardiomyocyte           Fb1           Fb2           Fb3          INT1       PeriFb1       PeriFb2 
            0             0             0             0             0             0             0 
         VSMC            NC          BEC1          BEC2        AdipoC          MCP1          MCP2 
            0             0             0             0             0             0             0 
           TC 
         4836 </code></pre>
<pre class="r"><code>## Plot Tcell sorted by sex

colsex &lt;- c(&quot;#dd1c77&quot;,&quot;#355C7D&quot;)
names(colsex) &lt;- c(&quot;female&quot;, &quot;male&quot;)

Idents(seuratTcell) &lt;- seuratTcell$sex
DimPlot(seuratTcell, reduction = &quot;umap&quot;, pt.size = 0.3, cols = colsex)</code></pre>
<p><img src="figure/T_cells.Rmd/Plot%20umaps%20by%20sex-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Plot Tcell sorted by Tcell condition
Idents(seuratTcell) &lt;- (seuratTcell$TC)

colTC &lt;- c(&quot;#BE3144&quot;,&quot;#de2d26&quot;,&quot;#fc9272&quot;, &quot;#fee0d2&quot;)
names(colTC) &lt;- c(&quot;Tcell High&quot;, &quot;Tcell Intermediate&quot;, &quot;Tcell Low&quot;, &quot;Tcell Very Low&quot;)

DimPlot(seuratTcell, reduction = &quot;umap&quot;, pt.size = 0.3, cols = colTC)</code></pre>
<p><img src="figure/T_cells.Rmd/Plot%20umaps%20by%20Tcell%20condition-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratTcell) &lt;- seuratTcell$clusterName

table(seuratM$clusterName)</code></pre>
<pre><code>
Cardiomyocyte           Fb1           Fb2           Fb3          INT1       PeriFb1       PeriFb2 
         8222         17699           694           380           882         12754           249 
         VSMC            NC          BEC1          BEC2        AdipoC          MCP1          MCP2 
         2156           905         13639          2365            91          5475          3723 
           TC 
         4836 </code></pre>
<pre class="r"><code>Idents (seuratTcell) &lt;- seuratTcell$RNA_snn_res.0.25

DimPlot(seuratTcell, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents (seuratTcell) &lt;- seuratTcell$RNA_snn_res.0.3

DimPlot(seuratTcell, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents (seuratTcell) &lt;- seuratTcell$RNA_snn_res.0.4

DimPlot(seuratTcell, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents (seuratTcell) &lt;- seuratTcell$RNA_snn_res.0.6

DimPlot(seuratTcell, reduction = &quot;umap&quot;, pt.size = 0.3, label = TRUE)</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-2-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Find all markers T cells, ILC, NK
DEGenesTcell &lt;- FindAllMarkers (seuratTcell, only.pos = TRUE, logfc.threshold = 0.2) %&gt;%
                                      dplyr::filter(p_val_adj &lt; 0.01)


#write.table(DEGenesTcell, file= &quot;/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenesTcell&quot;, sep=&quot;\t&quot;,quote=F,row.names=F,col.names=T)

#View(DEGenesTcell)</code></pre>
<pre class="r"><code>## Find marker genes for T cell subgroups

get_full_gene_name &lt;- function(gene, obj){
  return(grep(gene,rownames(obj), value =TRUE))
}
get_full_gene_name(&quot;XIST&quot;,seuratTcell) </code></pre>
<pre><code>[1] &quot;ENSG00000229807.XIST&quot;</code></pre>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000081237.PTPRC&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000168685.IL7R&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000010610.CD4&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000172116.CD8B&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;CD8A&quot;, seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000116824.CD2&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000167286.CD3D&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000198851.CD3E&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000178562.CD28&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;CD96&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;RIPOR2&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000227507.LTB&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Cluster 1 spec</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000069667.RORA&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Cluster 1</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;PATJ&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#cl 6: CD8+ T cells proliferating ASPM, RRM2, KNL1, FANC; TOP2A, CEP152, DIAPH3, KIF18B, KIF14, CDCA3, TROAP, SAC3D1, MS4A7, ZNF683, GEN1, SAC8D1, SKA3, RAD54L, UWOMT. SPC25, BRACA1, BUB1B, Gen1, FoxM1, SKA3, CCNB1, SAC8D1, FANCI, CIT, SPAG5, ZNF683, E2F2, ZWINT


FeaturePlot(seuratTcell, features = &quot;ENSG00000163599.CTLA4&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-15.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;LAG3&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# exhaustion</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;TIGIT&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-17.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;PDCD1&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;HAVCR2&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-19.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;FOXP3&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Tregs</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000136634.IL10&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-21.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;TGFB1&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-22.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;STAT5A&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-23.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;STAT5B&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-24.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;RORC&quot;, seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Th17 (and also IL3)</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-25.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;IL22&quot;, seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-26.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;STAT3&quot;, seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-27.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;IL17RA&quot;, seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# did not find IL17A, only R</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-28.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;CCR6&quot;, seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-29.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000162594.IL23R&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-30.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000138795.LEF1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-31.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000109906.ZBTB16&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))    </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-32.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000165810.BTNL9&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # MAIT</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-33.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;TBX21&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Th1</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-34.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;IFNG&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-35.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;IL12RB1&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))#</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-36.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;CXCR3&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-37.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;STAT4&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-38.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;GATA3&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Th2</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-39.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;BCL6&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# Tfh</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-40.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;IL21&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-41.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000163600.ICOS&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-42.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;CCR5&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-43.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000172116.CD8B&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-44.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;KIR2&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-45.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;GZMB&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-46.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;PRF1&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-47.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000104970.KIR3DX1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-48.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000111537.IFNG&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-49.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;TRGV9&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) #yd Tcells</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-50.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;TRDV2&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-51.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;TRGC1&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-52.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;KIR3DL1&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# NK</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-53.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;TXK&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-54.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;NCAM1&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) #CD56,NK marker</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-55.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;KLRD1&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # CD94</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-56.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;FCGR3A&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) #CD16</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-57.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;GZMB&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-58.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000189430.NCR1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-59.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000004468.CD38&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-60.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;JCHAIN&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# plasma cells</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-61.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000057657.PRDM1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-62.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000211899.IGHM&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))## proliferating cells </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-63.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000156738.MS4A1&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # CD20</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-64.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;CD19&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-65.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000148773.MKI67&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))## proliferating cells</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-66.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;TOP2A&quot;, seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-67.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;SELL&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) # SELL = CD62L# naive</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-68.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;PTPRC&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;)) #CD45</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-69.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;CCR7&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-70.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;CXCL12&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-71.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = &quot;ENSG00000110448.CD5&quot;, pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-72.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;STAT1&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-73.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;STAT2&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-74.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;STAT3&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-75.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;STAT4&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-76.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratTcell, features = get_full_gene_name(&quot;STAT5&quot;,seuratTcell) , pt.size = 1, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))# </code></pre>
<p><img src="figure/T_cells.Rmd/unnamed-chunk-4-77.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Mon Nov 25 22:49:58 2024&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.7.1             pathview_1.44.0             wordcloud_2.6              
 [4] textshaping_0.4.0           EnhancedVolcano_1.22.0      ggrepel_0.9.6              
 [7] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
[10] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
[13] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_7.5.1              
[16] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[19] multtest_2.60.0             metap_1.11                  scater_1.32.1              
[22] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[25] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[28] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[31] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[34] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[37] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[40] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[43] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[46] matrixStats_1.4.1           pheatmap_1.0.12             ggpubr_0.6.0               
[49] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[52] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             presto_1.0.0              parallelly_1.38.0        
[145] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[148] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[151] pbapply_1.7-2             httr2_1.0.5               spam_2.11-0              
[154] gson_0.1.0                utf8_1.2.4                graphlayouts_1.2.0       
[157] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[160] shiny_1.9.1               GenomeInfoDbData_1.2.12   glmmTMB_1.1.10           
[163] R.utils_2.12.3            RCurl_1.98-1.16           memoise_2.0.1            
[166] rmarkdown_2.28            scales_1.3.0              R.methodsS3_1.8.2        
[169] future_1.34.0             RANN_2.6.2                Cairo_1.6-2              
[172] spatstat.data_3.1-2       rstudioapi_0.16.0         whisker_0.4.1            
[175] cluster_2.1.6             mutoss_0.1-13             spatstat.utils_3.1-0     
[178] hms_1.1.3                 fitdistrplus_1.2-1        munsell_0.5.1            
[181] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.4              
[184] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.0               
[187] dotCall64_1.2             shinydashboard_0.7.2      ggforce_0.4.2            
[190] laeken_0.5.3              mgcv_1.9-1                xfun_0.48                
[193] e1071_1.7-16              TH.data_1.1-2             iterators_1.0.14         
[196] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[199] ps_1.8.0                  futile.options_1.0.1      bitops_1.0-9             
[202] Rdpack_2.6.1              promises_1.3.0            scatterpie_0.2.4         
[205] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-1           
[208] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[211] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[214] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[217] Rcpp_1.0.13               edgeR_4.2.1               BiocSingular_1.20.0      
[220] tensor_1.5                MASS_7.3-61               progress_1.2.3           
[223] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.3-2    
[226] R6_2.5.1                  fastmap_1.2.0             multcomp_1.4-26          
[229] fastmatch_1.1-4           rstatix_0.7.2             vipor_0.4.7              
[232] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[235] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-19              
[238] gtable_0.3.5              KernSmooth_2.23-24        miniUI_0.1.1.1           
[241] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[244] bit64_4.5.2               spatstat.explore_3.3-2    lifecycle_1.0.4          
[247] blme_1.0-6                processx_3.8.4            callr_3.7.6              
[250] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[253] robustbase_0.99-4-1       spatstat.geom_3.3-3       sn_2.1.1                 
[256] ggfun_0.1.6               future.apply_1.11.2       bslib_0.8.0              
[259] pillar_1.9.0              gplots_3.2.0              pcaMethods_1.96.0        
[262] locfit_1.5-9.10           jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.7.1             pathview_1.44.0             wordcloud_2.6              
 [4] textshaping_0.4.0           EnhancedVolcano_1.22.0      ggrepel_0.9.6              
 [7] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
[10] futile.logger_1.4.3         ggupset_0.4.0               gridExtra_2.3              
[13] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_7.5.1              
[16] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[19] multtest_2.60.0             metap_1.11                  scater_1.32.1              
[22] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[25] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[28] lubridate_1.9.3             forcats_1.0.0               stringr_1.5.1              
[31] purrr_1.0.2                 readr_2.1.5                 tidyr_1.3.1                
[34] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[37] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[40] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[43] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[46] matrixStats_1.4.1           pheatmap_1.0.12             ggpubr_0.6.0               
[49] ggplot2_3.5.1               Seurat_5.1.0                SeuratObject_5.0.2         
[52] sp_2.1-4                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] graph_1.82.0              igraph_2.0.3              ica_1.0-3                
  [4] plotly_4.10.4             Formula_1.2-5             zlibbioc_1.50.0          
  [7] tidyselect_1.2.1          bit_4.5.0                 doParallel_1.0.17        
 [10] clue_0.3-65               lattice_0.22-6            rjson_0.2.23             
 [13] blob_1.2.4                S4Arrays_1.4.1            pbkrtest_0.5.3           
 [16] parallel_4.4.1            png_0.1-8                 plotrix_3.8-4            
 [19] cli_3.6.3                 ggplotify_0.1.2           goftest_1.2-3            
 [22] VIM_6.2.2                 variancePartition_1.34.0  BiocNeighbors_1.22.0     
 [25] shadowtext_0.1.4          uwot_0.2.2                curl_5.2.3               
 [28] tidytree_0.4.6            mime_0.12                 evaluate_1.0.1           
 [31] leiden_0.4.3.1            ComplexHeatmap_2.20.0     stringi_1.8.4            
 [34] backports_1.5.0           XML_3.99-0.17             lmerTest_3.1-3           
 [37] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [40] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [43] sctransform_0.4.1         ggbeeswarm_0.7.2          DBI_1.2.3                
 [46] jquerylib_0.1.4           smoother_1.3              withr_3.0.1              
 [49] git2r_0.33.0              corpcor_1.6.10            systemfonts_1.1.0        
 [52] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [55] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [58] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [61] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [64] DESeq2_1.44.0             ranger_0.16.0             DEoptimR_1.1-3           
 [67] reticulate_1.39.0         hexbin_1.28.4             zoo_1.8-12               
 [70] XVector_0.44.0            knitr_1.48                ggplot.multistats_1.0.1  
 [73] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [76] foreach_1.5.2             fansi_1.0.6               patchwork_1.3.0          
 [79] caTools_1.18.3            ggtree_3.12.0             data.table_1.16.0        
 [82] R.oo_1.26.0               RSpectra_0.16-2           irlba_2.3.5.1            
 [85] gridGraphics_0.5-1        fastDummies_1.7.4         lazyeval_0.2.2           
 [88] yaml_2.3.10               survival_3.7-0            scattermore_1.2          
 [91] crayon_1.5.3              RcppAnnoy_0.0.22          Rgraphviz_2.48.0         
 [94] progressr_0.14.0          tweenr_2.0.3              later_1.3.2              
 [97] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
[100] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[103] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[106] KEGGgraph_1.64.0          TMB_1.9.15                spatstat.univar_3.0-1    
[109] mathjaxr_1.6-0            getPass_0.2-4             EnvStats_3.0.0           
[112] aplot_0.2.3               scatterplot3d_0.3-44      ape_5.8                  
[115] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[118] highr_0.11                plyr_1.8.9                httr_1.4.7               
[121] rbibutils_2.3             tools_4.4.1               globals_0.16.3           
[124] beeswarm_0.4.0            broom_1.0.7               nlme_3.1-166             
[127] lambda.r_1.2.4            lme4_1.1-35.5             digest_0.6.37            
[130] numDeriv_2016.8-1.1       Matrix_1.7-0              farver_2.1.2             
[133] tzdb_0.4.0                remaCor_0.0.18            reshape2_1.4.4           
[136] yulab.utils_0.1.7         glue_1.8.0                cachem_1.1.0             
[139] polyclip_1.10-7           generics_0.1.3            Biostrings_2.72.1        
[142] mvtnorm_1.3-1             presto_1.0.0              parallelly_1.38.0        
[145] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[148] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[151] pbapply_1.7-2             httr2_1.0.5               spam_2.11-0              
[154] gson_0.1.0                utf8_1.2.4                graphlayouts_1.2.0       
[157] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[160] shiny_1.9.1               GenomeInfoDbData_1.2.12   glmmTMB_1.1.10           
[163] R.utils_2.12.3            RCurl_1.98-1.16           memoise_2.0.1            
[166] rmarkdown_2.28            scales_1.3.0              R.methodsS3_1.8.2        
[169] future_1.34.0             RANN_2.6.2                Cairo_1.6-2              
[172] spatstat.data_3.1-2       rstudioapi_0.16.0         whisker_0.4.1            
[175] cluster_2.1.6             mutoss_0.1-13             spatstat.utils_3.1-0     
[178] hms_1.1.3                 fitdistrplus_1.2-1        munsell_0.5.1            
[181] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.4              
[184] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.0               
[187] dotCall64_1.2             shinydashboard_0.7.2      ggforce_0.4.2            
[190] laeken_0.5.3              mgcv_1.9-1                xfun_0.48                
[193] e1071_1.7-16              TH.data_1.1-2             iterators_1.0.14         
[196] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[199] ps_1.8.0                  futile.options_1.0.1      bitops_1.0-9             
[202] Rdpack_2.6.1              promises_1.3.0            scatterpie_0.2.4         
[205] RSQLite_2.3.7             qvalue_2.36.0             sandwich_3.1-1           
[208] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[211] GO.db_3.19.1              compiler_4.4.1            prettyunits_1.2.0        
[214] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[217] Rcpp_1.0.13               edgeR_4.2.1               BiocSingular_1.20.0      
[220] tensor_1.5                MASS_7.3-61               progress_1.2.3           
[223] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.3-2    
[226] R6_2.5.1                  fastmap_1.2.0             multcomp_1.4-26          
[229] fastmatch_1.1-4           rstatix_0.7.2             vipor_0.4.7              
[232] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[235] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-19              
[238] gtable_0.3.5              KernSmooth_2.23-24        miniUI_0.1.1.1           
[241] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[244] bit64_4.5.2               spatstat.explore_3.3-2    lifecycle_1.0.4          
[247] blme_1.0-6                processx_3.8.4            callr_3.7.6              
[250] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[253] robustbase_0.99-4-1       spatstat.geom_3.3-3       sn_2.1.1                 
[256] ggfun_0.1.6               future.apply_1.11.2       bslib_0.8.0              
[259] pillar_1.9.0              gplots_3.2.0              pcaMethods_1.96.0        
[262] locfit_1.5-9.10           jsonlite_1.8.9            GetoptLong_1.0.5         </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
