---
title: "DE genes per patient"
author: "Emily Payne"
date: "2024-11-04"
output: html_document
---

```{r setup, include = FALSE, fig.path = "custom/path", warning = FALSE, message = FALSE}

options (width = 100)
knitr::opts_chunk$set(warning = FALSE, echo = TRUE, message = FALSE, dev = c("png", "pdf"))
seed <- 1234

```

```{r load libraries, echo=FALSE, message=FALSE, warning=FALSE}

suppressPackageStartupMessages({
  library(ExploreSCdataSeurat3)
  library(runSeurat3)
  library(Seurat)
  library(ggpubr)
  library(pheatmap)
  library(SingleCellExperiment)
  library(dplyr)
  library(tidyverse)
  library(viridis)
  library(muscat)
  library(circlize)
  library(destiny)
  library(scater)
  library(metap)
  library(multtest)
  library(clusterProfiler)
  library(org.Hs.eg.db)
  library(msigdbr)
  library(enrichplot)
  library(DOSE)
  library(grid)
  library(gridExtra)
  library(ggupset)
  library(VennDiagram)
  library(NCmisc)
  library(RColorBrewer)
  library(EnhancedVolcano)
  library(textshaping)
  library(wordcloud)
  library(pathview)
  library(workflowr)
})

```

```{r load merged file}
##load merged file 
seuratM <- readRDS("~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/seuratM_all_metadata_withAMvsHealthy_ventricle_sex_TC con_27.10.2024.rds")

table(seuratM$dataset)
table(seuratM$RNA_snn_res.0.25)
table(seuratM$patient, seuratM$sex)
table(seuratM$orig.ident)
```

```{r}

seuratM$clusterName <- "clusterName"
Idents(seuratM) <- seuratM$clusterName
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "0" )] <- "Fb1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "1" )] <- "BEC1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "2" )] <- "PeriFb1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "3" )] <- "Cardiomyocyte"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "4" )] <- "MCP1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "5" )] <- "TC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "6" )] <- "MCP2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "7" )] <- "BEC2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "8" )] <- "VSMC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "9" )] <- "NC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "10" )] <- "INT1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "11" )] <- "Fb2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "12" )] <- "Fb3"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "13" )] <- "PeriFb2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "14" )] <- "AdipoC"


colclusterName <- c("#D53E4F", "#f4a582", "#FEE08B","#feb24c","#67001f", "#01665e","#66C2A5","#c7eae5","#BEAED4", "#355C7D","#3288BD","#8c510a" ,"#fde0dd","#B45B5C","#dd1c77")

                    
names(colclusterName) <- c("Cardiomyocyte","Fb1","Fb2","Fb3", "INT1", "PeriFb1", "PeriFb2", "VSMC","NC", "BEC1","BEC2", "AdipoC", "MCP1","MCP2","TC")

colpat_sub<- c("#dfc27d","#BE3144","#355C7D", "#779d8d")

seuratM$clusterName <- factor(seuratM$clusterName, levels=c("Cardiomyocyte", "Fb1", "Fb2", "Fb3", "INT1","PeriFb1", "PeriFb2", "VSMC", "NC", "BEC1", "BEC2", "AdipoC", "MCP1", "MCP2", "TC"))


Idents(seuratM) <- seuratM$clusterName
DimPlot(seuratM, reduction = "umap", pt.size = 0.1, label = TRUE, cols = colclusterName)

```

```{r}


Idents(seuratM) <- seuratM$patient

table(seuratM$patient, seuratM$clusterName)

DimPlot(seuratM, reduction = "umap", pt.size = 0.1)


```

```{r DE genes for all patients positive and negative}

DEGenesall_patient <- FindAllMarkers(seuratM, only.pos = FALSE, logfc.threshold = 0.2) %>%
  dplyr::filter(p_val_adj < 0.01)

#write.table(DEGenes_patient, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenes_patient positive markers_05.11.2024", sep="\t",quote=F,row.names=F,col.names=T)

#View(DEGenesall_patient)

##adjust table
DEGenesall_patient_adj <- DEGenesall_patient %>% 
  mutate(Gene=gene) %>%
  mutate(gene=gsub("^.*\\.", "", Gene))  %>%
  mutate(EnsID=gsub("\\..*","", Gene))

```






```{r Heatmap for patient clustering with marker genes, fig.width=10, fig.height = 10}

## Heatmap patient clustering with common marker genes

Idents(seuratM) <- seuratM$patient
seuratM$patient <- factor(seuratM$patient, levels = c("Healthy 01","Healthy 02","Healthy 03","Healthy 04", "Healthy 05", "Healthy 06", "Healthy 07", "Healthy 08","AM 01", "AM 02","AM 03", "AM 04", "AM 05","AM 06","AM 07", "AM 08", "AM 09", "AM 10", "AM 11", "AM 12", "AM 13", "AM 14", "AM 15", "AM 16", "AM 17", "AM 18", "CMP 01", "CMP 02", "CMP 03", "CMP 04", "CMP 05", "CMP 06", "CMP 07", "CMP 08", "CMP 09", "CMP 10", "CMP 11", "CMP 12", "CMP 13", "Sarcoidosis 01", "Sarcoidosis 02"))

genes <- data.frame(gene=rownames(seuratM)) %>% 
  mutate(geneID=gsub("^.*\\.", "", gene))

selGenes <- data.frame(geneID=rev(c("MYBPC3", "RYR2", "NEBL", "TNNT2", "CMYA5", "COL6A3", "DCN",  "FBN1", "C7", "PDGFRA", "CDH19", "PDGFRB", "BMP4", "GREM1", "ITGA7","RGS5", "NOTCH3", "MYH11", "ACTA2","PECAM1", "VWF", "EGFL7", "POSTN", "ITGA10", "CDH11","CCL21", "FLT4", "NRXN1", "ANK3", "PTPRZ1", "PLIN1", "GPAM", "CD163", "LYVE1", "MRC1", "SIGLEC1", "STAB1", "CSF1R","MERTK", "HLA-DQB1","CCR2", "CD79B","JCHAIN", "IL7R", "CD3E", "PTPRC", "CD2"))) %>% left_join(., genes, by="geneID") %>% filter(gene != "ENSG00000232995.RGS5") 


avgHeatmap <- function(seuratM, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL, cc=FALSE,
                       cr=FALSE, condCol=FALSE) {
  
  selGenes <- selGenes$gene
  

  clusterAssigned <- as.data.frame(Idents(seuratM)) %>%
    dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] <- "ident"
  seuratDat <- GetAssayData(seuratM)
  

  genes <- data.frame(gene=rownames(seuratM)) %>% 
    mutate(geneID=gsub("^.*\\.", "", gene)) %>% filter(geneID %in% selGenes)

 
  logNormExpres <- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres <- logNormExpres %>% 
    dplyr::mutate(cell=rownames(.)) %>%
    dplyr::left_join(., clusterAssigned, by=c("cell")) %>%
    dplyr::select(-cell) %>%
    dplyr::group_by(ident) %>%
    dplyr::summarise_all(mean)
  
  logNormExpresMa <- logNormExpres %>% dplyr::select(-ident) %>% as.matrix()
  rownames(logNormExpresMa) <- logNormExpres$ident
  logNormExpresMa <- t(logNormExpresMa)
  rownames(logNormExpresMa) <- gsub("^.*?\\.", "", rownames(logNormExpresMa))
  
  ind <- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa <- logNormExpresMa[!ind,]
  genes <- genes[!ind,]
  
  logNormExpresMa <- logNormExpresMa[selGenes,]
  if (is.null(ordVec)) {
    ordVec <- levels(seuratM$patient)
  }
  logNormExpresMa <- logNormExpresMa[, ordVec, drop=FALSE]
  
  pheatmap(logNormExpresMa, scale = "row", treeheight_row = 0, cluster_rows = cr, 
           cluster_cols = cc, color = colorRampPalette(c("#2166AC", "#F7F7F7", "#B2182B"))(50),
           annotation_col = NULL, cellwidth = 15, cellheight = 10,
           annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}

genesPlot <- data.frame(gene=c("MYBPC3", "RYR2", "NEBL", "TNNT2", "CMYA5", "COL6A3", "DCN",  "FBN1", "C7", "PDGFRA", "CDH19", "PDGFRB", "BMP4", "GREM1", "ITGA7","RGS5", "NOTCH3", "MYH11", "ACTA2","PECAM1", "VWF", "EGFL7", "POSTN", "ITGA10", "CDH11","CCL21", "FLT4", "NRXN1", "ANK3", "PTPRZ1", "PLIN1", "GPAM", "CD163", "LYVE1", "MRC1", "SIGLEC1", "STAB1", "CSF1R","MERTK", "HLA-DQB1", "CCR2", "CD79B","JCHAIN", "IL7R","CD3E", "PTPRC", "CD2"))

levels(seuratM)


avgHeatmap(seuratM, selGenes = genesPlot, colVecIdent = colVec)


```


```{r, fig.width=10, fig.height = 10}
## Heatmap for patients with DE genes

Idents(seuratM) <- seuratM$patient
seuratM$patient <- factor(seuratM$patient, levels = c("Healthy 01","Healthy 02","Healthy 03","Healthy 04", "Healthy 05", "Healthy 06", "Healthy 07", "Healthy 08","AM 01", "AM 02","AM 03", "AM 04", "AM 05","AM 06","AM 07", "AM 08", "AM 09", "AM 10", "AM 11", "AM 12", "AM 13", "AM 14", "AM 15", "AM 16", "AM 17", "AM 18", "CMP 01", "CMP 02", "CMP 03", "CMP 04", "CMP 05", "CMP 06", "CMP 07", "CMP 08", "CMP 09", "CMP 10", "CMP 11", "CMP 12", "CMP 13", "Sarcoidosis 01", "Sarcoidosis 02"))

genes <- data.frame(gene=rownames(seuratM)) %>% 
  mutate(geneID=gsub("^.*\\.", "", gene))

selGenes <- data.frame(geneID=rev(c("XIST", "RYR2","TNNT2","MALAT1", "COL8A1", "LAMA2", "SOX4", "NPPA", "GREM1", "GREM2", "BMP4", "BMPER", "TGFBR3", "ADAMTS5", "HIF3A","THBS1", "FOXO3", "EGR1", "HGF", "SERPINE1", "TIMP1", "FBN1", "IRS2", "GCGR", "SLC6A1", "POU3F3", "ZBTB16", "ACTA2","VWF", "POSTN", "CCL21", "LYVE1", "TNFAIP3","CXCL9", "HLA-B", "CCR2","JAK2", "STAT1", "TLR4", "IRF1", "GBP5", "IGHG3", "IL7R", "CD3E", "PTPRC", "CD2", "CD74", "IL17RA"))) %>% left_join(., genes, by="geneID") %>% filter(gene != "ENSG00000232995.RGS5") 


avgHeatmap <- function(seuratM, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=TRUE, gapVecR=NULL, gapVecC=NULL, cc=FALSE,
                       cr=FALSE, condCol=FALSE) {
  
  selGenes <- selGenes$gene
  

  clusterAssigned <- as.data.frame(Idents(seuratM)) %>%
    dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] <- "ident"
  seuratDat <- GetAssayData(seuratM)
  

  genes <- data.frame(gene=rownames(seuratM)) %>% 
    mutate(geneID=gsub("^.*\\.", "", gene)) %>% filter(geneID %in% selGenes)

 
  logNormExpres <- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres <- logNormExpres %>% 
    dplyr::mutate(cell=rownames(.)) %>%
    dplyr::left_join(., clusterAssigned, by=c("cell")) %>%
    dplyr::select(-cell) %>%
    dplyr::group_by(ident) %>%
    dplyr::summarise_all(mean)
  
  logNormExpresMa <- logNormExpres %>% dplyr::select(-ident) %>% as.matrix()
  rownames(logNormExpresMa) <- logNormExpres$ident
  logNormExpresMa <- t(logNormExpresMa)
  rownames(logNormExpresMa) <- gsub("^.*?\\.", "", rownames(logNormExpresMa))
  
  ind <- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa <- logNormExpresMa[!ind,]
  genes <- genes[!ind,]
  
  logNormExpresMa <- logNormExpresMa[selGenes,]
  if (is.null(ordVec)) {
    ordVec <- levels(seuratM$patient)
  }
  logNormExpresMa <- logNormExpresMa[, ordVec, drop=FALSE]
  
  pheatmap(logNormExpresMa, scale = "row", treeheight_row = 0, cluster_rows = cr, 
           cluster_cols = cc, color = colorRampPalette(c("#2166AC", "#F7F7F7", "#B2182B"))(50),
           annotation_col = NULL, cellwidth = 15, cellheight = 10,
           annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}

genesPlot <- data.frame(gene=c("XIST", "RYR2","TNNT2","MALAT1", "COL8A1", "LAMA2", "SOX4", "NPPA","GREM1", "GREM2", "BMP4", "BMPER", "TGFBR3", "ADAMTS5", "HIF3A","FOXO3", "EGR1", "HGF", "SERPINE1", "TIMP1", "IRS2", "GCGR", "SLC6A1", "POU3F3", "ZBTB16", "ACTA2","VWF", "POSTN", "CCL21", "LYVE1", "TNFAIP3","CXCL9", "HLA-B", "CCR2","JAK2", "STAT1", "TLR4", "IRF1", "GBP5", "IGHG3", "IL7R", "CD3E", "PTPRC", "CD2", "CD74", "IL17RA"))

levels(seuratM)


avgHeatmap(seuratM, selGenes = genesPlot, colVecIdent = colVec)

# AM 16-18 + CMP 13 = added = NA

# "XIST", "RYR2","TNNT2","MALAT1", "COL8A1", "LAMA2", "SOX4", "NPPA", "BMP4", "BMPER", "TGFBR3", "ADAMTS5", "HIF3A","THBS1", "FOXO3", "EGR1", "HGF", "SERPINE1", "TIMP1", "FBN1", "IRS2", "GCGR", "SLC6A1", "POU3F3", "ZBTB16", "ACTA2","VWF", "POSTN", "CCL21", "LYVE1", "TNFAIP3","CXCL9","CXCL10", "HLA-B", "CCR2","JAK2", "STAT1", "TLR4", "IRF1", "GBP5", "IGHM", "IGHG3", "IL7R", "CD3E", "PTPRC", "CD2", "CD74", "IL17RA"

```


```
```





```







```{r, DE Genes Volcano Plot AM 01, fig.width = 14, fig.height=14}


## Volcano Plot with pos. and neg. DE Genes for all conditions: 

##adjust table
DEGenesall_patient_adj <- DEGenesall_patient %>% 
  mutate(Gene=gene) %>%
  mutate(gene=gsub("^.*\\.", "", Gene))  %>%
  mutate(EnsID=gsub("\\..*","", Gene))

DEGenesallAM1 <- DEGenesall_patient_adj %>% filter(cluster == "AM 01")

res_AM1 <- as.data.frame(DEGenesallAM1)

EnhancedVolcano(res_AM1, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM1$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 01',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```



```{r DE Genes Volcano Plot AM 02, fig.width = 8, fig.height=8 }

## Volcano Plot with pos. and neg. DE Genes for Patient AM 02 

DEGenesallAM2 <- DEGenesall_patient_adj %>% filter(cluster == "AM 02")

res_AM2 <- as.data.frame(DEGenesallAM2)

EnhancedVolcano(res_AM2, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM2$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 02',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

## single numbers as gene names relate to pseudogenes or RNA

```

```{r DE Genes Volcano Plot AM 03, fig.width = 10, fig.height=7}

## Volcano Plot with DE Genes for Patient AM 03 


DEGenesallAM3 <- DEGenesall_patient_adj %>% filter(cluster == "AM 03")

res_AM3 <- as.data.frame(DEGenesallAM3)

EnhancedVolcano(res_AM3, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM3$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 03',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 04, fig.width = 10, fig.height=7}

## Volcano Plot with DE Genes for Patient AM 04 

DEGenesallAM4 <- DEGenesall_patient_adj %>% filter(cluster == "AM 04")

res_AM4 <- as.data.frame(DEGenesallAM4)

EnhancedVolcano(res_AM4, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM4$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 04',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 05, fig.width = 10, fig.height=9}

## Volcano Plot with DE Genes for Patient AM 05 


DEGenesallAM5 <- DEGenesall_patient_adj %>% filter(cluster == "AM 05")

res_AM5 <- as.data.frame(DEGenesallAM5)

EnhancedVolcano(res_AM5, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM5$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 05',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 06, fig.width = 12, fig.height=9}

## Volcano Plot with DE Genes for Patient AM 06

DEGenesallAM6 <- DEGenesall_patient_adj %>% filter(cluster == "AM 06")

res_AM6 <- as.data.frame(DEGenesallAM6)

EnhancedVolcano(res_AM6, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM6$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 06',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 07, fig.width = 14, fig.height=12}

## Volcano Plot with DE Genes for Patient AM 07
DEGenesallAM7 <- DEGenesall_patient_adj %>% filter(cluster == "AM 07")

res_AM7 <- as.data.frame(DEGenesallAM7)

EnhancedVolcano(res_AM7, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM7$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 07',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 08, fig.width = 11, fig.height=7}

## Volcano Plot with DE Genes for Patient AM 08
DEGenesallAM8 <- DEGenesall_patient_adj %>% filter(cluster == "AM 08")

res_AM8 <- as.data.frame(DEGenesallAM8)

EnhancedVolcano(res_AM8, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM8$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 08',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 09, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient AM 09
DEGenesallAM9 <- DEGenesall_patient_adj %>% filter(cluster == "AM 09")

res_AM9 <- as.data.frame(DEGenesallAM9)

EnhancedVolcano(res_AM9, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM9$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 09',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 10, fig.width = 8, fig.height=6}

## Volcano Plot with DE Genes for Patient AM 10
DEGenesallAM10 <- DEGenesall_patient_adj %>% filter(cluster == "AM 10")

res_AM10 <- as.data.frame(DEGenesallAM10)

EnhancedVolcano(res_AM10, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM10$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 10',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 11, fig.width = 10, fig.height=7}

## Volcano Plot with DE Genes for Patient AM 11
DEGenesallAM11 <- DEGenesall_patient_adj %>% filter(cluster == "AM 11")

res_AM11 <- as.data.frame(DEGenesallAM11)

EnhancedVolcano(res_AM11, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM11$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 11',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 12, fig.width = 10, fig.height=8}


## Volcano Plot with DE Genes for Patient AM 12

DEGenesallAM12 <- DEGenesall_patient_adj %>% filter(cluster == "AM 12")

# pos. and neg. DE genes 
res_AM12 <- as.data.frame(DEGenesallAM12)

EnhancedVolcano(res_AM12, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM12$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 12',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 13, fig.width = 10, fig.height=8}


## Volcano Plot with DE Genes for Patient AM 13
DEGenesallAM13 <- DEGenesall_patient_adj %>% filter(cluster == "AM 13")

# pos. and neg. DE genes
res_AM13 <- as.data.frame(DEGenesallAM13)

EnhancedVolcano(res_AM13, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM13$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 13',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 14, fig.width = 14, fig.height=10}

## Volcano Plots with DE Genes for Patient AM 14 

DEGenesallAM14 <- DEGenesall_patient_adj %>% filter(cluster == "AM 14")

# pos. and neg. DE genes
res_AM14 <- as.data.frame(DEGenesallAM14)

EnhancedVolcano(res_AM14, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM14$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 14',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

## single numbers as gene names refer to pseudogenes or RNA

```

```{r DE Genes Volcano Plot AM 15, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient AM 15
DEGenesallAM15 <- DEGenesall_patient_adj %>% filter(cluster == "AM 15")

# pos. and neg. DE genes
res_AM15 <- as.data.frame(DEGenesallAM15)

EnhancedVolcano(res_AM15, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM15$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 15',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 16, fig.width = 12, fig.height=12}

## Volcano Plot with DE Genes for Patient AM 16 

DEGenesallAM16 <- DEGenesall_patient_adj %>% filter(cluster == "AM 16")

# pos. and neg. DE genes
res_AM16 <- as.data.frame(DEGenesallAM16)

EnhancedVolcano(res_AM16, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM16$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 16',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 4.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 17, fig.width = 9, fig.height=6}

## Volcano Plot with DE Genes for Patient AM 17 

DEGenesallAM17 <- DEGenesall_patient_adj %>% filter(cluster == "AM 17")

res_AM17 <- as.data.frame(DEGenesallAM17)

EnhancedVolcano(res_AM17, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM17$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 17',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot AM 18, fig.width = 8, fig.height=6}

## Volcano Plot with DE Genes for Patient AM 18 

DEGenesallAM18 <- DEGenesall_patient_adj %>% filter(cluster == "AM 18")

res_AM18 <- as.data.frame(DEGenesallAM18)

EnhancedVolcano(res_AM18, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_AM18$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient AM 18',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP1, fig.width = 8, fig.height=6}

## Volcano Plot with DE Genes for Patient CMP 01

DEGenesallCMP1 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 01")

# pos. and neg. DE genes for AM
res_CMP1 <- as.data.frame(DEGenesallCMP1)

EnhancedVolcano(res_CMP1, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP1$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 01',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP2, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient CMP 02

DEGenesallCMP2 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 02")

res_CMP2 <- as.data.frame(DEGenesallCMP2)

EnhancedVolcano(res_CMP2, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP2$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 02',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP3, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient CMP 03


DEGenesallCMP3 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 03")

res_CMP3 <- as.data.frame(DEGenesallCMP3)

EnhancedVolcano(res_CMP3, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP3$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 03',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP4, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient CMP 04


DEGenesallCMP4 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 04")


res_CMP4 <- as.data.frame(DEGenesallCMP4)

EnhancedVolcano(res_CMP4, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP4$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 04',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP5, fig.width = 10, fig.height=10}

## Volcano Plot with DE Genes for Patient CMP 05


DEGenesallCMP5 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 05")

res_CMP5 <- as.data.frame(DEGenesallCMP5)

EnhancedVolcano(res_CMP5, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP5$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 05',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP6, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient CMP 06


DEGenesallCMP6 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 06")

res_CMP6 <- as.data.frame(DEGenesallCMP6)

EnhancedVolcano(res_CMP6, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP6$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 06',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP7, fig.width = 10, fig.height=10}

## Volcano Plot with DE Genes for Patient CMP 07


DEGenesallCMP7 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 07")

res_CMP7 <- as.data.frame(DEGenesallCMP7)

EnhancedVolcano(res_CMP7, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP7$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 07',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP8, fig.width = 10, fig.height=10}

## Volcano Plot with DE Genes for Patient CMP 08


DEGenesallCMP8 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 08")

res_CMP8 <- as.data.frame(DEGenesallCMP8)

EnhancedVolcano(res_CMP8, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP8$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 08',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP9, fig.width = 10, fig.height=6}

## Volcano Plot with DE Genes for Patient CMP 09


DEGenesallCMP9 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 09")

res_CMP9 <- as.data.frame(DEGenesallCMP9)

EnhancedVolcano(res_CMP9, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP9$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 09',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP10, fig.width = 10, fig.height=6}

## Volcano Plot with DE Genes for Patient CMP 10


DEGenesallCMP10 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 10")

res_CMP10 <- as.data.frame(DEGenesallCMP10)

EnhancedVolcano(res_CMP10, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP10$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 10',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP11, fig.width = 10, fig.height=6}

## Volcano Plot with DE Genes for Patient CMP 11


DEGenesallCMP11 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 11")

res_CMP11 <- as.data.frame(DEGenesallCMP11)

EnhancedVolcano(res_CMP11, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP11$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 11',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP12, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient CMP 12


DEGenesallCMP12 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 12")

res_CMP12 <- as.data.frame(DEGenesallCMP12)

EnhancedVolcano(res_CMP12, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP12$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 12',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot CMP13, fig.width = 8, fig.height=6}

## Volcano Plot with DE Genes for Patient CMP 13


DEGenesallCMP13 <- DEGenesall_patient_adj %>% filter(cluster == "CMP 13")

res_CMP13 <- as.data.frame(DEGenesallCMP13)

EnhancedVolcano(res_CMP13, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_CMP13$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient CMP 13',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot Sarcoidosis 01, fig.width = 12, fig.height=12}

## Volcano Plot with DE Genes for Patient Sarcoidosis 01

DEGenesallSarco1 <- DEGenesall_patient_adj %>% filter(cluster == "Sarcoidosis 01")

res_Sarco1 <- as.data.frame(DEGenesallSarco1)

EnhancedVolcano(res_Sarco1, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Sarco1$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Sarcoidosis 01',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.5,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot Sarcoidosis 02, fig.width = 12, fig.height=12}

## Volcano Plot with DE Genes for Patient Sarcoidosis 02

DEGenesallSarco2 <- DEGenesall_patient_adj %>% filter(cluster == "Sarcoidosis 02")

res_Sarco2 <- as.data.frame(DEGenesallSarco2)

EnhancedVolcano(res_Sarco2, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Sarco2$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Sarcoidosis 02',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.5,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot Healthy 01, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient Healthy 01

DEGenesallHealthy1 <- DEGenesall_patient_adj %>% filter(cluster == "Healthy 01")

res_Healthy1 <- as.data.frame(DEGenesallHealthy1)

EnhancedVolcano(res_Healthy1, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Healthy1$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Healthy 01',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot Healthy 02, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient Healthy 02

DEGenesallHealthy2 <- DEGenesall_patient_adj %>% filter(cluster == "Healthy 02")

res_Healthy2 <- as.data.frame(DEGenesallHealthy2)

EnhancedVolcano(res_Healthy2, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Healthy2$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Healthy 02',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot Healthy 03, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient Healthy 03

DEGenesallHealthy3 <- DEGenesall_patient_adj %>% filter(cluster == "Healthy 03")

res_Healthy3 <- as.data.frame(DEGenesallHealthy3)

EnhancedVolcano(res_Healthy3, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Healthy3$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient Healthy 03',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot Healthy 04, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient Healthy 04

DEGenesallHealthy4 <- DEGenesall_patient_adj %>% filter(cluster == "Healthy 04")

res_Healthy4 <- as.data.frame(DEGenesallHealthy4)

EnhancedVolcano(res_Healthy4, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Healthy4$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient Healthy 04',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot Healthy 05, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient Healthy 05

DEGenesallHealthy5 <- DEGenesall_patient_adj %>% filter(cluster == "Healthy 05")

res_Healthy5 <- as.data.frame(DEGenesallHealthy5)

EnhancedVolcano(res_Healthy5, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Healthy5$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient Healthy 05',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot Healthy 06, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient Healthy 06

DEGenesallHealthy6 <- DEGenesall_patient_adj %>% filter(cluster == "Healthy 06")

res_Healthy6 <- as.data.frame(DEGenesallHealthy6)

EnhancedVolcano(res_Healthy6, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Healthy6$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient Healthy 06',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot Healthy 07, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient Healthy 07

DEGenesallHealthy7 <- DEGenesall_patient_adj %>% filter(cluster == "Healthy 07")

res_Healthy7 <- as.data.frame(DEGenesallHealthy7)

EnhancedVolcano(res_Healthy7, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Healthy7$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient Healthy 07',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```

```{r DE Genes Volcano Plot Healthy 08, fig.width = 10, fig.height=8}

## Volcano Plot with DE Genes for Patient Healthy 08
## "Healthy 08" shows inflammatory signature

DEGenesallHealthy8 <- DEGenesall_patient_adj %>% filter(cluster == "Healthy 08")

res_Healthy8 <- as.data.frame(DEGenesallHealthy8)

EnhancedVolcano(res_Healthy8, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Healthy8$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Patient Healthy 08',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 2.0,
    labSize = 3.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)


```

```{r Find DE genes for each patient GSEA and gene ontology}

# Gene Ontology and GSEA for all Patients, positive markers

DEGenes_patient <- FindAllMarkers(seuratM, only.pos = TRUE, logfc.threshold = 0.2) %>%
  dplyr::filter(p_val_adj < 0.01)

#head(DEGenes_patient)

#write.table(DEGenes_patient, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenes_patient positive markers_05.11.2024", sep="\t",quote=F,row.names=F,col.names=T)

##adjust table
DEGenes_patient <- DEGenes_patient %>% 
  mutate(Gene=gene) %>%
  mutate(gene=gsub("^.*\\.", "", Gene))  %>%
  mutate(EnsID=gsub("\\..*","", Gene))

#View(DEGenes_patient)


```

```{r, fig.width = 9, fig.height = 8}
# Patient Gene Ontology AM 01
DEGenesAM1 <- DEGenes_patient %>% filter(cluster == "AM 01")

egoAM1 <- enrichGO(gene = unique(DEGenesAM1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM1 <- setReadable(egoAM1, OrgDb = org.Hs.eg.db)

#View(egoAM1)

dotplot(egoAM1, showCategory=20, title = "Patient AM 01 Biological Process", font.size = 10)
barplot(egoAM1, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 01 Biological Process",
        font.size = 10)

egoAM1 <- enrichGO(gene = unique(DEGenesAM1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM1 <- setReadable(egoAM1, OrgDb = org.Hs.eg.db)
dotplot(egoAM1, showCategory=20, title = "Patient AM 01 Metabolic Function", font.size = 10)

egoAM1 <- enrichGO(gene = unique(DEGenesAM1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM1 <- setReadable(egoAM1, OrgDb = org.Hs.eg.db)
dotplot(egoAM1, showCategory=20, title = "Patient AM 01 Cellular Component", font.size = 10)
```


```{r, egoAM categories}

### Gene patterns and signature sorted by function found in Patient AM 01


# Genes associated with T cell differentiation: PTPRC/B2M/IRF1/CD74/NCKAP1L/IL7R/CD2/DOCK2/JAK3/HLA-DRA/BCL11B/HLA-DRB1/IKZF3/TCIRG1/RUNX3/SPN/PREX1/PTPN22/LILRB2/ZAP70/ITK/PIK3CD/LCK/FGL2/LAG3/LILRB4/IL2RG/CD3G/STAT4/KLRC1/VAV1/CD3E/RHOH/CAMK4/ADAM8/CD1D/ZNF683/SASH3/THEMIS/IL12RB1/LY9/IL4I1/CARD11/CD3D/CD8A/CD4/SLAMF6/FANCA/CTLA4/CYLD/LGALS9/TBX21/TESPA1/SPI1/PTPN2/BTN2A2/NFKBID/CCR7/CD27/TOX/LEF1/TNFRSF9/SYK/GATA3/RASGRP1/HLA-DOA/PRDM1/TMEM131L/CD28/TNFSF8/MAFB/SEMA4A/ZFP36L2/EOMES/IFNG/TGFB1/RUNX1/JAK1/BATF/CD83/STAT6/FANCD2/ACTB/CCR2/MR1/AP3B1/ZFP36L1/RUNX2/IL15/FCER1G/NFATC3/IRF4/ASCL2/RIPK3/IL4R/CD80/TCF7/CD86/NLRP3/WNT10B/JUNB/BCL2/ZC3H12A/PIK3R6/IL18/ZBTB7B/SMARCC1/SOS1/PRKDC/GPR183/PTGER4/RSAD2/BRD2/ARID1A/LFNG/CRTAM/DUSP10/IL6R/VSIR/NCAPH2/PBRM1/RELB/KMT2A/AP3D1/LOXL3/IL18R1/IL2RA/CLEC4E/STAT5B/SOCS1/MALT1/DROSHA/RORC/GPR18/ANXA1/ATG5/SMARCA4/PRKCZ/RIPK2/TP53/PNP/RC3H2

# TCR signaling: GBP1/PTPN22/LCK/PTPN6/LILRB4/LAPTM5/CD226/CD300A/KCNN4/CARD11/UBASH3A/PTPRJ/CYLD/TESPA1/PTPN2/BTN2A2/CCR7/TRAT1/SH2D1A/DGKZ/RAB29/SLA2/PRKD2/MALT1/IKBKG

# Th 17 response: JAK2/CLEC7A/IL12RB1/LY9/PRKCQ/SLAMF6/TBX21/IL17RA/CARD9/IL27RA/NFKBID/JAK1/BATF/TYK2/IRF4/ASCL2/JUNB/ZC3H12A/ZBTB7B/BRD2/IL6R/ARID5A/LOXL3/MALT1/RORC/RC3H2

# type II interferon response STAT1/GBP5/IRF1/GBP1/NLRC5/JAK2/SP100/GBP2/GBP4/HLA-DPA1/WAS/IRF8/CCL5/CCL4/IL12RB1/CCL3/TNF/AIF1/TLR2/LGALS9/PARP9/PTPN2/PARP14/CASP1/TXK/HCK/GBP3/FASLG/NR1H3/IFNG/CCL4L2/ACTR3/JAK1/TYK2/CD47/DAPK1/IFNGR2/STX4/ZYX/KIF5B/CCL8/ACTR2/STXBP3/TLR4/SIRPA/SLC26A6/TP53/RPL13A/SYNCRIP

# B cell activation: PTPRC/CD74/NCKAP1L/IL7R/DOCK10/JAK3/ITGA4/ATM/TBC1D10C/INPP5D/PTK2B/IKZF3/TCIRG1/CASP8/ZAP70/PIK3CD/TNFAIP3/PRKCB/EZH2/PTPN6/SAMSN1/BTK/IRF8/IL2RG/LAPTM5/MNDA/CD300A/SASH3/IL4I1/CARD11/MS4A1/THEMIS2/TNFSF13B/PTPRJ/LAT2/CTLA4/TYROBP/CYLD/TBX21/SPI1/PTPN2/SLAMF8/BANK1/IL27RA/LAX1/PRKCD/CD27/LEF1/NOD2/SYK/CD38/RASGRP1/CD40LG/CD28/SKAP2/ZFP36L2/TGFB1/TFRC/FLT3/CD70/BATF/STAT6/PLCG2/BAX/FCRL1/BLNK/MZB1/ID2/NFAM1/ZFP36L1/DOCK11/NFATC2/LYN/RBPJ/TNFRSF13C/CDKN1A/CD86/GAPT/BCL2/DCLRE1C/CD22/PKN1/PRKDC/GPR183/SH3KBP1/AKAP17A/CD180/VCAM1/LFNG/NSD2/FCRL3/NOTCH2/SLC39A10/ATAD5/MFNG/ZBTB7A/STAT5B/SUPT6H/CD79B/TLR4/BST2/VAV3/KLF6/RIF1/TICAM1/TPD52/PARP3/CMTM7/MALT1/DCAF1/TNFRSF4/EP300/LYL1/PPP2R3C/HHEX/ITFG2/AKIRIN2/NBN/TP53

# Innate immune responses: GBP5/IRF1/TRAF3IP3/CTSS/CYBA/USP15/GBP2/BIRC3/NAGK/ARRB2/PTPN22/NLRC3/PYHIN1/TNFAIP3/FCN1/CLEC7A/ZBP1/OAS1/BTK/HAVCR2/KLRC1/MNDA/CD300A/TASL/AIM2/IRF7/IFI16/TNIP3/TNF/UNC93B1/TLR2/TYROBP/CYLD/OASL/GFI1/CASP1/KLRD1/DDX60/MYD88/PIK3AP1/CD300LF/TRAF3/TLR6/IFIH1/NOD2/SCIMP/PAK1/IRAK1/HCK/SYK/NMI/CGAS/KLRC2/IRAK4/DDX3X/NR1H3/KLRK1/NLRP1/TLR10/AURKB/CARD8/NLRC4/PLCG2/OAS3/AP3B1/TLR8/IRF4/TLR1/IKBKB/FYN/MEFV/DHX58/ZNFX1/LYN/TREM2/RNF135/IRF3/ALPK1/NLRP3/SLC15A3/CD274/IFI35/TLR7/PARP1/PAK2/PRKDC/RSAD2/RIGI/XRCC5/SRC/KLRC4/EIF2AK2/PYCARD/FCRL3/CD14/TAX1BP1/LRCH4/ERBIN/TRIM11/CLEC4E/HSP90AA1/WDFY1/NAIP/TRIM25/GPR108/AKT1/TLR4/LILRA2/RBM14/RIOK3/IRAK2/SLC15A2/TICAM1/TKFC/RPS6KA3/PDPK1/SIN3A/UFD1/EP300/ITCH/ZDHHC18/RIPK2/LACC1

# NK activation HLA-E/JAK2/PTPN22/HAVCR2/FGR/KLRC1/ZNF683/TYROBP/KLRD1/TOX/RASGRP1/CLNK/PRDM1/KLRC2/FCGR3A/TYK2/IL15/IL18/STAT5B/DCAF15/TICAM1/IL15RA

# phagocytosis: PTPRC/ITGAL/NCKAP1L/DOCK2/ITGB2/CORO1A/CYBA/BIN2/ABCA7/LYST/MYO1G/P2RY6/FCN1/CLEC7A/P2RX7/ARHGAP25/BTK/IRF8/IL2RG/UNC13D/FGR/VAV1/SH3BP1/CD300A/IL2RB/ICAM3/SIRPB1/SLAMF1/SIRPG/PTPRJ/MSR1/AIF1/NCF4/TLR2/TYROBP/CDC42SE2/ITGAM/CSK/FCGR1A/PRKCD/NCF2/MYD88/CD300LF/NOD2/PAK1/C2/HCK/SYK/LIMK1/MYO7A/NR1H3/IFNG/PLCG2/CCR2/SPG11/IL15/FCER1G/CD47/FYN/CNN2/LYN/TREM2/RAB27A/CDC42SE1/DNM2/DYSF/ATG3/BLTP1/SLC11A1/ELMO1/RAB31/CFP/SRC/TMEM175/PYCARD/CD14/CEACAM4/SCARB1/RUBCN/CAMK1D/ITGA2/EIF2AK1/CLN3/TLR4/VAV3/SPHK1/RAB39A/FCGR2A/SIRPA/ELMO2/ELMO3/MYH9/LYAR/IL15RA/ANXA1/ATG5/PLA2G6

# IL6 reaaction: HLA-B/CD74/NCKAP1L/CYBA/POU2F2/INPP5D/CD84/ARRB2/PTPN22/NLRC3/LILRB2/TNFAIP3/CLEC7A/P2RX7/PTPN6/LILRB4/BTK/HAVCR2/SLAMF1/TNF/UNC93B1/AIF1/TLR2/TYROBP/LGALS9/CSK/MBP/IL17RA/BANK1/CARD9/IL27RA/SIGLEC16/MYD88/TLR6/IFIH1/NOD2/SCIMP/SYK/IFNG/TGFB1/TMEM106A/IL16/PLCG2/TLR8/TLR1/CD47/PTAFR/TREM2/TLR7/ZC3H12A/AGER/ARHGEF2/RIGI/PYCARD/IL6R/ELF4/ARID5A/ARRB1/TLR4/LILRA2/CD200R1/MAPK13/GHRL/SIRPA/TICAM1/DHX9/TRPV4/AKIRIN2/RIPK2

# response to bacterial origin: GBP5/B2M/IL10RA/JAK2/GBP2/LILRB1/CASP8/CD6/PTPN22/CYRIB/LILRB2/TAP2/TNFAIP3/P2RX7/CD96/BTK/IRF8/HAVCR2/CXCL10/CXCL9/TNIP3/TNF/TLR2/LGALS9/ACP5/GFI1/IRF5/CARD9/CCR7/CASP1/MYD88/TLR6/NOD2/SCIMP/IRAK1/C2/HCK/TNFRSF1B/GCH1/GBP3/PDE4B/FASLG/IRAK4/KMO/NR1H3/KLRK1/CSF2RB/SIGIRR/NFKB2/TLR10/SBNO2/CARD8/LTA/PLCG2/CCR5/LY86/TLR1/PTGER2/LYN/PTAFR/TREM2/CD80/IRF3/CD86/NLRP3/CD274/IDO1/PDCD4/ZC3H12A/IL18BP/FOS/SLC11A1/PTGER4/TRIM41/CD180/SRC/VCAM1/PYCARD/DUSP10/IL12RB2/CD14/SCARB1/ARID5A/ERBIN/NUGGC/CARD16/CDK19/HPGD/TAB2/AKT1/TLR4/LILRA2/ZFP36/CNR2/MAP2K3/PCK2/IRAK2/SIRPA/TICAM1/CMPK2/MALT1/RPS6KA3/OTUD5/CD68/SLC7A5/NFKB1/AKIRIN2/RIPK2/RPL13A/HMGB2/AKAP8

# viral genome replication: MX1/RESF1/GRK2/APOBEC3G/OAS1/CCL5/IFI16/APOBEC3H/ATG16L2/ZC3HAV1/TNF/OASL/OAS2/ISG20/FAM111A/IFIH1/TOP2A/PABPC1/CD28/DDX3X/TRIM38/ADAR/OAS3/ZNFX1/APOBEC3D/SHFL/ATG16L1/CXCR6/DEK/APOBEC3C/BCL2/ZC3H12A/ILF3/ISG15/RSAD2/EIF2AK2/CTBP1/APOBEC3F/CCL8/EEF1A1/SRPK1/FMR1/SETDB1/BST2/MPHOSPH8/LARP1/PKN2/ZBED1/SMC6/ATG5/TASOR


```

```{r, fig.width = 9, fig.height = 8}

# Patient AM 02
DEGenesAM2 <- DEGenes_patient %>% filter(cluster == "AM 02")


egoAM2 <- enrichGO(gene = unique(DEGenesAM2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

egoAM2 <- setReadable(egoAM2, OrgDb = org.Hs.eg.db)

#View(egoAM2)

dotplot(egoAM2, showCategory=20, title = "Patient AM 02 Biological Process", font.size = 10)
barplot(egoAM2, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 02",
        font.size = 10)

egoAM2 <- enrichGO(gene = unique(DEGenesAM2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

egoAM2 <- setReadable(egoAM2, OrgDb = org.Hs.eg.db)
dotplot(egoAM2, showCategory=20, title = "Patient AM 02 Metabolic Function", font.size = 10)

egoAM2 <- enrichGO(gene = unique(DEGenesAM2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

egoAM2 <- setReadable(egoAM2, OrgDb = org.Hs.eg.db)
dotplot(egoAM2, showCategory=20, title = "Patient AM 02 Cellular Component", font.size = 10)

# heart contraction: TTN/MYBPC3/TPM1/MYH7/RYR2/TNNT2/DMD/SCN7A/MYH6/ATP2A2/TNNI3/PKP2/MYL7/GLP1R/GSN/CYP2J2/AKAP9/SCN5A/CHRM2/MYL3/ANK2/KCNH2/SLC4A3/SNTA1/MYL2/CASQ2/CACNA1C/POPDC2/ACTC1/ATP1A2/TNNC1/RNF207/TBX5/TBX2/DSP/ADRB1/GJA1/THRA/FGF12/JUP/CTNNA3/TNNI3K/TRDN/CACNB2/TMEM38A/TRPC1/DES/PPP1R13L/PLN/SPTBN4/DMPK/SCN2B/GATA6/TMEM65/CAMK2D/FLNA/AKAP13/KCND3/ACE/CORIN/SLC8A1/BVES/SPX

# action potential: RYR2/DMD/SCN7A/ATP2A2/PKP2/AKAP9/SCN5A/SLMAP/ANK2/KCNH2/SLC4A3/SNTA1/CACNA1C/ATP1A2/RNF207/DSP/GJA1/FGF12/JUP/CTNNA3/ANK3/CACNB2/SCN2B/CAMK2D/FLNA/KCND3/SLC8A1

# muscle tissue

```

```{r, fig.width = 9, fig.height = 8}


# Patient AM 03
DEGenesAM3 <- DEGenes_patient %>% filter(cluster == "AM 03")


egoAM3 <- enrichGO(gene = unique(DEGenesAM3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM3 <- setReadable(egoAM3, OrgDb = org.Hs.eg.db)

#View(egoAM3)
dotplot(egoAM3, showCategory=20, title = "Patient AM 03 Biological Process", font.size = 10)
barplot(egoAM3, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 03 Biological Process",
        font.size = 10)

egoAM3 <- enrichGO(gene = unique(DEGenesAM3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM3 <- setReadable(egoAM3, OrgDb = org.Hs.eg.db)
dotplot(egoAM3, showCategory=20, title = "Patient AM 03 Metabolic Function", font.size = 10)

egoAM3 <- enrichGO(gene = unique(DEGenesAM3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM3 <- setReadable(egoAM3, OrgDb = org.Hs.eg.db)
dotplot(egoAM3, showCategory=20, title = "Patient AM 03 Cellular Component", font.size = 10)


## Genes associated with different pathways:
# BMP signaling pathway: BMPER/PCSK6/USP9Y/GPC3/TMEM100/SMAD5/NFIA/BMP4/ENG/HTRA3/TGFBR3/GATA6/PDCD4/SMAD5-AS1/CTDSPL2/GATA4/PPM1A/SMAD6/SMAD2/TBX20/SFRP1/SMAD7/BMPR1A/BMPR2/HTRA1/TOB1/SMAD9/ACVR1/ACVR2A/NEO1/NBL1/FST/HFE

# renal axis and development: BMPER/ALDH1A2/POU3F3/FOXC1/MAGI2/COL4A4/ITGA8/COL4A3/PBX1/TBX18/PDGFD/GFRA1/GPC3/FGFR1/SMAD5/NFIA/PDGFRA/BMP4/WWTR1/CALB1/CC2D2A/ROBO2/DYNC2H1/TCF21/FGF10/BICC1/CDKN1C/FLCN/RGN/PCSK5/PPP3CA/EFNB2/WDPCP/IFT20/ACAT1/SMAD6/SMAD2/SFRP1/SALL1/SMAD7/PODXL/EDNRB/RET/OSR2/RARB/MME/C1GALT1/WNT9B/LGR4/KLHL3/CD24/IFT25/PECAM1/ACTA2/SOX9/PYGO1/FGF2/AKR1B1/TRAF3IP2/ERBB4/SULF2/GDNF/TRAF3IP1/IRX3/WNT2B/ANKS6/AQP11

# endocardial cushion development/aortic vavle morphogenesis: ADAMTS5/TMEM100/BMP4/ENG/ROBO2/GATA4/NOS3/SMAD2/TBX20/BMPR1A/BMPR2/APC/SOX9/ACVR1/SNAI2/CRELD1/ROBO1/APLNR/TWIST1

# determination left/right symmetry PCSK6/HAND2/ENG/C2CD3/MMP21/BBS5/CC2D2A/DYNC2H1/IFT172/FGF10/BICC1/MICAL2/GATA4/DYNC2LI1/MIB1/SMAD2/TBX20/ALG5/CCDC39/GAS8/ACVR1/APLNR/GJA1/TMED2/ACVR2A/IFT57/DISP1/BBS7/PSKH1/SOX18/ANKS6

# septum development ZFPM2/HEG1/MATR3/BMP4/ENG/TGFBR3/GATA6/ROBO2/SLIT3/PARVA/GATA4/SMAD6/NOS3/ANK2/TBX20/SALL1/SMAD7/ZFPM1/BMPR1A/BMPR2/EGLN1/ACVR1/CRELD1/ROBO1/APLNR

```

```{r, fig.width = 9, fig.height = 8}

# Patient AM 04
DEGenesAM4 <- DEGenes_patient %>% filter(cluster == "AM 04")


egoAM4 <- enrichGO(gene = unique(DEGenesAM4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM4 <- setReadable(egoAM4, OrgDb = org.Hs.eg.db)
dotplot(egoAM4, showCategory=20, title = "Patient AM 04 Biological Process", font.size = 10)
barplot(egoAM4, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 04 Biological Process",
        font.size = 10)

egoAM4 <- enrichGO(gene = unique(DEGenesAM4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM4 <- setReadable(egoAM4, OrgDb = org.Hs.eg.db)
dotplot(egoAM4, showCategory=20, title = "Patient AM 04 Metabolic Function", font.size = 10)

egoAM4 <- enrichGO(gene = unique(DEGenesAM4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM4 <- setReadable(egoAM4, OrgDb = org.Hs.eg.db)
dotplot(egoAM4, showCategory=20, title = "Patient AM 04 Cellular Component", font.size = 10)

# genes --> cardiac muscle

```

```{r, fig.width = 9, fig.height = 8}


# Patient AM 05
DEGenesAM5 <- DEGenes_patient %>% filter(cluster == "AM 05")


egoAM5 <- enrichGO(gene = unique(DEGenesAM5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM5 <- setReadable(egoAM5, OrgDb = org.Hs.eg.db)
dotplot(egoAM5, showCategory=20, title = "Patient AM 05 Biological Process", font.size = 10)
barplot(egoAM5, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 05",
        font.size = 10)

egoAM5 <- enrichGO(gene = unique(DEGenesAM5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM5 <- setReadable(egoAM5, OrgDb = org.Hs.eg.db)
dotplot(egoAM5, showCategory=20, title = "Patient AM 05 Metabolic Function", font.size = 10)

egoAM5 <- enrichGO(gene = unique(DEGenesAM5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM5 <- setReadable(egoAM5, OrgDb = org.Hs.eg.db)
dotplot(egoAM5, showCategory=20, title = "Patient AM 05 Cellular Component", font.size = 10)

# ECM, adhesion, WNT


```

```{r, fig.width = 9, fig.height = 9}

# Patient AM 06
DEGenesAM6 <- DEGenes_patient %>% filter(cluster == "AM 06")


egoAM6 <- enrichGO(gene = unique(DEGenesAM6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM6 <- setReadable(egoAM6, OrgDb = org.Hs.eg.db)

#View(egoAM6)

dotplot(egoAM6, showCategory=20, title = "Patient AM 06 Biological Process", font.size = 10)
barplot(egoAM6, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 06 Biological Process",
        font.size = 10)

egoAM6 <- enrichGO(gene = unique(DEGenesAM6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM6 <- setReadable(egoAM6, OrgDb = org.Hs.eg.db)
dotplot(egoAM6, showCategory=20, title = "Patient AM 06 Metabolic Function", font.size = 10)


egoAM6 <- enrichGO(gene = unique(DEGenesAM6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM6 <- setReadable(egoAM6, OrgDb = org.Hs.eg.db)
dotplot(egoAM6, showCategory=20, title = "Patient AM 06 Cellular Component", font.size = 10)

# rRNA: MRTO4/RPLP0/RPL7/RPS4X/EIF1AX/RPS17/RPS5/MDN1/RPL26/RPS27/PDCD11/RPL10A/RPS14/PRKDC/RPL10/RPS3A/RPL35/RPL23A/HEATR1/RPS24/RPS16/RPL27/RPL11/WDR75/RPS6/RPS12/DDX10/RPS28/RPS13/WDR12/RPS19/RPL38/CUL4B/GTPBP10/WDR36/RPS8/RPS23/KRI1/RPS15/MTG2/RPL7A/NHP2/RPS7/NOL9/NPM1/NOP58/RPL35A/RPS11/TSR1/CUL4A/RPS9/RPS27L/PA2G4/DIMT1/GLUL/DDX3X/DKC1/C1QBP/GNL3L/RPS21/RPL5/UTP20/EIF2A/NUP88/SBDS/FTSJ3/PWP2/FCF1/URB1/RAN/NOP56/RPSA/NOL10/GTPBP4/DDX52/WDR46/EXOSC2/BUD23/WDR18/METTL5/FASTKD2/LAS1L/MRM1/RIOX2/SNU13/TSC1/MYBBP1A/ABCE1/RRP12/WDR3/LTV1/NOP53/EMG1/MRPS2/NOL6/FRG1/RPS25/MPHOSPH6/RSL1D1/AFG2A/NAT10/MTERF3/BRIX1/SLX9/RPP25/EXOSC9/RRP36/PIH1D1/ERI3/EXOSC7

# purine and nucleotide synthesis: PAPSS2/ND4/ND1/ND5/ND2/ND3/ND6/GCDH/ATP6/ATP8/NNMT/ENO1/NDUFA13/AFMID/PFAS/ACACA/PDK4/AK5/DMAC2L/IMPDH2/PTGIS/HSD17B12/NAMPT/AK3/ADSL/ACAT1/ATIC/AK2/ND4L/PID1/MPC2/NDUFAB1/NDUFS4/TMSB4X/ATP5F1D/PRPS1/ACSS2/VCP/ATP5F1A/ALDOA/SDHA/ELOVL5/ATP5MJ/NDUFS8/NMRK2/PRPSAP1/ATP5PF/PAICS/PDK1/NDUFB8/NDUFB3/NADK/ADSS2/NOS1/ACSL1/NME7/ATP5ME/ACLY/ATP6V1A/AK1/PINK1/NDUFB7/ATP5F1C/NDUFS3/APRT/SDHB/ADCY4/ATP5MK/NMNAT3/ATP5MC1/NPR1/NDUFV1/NDUFA1

# ATP metabolic process: MYH6/ND4/ND1/ND5/ND2/ND3/ND6/ATP6/ATP8/ENO1/NDUFA13/ENO3/HSPA8/AK5/DMAC2L/ALDOC/MLST8/PRKAA1/GAPDH/AK2/GPI/HIF1A/ND4L/PID1/PGK1/NDUFAB1/NDUFS4/HDAC4/TMSB4X/ATP5F1D/VCP/ATP5F1A/ALDOA/SDHA/ATP5MJ/NDUFS8/PRKAA2/TPI1/ATP5PF/FKRP/NDUFB8/PRKAG1/CLPX/NDUFB3/NADK/APP/LDHA/OGT/ATP1B1/ATP5ME/ARL2/ATP6V1A/AK1/PINK1/MYH4/NDUFB7/ATP5F1C/MYH7/NDUFS3/SDHB/ATP5IF1/ATP5MK/ATP5MC1/IER3/NDUFV1/NDUFA1/RPTOR

```

```{r, fig.width = 9, fig.height = 9}

# Patient AM 07
DEGenesAM7 <- DEGenes_patient %>% filter(cluster == "AM 07")


egoAM7 <- enrichGO(gene = unique(DEGenesAM7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM7 <- setReadable(egoAM7, OrgDb = org.Hs.eg.db)

#View(egoAM7)
dotplot(egoAM7, showCategory=20, title = "Patient AM 07 Biological Process", font.size = 10)
barplot(egoAM7, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 07 Biological Process",
        font.size = 10)

egoAM7 <- enrichGO(gene = unique(DEGenesAM7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM7 <- setReadable(egoAM7, OrgDb = org.Hs.eg.db)
dotplot(egoAM7, showCategory=20, title = "Patient AM 07 Metabolic Function", font.size = 10)


egoAM7 <- enrichGO(gene = unique(DEGenesAM7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM7 <- setReadable(egoAM7, OrgDb = org.Hs.eg.db)
dotplot(egoAM7, showCategory=20, title = "Patient AM 07 Cellular Component", font.size = 10)

# Response to hypoxia: VEGFA/THBS1/EGLN3/EGR1/ANGPTL4/ADM/ENO1/HK2/STC2/FOSL2/PDK1/P4HB/PLOD2/LOXL2/HSP90B1/HMOX1/LONP1/ERO1A/HILPDA/SLC2A1/SMAD3/STC1/BNIP3/HIF3A/ND4/PGK1/LIF/AK4/NDRG1/PTGS2/PLOD1/HYOU1/ACVRL1/NF1/HIPK2/ND5/LMNA/ND2/PLEKHN1/MMP14/RORA/BNIP3L/FOXO3/HIF1A/MYC/COX2/ATF4/ND1/DDR2/DDAH1/TGFBR3/MDM4/PTGIS/SCFD1/TMBIM6/FOS/SRF/TBL2/ADAM17/HSPG2/VHL/COX1/SLC9A1/NOL3/PRKCE/SIRT1/ADSL/LIMD1/ALKBH5/CYP1A1/RYR2/PPARA/COMT/POSTN/GATA6/MMP2/TGFB3/SOD3/ARNT/TP53/FAM162A/RWDD3/MYOCD/PLK3/DDIT4/TGFB1/ERCC2/PMAIP1/AGTRAP/ANGPT2/NOP53/LTA/PLAU/VASN/TFAM/NR4A2

# Wnt signaling: EGR1/SMAD3/COL1A1/JADE1/NFATC4/PLEKHA4/IGFBP4/CCN4/EDNRB/NID1/LRP1/VCP/BICC1/UBR5/PLPP3/PKD1/COL6A1/FOXO3/G3BP1/UBE2B/CSNK2A1/RBMS3/ALPK2/PRKAA2/SKI/DDIT3/PPP2CA/ABL1/SNAI2/MAGI2/MDFI/DDB1/TLE2/STRN/DDX3X/SMARCA4/FOXO1/PTK7/RYK/ZNRF3/LMBR1L/ADNP/DAPK3/YAP1/FZD8/NKD2/WNK1/PYGO1/CDK14/EGF/ZRANB1/XIAP/ARHGEF19/KLF15/AMOTL1/NFKB1/PRICKLE1/LIMD1/CDH2/EXT1/SNAI1/CUL3/TGFB1I1/LATS2/SMURF2/PPM1A/PPP2R1A/CSNK1G2/ASPM/PIAS4/PTPRU/CCAR2/TTC21B/GID8/ITGA3/EMD/ROR2/APOE/BCL9L/SMURF1/RPS12/ZNF703/NPHP4/TBL1X/NLK/BMAL1/PRICKLE2/GLI3/DACT3/LGR6/TGFB1/NOTUM/NKD1/ATP6AP2/RAB5A/RNF138/LGR4/TAX1BP3/FGF9/AMFR/DVL2/PPM1B/AXIN2/TBX18/IGFBP2/LRRK1/AMOTL2/SOX9/CELSR1/FZD3/RUVBL1/FZD1/GSKIP/RUVBL2/HIC1/NR4A2

# TGFbeta:THBS1/LOX/GDF15/HSPA5/COL3A1/SMAD3/FBN1/COL1A2/ADAM9/ACVRL1/HIPK2/FLCN/SKIL/FMOD/LTBP1/SKI/SPRY1/HSP90AB1/SNX25/STAT3/TGFBR3/USP9X/ZEB1/ADAMTSL2/SPRED3/FOS/JUN/TGFBR2/SMAD2/ADAM17/OGT/FKBP1A/ZBTB7A/HDAC2/ZNF451/SMAD5/SIRT1/LEMD3/SPRY2/FBN2/TGFBR1/TGFB1I1/LATS2/SMURF2/PPM1A/VEPH1/PPARA/ITGB5/TGFB3/ZYX/ITGA3/TP53/BCL9L/SMURF1/APPL1/ZNF703/STRAP/NLK/MYOCD/PDPK1/TGFB1/SINHCAF/SDCBP/LTBP2/ITGB8/CILP/ACVR1/ING1/VASN/BRMS1

```

```{r, fig.width = 9, fig.height = 10}

# Patient AM 08
DEGenesAM8 <- DEGenes_patient %>% filter(cluster == "AM 08")


egoAM8 <- enrichGO(gene = unique(DEGenesAM8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM8 <- setReadable(egoAM8, OrgDb = org.Hs.eg.db)
dotplot(egoAM8, showCategory=20, title = "Patient AM 08 Biological Process", font.size = 10)
barplot(egoAM8, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 08 Biological Process",
        font.size = 10)

egoAM8 <- enrichGO(gene = unique(DEGenesAM8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM8 <- setReadable(egoAM8, OrgDb = org.Hs.eg.db)
dotplot(egoAM8, showCategory=20, title = "Patient AM 08 Metabolic Function", font.size = 10)


egoAM8 <- enrichGO(gene = unique(DEGenesAM8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM8 <- setReadable(egoAM8, OrgDb = org.Hs.eg.db)
dotplot(egoAM8, showCategory=20, title = "Patient AM 08 Cellular Component", font.size = 10)

```

```{r, fig.width = 9, fig.height = 10}


# Patient AM 09
DEGenesAM9 <- DEGenes_patient %>% filter(cluster == "AM 09")


egoAM9 <- enrichGO(gene = unique(DEGenesAM9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM9 <- setReadable(egoAM9, OrgDb = org.Hs.eg.db)
dotplot(egoAM9, showCategory=20, title = "Patient AM 09 Biological Process", font.size = 10)
barplot(egoAM9, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 09 Biological Process",
        font.size = 10)

egoAM9 <- enrichGO(gene = unique(DEGenesAM9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM9 <- setReadable(egoAM9, OrgDb = org.Hs.eg.db)
dotplot(egoAM9, showCategory=20, title = "Patient AM 09 Metabolic Function", font.size = 10)


egoAM9 <- enrichGO(gene = unique(DEGenesAM9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM9 <- setReadable(egoAM9, OrgDb = org.Hs.eg.db)
dotplot(egoAM9, showCategory=20, title = "Patient AM 09 Cellular Component", font.size = 10)

```

```{r, fig.width = 9, fig.height = 10}

# Patient AM 10
DEGenesAM10 <- DEGenes_patient %>% filter(cluster == "AM 10")


egoAM10 <- enrichGO(gene = unique(DEGenesAM10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM10 <- setReadable(egoAM10, OrgDb = org.Hs.eg.db)

#View(egoAM10@result)

dotplot(egoAM10, showCategory=20, title = "Patient AM 10 Biological Process", font.size = 10)
barplot(egoAM10, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 10 Biological Process",
        font.size = 10)

egoAM10 <- enrichGO(gene = unique(DEGenesAM10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM10 <- setReadable(egoAM10, OrgDb = org.Hs.eg.db)
dotplot(egoAM10, showCategory=20, title = "Patient AM 10 Metabolic Function", font.size = 10)


egoAM10 <- enrichGO(gene = unique(DEGenesAM10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM10 <- setReadable(egoAM10, OrgDb = org.Hs.eg.db)
dotplot(egoAM10, showCategory=20, title = "Patient AM 10 Cellular Component", font.size = 10)

# wound healing: SERPINE1/TIMP1/TNFRSF12A/FN1/THBS1/ITGAV/COL3A1/F2RL1/ANXA2/ANXA5/CCN1/TFPI/PPL/SMAD3/DST/AHNAK/CD59/SERPINE2/VCL/GNA12/CAV1/AXL/CSRP1/IGF1/MYH9/ITGA5/CHMP1A/DCBLD2/FLNA/ITGB5/SYT7/ACTB/PDIA3/COL5A1/FKBP10/PDGFRA/EMILIN1/F2R/HSPB1/SVEP1/TGFB1/MYOF/ANO6/EXT1/MRTFA/YAP1/PIK3CA/XBP1/ARFGEF1/NF1/STXBP1/FGFR1OP2/PAPSS2/DDR2/ACTG1/LMAN1/NFE2L2/CLASP2/SDC4/FBLN1/KANK1/DGKI/NOTCH2/PLPP3/PLEC/VAV3/CD34/NFATC1/IL6ST/DAG1/SYTL4

# integrin mediated pathway: TIMP1/LAMA2/NID1/LAMC1/CD63/LIMS1/LAMA1/FLNA/LAMB1/PRKD1

# cell adhesion: SERPINE1/FN1/THBS1/COL8A1/ITGAV/CCN2/MICALL2/COL3A1/ANXA2/ITGBL1/ABI3BP/FBLN5/COL1A1/CCN1/NID1/SMAD3/LAMC1/CALR/SPOCK1/VCL/CD63/LIMS1/AXL/SLK/ITGA5/SNED1/ITGA11/FLNA/ACTN1/ITGB5/LAMB1/P4HB/EMILIN1/CCDC80/CORO1C/VIT/SVEP1/ROCK2/BRAF/HAS2/ANGPT1/BVES/NF1/COL16A1/MYADM/POSTN/ACTG1/ADAM9/MMP14/FER/CLASP2/SDC4/TRIOBP/EPHA3/FBLN1/KANK1/MKLN1/DAPK3/MELTF/PXN/PEAK1/BCL6/ABL1/TNXB/CD34/DAG1

# ECM organisation: OL8A1/LAMA2/CYP1B1/COL1A2/COL4A1/COL3A1/COL4A2/ANXA2/ABI3BP/HMCN1/FBLN5/COL1A1/ADAMTS2/COL12A1/CCN1/ADAMTS5/NID1/COL5A2/ELN/SMAD3/LAMC1/PXDN/COL18A1/TGFBR3/CAV1/PHLDB1/LOXL1/LAMA1/ADAMTS19/LAMB1/FURIN/COL5A1/LRP1/FKBP10/MMP2/PDGFRA/EMILIN1/SOX9/CCDC80/VIT/TGFB1/EXT1/SERPINH1/HAS2/NF1/MMP16/COL16A1/COL6A1/DDR2/POSTN/MMP14/AEBP1/CLASP2/FBLN1/ADAMTS10/MELTF/QSOX1/TLL1/ABL1/SH3PXD2B/TNXB/PRDM5/DAG1

```

```{r, fig.width = 9, fig.height = 10}


# Patient AM 11
DEGenesAM11 <- DEGenes_patient %>% filter(cluster == "AM 11")


egoAM11 <- enrichGO(gene = unique(DEGenesAM11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM11 <- setReadable(egoAM11, OrgDb = org.Hs.eg.db)

#View(egoAM11@result)

dotplot(egoAM11, showCategory=20, title = "Patient AM 11 Biological Process", font.size = 10)
barplot(egoAM11, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 11 Biological Process",
        font.size = 10)

egoAM11 <- enrichGO(gene = unique(DEGenesAM11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM11 <- setReadable(egoAM11, OrgDb = org.Hs.eg.db)
dotplot(egoAM11, showCategory=20, title = "Patient AM 11 Metabolic Function", font.size = 10)


egoAM11 <- enrichGO(gene = unique(DEGenesAM11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM11 <- setReadable(egoAM11, OrgDb = org.Hs.eg.db)
dotplot(egoAM11, showCategory=20, title = "Patient AM 11 Cellular Component", font.size = 10)

# angiogenesis/vasculature development TNN/SFRP1/TSPAN12/BMPER/GAB1/TIE1/ADAM10/ANGPTL4/PLK2/TJP1/GTF2I/PTPRM/FOXC1/PRKD1/CXCR2/HSPB1/TEK/ATP2B4/ECSCR/TSPAN18/CCBE1/CYSLTR1/GREM1/CD34/SASH1/RAMP2/SPINK5/FGF18/RECK

# heart growth AKAP6/FOXC1/PDLIM5/PRKG1/HEG1/COL14A1/HEY2/ACACB/FOXP1/SORBS2/ADPRHL1/MEF2C/VGLL4

# action potential: PTPN3/KCNJ2/SLMAP/CACNB2/YWHAH/SCN11A/CACNA2D1

# lymph vessel development: BMPR2/TIE1/FOXC1/HEG1/EFNB2/CCBE1

# muscle apoptosis: GF1R/KIFAP3/BMP7/SIRT5/PDE1A/CXCR2/MYOCD/MAP3K5/HAND2/LRP6/HEY2



```

```{r, fig.width = 9, fig.height = 10}

# Patient AM 12
DEGenesAM12 <- DEGenes_patient %>% filter(cluster == "AM 12")


egoAM12 <- enrichGO(gene = unique(DEGenesAM12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM12 <- setReadable(egoAM12, OrgDb = org.Hs.eg.db)
#View(egoAM12@result)


dotplot(egoAM12, showCategory=20, title = "Patient AM 12 Biological Process", font.size = 10)
barplot(egoAM12, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 12 Biological Process",
        font.size = 10)

egoAM12 <- enrichGO(gene = unique(DEGenesAM12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM12 <- setReadable(egoAM12, OrgDb = org.Hs.eg.db)
dotplot(egoAM12, showCategory=8, title = "Patient AM 12 Metabolic Function", font.size = 10)


egoAM12 <- enrichGO(gene = unique(DEGenesAM12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM12 <- setReadable(egoAM12, OrgDb = org.Hs.eg.db)
dotplot(egoAM12, showCategory=20, title = "Patient AM 12 Cellular Component", font.size = 10)

# cardiac muscle
# sympathetic nerve development: SOX4/NTRK1/HAND2/NRP1/EDNRA/SOX11

```

```{r, fig.width = 9, fig.height = 10}


# Patient AM 13
DEGenesAM13 <- DEGenes_patient %>% filter(cluster == "AM 13")


egoAM13 <- enrichGO(gene = unique(DEGenesAM13$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM13 <- setReadable(egoAM13, OrgDb = org.Hs.eg.db)
#View(egoAM13@result)

dotplot(egoAM13, showCategory=30, title = "Patient AM 13 Biological Process", font.size = 10)
barplot(egoAM13, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 13 Biological Process",
        font.size = 10)

egoAM13 <- enrichGO(gene = unique(DEGenesAM13$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM13 <- setReadable(egoAM13, OrgDb = org.Hs.eg.db)
dotplot(egoAM13, showCategory=20, title = "Patient AM 13 Metabolic Function", font.size = 10)

# WNT; adhesion
# Response to Glucocorticosteroid: FOXO3/EGFR/KLF9/DDIT4/FIBIN/PIK3CA/CFLAR/FBXO32/CDKN1A/PDCD7/FOSL2/BCL2/SERPINF1/PTGDS/FAM107A/PAPPA/IGF1R/UCN

# response to hypoxia: HIF3A/MALAT1/FOXO3/MDM2/KCNMA1/DDIT4/TGFBR3/CFLAR/CAT/AQP3/BNIP3L/PDK3/HIPK2/VEGFD/SMAD3/NFE2L2/ANGPTL4/RGCC/FOSL2/BCL2/NDRG1/SMAD4/SIRT1/BNIP3/CRYAB/ABAT/ANG/SOD3/BMP2

# Insulin R signaling: PDK4/TNS2/PID1/RHOQ/IRS2/PIK3CA/PIK3R1/PDK2/SORBS1/FOXO1/PIP4K2A/KANK1/SIRT1/GKAP1/NCL/IGF1R/GRB10/BAIAP2/FOXC2/SOS2/STXBP4

# renal system: ZBTB16/NID1/TNS2/NPNT/ARID5B/EDNRB/ACTA2/COL4A1/CFLAR/CAT/IL6R/KLF15/PKD2/ITGA8/FMN1/SMAD3/BCL2/BMPER/SERPINF1/ODC1/WNT9B/IRX3/MTSS1/SMAD4/FGF10/GPC3/PTCH1/AGTR1/ERBB4/GFRA1/GDF11/JMJD6/FOXC2/PDGFA/MPST/CALB1/BMP2/PPP3CA

# WNT: ADE1/FOXO3/SHISA3/NID1/PTEN/EDNRB/EGFR/CALCOCO1/NKD1/IGFBP6/RBMS3/TRABD2B/NDRG2/KLF15/HBP1/PPM1B/PKD2/SIAH2/DAAM2/FOXO1/TLE1/SMAD3/EMD/OTULIN/LGR6/WNT9B/UBE2B/SMAD4/FGF10/APOE/KANK1/FRAT1/GPC3/BMAL1/FBXW4/TCF7/WNK1/SKIC8/LDB1/BAMBI/STK4/DAAM1/GRB10/DKK3/DIXDC1/BRD7/BMP2


```

```{r, fig.width = 9, fig.height = 11}

# Patient AM 14
DEGenesAM14 <- DEGenes_patient %>% filter(cluster == "AM 14")


egoAM14 <- enrichGO(gene = unique(DEGenesAM14$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM14 <- setReadable(egoAM14, OrgDb = org.Hs.eg.db)
dotplot(egoAM14, showCategory=30, title = "Patient AM 14 Biological Process", font.size = 10)
barplot(egoAM14, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 14 Biological Process",
        font.size = 10)

egoAM14 <- enrichGO(gene = unique(DEGenesAM14$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM14 <- setReadable(egoAM14, OrgDb = org.Hs.eg.db)
dotplot(egoAM14, showCategory=8, title = "Patient AM 14 Metabolic Function", font.size = 10)


egoAM14 <- enrichGO(gene = unique(DEGenesAM14$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM14 <- setReadable(egoAM14, OrgDb = org.Hs.eg.db)
dotplot(egoAM14, showCategory=20, title = "Patient AM 14 Cellular Component", font.size = 10)

```

```{r, fig.width = 9, fig.height = 10}


# Patient AM 15
DEGenesAM15 <- DEGenes_patient %>% filter(cluster == "AM 15")


egoAM15 <- enrichGO(gene = unique(DEGenesAM15$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM15 <- setReadable(egoAM15, OrgDb = org.Hs.eg.db)
dotplot(egoAM15, showCategory=20, title = "Patient AM 15 Biological Process", font.size = 10)
barplot(egoAM15, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 15 Biological Process",
        font.size = 10)

egoAM15 <- enrichGO(gene = unique(DEGenesAM15$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM15 <- setReadable(egoAM15, OrgDb = org.Hs.eg.db)
dotplot(egoAM15, showCategory=8, title = "Patient AM 15 Metabolic Function", font.size = 10)


egoAM15 <- enrichGO(gene = unique(DEGenesAM15$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM15 <- setReadable(egoAM15, OrgDb = org.Hs.eg.db)
dotplot(egoAM15, showCategory=20, title = "Patient AM 15 Cellular Component", font.size = 10)

```

```{r, fig.width = 9, fig.height = 10}

# Patient AM 16
DEGenesAM16 <- DEGenes_patient %>% filter(cluster == "AM 16")


egoAM16 <- enrichGO(gene = unique(DEGenesAM16$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM16 <- setReadable(egoAM16, OrgDb = org.Hs.eg.db)

#View(egoAM16@result)
dotplot(egoAM16, showCategory=20, title = "Patient AM 16 Biological Process", font.size = 10)
barplot(egoAM16, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 16 Biological Process",
        font.size = 10)

egoAM16 <- enrichGO(gene = unique(DEGenesAM16$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM16 <- setReadable(egoAM16, OrgDb = org.Hs.eg.db)
dotplot(egoAM16, showCategory=8, title = "Patient AM 16 Metabolic Function", font.size = 10)


egoAM16 <- enrichGO(gene = unique(DEGenesAM16$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM16 <- setReadable(egoAM16, OrgDb = org.Hs.eg.db)
dotplot(egoAM16, showCategory=20, title = "Patient AM 16 Cellular Component", font.size = 10)

# T lymphocyte differentiation: CD74/IRF1/HLA-DRB1/HLA-DRA/B2M/PTPRC/ACTB/FGL2/LILRB2/NCKAP1L/TCIRG1/JAK3/RUNX1/DOCK2/SOCS1/LGALS9/BTN2A2/IL15/FCER1G/CD2/RSAD2/LILRB4/SYK/SPI1/RUNX3/IL4I1/LAG3/CD4/PTPN22/IL2RG/HLA-DOA/ASCL2/CD86/CLEC4E/CCR2/CD80/PTPN2/CD8A/RHOH/IL12RB1/IKZF3/BATF/PREX1/BCL11B/MR1/IL18/CD3D/SASH3/TBX21/ITK/VAV1/SPN/FANCA/TGFB1/SEMA4A/CYLD/IL27/IRF4/PIK3CD/THEMIS/CD3G/CD1D/CDK6/LCK/EOMES/MAFB/CD3E/RHOA/PRELID1/ZC3H12A/IL4R/RELB/PRDM1/RIPK3/IL1B/NFKBID/ADAM8/CD83/KLRC1/RIPK2/ZAP70/SOCS3/IFNG/MYB/IL7R/PTGER4/STAT6/RASGRP1/PNP/TMEM131L/CTLA4/CLEC4D/ANXA1/LIPA/RC3H2/SLAMF6/SOS1/JUNB/BCL3/GATA3/CAMK4/STAT4/BRD2/FOSL2/SP3/MALT1/ARID1A/IL6R/CTNNB1/CD27/JAK1/VSIR/PBRM1/FANCD2/RUNX2/ZBTB7B/PRKDC/ZBTB1/CTSL/TGFBR2/PIK3R6/NFKBIZ/RPL22/CBFB/TNFRSF9/DUSP10/FOXP3/ADAM17/LOXL3/BCL2/IL18R1/CRTAM/BRD7/AP3B1/NFATC3/LIG4/NLRP3/XBP1/RC3H1/TP53/KMT2A/ZFP36L2/FZD5/MDK/PKNOX1/CARD11/SMAD7/TESPA1/METTL3/SLC4A2/CCL19/ATG5/DROSHA/GPR89B/GPR89B/EGR3/EGR1/PATZ1/SRF/MEN1/LY9/TOX/LEF1/FADD/GPR183/TNFSF8/CD28/IL7/CCR7/PIK3R3/ACTL6A/ZNF683/IL2RA/PDP2/GPR18/WNT10B

# TCR activation: RF1/HLA-A/HLA-DRB1/HLA-DRA/PTPRC/HLA-E/JAK2/NCKAP1L/TCIRG1/JAK3/LILRB1/RUNX1/DOCK2/SOCS1/CD274/LGALS9/IL15/RSAD2/LILRB4/SYK/RUNX3/IL2RG/CD300A/NKG7/ASCL2/CD86/CCR2/CD80/EBI3/IL12RB1/BATF/TRBC2/BCL11B/IL18/CD3D/SASH3/TBX21/HSPH1/ITK/SPN/SEMA4A/RASAL3/WDFY4/IL27/IRF4/TRAC/CD3G/CD247/EOMES/CD3E/ELF4/RHOA/ZC3H12A/IL4R/TNFRSF14/RELB/PRDM1/NFKBID/CD83/RIPK2/ZAP70/SOCS3/IFNG/MYB/PTGER4/STAT6/PNP/ANXA1/RC3H2/SLAMF6/JUNB/BCL3/GATA3/STAT4/BRD2/CD81/FOSL2/MALT1/IL6R/JAK1/VSIR/ZBTB7B/CTSL/TGFBR2/NFKBIZ/PRKCQ/RPL22/CBFB/FOXP3/LOXL3/BCL2/IL18R1/CRTAM/TYK2/AP3B1/NLRP3/RC3H1/KMT2A/NDFIP1/AGER/CARD11/SMAD7/CBLB/SLC4A2/CCL19/CLEC4A/TRBC1/IL12A/MEN1/LY9/TOX/LEF1/GPR183/TNFSF8/CD55/CD28/ZNF683/IL2RA/CD160/PDP2/GPR18

# B lymphocyte activation: CD74/PTPRC/CD38/TNFAIP3/NCKAP1L/TCIRG1/JAK3/DOCK10/TYROBP/LAPTM5/TLR4/TNFSF13B/ITGA4/SLAMF8/PTK2B/SAMSN1/SKAP2/LYN/CASP8/IRF8/PTPRJ/PRKCB/LAT2/BTK/PTPN6/SYK/SPI1/MNDA/IL4I1/IL2RG/CD300A/CD86/NFAM1/PTPN2/EZH2/IKZF3/BATF/CD40/BST2/CDKN1A/BLNK/THEMIS2/SASH3/TBX21/PRKCD/INPP5D/NOD2/TGFB1/MZB1/TBC1D10C/NBN/CYLD/VCAM1/PIK3CD/KLF6/BAX/DOCK11/PLCG2/IL27RA/ZAP70/IL7R/NOTCH2/STAT6/RASGRP1/CTLA4/DCLRE1C/AKIRIN2/ATM/EXO1/BCL3/TNIP2/RBPJ/LAX1/CD81/TNFRSF4/TXLNA/FOSL2/HSPD1/SP3/HHEX/CMTM7/AHR/NSD2/CASP3/MALT1/FLT3/TFRC/CD27/CD70/PPP2R3C/ATAD5/LGALS1/PRKDC/ZBTB1/NFKBIZ/LYL1/SLC15A4/ID2/NFATC2/ZBTB7A/BAK1/FOXP3/ADAM17/BCL2/EXOSC6/PKN1/LIG4/VAV3/IL10/XBP1/RC3H1/PLCL2/SLC25A5/TP53/TICAM1/MIF/ZFP36L2/SHLD2/NDFIP1/BST1/CARD11/TOP2B/FCGR2B/BANK1/SYVN1/MIR17HG/HDAC4/MLH1/PHB2/TNFSF13/CD180/MSH2/HMGB3/DCAF1/MFNG/TCF3/IGBP1/SHLD1/CTPS1/EPHB2/FCRL3/POU2AF1/SH3KBP1/TPD52/PARP3/LEF1/PAXIP1/GPR183/PHB1/CD28/IL7/DNAJB9/SHLD3/GAPT/CD79B/PIK3R3/TNFRSF13C/MAD2L2/TIRAP/CD40LG/MS4A1

# Innate: GBP5/IRF1/GBP2/CTSS/IFI16/BIRC3/TNFAIP3/CYBA/DDX60/HSPA8/ALPK1/TYROBP/IFIH1/OAS1/OAS3/TLR4/CLEC7A/CASP1/TLR2/IRF7/CD274/CGAS/LYN/IFI35/RSAD2/BTK/MYD88/HAVCR2/UNC93B1/NMI/SYK/SCIMP/MNDA/S100A9/TLR1/PTPN22/PIK3AP1/CD300A/S100A8/AIM2/CD300LF/TLR8/CLEC4E/SLC15A3/HCK/PAK1/TRAF3/TIFA/TNF/IRAK1/ZBP1/TASL/TRAF3IP3/USP15/ACOD1/TLR6/MEFV/KLRD1/FFAR2/PYCARD/WDFY1/NOD2/PYHIN1/LACC1/CYLD/IRF4/ZNFX1/ARRB2/HSP90AA1/DDX3X/NAGK/RIGI/GFI1/CARD8/TRIM25/DHX58/OASL/NFKBIA/AURKB/PLCG2/NINJ1/IRAK4/OTUD4/KLRC1/RIPK2/EIF2AK2/LY96/FCN1/TLR7/TBK1/ERBIN/RTN4/SRC/IRAK3/NLRC4/OTULIN/TNIP3/NR1H3/TNIP2/BIRC2/NLRC3/IKBKB/RNF135/ZDHHC5/HSPD1/NAIP/XIAP/TRIM5/SLC15A2/UFD1/PPP2CA/EPG5/BECN1/PAK2/RPS6KA3/PRKDC/NFKBIZ/PUM2/ZCCHC3/PDPK1/RELA/SLC15A4/UBQLN1/TRIM11/NOD1/PARP1/BCL10/ZDHHC12/MAP3K7/CD14/RNF34/TNIP1/TREM2/TICAM2/AP3B1/NLRP3/ZDHHC18/PSPC1/AKT1/LILRA2/TICAM1/KLRK1/TLR10/MAPKAPK2/TRAF6/SIN3A/LSM14A/KLRC3/TLR3/BANF1/IRAK2/FYN/SARM1/PHB2/NOP53/RAB11FIP2/HSPA1A/MFHAS1/RNF125/CHUK/FCRL3/C1QBP/TOMM70/HSPA1B/MAPKAPK3/PHB1/KLRC4/PPP6C/BRCC3/FCN3/LRCH4/TKFC/LRRC14/KLRC2/TIRAP

# NFKAPPB: STAT1/CD74/HLA-DRB1/TMSB4X/TRIM22/TNFSF10/APOL3/BIRC3/TNFAIP3/SLC20A1/ALPK1/TLR4/CLEC7A/CASP1/TLR2/CASP8/LGALS9/PRDX1/PRKCB/LILRB4/MYD88/IKBKE/CD4/SLC39A8/SECTM1/CARD16/TLR8/UBD/RHOH/CD40/TRAF3/BST2/TIFA/TNF/IRAK1/RASSF2/TMEM106A/TLR6/TANK/PYCARD/LITAF/TRIM38/NOD2/TGFB1/RBCK1/ARRB2/SIRPA/CARD9/CARD8/TRIM25/RHOA/NFKBIA/ZC3H12A/CASP10/RELB/NUP62/IL1B/NFKB1/PLCG2/ADAM8/HMOX1/IRAK4/RIPK2/NFKB2/TLR7/TBK1/TRIM21/FASLG/FKBP1A/ECT2/GAPDH/IFIT5/TRAF2/CLEC4D/S100A4/TNIP3/BCL3/TNIP2/BIRC2/REL/TAB3/TFG/NLRC3/IKBKB/LTB/XIAP/TNFRSF1A/CX3CL1/TRIM5/LIMS1/MALT1/TRAF1/MTDH/TFRC/NEK6/RIPK1/CUL1/ATP2C1/USP10/LGALS1/TRIM13/TRADD/PDPK1/RELA/OPTN/PPM1B/NFAT5/TNFRSF10B/TAB2/NOD1/NKIRAS2/PARP1/BCL10/DHX15/MIER1/MAP3K7/ZNF675/TNIP1/PIM2/HDAC1/DDX21/TREM2/TRIM59/TICAM2/SHISA5/GOLT1B/ZDHHC13/F2R/AKT1/TICAM1/IKBKG/TRAF5/NDFIP1/CANT1/TRAF6/RBX1/CARD11/ZFAND6/SLC35B2/TLR3/CCL19/IRAK2/FLOT2/GPR89B/UBE2N/GPR89B/NFKBIB/USP20/CHUK/LAMTOR5/CX3CR1/CARD19/UBE2V1/FBXW11/EDN1/TMEM9B/FADD/SUMO4/CCR7/JMJD8/NDFIP2/TIRAP/TMEM101

```

```{r, fig.width = 9, fig.height = 10}


# Patient AM 17
DEGenesAM17 <- DEGenes_patient %>% filter(cluster == "AM 17")


egoAM17 <- enrichGO(gene = unique(DEGenesAM17$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM17 <- setReadable(egoAM17, OrgDb = org.Hs.eg.db)
dotplot(egoAM17, showCategory= 30, title = "Patient AM 17 Biological Process", font.size = 10)
barplot(egoAM17, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 17 Biological Process",
        font.size = 10)

egoAM17 <- enrichGO(gene = unique(DEGenesAM17$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM17 <- setReadable(egoAM17, OrgDb = org.Hs.eg.db)
dotplot(egoAM17, showCategory=20, title = "Patient AM 17 Metabolic Function", font.size = 10)


egoAM17 <- enrichGO(gene = unique(DEGenesAM17$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM17 <- setReadable(egoAM17, OrgDb = org.Hs.eg.db)
#View(egoAM17@result)
dotplot(egoAM17, showCategory=20, title = "Patient AM 17 Cellular Component", font.size = 10)

# IGHG3/IGKC/IGHG1
# IGHM/IGKC
# presynaptic: GPM6A/LPAR2/KCNJ3/OPRD1
# voltage gated channels: KCNJ3/CNTNAP2/KCNG1/SCNN1A


```

```{r, fig.width = 9, fig.height = 11}

# Patient AM 18
DEGenesAM18 <- DEGenes_patient %>% filter(cluster == "AM 18")


egoAM18 <- enrichGO(gene = unique(DEGenesAM18$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM18 <- setReadable(egoAM18, OrgDb = org.Hs.eg.db)
dotplot(egoAM18, showCategory=30, title = "Patient AM 18 Biological Process", font.size = 10)
barplot(egoAM18, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient AM 18 Biological Process",
        font.size = 10)

egoAM18 <- enrichGO(gene = unique(DEGenesAM18$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM18 <- setReadable(egoAM18, OrgDb = org.Hs.eg.db)
dotplot(egoAM18, showCategory=20, title = "Patient AM 18 Metabolic Function", font.size = 10)


egoAM18 <- enrichGO(gene = unique(DEGenesAM18$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoAM18 <- setReadable(egoAM18, OrgDb = org.Hs.eg.db)
dotplot(egoAM18, showCategory=20, title = "Patient AM 18 Cellular Component", font.size = 10)

```

```{r, CMP 01}

# Patient CMP 01
DEGenesCMP1 <- DEGenes_patient %>% filter(cluster == "CMP 01")


egoCMP1 <- enrichGO(gene = unique(DEGenesCMP1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP1 <- setReadable(egoCMP1, OrgDb = org.Hs.eg.db)
dotplot(egoCMP1, showCategory=30, title = "Patient CMP 01 Biological Process", font.size = 10)


egoCMP1 <- enrichGO(gene = unique(DEGenesCMP1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP1 <- setReadable(egoCMP1, OrgDb = org.Hs.eg.db)
dotplot(egoCMP1, showCategory=20, title = "Patient CMP 01 Metabolic Function", font.size = 10)


egoCMP1 <- enrichGO(gene = unique(DEGenesCMP1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP1 <- setReadable(egoCMP1, OrgDb = org.Hs.eg.db)
dotplot(egoCMP1, showCategory=20, title = "Patient CMP 01 Cellular Component", font.size = 10)

```

```{r, fig.width = 12, fig.height = 15}

# Patient CMP 02
DEGenesCMP2 <- DEGenes_patient %>% filter(cluster == "CMP 02")


egoCMP2 <- enrichGO(gene = unique(DEGenesCMP2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP2 <- setReadable(egoCMP2, OrgDb = org.Hs.eg.db)
dotplot(egoCMP2, showCategory=30, title = "Patient CMP 02 Biological Process", font.size = 10)
barplot(egoCMP2, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 02 Biological Process",
        font.size = 10)

egoCMP2 <- enrichGO(gene = unique(DEGenesCMP2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP2 <- setReadable(egoCMP2, OrgDb = org.Hs.eg.db)
dotplot(egoCMP2, showCategory=20, title = "Patient CMP 02 Metabolic Function", font.size = 10)


egoCMP2 <- enrichGO(gene = unique(DEGenesCMP2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP2 <- setReadable(egoCMP2, OrgDb = org.Hs.eg.db)
dotplot(egoCMP2, showCategory=20, title = "Patient CMP 02 Cellular Component", font.size = 10)

# transmembrane receptor signaling, TGFbeta and BMP pathway

```

```{r, fig.width = 8, fig.height = 10}


# Patient CMP 03
DEGenesCMP3 <- DEGenes_patient %>% filter(cluster == "CMP 03")


egoCMP3 <- enrichGO(gene = unique(DEGenesCMP3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP3 <- setReadable(egoCMP3, OrgDb = org.Hs.eg.db)
#View(egoCMP3@result)

dotplot(egoCMP3, showCategory=30, title = "Patient CMP 03 Biological Process", font.size = 10)
barplot(egoCMP3, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 03 Biological Process",
        font.size = 10)

egoCMP3 <- enrichGO(gene = unique(DEGenesCMP3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP3 <- setReadable(egoCMP3, OrgDb = org.Hs.eg.db)
dotplot(egoCMP3, showCategory=20, title = "Patient CMP 03 Metabolic Function", font.size = 10)


egoCMP3 <- enrichGO(gene = unique(DEGenesCMP3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP3 <- setReadable(egoCMP3, OrgDb = org.Hs.eg.db)
dotplot(egoCMP3, showCategory=20, title = "Patient CMP 03 Cellular Component", font.size = 10)

# response to peptide hormone: GCGR/TGFBR3/IRS2/CYTB/TNS2/CDO1/ADM/DDR2/INSR/FBN1/SORL1/COL3A1/LPIN1/TRIM72/STK11/AGTRAP/TBC1D4/PID1/OGT/NCOA1/EIF4EBP2/SESN3/AGTR1/EDNRB/HMGCS2/ARID1B/CFLAR/IRS1/PXN/IGF1R/COL6A1/MAPK14/ZBTB7B/MAPK3/STAT6/CTSD/FOXO1/C2CD5/SLC39A14/PDK4/PIP4K2A/CTSK/COL1A1/SLC27A1/STAT5B/PDK2/SCAP/PHIP/HDAC5/BCAR1/PIK3CA/PIK3R2/GHR/MAPK1/SOS2/CAMK2A/EREG/RANGAP1/CSK/JAK3/KLF15/TLR2/RHOQ/TSC1/LPIN3/DENND4C

# response to ROS: POD/HBA2/HBA1/HBB/FBLN5/MMP9/ND5/LRRK2/DDR2/MAP1LC3A/FOXO3/MMP2/BNIP3/PDGFRA/BCL2/SESN3/ANKZF1/PTPRK/PXN/TOP2B/COL6A1/KDM6B/MAPK3/STAT6/TXNRD2/ZNF580/FOXP1/COL1A1/TXNIP/PDK2/IL18RAP/EGFR/FXN/MAPK1/CCS/MAP3K5/ZNF277/MAP2K4/SETX/OSER1

# insulin pathway: IRS2/TNS2/ADM/INSR/SORL1/LPIN1/TRIM72/TBC1D4/PID1/OGT/NCOA1/EIF4EBP2/SESN3/HMGCS2/CFLAR/IRS1/IGF1R/COL6A1/MAPK14/ZBTB7B/MAPK3/CTSD/FOXO1/C2CD5/SLC39A14/PDK4/PIP4K2A/CTSK/COL1A1/SLC27A1/PDK2/SCAP/PHIP/HDAC5/BCAR1/PIK3CA/PIK3R2/GHR/MAPK1/SOS2/KLF15/TLR2/RHOQ/TSC1/LPIN3/DENND4C

# glucose homeostasis:  GCGR/FFAR2/SERPINF1/IRS2/NOX4/INSR/FOXO3/ZBTB20/STK11/OGT/PIM3/MLXIPL/FOSL2/SESN3/PLA2G6/NEUROD1/IRS1/NADK/IGF1R/CRTC2/FOXO1/LRRC8A/PFKM/SLC39A14/PDK4/COL1A1/PDK2/SLC12A6/RAB11B/ASPSCR1/PIK3CA/PIK3R2/DHPS/ALOX5/PIH1D1/KLF15

# Fatty acid: ACSM3/CYP4F3/CES1/AKR1C2/ALDH3A2/ECHDC2/IRS2/ACSL1/GGT5/C3/CYP4B1/NR1H3/GPX4/ACADL/ABHD5/LPIN1/ALOX5AP/AKR1C1/PTGR3/MLXIPL/BDH2/PLIN5/ACSS1/ABCD2/ACADVL/ACADSB/ADH5/IRS1/CPT1B/GCDH/PTGIS/PLA2G5/MAPK14/PTGDS/CYP4Z1/ACACB/PDK4/ABCB11/SLC27A1/PDK2/SCAP/FADS3/PLA2G15/ETFDH/ABHD2/SOX9/ABCD1/FMO4/ALOX5/AACS/GPAT4/PTGR1/PRKAG2/ABCD3/LPIN3

# organic acid metabolism: CYP4F3/ALDH1L1/ECHDC2/IRS2/AASS/CDO1/HAL/MPST/ACADL/GLUL/LPIN1/BDH2/PLIN5/ABCD2/ALDH6A1/ACADVL/ACADSB/BCKDHB/IRS1/BCAT2/ENOSF1/CPT1B/GCDH/PGD/DCXR/ACACB/ABCB11/ARG1/CBS/DDO/PLA2G15/ETFDH/ABHD2/GPT2/ABCD1/DBT/ABCD3/LPIN3/PPM1K

```



```{r, fig.width = 12, fig.height = 14}

# Patient CMP 04
DEGenesCMP4 <- DEGenes_patient %>% filter(cluster == "CMP 04")


egoCMP4 <- enrichGO(gene = unique(DEGenesCMP4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP4 <- setReadable(egoCMP4, OrgDb = org.Hs.eg.db)
dotplot(egoCMP4, showCategory=30, title = "Patient CMP 04 Biological Process", font.size = 10)
barplot(egoCMP4, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 04 Biological Process",
        font.size = 10)

egoCMP4 <- enrichGO(gene = unique(DEGenesCMP4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP4 <- setReadable(egoCMP4, OrgDb = org.Hs.eg.db)
dotplot(egoCMP4, showCategory=20, title = "Patient CMP 04 Metabolic Function", font.size = 10)


egoCMP4 <- enrichGO(gene = unique(DEGenesCMP4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP4 <- setReadable(egoCMP4, OrgDb = org.Hs.eg.db)
dotplot(egoCMP4, showCategory=20, title = "Patient CMP 04 Cellular Component", font.size = 10)

# angiogenesis, lymph vessel development
```

```{r, fig.width = 12, fig.height = 14}


# Patient CMP 05
DEGenesCMP5 <- DEGenes_patient %>% filter(cluster == "CMP 05")


egoCMP5 <- enrichGO(gene = unique(DEGenesCMP5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP5 <- setReadable(egoCMP5, OrgDb = org.Hs.eg.db)
dotplot(egoCMP5, showCategory=30, title = "Patient CMP 05 Biological Process", font.size = 10)
barplot(egoCMP5, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 05 Biological Process",
        font.size = 10)

egoCMP5 <- enrichGO(gene = unique(DEGenesCMP5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP5 <- setReadable(egoCMP5, OrgDb = org.Hs.eg.db)
dotplot(egoCMP5, showCategory=20, title = "Patient CMP 05 Metabolic Function", font.size = 10)


egoCMP5 <- enrichGO(gene = unique(DEGenesCMP5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP5 <- setReadable(egoCMP5, OrgDb = org.Hs.eg.db)
dotplot(egoCMP5, showCategory=20, title = "Patient CMP 05 Cellular Component", font.size = 10)

```

```{r, fig.width = 12, fig.height = 14}
# Patient CMP 06
DEGenesCMP6 <- DEGenes_patient %>% filter(cluster == "CMP 06")


egoCMP6 <- enrichGO(gene = unique(DEGenesCMP6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP6 <- setReadable(egoCMP6, OrgDb = org.Hs.eg.db)
dotplot(egoCMP6, showCategory=30, title = "Patient CMP 06 Biological Process", font.size = 10)
barplot(egoCMP6, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 06 Biological Process",
        font.size = 10)

egoCMP6 <- enrichGO(gene = unique(DEGenesCMP6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP6 <- setReadable(egoCMP6, OrgDb = org.Hs.eg.db)
dotplot(egoCMP6, showCategory=20, title = "Patient CMP 06 Metabolic Function", font.size = 10)

#response to peptide, RNA splicing, insulin, circadian rhythm, 


```

```{r}

egoCMP6 <- enrichGO(gene = unique(DEGenesCMP6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP6 <- setReadable(egoCMP6, OrgDb = org.Hs.eg.db)
dotplot(egoCMP6, showCategory=20, title = "Patient CMP 06 Cellular Component", font.size = 10)

```

```{r, fig.width = 12, fig.height = 14}

# Patient CMP 07
DEGenesCMP7 <- DEGenes_patient %>% filter(cluster == "CMP 07")


egoCMP7 <- enrichGO(gene = unique(DEGenesCMP7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP7 <- setReadable(egoCMP7, OrgDb = org.Hs.eg.db)
dotplot(egoCMP7, showCategory=30, title = "Patient CMP 07 Biological Process", font.size = 10)
barplot(egoCMP7, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 07 Biological Process",
        font.size = 10)

egoCMP7 <- enrichGO(gene = unique(DEGenesCMP7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP7 <- setReadable(egoCMP7, OrgDb = org.Hs.eg.db)
dotplot(egoCMP7, showCategory=20, title = "Patient CMP 07 Metabolic Function", font.size = 10)


egoCMP7 <- enrichGO(gene = unique(DEGenesCMP7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP7 <- setReadable(egoCMP7, OrgDb = org.Hs.eg.db)
dotplot(egoCMP7, showCategory=20, title = "Patient CMP 07 Cellular Component", font.size = 10)

```

```{r, fig.width = 12, fig.height = 14}

# Patient CMP 08
DEGenesCMP8 <- DEGenes_patient %>% filter(cluster == "CMP 08")


egoCMP8 <- enrichGO(gene = unique(DEGenesCMP8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP8 <- setReadable(egoCMP8, OrgDb = org.Hs.eg.db)
dotplot(egoCMP8, showCategory=30, title = "Patient CMP 08 Biological Process", font.size = 10)
barplot(egoCMP8, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 08 Biological Process",
        font.size = 10)

egoCMP8 <- enrichGO(gene = unique(DEGenesCMP8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP8 <- setReadable(egoCMP8, OrgDb = org.Hs.eg.db)
dotplot(egoCMP8, showCategory=20, title = "Patient CMP 08 Metabolic Function", font.size = 10)


egoCMP8 <- enrichGO(gene = unique(DEGenesCMP8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP8 <- setReadable(egoCMP8, OrgDb = org.Hs.eg.db)
dotplot(egoCMP8, showCategory=20, title = "Patient CMP 08 Cellular Component", font.size = 10)

```

```{r, fig.width = 12, fig.height = 14}

# Patient CMP 09
DEGenesCMP9 <- DEGenes_patient %>% filter(cluster == "CMP 09")


egoCMP9 <- enrichGO(gene = unique(DEGenesCMP9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP9 <- setReadable(egoCMP9, OrgDb = org.Hs.eg.db)
dotplot(egoCMP9, showCategory=30, title = "Patient CMP 09 Biological Process", font.size = 10)
barplot(egoCMP9, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 09 Biological Process",
        font.size = 10)

egoCMP9 <- enrichGO(gene = unique(DEGenesCMP9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP9 <- setReadable(egoCMP9, OrgDb = org.Hs.eg.db)
dotplot(egoCMP9, showCategory=20, title = "Patient CMP 09 Metabolic Function", font.size = 10)

```

```{r, Patient CMP 09}

egoCMP9 <- enrichGO(gene = unique(DEGenesCMP9$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP9 <- setReadable(egoCMP9, OrgDb = org.Hs.eg.db)
dotplot(egoCMP9, showCategory=20, title = "Patient CMP 09 Cellular Component", font.size = 10)

#View(egoCMP9@result)

# platelet alpha granule: CFD/SPARC/IGF1/CD9/MMRN1/CLU/APP/APLP2/MAGED2/ISLR/PCDH7/THBS2/SYTL4/PROS1/SELP/HGF/PHACTR2/VEGFC/CD109

# postsynaptic membrane: LRRC4C/GRID2/PRRT2/SSPN/ITGA8/FAIM2/NECTIN3/SLC6A1/CNKSR2/ADGRB3/EFNB2/CHRM2/DLG2/FXYD6/NRP2/KCNC4/IQSEC3/TIAM1/LRFN5/ATP6AP2/GABRE/KCTD12/GRM3/RNF10/SLC6A4/PCDH17/IL1RAPL1/ERBB4/P2RY1/ANK2/TRAPPC4/NLGN4X/HIP1/LRRC7/LRRTM3/CHRNB1/TRPV1/ABHD17B/COL13A1/PTPRT/SHISA6

```

```{r}

# Patient CMP 10
DEGenesCMP10 <- DEGenes_patient %>% filter(cluster == "CMP 10")


egoCMP10 <- enrichGO(gene = unique(DEGenesCMP10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP10 <- setReadable(egoCMP10, OrgDb = org.Hs.eg.db)
dotplot(egoCMP10, showCategory=30, title = "Patient CMP 10 Biological Process", font.size = 10)
barplot(egoCMP10, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 10 Biological Process",
        font.size = 10)

# BMP signaling pathway!

```





```{r}

egoCMP10 <- enrichGO(gene = unique(DEGenesCMP10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP10 <- setReadable(egoCMP10, OrgDb = org.Hs.eg.db)
dotplot(egoCMP10, showCategory=20, title = "Patient CMP 10 Metabolic Function", font.size = 10)


egoCMP10 <- enrichGO(gene = unique(DEGenesCMP10$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP10 <- setReadable(egoCMP10, OrgDb = org.Hs.eg.db)
dotplot(egoCMP10, showCategory=20, title = "Patient CMP 10 Cellular Component", font.size = 10)

```

```{r, fig.width = 12, fig.height = 14}

# Patient CMP 11
DEGenesCMP11 <- DEGenes_patient %>% filter(cluster == "CMP 11")


egoCMP11 <- enrichGO(gene = unique(DEGenesCMP11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP11 <- setReadable(egoCMP11, OrgDb = org.Hs.eg.db)
dotplot(egoCMP11, showCategory=30, title = "Patient CMP 11 Biological Process", font.size = 10)
barplot(egoCMP11, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 11 Biological Process",
        font.size = 10)

egoCMP11 <- enrichGO(gene = unique(DEGenesCMP11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP11 <- setReadable(egoCMP11, OrgDb = org.Hs.eg.db)
dotplot(egoCMP11, showCategory=20, title = "Patient CMP 11 Metabolic Function", font.size = 10)


egoCMP11 <- enrichGO(gene = unique(DEGenesCMP11$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP11 <- setReadable(egoCMP11, OrgDb = org.Hs.eg.db)
dotplot(egoCMP11, showCategory=20, title = "Patient CMP 11 Cellular Component", font.size = 10)

```

```{r, fig.width = 12, fig.height = 14}

# Patient CMP 12
DEGenesCMP12 <- DEGenes_patient %>% filter(cluster == "CMP 12")


egoCMP12 <- enrichGO(gene = unique(DEGenesCMP12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP12 <- setReadable(egoCMP12, OrgDb = org.Hs.eg.db)
dotplot(egoCMP12, showCategory=30, title = "Patient CMP 12 Biological Process", font.size = 10)
barplot(egoCMP12, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Patient CMP 12 Biological Process",
        font.size = 10)

#View(egoCMP12@result)

egoCMP12 <- enrichGO(gene = unique(DEGenesCMP12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP12 <- setReadable(egoCMP12, OrgDb = org.Hs.eg.db)
dotplot(egoCMP12, showCategory=20, title = "Patient CMP 12 Metabolic Function", font.size = 10)


egoCMP12 <- enrichGO(gene = unique(DEGenesCMP12$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoCMP12 <- setReadable(egoCMP12, OrgDb = org.Hs.eg.db)
dotplot(egoCMP12, showCategory=20, title = "Patient CMP 12 Cellular Component", font.size = 10)

```

```{r CMP 13}

# Patient CMP 13
# p adjusted not significant

```

```{r, fig.width = 12, fig.height = 12}

# Patient Sarcoidosis 01
DEGenesSarco1 <- DEGenes_patient %>% filter(cluster == "Sarcoidosis 01")


egoSarco1 <- enrichGO(gene = unique(DEGenesSarco1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco1 <- setReadable(egoSarco1, OrgDb = org.Hs.eg.db)
dotplot(egoSarco1, showCategory=20, title = "Patient Sarcoidosis 01 Biological Process", font.size = 10)
barplot(egoSarco1, 
        drop = TRUE, 
        showCategory = 10, 
        title = "Patient Sarcoidosis 01 Biological Process",
        font.size = 10)

egoSarco1 <- enrichGO(gene = unique(DEGenesSarco1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco1 <- setReadable(egoSarco1, OrgDb = org.Hs.eg.db)
dotplot(egoSarco1, showCategory=20, title = "Patient Sarcoidosis 01 Metabolic Function", font.size = 10)


egoSarco1 <- enrichGO(gene = unique(DEGenesSarco1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco1 <- setReadable(egoSarco1, OrgDb = org.Hs.eg.db)
dotplot(egoSarco1, showCategory=20, title = "Patient Sarcoidosis 01 Cellular Component", font.size = 10)

# rRNA processing

```

```{r, fig.width = 12, fig.height = 12}

# Patient Sarcoidosis 02
DEGenesSarco2 <- DEGenes_patient %>% filter(cluster == "Sarcoidosis 02")


egoSarco2 <- enrichGO(gene = unique(DEGenesSarco2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco2 <- setReadable(egoSarco2, OrgDb = org.Hs.eg.db)
dotplot(egoSarco2, showCategory=20, title = "Patient Sarcoidosis 02 Biological Process", font.size = 10)
barplot(egoSarco2, 
        drop = TRUE, 
        showCategory = 10, 
        title = "Patient Sarcoidosis 02 Biological Process",
        font.size = 10)

egoSarco2 <- enrichGO(gene = unique(DEGenesSarco2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco2 <- setReadable(egoSarco2, OrgDb = org.Hs.eg.db)
dotplot(egoSarco2, showCategory=20, title = "Patient Sarcoidosis 02 Metabolic Function", font.size = 10)


egoSarco2 <- enrichGO(gene = unique(DEGenesSarco2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoSarco2 <- setReadable(egoSarco2, OrgDb = org.Hs.eg.db)
dotplot(egoSarco2, showCategory=20, title = "Patient Sarcoidosis 02 Cellular Component", font.size = 10)

# angiogenesis, adhesion, TGFbeta, SMAD

```

```{r, fig.width = 12, fig.height = 8}

# Patient Healthy 01
DEGenesHealthy1 <- DEGenes_patient %>% filter(cluster == "Healthy 01")


egoHealthy1 <- enrichGO(gene = unique(DEGenesHealthy1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy1 <- setReadable(egoHealthy1, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy1, showCategory=20, title = "Patient Healthy 01 Biological Process", font.size = 10)
barplot(egoHealthy1, 
        drop = TRUE, 
        showCategory = 10, 
        title = "Patient Healthy 02 Biological Process",
        font.size = 10)

egoHealthy1 <- enrichGO(gene = unique(DEGenesHealthy1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy1 <- setReadable(egoHealthy1, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy1, showCategory=20, title = "Patient Healthy 01 Metabolic Function", font.size = 10)


egoHealthy1 <- enrichGO(gene = unique(DEGenesHealthy1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy1 <- setReadable(egoHealthy1, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy1, showCategory=20, title = "Patient Healthy 01 Cellular Component", font.size = 10)



```

```{r, fig.width = 12, fig.height = 12}

# Patient Healthy 02
DEGenesHealthy2 <- DEGenes_patient %>% filter(cluster == "Healthy 02")


egoHealthy2 <- enrichGO(gene = unique(DEGenesHealthy2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy2 <- setReadable(egoHealthy2, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy2, showCategory=20, title = "Patient Healthy 02 Biological Process", font.size = 10)
barplot(egoHealthy2, 
        drop = TRUE, 
        showCategory = 10, 
        title = "Patient Healthy 02 Biological Process",
        font.size = 10)

egoHealthy2 <- enrichGO(gene = unique(DEGenesHealthy2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy2 <- setReadable(egoHealthy2, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy2, showCategory=20, title = "Patient Healthy 02 Metabolic Function", font.size = 10)


egoHealthy2 <- enrichGO(gene = unique(DEGenesHealthy2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy2 <- setReadable(egoHealthy2, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy2, showCategory=20, title = "Patient Healthy 02 Cellular Component", font.size = 10)



```

```{r, fig.width = 12, fig.height = 12}

# Patient Healthy 03
DEGenesHealthy3 <- DEGenes_patient %>% filter(cluster == "Healthy 03")


egoHealthy3 <- enrichGO(gene = unique(DEGenesHealthy3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy3 <- setReadable(egoHealthy3, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy3, showCategory=20, title = "Patient Healthy 03 Biological Process", font.size = 10)
barplot(egoHealthy3, 
        drop = TRUE, 
        showCategory = 10, 
        title = "Patient Healthy 03 Biological Process",
        font.size = 10)

egoHealthy3 <- enrichGO(gene = unique(DEGenesHealthy3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy3 <- setReadable(egoHealthy3, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy3, showCategory=20, title = "Patient Healthy 03 Metabolic Function", font.size = 10)


egoHealthy3 <- enrichGO(gene = unique(DEGenesHealthy3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy3 <- setReadable(egoHealthy3, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy3, showCategory=20, title = "Patient Healthy 03 Cellular Component", font.size = 10)



```

```{r, fig.width = 12, fig.height = 12}

# Patient Healthy 04
DEGenesHealthy4 <- DEGenes_patient %>% filter(cluster == "Healthy 04")


egoHealthy4 <- enrichGO(gene = unique(DEGenesHealthy4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy4 <- setReadable(egoHealthy4, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy4, showCategory=20, title = "Patient Healthy 04 Biological Process", font.size = 10)
barplot(egoHealthy4, 
        drop = TRUE, 
        showCategory = 10, 
        title = "Patient Healthy 04 Biological Process",
        font.size = 10)

egoHealthy4 <- enrichGO(gene = unique(DEGenesHealthy4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy4 <- setReadable(egoHealthy4, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy4, showCategory=20, title = "Patient Healthy 04 Metabolic Function", font.size = 10)


egoHealthy4 <- enrichGO(gene = unique(DEGenesHealthy4$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy4 <- setReadable(egoHealthy4, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy4, showCategory=20, title = "Patient Healthy 04 Cellular Component", font.size = 10)



```

```{r, fig.width = 12, fig.height = 12}

# Patient Healthy 05
DEGenesHealthy5 <- DEGenes_patient %>% filter(cluster == "Healthy 05")


egoHealthy5 <- enrichGO(gene = unique(DEGenesHealthy5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy5 <- setReadable(egoHealthy5, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy5, showCategory=20, title = "Patient Healthy 05 Biological Process", font.size = 10)
barplot(egoHealthy5, 
        drop = TRUE, 
        showCategory = 10, 
        title = "Patient Healthy 05 Biological Process",
        font.size = 10)

egoHealthy5 <- enrichGO(gene = unique(DEGenesHealthy5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy5 <- setReadable(egoHealthy5, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy5, showCategory=10, title = "Patient Healthy 05 Metabolic Function", font.size = 10)


egoHealthy5 <- enrichGO(gene = unique(DEGenesHealthy5$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy5 <- setReadable(egoHealthy5, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy5, showCategory=10, title = "Patient Healthy 05 Cellular Component", font.size = 10)



```

```{r, fig.width = 10, fig.height = 8}

# Patient Healthy 06
DEGenesHealthy6 <- DEGenes_patient %>% filter(cluster == "Healthy 06")


egoHealthy6 <- enrichGO(gene = unique(DEGenesHealthy6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy6 <- setReadable(egoHealthy6, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy6, showCategory=20, title = "Patient Healthy 06 Biological Process", font.size = 10)
barplot(egoHealthy6, 
        drop = TRUE, 
        showCategory = 10, 
        title = "Patient Healthy 06 Biological Process",
        font.size = 10)

egoHealthy6 <- enrichGO(gene = unique(DEGenesHealthy6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy6 <- setReadable(egoHealthy6, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy6, showCategory=20, title = "Patient Healthy 06 Metabolic Function", font.size = 10)


egoHealthy6 <- enrichGO(gene = unique(DEGenesHealthy6$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy6 <- setReadable(egoHealthy6, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy6, showCategory=20, title = "Patient Healthy 06 Cellular Component", font.size = 10)


```

```{r, fig.width = 10, fig.height = 8}

# Patient Healthy 07
DEGenesHealthy7 <- DEGenes_patient %>% filter(cluster == "Healthy 07")


egoHealthy7 <- enrichGO(gene = unique(DEGenesHealthy7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy7 <- setReadable(egoHealthy7, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy7, showCategory=20, title = "Patient Healthy 07 Biological Process", font.size = 10)
barplot(egoHealthy7, 
        drop = TRUE, 
        showCategory = 10, 
        title = "Patient Healthy 07 Biological Process",
        font.size = 10)

egoHealthy3 <- enrichGO(gene = unique(DEGenesHealthy3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy7 <- setReadable(egoHealthy7, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy7, showCategory=20, title = "Patient Healthy 07 Metabolic Function", font.size = 10)


egoHealthy7 <- enrichGO(gene = unique(DEGenesHealthy7$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy7 <- setReadable(egoHealthy7, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy7, showCategory=20, title = "Patient Healthy 07 Cellular Component", font.size = 10)



```

```{r, fig.width = 10, fig.height = 8}

# Patient Healthy 08
DEGenesHealthy8 <- DEGenes_patient %>% filter(cluster == "Healthy 08")


egoHealthy8 <- enrichGO(gene = unique(DEGenesHealthy8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy8 <- setReadable(egoHealthy8, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy8, showCategory=10, title = "Patient Healthy 08 Biological Process", font.size = 10)
barplot(egoHealthy8, 
        drop = TRUE, 
        showCategory = 10, 
        title = "Patient Healthy 08 Biological Process",
        font.size = 10)

egoHealthy8 <- enrichGO(gene = unique(DEGenesHealthy8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy8 <- setReadable(egoHealthy8, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy8, showCategory=20, title = "Patient Healthy 08 Metabolic Function", font.size = 10)


egoHealthy8 <- enrichGO(gene = unique(DEGenesHealthy8$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoHealthy8 <- setReadable(egoHealthy8, OrgDb = org.Hs.eg.db)
dotplot(egoHealthy8, showCategory=20, title = "Patient Healthy 08 Cellular Component", font.size = 10)

# Healthy 08 is apparently not healthy, but has inflammatory signature (response LPS, innate, hypoxia)

```

```{r}

date()
sessionInfo()

```
