---
title: "Fibroblasts Subgroups Preliminary"
output: html_document
date: "2024-09-19"
---



```{r setup, include = FALSE}
options (width = 100)
knitr::opts_chunk$set(warning = FALSE, echo = TRUE,message = FALSE, dev = c("png", "pdf"))
seed <- 1234

```

```{r load libraries, echo=FALSE, message=FALSE, warning=FALSE}

suppressPackageStartupMessages({
  library(ExploreSCdataSeurat3)
  library(runSeurat3)
  library(Seurat)
  library(ggpubr)
  library(pheatmap)
  library(SingleCellExperiment)
  library(dplyr)
  library(tidyverse)
  library(viridis)
  library(muscat)
  library(circlize)
  library(destiny)
  library(scater)
  library(metap)
  library(multtest)
  library(clusterProfiler)
  library(org.Hs.eg.db)
  library(msigdbr)
  library(enrichplot)
  library(DOSE)
  library(grid)
  library(gridExtra)
  library(ggupset)
  library(VennDiagram)
  library(NCmisc)
  library(RColorBrewer)
  library(EnhancedVolcano)
  library(textshaping)
  library(wordcloud)
  library(pathview)
  library(workflowr)
})

```

```{r load merged file}

##load merged file 
seuratM <- readRDS("/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_allfiltered_12.11.24.rds")


table(seuratM$dataset)

table(seuratM$dataset)
table(seuratM$RNA_snn_res.0.25)
table(seuratM$patient)
table(seuratM$orig.ident)
```



```{r}

#### Rename clusters

seuratM$clusterName <- "clusterName"
Idents(seuratM) <- seuratM$clusterName
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "0" )] <- "Fb1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "1" )] <- "BEC1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "2" )] <- "PeriFb1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "3" )] <- "Cardiomyocyte"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "4" )] <- "MCP1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "5" )] <- "TC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "6" )] <- "MCP2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "7" )] <- "BEC2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "8" )] <- "VSMC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "9" )] <- "NC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "10" )] <- "INT1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "11" )] <- "Fb2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "12" )] <- "Fb3"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "13" )] <- "PeriFb2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "14" )] <- "AdipoC"


colclusterName <- c("#D53E4F", "#f4a582", "#FEE08B","#feb24c","#67001f", "#01665e","#66C2A5","#c7eae5","#BEAED4", "#355C7D","#3288BD","#8c510a" ,"#fde0dd","#B45B5C","#dd1c77")

                    
names(colclusterName) <- c("Cardiomyocyte","Fb1","Fb2","Fb3", "INT1", "PeriFb1", "PeriFb2", "VSMC","NC", "BEC1","BEC2", "AdipoC", "MCP1","MCP2","TC")

colpat_sub<- c("#dfc27d","#BE3144","#355C7D", "#779d8d")

seuratM$clusterName <- factor(seuratM$clusterName, levels=c("Cardiomyocyte", "Fb1", "Fb2", "Fb3", "INT1","PeriFb1", "PeriFb2", "VSMC", "NC", "BEC1", "BEC2", "AdipoC", "MCP1", "MCP2", "TC"))


Idents(seuratM) <- seuratM$clusterName
DimPlot(seuratM, reduction = "umap", pt.size = 0.1, label = TRUE, cols = colclusterName)

```
```{r}

# DE Genes 

DEGenesCluster <- FindAllMarkers (seuratM, only.pos = TRUE, logfc.threshold = 0.2) %>%
                                      dplyr::filter(p_val_adj < 0.01)


#write.table(DEGenesCluster, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenes_condition pos. and neg. AM, Healthy, CMP, Sarcoidosis", sep="\t",quote=F,row.names=F,col.names=T)


##adjust table
DEGenesCluster_adj <- DEGenesCluster %>% 
  mutate(Gene=gene) %>%
  mutate(gene=gsub("^.*\\.", "", Gene))  %>%
  mutate(EnsID=gsub("\\..*","", Gene))

#View(DEGenesCluster)

```



```{r Fibroblast Cluster 2, fig.width = 11, fig.height= 8}
##GSEA for Fb2
DEGenesFb2 <- DEGenesCluster_adj %>% filter(cluster == "Fb2")
#head(DEGenesFb2)

#write.table(DEGenesFb2, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DEGenesFb", sep="\t",quote=F,row.names=F,col.names=T)

egoFb2 <- enrichGO(gene = unique(DEGenesFb2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoFb2 <- setReadable(egoFb2, OrgDb = org.Hs.eg.db)
dotplot(egoFb2, showCategory=20, title = "Fibroblast 2 Cluster Biological Process", font.size = 10)
```


```{r }
#Barplot Fb2
barplot(egoFb2, 
        drop = TRUE, 
        showCategory = 5, 
        title = "Fibroblast 2 Cluster",
        font.size = 10)
```


```{r fig.width = 11, fig.height= 8}
egoFb2 <- enrichGO(gene = unique(DEGenesFb2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "CC",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoFb2 <- setReadable(egoFb2, OrgDb = org.Hs.eg.db)


dotplot(egoFb2, showCategory=20, title = "Fibroblast 2 Cluster Cellular Component", font.size = 10)


egoFb2 <- enrichGO(gene = unique(DEGenesFb2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "MF",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoFb2 <- setReadable(egoFb2, OrgDb = org.Hs.eg.db)
dotplot(egoFb2, showCategory=20, title = "Fibroblast 2 Metabolic Function", font.size = 10)


```





```{r fig.width = 10, fig.height = 9}
##GSEA for Fb1
DEGenesFb1 <- DEGenesCluster_adj %>% filter(cluster == "Fb1")
#head(DEGenesFb2)

egoFb1 <- enrichGO(gene = unique(DEGenesFb1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoFb1 <- setReadable(egoFb1, OrgDb = org.Hs.eg.db)
dotplot(egoFb1, showCategory=30, title = "Fibroblast 1 Cluster", font.size = 10)
```


```{r}

# Barplot Fb1
barplot(egoFb1, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Fibroblast 1 Cluster",
        font.size = 10)


```

```{r fig.width=9, fig.height=7}
##GSEA for Fb3
DEGenesFb3 <- DEGenesCluster_adj %>% filter(cluster == "Fb3")
#head(DEGenesFb3)

egoFb3 <- enrichGO(gene = unique(DEGenesFb3$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoFb3 <- setReadable(egoFb3, OrgDb = org.Hs.eg.db)
dotplot(egoFb3, showCategory=10, title = "Fibroblast 3 Cluster", font.size = 10)
```


```{r fig.width=9, fig.height=7}

# Barplot Fb3
barplot(egoFb3, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Fibroblast 3 Cluster",
        font.size = 10)
```

```{r fig.width=10, fig.height=8}

##GSEA for Int non significant
##GSEA for PeriFb1
DEGenesPeriFb1 <- DEGenesCluster_adj %>% filter(cluster == "PeriFb1")
#head(DEGenesPeriFb1)

egoPeriFb1 <- enrichGO(gene = unique(DEGenesPeriFb1$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoPeriFb1 <- setReadable(egoPeriFb1, OrgDb = org.Hs.eg.db)
dotplot(egoPeriFb1, showCategory=10, title = "Perivascular Fb 1", font.size = 10)
barplot(egoPeriFb1, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Perivascular Fb 1",
        font.size = 10)
```


```{r fig.width=10, fig.height=8}


##GSEA for PeriFb2
DEGenesPeriFb2 <- DEGenesCluster_adj %>% filter(cluster == "PeriFb2")
#head(DEGenesPeriFb1)

egoPeriFb2 <- enrichGO(gene = unique(DEGenesPeriFb2$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoPeriFb2 <- setReadable(egoPeriFb1, OrgDb = org.Hs.eg.db)
dotplot(egoPeriFb2, showCategory=10, title = "Perivascular Fb 2", font.size = 10)
barplot(egoPeriFb2, 
        drop = TRUE, 
        showCategory = 20, 
        title = "Perivascular Fb 2",
        font.size = 10)
```


```{r fig.width=10, fig.height=9}


##GSEA for PeriFb2
DEGenesVsmc <- DEGenesCluster_adj %>% filter(cluster == "VSMC")
#head(DEGenesVsmc)

egoVsmc <- enrichGO(gene = unique(DEGenesVsmc$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoPeriVsmc <- setReadable(egoVsmc, OrgDb = org.Hs.eg.db)
dotplot(egoPeriVsmc, showCategory=10, title = "VSMC", font.size = 10)
barplot(egoPeriVsmc, 
        drop = TRUE, 
        showCategory = 20, 
        title = "VSMC",
        font.size = 10)
```





```{r}
DEGenesClusterall <- FindAllMarkers(seuratM, only.pos = FALSE, logfc.threshold = 0.2) %>%
  dplyr::filter(p_val_adj < 0.01)


##adjust table
DEGenesClusterall_adj <- DEGenesClusterall %>% 
  mutate(Gene=gene) %>%
  mutate(gene=gsub("^.*\\.", "", Gene))  %>%
  mutate(EnsID=gsub("\\..*","", Gene))

```


```{r Volcano Plot, fig.width = 8, fig.height = 8}

## Volcano Plot with pos. and neg. DE Genes Fb2

DEGenesFb2all <- DEGenesClusterall_adj %>% filter(cluster == "Fb2")

head(DEGenesFb2all)

#View(DEGenesFb2all)

res_Fb2 <- as.data.frame(DEGenesFb2all)

EnhancedVolcano(res_Fb2, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Fb2$gene,
                # xlim =c(-6, 6), 
                title = 'DE Genes Fibroblast 2 Cluster',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 1.0,
    labSize = 5.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)
```


```{r fig.width=8, fig.height = 8}

EnhancedVolcano(res_Fb2, 
                x = 'avg_log2FC',
                y = 'p_val_adj',
                lab = res_Fb2$gene,
                selectLab = c("CXCL6","STAT1", "VCAN",  "GBP1P1", "GBP2", "SRPX2", "SLIT2", "CCL19","CLSPN", "SLIT2","GBP2", "VDR", "CCL11","CCL2", 
                             "CXCL9", "CXCL10", "PDGFA", "CXCL1", "FN1","ABI3BP","FBN1","TIMP", "GBP1", "C1GALT1", "EFEMP1", "APOL1"),
                # xlim =c(-6, 6), 
                title = 'DE Genes Fibroblast 2 Cluster',
                xlab = bquote(~Log[2]~ 'fold change'),
                pCutoff = 10e-1,
                FCcutoff = 1.5,
    cutoffLineType = 'twodash',
    cutoffLineWidth = 0.8,
    pointSize = 1.0,
    labSize = 5.0,
    # shape = c(1,4, 23,25),
    colAlpha = 1,
    legendLabels=c('Not sig.','Log (base 2) FC','p-value',
      'p-value & Log (base 2) FC'),
    legendPosition = 'right',
    legendLabSize = 12,
    legendIconSize = 5.0,
    # drawConnectors = TRUE,
    # widthConnectors = 0.75,
    # ridlines.major = FALSE,
    gridlines.minor = FALSE)

```



```{r}

# Select only Fibroblast 2 Cluster
seuratFb2 <- subset(seuratM, clusterName %in% "Fb2")

table(seuratFb2$clusterName)
table(seuratFb2$patient)

table(seuratFb2$orig.ident)

# nuclei originate from AM 01 and AM 16 primarily

```




```{r}


Idents(seuratM) <- seuratM$clusterName

seuratFb <- subset(seuratM, clusterName %in% c("Fb1", "Fb2", "Fb3", "INT1", "PeriFb1", "PeriFb2", "VSMC"))

table(seuratFb$patient)

levels(seuratFb)
DimPlot(seuratFb, reduction = "umap", cols = colclusterName, pt.size = 0.3)

Idents(seuratFb) <- seuratFb$pat_sub
DimPlot(seuratFb, reduction = "umap", cols = colpat_sub, pt.size = 0.3)

Fb_markers_patsub <- FindAllMarkers(seuratFb, only.pos=T) %>% 
  dplyr::filter(p_val_adj < 0.01)

#View(Fb_markers_patsub)

```


```{r}

Idents(seuratFb) <- seuratFb$clusterName

Fbmarkers <- FindAllMarkers(seuratFb, only.pos=T) %>% 
  dplyr::filter(p_val_adj < 0.01)
  

```

```{r}
## 

Idents(seuratFb) <- seuratFb$clusterName
seuratFb$clusterName <- factor(seuratFb$clusterName, levels =c("Fb1","Fb2","Fb3", "INT1", "PeriFb1", "PeriFb2", "VSMC"))


```




```{r Dotplot Fb, fig.width = 8, fig.height = 7}

## Create Heatmap

seuratFb$clusterName <- factor(seuratFb$clusterName, levels =c("Fb1","Fb2","Fb3", "INT1", "PeriFb1", "PeriFb2", "VSMC"))

genes <- data.frame(gene=rownames(seuratFb)) %>% 
  mutate(geneID=gsub("^.*\\.", "", gene))


selGenesFb <- data.frame(geneID=rev(c("XIST", "ABI3BP", "NOX4", "FBN1", "GBP1", "FN1", "TIMP1", "MICALL2", "CCL2", "SRPX2", "CXCL6", "CXCL9", "CXCL10", "CXCL11", "SPHK1",  "HIF3A",  "LAMA2", "STAT1", "PDGFRA", "MEG3", "ABCA8", "LAMB1", "C7", "BMP4", "GREM1", "CARMN", "COL5A3", "PDGFRB","ACTA2", "NOTCH3","ADAP1", "ADAMTS4", "RGS5", "HLA-B")))  %>% left_join(., genes, by="geneID") %>% distinct(geneID, .keep_all = TRUE)


DotPlot(seuratFb, features = selGenesFb, group.by= "clusterName") + RotatedAxis() + scale_color_viridis(option="T") + coord_flip()


```









```{r Heatmap Fb cell clusters, fig.width=10, fig.height = 7}


avgHeatmap <- function(seuratFb, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  selGenes <- selGenes$gene
  
  ## assay data
  clusterAssigned <- as.data.frame(Idents(seuratFb)) %>%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] <- "ident"
  seuratDat <- GetAssayData(seuratFb)
  
  ## genes of interest
  genes <- data.frame(gene=rownames(seuratFb)) %>% 
    mutate(geneID=gsub("^.*\\.", "", gene)) %>% filter(geneID %in% selGenes)

  ## matrix with averaged cnts per ident
  logNormExpres <- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres <- logNormExpres %>% dplyr::mutate(cell=rownames(.)) %>%
    dplyr::left_join(.,clusterAssigned, by=c("cell")) %>%
    dplyr::select(-cell) %>% dplyr::group_by(ident) %>%
    dplyr::summarise_all(mean)
  logNormExpresMa <- logNormExpres %>% dplyr::select(-ident) %>% as.matrix()
  rownames(logNormExpresMa) <- logNormExpres$ident
  logNormExpresMa <- t(logNormExpresMa)
  rownames(logNormExpresMa) <- gsub("^.*?\\.","",rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind <- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa <- logNormExpresMa[!ind,]
  genes <- genes[!ind,]
  
  ## color columns according to cluster
  annotation_col <- as.data.frame(gsub("(^.*?_)","",
                                       colnames(logNormExpresMa)))%>%
    dplyr::mutate(celltype=gsub("(_.*$)","",colnames(logNormExpresMa)))
  colnames(annotation_col)[1] <- "col1"
  annotation_col <- annotation_col %>%
    dplyr::mutate(cond = gsub("(^[0-9]_?)","",col1)) %>%
    dplyr::select(cond, celltype)
  rownames(annotation_col) <- colnames(logNormExpresMa) 

  ann_colors = list(
      celltype=colVec)
  if(is.null(ann_colors$cond)){
    annotation_col$cond <- NULL
  }
  
  
  ## adjust order
  logNormExpresMa <- logNormExpresMa[selGenes,]
  if(is.null(ordVec)){
  ordVec <- levels(seuratFb)
  ordVec <- c(c("Fb1","Fb2","Fb3", "INT1", "PeriFb1", "PeriFb2", "VSMC"))
  }
  logNormExpresMa <- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale="row" ,treeheight_row = 0,cluster_rows = cr, 
         cluster_cols = cc, color = colorRampPalette(c("#2166AC", "#F7F7F7", "#B2182B"))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}

genesPlot <- data.frame(gene=c("XIST", "ABI3BP", "NOX4", "FBN1", "GBP1", "FN1", "TIMP1", "MICALL2", "CCL2", "SRPX2", "CXCL6", "CXCL9", "CXCL10", "CXCL11", "SPHK1",  "HIF3A",  "LAMA2", "STAT1", "PDGFRA", "MEG3", "ABCA8", "LAMB1", "C7", "BMP4", "GREM1", "CARMN", "COL5A3", "PDGFRB", "NOTCH3","ADAP1", "ADAMTS4", "RGS5", "HLA-B"))

levels(seuratFb)


colVec <- colclusterName

# colVec <- c(colPal, colPal, colPal)
# colVec <- c("blue", "red")

avgHeatmap(seuratFb, selGenes = genesPlot, colVecIdent = colVec)


```





```{r}

# Filter for clusters Fb1, Fb2, Fb3, Int1, PeriFb1,2,3

seuratFb <- subset(seuratM, clusterName %in% c("Fb1", "Fb2", "Fb3", "INT1", "PeriFb1", "PeriFb2", "VSMC"))
# cell count N = 21641

DimPlot(seuratFb, reduction = "umap", cols = colclusterName, pt.size = 0.1)

table(seuratFb$orig.ident)
table(seuratFb$patient)

Fb_markers <- FindAllMarkers(seuratFb, only.pos=T) %>% 
  dplyr::filter(p_val_adj < 0.01)
#
#View(Fb_markers)
#write.table(Fb_markers, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Fb markers after HH2_HH3 removal", sep="\t",quote=F,row.names=F,col.names=T)


```


```{r}

datList <- NULL
for(con in unique(seuratFb$patient)){
  seuratSub <- subset(seuratFb, patient==con)
  print(dim(seuratSub))
  dat_con <- as.data.frame(table(seuratSub$clusterName)) %>%
  mutate(percent=Freq/ncol(seuratSub)) %>% mutate(patient=con)
  datList[[con]] <- dat_con
}


dat_all <- do.call("rbind", datList)

# Order of patient levels 
patient_order <- c("Healthy 01", "Healthy 02", "Healthy 03", "Healthy 04", "Healthy 05", 
                   "Healthy 06", "Healthy 07", "Healthy 08", "AM 01", "AM 02", 
                   "AM 03", "AM 04", "AM 05", "AM 06", "AM 07", 
                   "AM 08", "AM 09", "AM 10", "AM 11", "AM 12", "AM 13", "AM 14", 
                   "AM 15", "AM 16", "AM 17", "AM 18", "CMP 01", "CMP 02", "CMP 03", 
                   "CMP 04", "CMP 05", "CMP 06", "CMP 07", "CMP 08", "CMP 09", 
                   "CMP 10", "CMP 11", "CMP 12", "CMP 13", "Sarcoidosis 01", "Sarcoidosis 02")

View(dat_all)



# Convert the patient column to a factor with the desired order
dat_all$patient <- factor(dat_all$patient, levels = patient_order)

## plot abundance
ggbarplot(dat_all, x= "patient", y= "percent", fill = "Var1", legend = "right", legend.titel = "cluster", ylab = "frequency", palette = colclusterName )  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


```




```{r Relative abundance dis con}

datList <- NULL
for(con in unique(seuratFb$pat_sub)){
  seuratSub <- subset(seuratFb, pat_sub==con)
  print(dim(seuratSub))
  dat_con <- as.data.frame(table(seuratSub$clusterName)) %>%
  mutate(percent=Freq/ncol(seuratSub)) %>% mutate(diseaseCond=con)
  datList[[con]] <- dat_con
  
}

dat_all <- do.call("rbind", datList)

order_patsub <- c("Healthy", "AM", "CMP", "Sarcoidosis")

ggbarplot(dat_all, x = "diseaseCond", y = "percent", fill = "Var1", palette = colclusterName, legend = "right", legend.titel = "cluster", xlab = "condition", ylab = "frequency") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_x_discrete(limits = order_patsub)


```





```{r GSEA}

Idents(seuratM) <- seuratM$clusterName
seuratFb <- subset(seuratM, clusterName %in% c("Fb1", "Fb2", "Fb3", "INT1", "PeriFb1", "PeriFb2", "VSMC"))
levels(seuratFb)
DimPlot(seuratFb, reduction = "umap", cols = colclusterName, pt.size = 0.1, label = TRUE)

seuratFb <- NormalizeData (object = seuratFb)
seuratFb <- FindVariableFeatures(object = seuratFb)
seuratFb <- ScaleData(object = seuratFb, verbose = TRUE)
seuratFb <- RunPCA(object=seuratFb, npcs = 30, verbose = FALSE)
seuratFb <- RunTSNE(object=seuratFb, reduction="pca", dims = 1:20)
seuratFb <- RunUMAP(object=seuratFb, reduction="pca", dims = 1:20)
seuratFb <- FindNeighbors(object = seuratFb, reduction = "pca", dims= 1:20)

res <- c(0.25, 0.6, 0.8, 0.4)
for (i in 1:length(res)) {
  seuratFb <- FindClusters(object = seuratFb, resolution = res[i], random.seed = 1234)
}

table(seuratFb$orig.ident)


```



```{r}


Idents(seuratFb) <- seuratFb$RNA_snn_res.0.25
DimPlot(seuratFb, reduction = "umap", pt.size = 0.1, label = TRUE)

Idents(seuratFb) <- seuratFb$pat_sub
DimPlot(seuratFb, reduction = "umap", pt.size = 0.1, repel = TRUE, cols = colpat_sub)

```






```{r}


seuratFb$clusterName <- "clusterName"
Idents(seuratFb) <- seuratFb$clusterName
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "0" )] <- "PeriFb1"
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "1" )] <- "Fb1"
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "2" )] <- "Fb3"
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "3" )] <- "VSMC"
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "4" )] <- "Fb4"
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "5" )] <- "Fb5"
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "6" )] <- "Fb8"
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "7" )] <- "Fb6"
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "8" )] <- "Fb2"
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "9" )] <- "PeriFb2"
seuratFb$clusterName[which(seuratFb$RNA_snn_res.0.25 %in% "10" )] <- "Fb7"


seuratFb$clusterName <- factor(seuratFb$clusterName, levels =c("Fb1","Fb2","Fb3","Fb4", "Fb5", "Fb6", "Fb7", "Fb8", "PeriFb1", "PeriFb2", "VSMC"))

table(seuratFb$patient,seuratFb$clusterName)

table(seuratFb$clusterName)

print(levels(seuratFb$clusterName))

                    
colFb <- c("#f4a582", "#FEE08B","#feb24c", "#9ebcda", "#2166AC" , "#756bb1", "#3182bd", "#c994c7", "#01665e","#66C2A5","#c7eae5")

names(colFb) <- c("Fb1","Fb2","Fb3","Fb4", "Fb5", "Fb6", "Fb7", "Fb8", "PeriFb1", "PeriFb2", "VSMC")

Idents(seuratFb) <- seuratFb$clusterName
DimPlot(seuratFb, reduction = "umap", pt.size = 0.1, label = FALSE, cols =colFb)

```







```{r Abundance per patient Fibroblast subgroup analysis}


datList <- NULL
for(con in unique(seuratFb$patient)){
  seuratSub <- subset(seuratFb, patient==con)
  print(dim(seuratSub))
  dat_con <- as.data.frame(table(seuratSub$clusterName)) %>%
  mutate(percent=Freq/ncol(seuratSub)) %>% mutate(patient=con)
  datList[[con]] <- dat_con
}


dat_all <- do.call("rbind", datList)

# Order 
patient_order <- c("Healthy 01", "Healthy 02", "Healthy 03", "Healthy 04", "Healthy 05", 
                   "Healthy 06", "Healthy 07", "Healthy 08", "Healthy 09", "Healthy 10", 
                   "AM 01", "AM 02", "AM 03", "AM 04", "AM 05", "AM 06", "AM 07", 
                   "AM 08", "AM 09", "AM 10", "AM 11", "AM 12", "AM 13", "AM 14", 
                   "AM 15", "AM 16", "AM 17", "AM 18", "CMP 01", "CMP 02", "CMP 03", 
                   "CMP 04", "CMP 05", "CMP 06", "CMP 07", "CMP 08", "CMP 09", 
                   "CMP 10", "CMP 11", "CMP 12", "CMP 13", "Sarcoidosis 01", "Sarcoidosis 02")

#View(dat_all)

#write.table(dat_all, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers_dat_all_clusters_frequency_relative abundance_V2", sep = "\t",quote=F,row.names=T,col.names=T)


dat_all$patient <- factor(dat_all$patient, levels = patient_order)

## plot abundance
ggbarplot(dat_all, x= "patient", y= "percent", fill = "Var1", legend = "right", legend.titel = "cluster", ylab = "frequency", palette = colFb )  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))



```




```{r Relative abundance per disease condition}

datList <- NULL
for(con in unique(seuratFb$pat_sub)){
  seuratSub <- subset(seuratFb, pat_sub==con)
  print(dim(seuratSub))
  dat_con <- as.data.frame(table(seuratSub$clusterName)) %>%
  mutate(percent=Freq/ncol(seuratSub)) %>% mutate(diseaseCond=con)
  datList[[con]] <- dat_con
  
}

dat_all <- do.call("rbind", datList)

order_patsub <- c("Healthy", "AM", "CMP", "Sarcoidosis")

ggbarplot(dat_all, x = "diseaseCond", y = "percent", fill = "Var1", palette = colFb, legend = "right", legend.titel = "cluster", xlab = "condition", ylab = "frequency") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_x_discrete(limits = order_patsub)


```




```{r}


##cluster markers
Fbmarkers <- FindAllMarkers(seuratFb, only.pos=F, logfc.threshold = 0.2) %>% 
  dplyr::filter(p_val_adj < 0.01)

#View(Fbmarkers)

# write.table(Fbmarkers, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/allFb_PeriFb_vsmc_int markers", sep="\t",quote=F,row.names=F,col.names=T)

```














```{r find markers, fig.width = 8, fig.height = 8}

Idents(seuratFb) <- seuratFb$clusterName

genes <- data.frame(gene = rownames (seuratFb)) %>%
  mutate(geneID =gsub("^.*\\.", "", gene))

# Selected genes for DotPlot
selGenes <- data.frame(geneID=rev(c("MEG3", "ABCA6", "ABCA10", "COL6A3","C7", "PDGFRA", "ACTA2","PECAM1", "VWF", "BMP4", "XIST", "GBP1", "GBP5", "CXCL9", "IRF8", "VMP1", "HLA-DMB", "APOL2", "IL7R", "CXCR2P1", "JAK2", "SLAMF8", "CD2", "CCL8", "CXCL6", "IL2RG", "COL1A2", "TNFSF10", "COl1A2", "SNX10", "XAF1","ZBTB16", "APOD", "ADAMTS5", "ADAMTS1", "RGS2", "GRAMD2B", "WNK3", "SLC7A8", "PLEKHG4", "S100A8")))  %>% left_join(., genes, by="geneID") %>% filter(gene != "ENSG00000232995.RGS5") 

DotPlot(seuratFb, features = selGenes, group.by= "clusterName") + RotatedAxis() + scale_color_viridis(option="T") + coord_flip()


## overlap with mcp genes
####selGenes <- data.frame(geneID=rev(c("MEG3", "ABCA6", "ABCA10", "COL6A3", "DCN",  "FBN1", "C7", "PDGFRA", "BMP4", "GREM1","GREM2", "HLA-B", "GBP1", "STAT1", "CDH19", "PDGFRB","ITGA7","RGS5", "NOTCH3", "MYH11","CARMN", "CD36", "ACTA2","PECAM1", "VWF", "SMOC1", "EGFL7", "ITGA10", "CXCL9", "CXCL10", "CDH11","CCL21", "PROX1", "FLT4", "EGFL71", "NOTCH41","SHANK31", "BTLN9", "CD163", "LYVE1", "MRC1", "SIGLEC1", "STAB1","MS4A6A", "CSF1R","MERTK", "HLA-DQB1", "CCR2", "IL7R", "CD2", "CD79B", "NRXN1", "TTN"))) %>% left_join(., genes, by="geneID") %>% filter(gene != "ENSG00000232995.RGS5") 

## Fb1: ECM "normal" fibroblasts

## Fb2
# HLA-B and HLA-A
# STAT1 signal transducer and activator of transcription, can be activated by inferon, IL-6, involved in immune responses
# CD74 MHCII antigen presentation
# LAP3: Metallopeptidase
# WARS1: catalyzes the aminoacylation of tRNA with tryptophan and is induced by interferon
# NLRC5: cytokine response and antiviral immunity
# CXCL9 and CXCL10

## Fb3
# STAB1 Stabilin, angiogenesis, lymphocyte homing, cell adhesion, binds to gram pos and neg. bacteria, defense against bacterial infections
# CD163 scavenger receptor family expressed in monocytes and macrophages, acute phase regulated receptor, acute phase reaction
# MS4A7 CD20 antigen like 4, high affinity immunoglobulin epsilon receptor beta subunit, associated with cellular function in the monocytic lineage
# MRC1 Mannose receptor type 1, cell-cell recognition, neutroalizuation pathogens, mediates endocytosis by macrophages
# F13A1 coagulation factor XIII, crosslinking between fibrin molecules, stabilizes fibrin clots
# SIGLEC1  Sialic acid binding immunoglobulin like lectin: on subpopulation of macrophages, mediates cell-cell interactions, bacterial and viral infection
# LILRB5 Leukocyte immunoglobulin like receptor subfamily B, innate immune system, antigen processing
# CYTH4 Cytohesin 4: vesicle mediated transport. Golgi and ER
# FGD2 fuanine nucleotide exchange factors, cytoskeleton dependent membrane rearrancgement, B lymphocytes, macrophages, dendritic cells
# LCGR2C: not found
# LYVE1: lymphatic vessel endothelial hyaluronan receptor 1: membrane glycoproteine, lymphatic hyaluronan transport
# NCF4: cytosolic regulatory component of NADPH oxidase; important for host defense

###
## PeriFb1
# CARMN cardiac mesoderm enhancer associated non coding RNA: responsible for contractile phenotype of VSMCs
# CD36: glycoprotein on platelet surfaces, cell adhesion, binds collagen, trhobmospondin, angiogenesis, inflammatroy responses, muscle lipid utilization
# ITGA7: Integrin subunit alpha, mainly expressed in skeletal and cardiac muscles; receptor for Lamini-1, myofiber cytoarchitecture maintenance
# ABCC9: ATP sensitive potassium channel in cardiac skeletal and vascular and non-vascular smooth muscle
# RGS5: GTPasa activity Gi
# DLC1: deleted in liver cancer 1, GTPAse, regulatory processes cytoskeletal changes


##PeriFb3
#GBP1: Interferon induced GTPase: innate immunity; inflammasome assembly Guanlyate binding protein expression is induced by interferon
#GBP4: Interferon induced GTPase
#TAP1: Antigen transporter associated with antigen processing
#CXCL9: binds CXCR3, monokine induced by Interferon-Gamma; immunoregulatory and inflammatory processes, T cell trafficking
#CXCL10: cytokine storm, binds CXCR3, stimulates monocytes, NK, T cells
#ADAP1: GTPase activator activity
#CXCL11: induced chemotactic response in activated T cells, binds to CXCR3
#CCL8: Monocyte chemotactic protein 2: chemotactic activity for monocytes, lymphocytes, basophils and eosinophils, contributes to antiviral state
#HLA-B: antigen presentation

###INT1
# EGFL71: secreted endothelial cell protein that contains two epidermal growth factor-like domains. The encoded protein may play a role in regulating vasculogenesis. EGFL7 Secreted By Human Bone Mesenchymal Stem Cells Promotes Osteoblast Differentiation
# NOTCH41: not found in gene database (only Notch4), epidermal growth factor vascular, renal and hepatic development
# SHANK31: not found in gene database (only shank3synapse formation and dendritic spine maturation)
# BTLN9: Butyrophylin like 9 Predicted to be involved in T cell receptor signaling pathway and regulation of cytokine production
# SLC9A3R2 nf --> SLC9A3 (Solute Carrier Family 9 Member A3) sodium channel
# CR381670.1 ankyrin repeat domain 20 family, member A3 (ANKRD20A3) pseudogene
# PTPRB cell growth, differentiation, mitotic cycle, and oncogenic transformation. 
# SYNE22 nf
# ADGRF52 nf, ADGRF5 glomerular filtration; pharyngeal arch artery morphogenesis; and surfactant homeostasis
# VWF platelet aggregation endothelial cells
# PECAM12: Platelet And Endothelial Cell Adhesion Molecule 1, PECAM1 The protein encoded by this gene is found on the surface of platelets, monocytes, neutrophils, and some types of T-cells, and makes up a large portion of endothelial cell intercellular junction

## PeriFb2 first 7 highly expressed genes identical to INT1:
#EGFL7
#SHANK3
#NOTCH4

## VSMC

```











`













```{r}
date()
sessionInfo()
```


