---
title: "Interactome Analysis Loading"
author: "Emily Payne"
date: "2024-11-23"
output: html_document
---


```{r setup, include = FALSE, fig.path = "custom/path", warning = FALSE, message = FALSE}

options (width = 100)
knitr::opts_chunk$set(warning = FALSE, echo = TRUE, message = FALSE, dev = c("png", "pdf"))
seed <- 1234

```


```{r load libraries, echo=FALSE, message=FALSE, warning=FALSE}

#devtools::install_github("sqjin/CellChat")
library(CellChat)
library(patchwork)
options(stringsAsFactors = FALSE)

suppressPackageStartupMessages({
  library(ExploreSCdataSeurat3)
  library(runSeurat3)
  library(Seurat)
  library(ggpubr)
  library(pheatmap)
  library(SingleCellExperiment)
  library(dplyr)
  library(tidyverse)
  library(viridis)
  library(muscat)
  library(circlize)
  library(destiny)
  library(scater)
  library(metap)
  library(multtest)
  library(clusterProfiler)
  library(org.Hs.eg.db)
  library(msigdbr)
  library(enrichplot)
  library(DOSE)
  library(grid)
  library(gridExtra)
  library(ggupset)
  library(VennDiagram)
  library(NCmisc)
  library(RColorBrewer)
  library(EnhancedVolcano)
  library(textshaping)
  library(wordcloud)
  library(pathview)
  library(workflowr)
  library(CellChat)
  library(patchwork)
  options(stringsAsFactors = FALSE)
})


```

```{r load merged file}
##load merged file 
seuratM <- readRDS("~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_allfiltered_V3_15.11.2024.rds")

table(seuratM$RNA_snn_res.0.25)
table(seuratM$patient)
table(seuratM$orig.ident)

```

```{r}

seuratM$clusterName <- "clusterName"
Idents(seuratM) <- seuratM$clusterName
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "0" )] <- "Fb1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "1" )] <- "BEC1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "2" )] <- "PeriFb1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "3" )] <- "Cardiomyocyte"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "4" )] <- "MCP1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "5" )] <- "TC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "6" )] <- "MCP2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "7" )] <- "BEC2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "8" )] <- "VSMC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "9" )] <- "NC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "10" )] <- "INT1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "11" )] <- "Fb2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "12" )] <- "Fb3"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "13" )] <- "PeriFb2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "14" )] <- "AdipoC"


colclusterName <- c("#D53E4F", "#f4a582", "#FEE08B","#feb24c","#67001f", "#01665e","#66C2A5","#c7eae5","#BEAED4", "#355C7D","#3288BD","#8c510a" ,"#fde0dd","#B45B5C","#dd1c77")

                    
names(colclusterName) <- c("Cardiomyocyte","Fb1","Fb2","Fb3", "INT1", "PeriFb1", "PeriFb2", "VSMC","NC", "BEC1","BEC2", "AdipoC", "MCP1","MCP2","TC")

colpat_sub<- c("#dfc27d","#BE3144","#355C7D", "#779d8d")

seuratM$clusterName <- factor(seuratM$clusterName, levels=c("Cardiomyocyte", "Fb1", "Fb2", "Fb3", "INT1","PeriFb1", "PeriFb2", "VSMC", "NC", "BEC1", "BEC2", "AdipoC", "MCP1", "MCP2", "TC"))


Idents(seuratM) <- seuratM$clusterName
DimPlot(seuratM, reduction = "umap", pt.size = 0.1, label = TRUE, cols = colclusterName)

```


```{r}
#convert to sce to change rownames

sce <- as.SingleCellExperiment(seuratM)
#scecko <- as.SingleCellExperiment(seuratcko)
#scewt <- as.SingleCellExperiment(seuratwt)
#rownames(sce) = gsub("\\..*","",rownames(sce))
rownames(sce) = gsub("^.*\\.","",rownames(sce))
#rownames(scecko) = gsub("^.*\\.","",rownames(scecko))
#rownames(scewt) = gsub("^.*\\.","",rownames(scewt))

cellchat <- createCellChat(object = sce, group.by = "clusterName")

CellChatDB <- CellChatDB.human # use CellChatDB.mouse if running on mouse data
showDatabaseCategory(CellChatDB)

# Show the structure of the database
dplyr::glimpse(CellChatDB$interaction)

# use a subset of CellChatDB for cell-cell communication analysis
#CellChatDB.use <- subsetDB(CellChatDB, search = "Secreted Signaling") # use Secreted Signaling

# use all CellChatDB for cell-cell communication analysis
CellChatDB.use <- CellChatDB # simply use the default CellChatDB

# set the used database in the object
cellchat@DB <- CellChatDB.use

```


```{r}
# subset the expression data of signaling genes for saving computation cost
cellchat <- subsetData(cellchat) # This step is necessary even if using the whole database
future::plan("multisession", workers = 4) # do parallel
options(future.globals.maxSize = 1000 * 1024^2)
#> Warning: [ONE-TIME WARNING] Forked processing ('multicore') is disabled
#> in future (>= 1.13.0) when running R from RStudio, because it is
#> considered unstable. Because of this, plan("multicore") will fall
#> back to plan("sequential"), and plan("multiprocess") will fall back to
#> plan("multisession") - not plan("multicore") as in the past. For more details,
#> how to control forked processing or not, and how to silence this warning in
#> future R sessions, see ?future::supportsMulticore
cellchat <- identifyOverExpressedGenes(cellchat)
cellchat <- identifyOverExpressedInteractions(cellchat)
# project gene expression data onto PPI network (optional)
#cellchat <- projectData(cellchat, PPI.human)
```


```{r}
cellchat <- computeCommunProb(cellchat, type =  "triMean")
cellchatH <- computeCommunProb(cellchat, type =  "truncatedMean", trim= 0.1) #set truncated mean to 10% - average gene expression is zero if less than 10% of cell in one group express gene (default is 25%)
# Filter out the cell-cell communication if there are only few number of cells in certain cell groups
# cellchat <- filterCommunication(cellchat, min.cells = 35)

```


```{r}

# Infer the cell-cell communication at a signaling pathway level CellChat computes the communication probability on signaling pathway level by summarizing the communication probabilities of all ligands-receptors interactions associated with each signaling pathway.

cellchat <- computeCommunProbPathway(cellchat)
cellchatH <- computeCommunProbPathway(cellchatH)
cellchat <- aggregateNet(cellchat)
cellchatH <- aggregateNet(cellchatH)

```


```{r}

future.seed=TRUE
cellchat <- netAnalysis_computeCentrality(cellchat, slot.name = "netP") # the slot 'netP' means the inferred intercellular communication network of signaling pathways
cellchatH <- netAnalysis_computeCentrality(cellchatH, slot.name = "netP") # the slot 'netP' means the inferred intercellular communication network of signaling pathways
saveRDS(cellchat, file = "~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/cellchat.rds")
saveRDS(cellchatH, file = "~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/_cellchatH.rds")

```


```{r}

cellchat <- readRDS("~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/cellchat.rds")
cellchatH <- readRDS("~/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/cellchat.rds")


```



```{r Session info}


date()
sessionInfo()

```

