---
title: "T cell abundancance Differential expression"
output: html_document
date: "2024-09-21"
---



```{r setup, include = FALSE}
options (width = 100)
knitr::opts_chunk$set(warning = FALSE, meassage = FALSE, dev = c("png", "pdf"))
seed <- 1234
```

```{r Load library}
suppressPackageStartupMessages({
  library(ExploreSCdataSeurat3)
  library(runSeurat3)
  library(Seurat)
  library(ggpubr)
  library(pheatmap)
  library(SingleCellExperiment)
  library(dplyr)
  library(tidyverse)
  library(viridis)
  library(muscat)
  library(circlize)
  library(destiny)
  library(scater)
  library(metap)
  library(multtest)
  library(clusterProfiler)
  library(org.Hs.eg.db)
  library(msigdbr)
  library(enrichplot)
  library(DOSE)
  library(grid)
  library(gridExtra)
  library(ggupset)
  library(VennDiagram)
  library(NCmisc)
  library(RColorBrewer)
  library(EnhancedVolcano)
  library(textshaping)
  library(wordcloud)
  library(pathview)
  library(workflowr)
})


```


```{r}
##load merged file 
seuratM <- readRDS("/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_by_Tc_condition_metadata_04.10.24.rds")


table(seuratM$dataset)
```


```{r}
table(seuratM$RNA_snn_res.0.25)
table(seuratM$orig.ident)
```


```{r Plot umap}
Idents(seuratM) <- seuratM$RNA_snn_res.0.25
DimPlot(seuratM, reduction = "umap", pt.size = 0.1, label = TRUE)

unique(seuratM$clusterName)
```

```{r}
#### Rename clusters

seuratM$clusterName <- "clusterName"
Idents(seuratM) <- seuratM$clusterName
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "0" )] <- "Fb1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "1" )] <- "BEC1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "2" )] <- "PeriFb1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "3" )] <- "Cardiomyocyte"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "4" )] <- "MCP1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "5" )] <- "TC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "6" )] <- "MCP2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "7" )] <- "BEC2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "8" )] <- "VSMC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "9" )] <- "NC"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "10" )] <- "INT1"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "11" )] <- "Fb2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "12" )] <- "Fb3"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "13" )] <- "PeriFb2"
seuratM$clusterName[which(seuratM$RNA_snn_res.0.25 %in% "14" )] <- "AdipoC"


colclusterName <- c("#D53E4F", "#f4a582", "#FEE08B","#feb24c","#67001f", "#01665e","#66C2A5","#c7eae5","#BEAED4", "#355C7D","#3288BD","#8c510a" ,"#fde0dd","#B45B5C","#dd1c77")

                    
names(colclusterName) <- c("Cardiomyocyte","Fb1","Fb2","Fb3", "INT1", "PeriFb1", "PeriFb2", "VSMC","NC", "BEC1","BEC2", "AdipoC", "MCP1","MCP2","TC")

colpat_sub<- c("#dfc27d","#BE3144","#355C7D", "#779d8d")

seuratM$clusterName <- factor(seuratM$clusterName, levels=c("Cardiomyocyte", "Fb1", "Fb2", "Fb3", "INT1","PeriFb1", "PeriFb2", "VSMC", "NC", "BEC1", "BEC2", "AdipoC", "MCP1", "MCP2", "TC"))


Idents(seuratM) <- seuratM$clusterName
DimPlot(seuratM, reduction = "umap", pt.size = 0.1, label = TRUE, cols = colclusterName)
```


```{r}

#saveRDS(seuratM, file="/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_RDS_04.10.2024.rds")

```



```{r}
###patient
datList <- NULL
for(con in unique(seuratM$patient)){
  seuratSub <- subset(seuratM, patient==con)
  print(dim(seuratSub))
  dat_con <- as.data.frame(table(seuratSub$clusterName)) %>%
  mutate(percent=Freq/ncol(seuratSub)) %>% mutate(patient=con)
  datList[[con]] <- dat_con
}
dat_all <- do.call("rbind", datList)

#View(dat_all)
```


```{r}
###patient_patsub

datList2 <- NULL
for(con in unique(seuratM$patient_pat_sub)){
  seuratSub <- subset(seuratM, patient_pat_sub==con)
  print(dim(seuratSub))
  dat_con <- as.data.frame(table(seuratSub$clusterName)) %>%
  mutate(percent=Freq/ncol(seuratSub)) %>% mutate(patient_pat_sub=con)
  datList2[[con]] <- dat_con
}
dat_all2 <- do.call("rbind", datList2)

library(stringr)
# Split patient_pat_sub into patient and pat_sub 
dat_all2 <- as.data.frame(dat_all2) 
 
dat_all2[c('patient', 'pat_sub')] <- str_split_fixed(dat_all2$patient_pat_sub, '_', 2)

# View(dat_all2)
 
#write.table(dat_all2, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/dat_all2_abundance by condition_4-10%_04.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

```




```{r subgroup Tcell conditions}
# Filter "TC" cluster
dat_all2 <- filter(dat_all2, Var1 == "TC")

# New column
dat_all2$Tgrp <- NA  

# T cell categories
dat_all2$Tgrp[dat_all2$percent > 0.1] <- "Tcell High"
dat_all2$Tgrp[dat_all2$percent < 0.1 & dat_all2$percent >= 0.05] <- "Tcell Intermediate"
dat_all2$Tgrp[dat_all2$percent < 0.05 & dat_all2$percent >= 0.02] <- "Tcell Low"
dat_all2$Tgrp[dat_all2$percent < 0.02] <- "Tcell Very Low"

table(dat_all2$Tgrp)

#write.table(dat_all2, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/dat_all2_con_pat_Tc highvs.low_04.10.2024_final", sep="\t",quote=F,row.names=F,col.names=T)

ggplot(dat_all2, aes(x = Tgrp, y = percent, fill = Tgrp)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.5) +  
  geom_jitter(size = 0.8, width = 0.2, aes(color = Tgrp)) +  
  theme_classic() +
  ggtitle("Frequencies of T cells") +
  xlab("T cell groups") +
  ylab("T Cell Frequency") +
  stat_compare_means(method = "kruskal", label = "p.signif", hide.ns = FALSE) +
  stat_compare_means(label = "p.format", comparisons = list(c("Tcell High", "Tcell Intermediate", "Tcell Low", "T cell Very Low")), method = "kruskal")  

## with cutoffs high >10%, intermediate 4-10%, low 2-4%, very low <2% -> 
# N patients: high N = 5, intermediate N = 6, low N = 10, very low N = 20 
# N nuclei: high = 9034, intermediate = 10840, low = 18066 , very low = 36130 


## with cutoffs high >10%, intermediate 5-10%, low 2-5%, very low <2% -> 
# N patients: high N = 5, intermediate N = 4, low N = 12, very low N = 20 
# N nuclei: high = 9034, intermediate = 7227, low = 21679 , very low = 36130 

#View(dat_all2)

```





```{r add data2 to meta.data}

#seuratM$clusterName <- dat_all2$Var1

#seuratM@meta.data <- dat_all2

# Factor in meta.data with TC condition
seuratM$TC <- dat_all2$Tgrp
Idents(seuratM) <- seuratM$TC
TC_condition <- seuratM$TC
Idents(seuratM) <- seuratM@meta.data$TC

table(seuratM$TC)
TC_condition <- c("Tcell Very High", "Tcell High", "Tcell Intermediate", "Tcell Low", "Tcell Very Low")

seuratM$pat_TC <- paste0(dat_all2$Tgrp, '_', dat_all2$patient)

seuratM$pat_percent_TC <- paste0(dat_all2$percent, '_', dat_all2$patient)

seuratM$pat_sub_TC <- paste0(dat_all2$Tgrp, '_', dat_all2$pat_sub)

seuratM$pat_sub_percent_TC <- paste0(dat_all2$percent, '_', dat_all2$pat_sub)

seuratM$TC <- paste0(dat_all2$Tgrp)


```
```{r}
table(seuratM$pat_TC)
```


```{r relative abundance T cells per patient}
table(seuratM$pat_percent_TC)
```


```{r Nuclei number per disease and Tcell condition}
table(seuratM$pat_sub_TC)
```


```{r relative abundance of T cells per disease condition}
table(seuratM$pat_sub_percent_TC)

```

```{r add condition to list}
Idents(seuratM) <- seuratM$TC

colTC <- c("#BE3144","#de2d26","#fc9272", "#fee0d2")
names(colTC) <- c("Tcell High", "Tcell Intermediate", "Tcell Low", "Tcell Very Low")

seuratM$TC <- factor(seuratM$TC, levels = c("Tcell High", "Tcell Intermediate", "Tcell Low", "Tcell Very Low"))

DimPlot(seuratM, reduction = "umap", pt.size = 0.1, group.by = "pat_sub_TC")

DimPlot(seuratM, reduction = "umap", pt.size = 0.1, group.by = "TC", cols = colTC)

DimPlot(seuratM, reduction = "umap", pt.size = 0.1, group.by = "pat_sub")



Idents(seuratM) <- seuratM$clusterName
DimPlot(seuratM, reduction = "umap", pt.size = 0.1, label = TRUE, cols = colclusterName)
```






```{r}

Idents(seuratM) <- seuratM$TC
orderTC <- c("Tcell Very High", "Tcell High", "Tcell Intermediate", "Tcell Low", "Tcell Very Low")

DimPlot(seuratM, reduction = "umap", pt.size = 0.1, group.by = "TC", cols = colTC)


seuratTC_high <- subset(seuratM, TC == "Tcell High")
DimPlot(seuratTC_high, reduction = "umap", pt.size = 0.1, cols = colTC, raster=FALSE)

seuratTC_intermediate <- subset(seuratM, TC == "Tcell Intermediate")
DimPlot(seuratTC_intermediate, reduction = "umap", pt.size = 0.1, cols = colTC, raster=FALSE)

seuratTC_low <- subset(seuratM, TC == "Tcell Low")
DimPlot(seuratTC_low, reduction = "umap", pt.size = 0.1, cols = colTC, raster=FALSE)

seuratTC_verylow <- subset(seuratM, TC == "Tcell Very Low")
DimPlot(seuratTC_verylow, reduction = "umap", pt.size = 0.05, cols = colTC, raster=FALSE)

### label split by T cell Grp

DimPlot(seuratM, reduction = "umap", group.by = "clusterName", cols=colclusterName,
        split.by = "TC", shuffle = T)+
  theme_void()


```





```{r Cellcount per Tcell condition}


seuratTC_vhigh_vs_vlow <- subset(seuratM, TC %in% c("Tcell Very High", "Tcell Very Low"))

seuratTC_high_vs_vlow <- subset(seuratM, TC %in% c("Tcell High", "Tcell Very Low"))


levels(seuratTC_high_vs_vlow)
Idents(seuratTC_high_vs_vlow) <- seuratTC_high_vs_vlow$clusterName

table(seuratTC_high_vs_vlow$orig.ident)
# 45164 nuclei for Tcell High vs. Very Low

DimPlot(seuratTC_high_vs_vlow, reduction = "umap", pt.size = 0.1, raster=FALSE, cols = colclusterName, label = T)

cellcount_TC_high_vs_vlow <- data.frame(table(seuratTC_high_vs_vlow$RNA_snn_res.0.25))
colnames(cellcount_TC_high_vs_vlow) <- c("clusterName", "Freq")
hsize <- 1.5

ggplot(cellcount_TC_high_vs_vlow, aes(x = clusterName, y = Freq, fill = clusterName)) +
  geom_col(color = "white") + 
  theme_classic() +  
  ggtitle("Cell Number by Cluster") +
  theme(plot.title = element_text(hjust = 0.5, size = 15)) +
  geom_text(aes(label = Freq), vjust = -0.3, size = 3) +  
  xlab("Cluster Names") +  
  ylab("Frequency (Cell Count)")  

#cell count per patient, greatest contribution
cellcount_TC_high_vs_vlow_pat <- data.frame(table(seuratTC_high_vs_vlow$patient))
colnames(cellcount_TC_high_vs_vlow_pat) <- c("patient", "Freq")
hsize <- 1.5

ggplot(cellcount_TC_high_vs_vlow_pat, aes(x = patient, y = Freq, fill = patient)) +
  geom_col(color = "white") +  
  theme_classic() +  
  ggtitle("Cell Number by Patient") +
  theme(plot.title = element_text(hjust = 0.5, size = 15)) +
  theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.5)) +
  geom_text(aes(label = Freq), vjust = -0.3, size = 3) +  
  xlab("Patients") +  
  ylab("Frequency (Cell Count)")  

#cell count per condition
cellcount_TC_high_vs_vlow_con <- data.frame(table(seuratTC_high_vs_vlow$pat_sub))
colnames(cellcount_TC_high_vs_vlow_con) <- c("Condition", "Freq")
hsize <- 1.5

ggplot(cellcount_TC_high_vs_vlow_con, aes(x = hsize, y = Freq, fill = Condition)) +
  #scale_fill_manual(values = colpat2) +
  geom_col(color = "white") +
  coord_polar(theta = "y") +
  xlim(c(0.2, hsize + 0.5)) +
  theme_void() +
  ggtitle("Tcell High vs. Very Low per condition") +
  theme(plot.title = element_text(hjust = 0.5, size = 15)) +
  geom_text(aes(label = Freq), position = position_stack(vjust = 0.5))

# Healthy 17250, AM 17730, CMP 7095, Sarcoidosis 3089, highest nucleus number patient Healthy 04

```







```{r Subset Macrophage 2 cluster)}

seuratMCP2 <- subset(seuratM, clusterName %in% "MCP2")
table(seuratMCP2$clusterName)

Idents(seuratMCP2) <- seuratMCP2$TC

DimPlot(seuratMCP2, reduction = "umap")

levels(seuratMCP2)

table(seuratMCP2$TC)

# for MCP2 Tcell High N = 447, Tcell Intermediate N = 372, Tcell Low = 1098, Tcell Very Low = 1806 

DEGenesMCP2_adj <- FindAllMarkers(seuratMCP2, only.pos = T, logfc.threshold = 0.2) %>% 
  dplyr::filter(p_val_adj < 0.05)


head(DEGenesMCP2_adj)

DEGenesMCP2 <- FindAllMarkers(seuratMCP2, only.pos = T, logfc.threshold = 0.2) 

head(DEGenesMCP2)

#View(DEGenesMCP2)

#write.table(DEGenesMCP2, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/DEGenesMCP2_all markers for all TC conditions 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

# Notes: MCP2 enriched genes for Tcell High condition:


#LINCO1460: RNA gene
# Y-RNA:RNA gene
# SERP2: SERP2 (Stress Associated Endoplasmic Reticulum Protein Family Member 2).  Protects unfolded target proteins against degradation during ER stress
# GOLGA8K: Golgi organization

# non significant according to adjusted p value: 

#ST7-OT4 (ST7 Overlapping Transcript 4) is an RNA Gene, and is affiliated with the lncRNA class. Diseases associated with ST7-OT4 include Type C Thymoma.
# GALNT5 catalyzes the first step in the mucin-type O-glycosylation of Golgi proteins
# BNIP1 This gene is a member of the BCL2/adenovirus E1B 19 kd-interacting protein (BNIP) family. ER stress associated apoptosis
# SLC5A2: sodium sugar transporter
# LRRC37A3: encodes membrane protein
# GPX1 Catalyzes the reduction of hydroperoxides in a glutathione-dependent manner thus regulating cellular redox homeostasis
# NUP88 nuclear protein comples
# SP100: tumor suppressor, also plays a role for virus induced infections (CMV, EBV)
# GTF2IRD1: regeneration of muscle
# MST1R: receptor for  macrophage-stimulating protein (MSP) with tyrosine kinase activity.
# ZFPM1: Zink finger protein: erythroid and megakaryocytic cell differentiation, GATA pathway
# LCP1: Plays a role in the activation of T cells in response to costimulation through TCR/CD3 and CD2 or CD28
# COL4A3
# TTYH1
# PPIA associated with cyclosporin A-mediated immunosuppression.
# TP 73 Participates in the apoptotic response to DNA damage. Isoforms containing the transactivation domain are pro-apoptotic, isoforms lacking the domain are anti-apoptotic and block the function of p53 and transactivating p73 isoforms. May be a tumor suppressor protein. Is an activator of FOXJ1 expression (By similarity)
# CLEC12B (C-Type Lectin Domain Family 12 Member B): NK negative regulation of natural killer cell mediated cytotoxicity; and negative regulation of signaling receptor activity
# CD46 Acts as a cofactor for complement factor I, costimulatory factor for T cells --> induces the differentiation of CD4+ into T regulatory 1 cells
# NFIC: adenovirus replication
# FPR1: High affinity receptor for N-formyl-methionyl peptides (fMLP), which are powerful neutrophil chemotactic factors
# LACTB
#IL2RG and CCRL2, NFKB1
# C1orf127 Predicted to be involved in heart development.
# CD247; IL6R, CCRL2

#  TC intermediate
# IRF2BP1, VWA8, APOL4,MAP3K10
# osmotic stress SLC12A6/XRCC6/MAP7
# cell adhesion integrin TESC/ADAM9/ITGAE


## TC Low
# CCL19/IL1R1/BCL3
# CLPTM1 (T cell development)/ZC3H8 (repressor GATA3 Induces thymocyte apoptosis when overexpressed, which may indicate a role in regulation of thymocyte homeostasis)

# TC Very Low
# TNFSF13, WDFY1 (TLR4 regulation), NFKBID, WFIKKN1/SUDS3, OAS2 (IFNbeta regulation), STMN3/KPNB1
# ZFPM1/CD46/PIK3CD/IL2RG/IL6R/FANCD2
# CRB2 positive regulation of BMP pathway
```


```{r Inflammatory Macrophages (MCP2) cluster for Tcell High}


DEGenesMCP2 <- DEGenesMCP2 %>% 
  mutate(Gene=gene) %>%
  mutate(gene=gsub("^.*\\.", "", Gene))  %>%
  mutate(EnsID=gsub("\\..*","", Gene))

##GSEA Macrophage type 2 (inflammatory)
DEGenes_MCP2_TChigh <- DEGenesMCP2 %>% filter(cluster == "Tcell High")
egoMCP2 <- enrichGO(gene = unique(DEGenes_MCP2_TChigh$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)
egoMCP2 <- setReadable(egoMCP2, OrgDb = org.Hs.eg.db)

#write.table(egoMCP2@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/egoMCP2 for TC high 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)


summary(egoMCP2@result$p.adjust)
summary(egoMCP2@result$pvalue)

egoMCP2_df <- as.data.frame(egoMCP2@result)

egoMCP2_df <- egoMCP2_df %>%
  arrange(desc(zScore))

ggplot(egoMCP2_df[1:12, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Macrophage type 2 Differential Expression by Z-score for T cell High Condition", y = "Z-score", x = "Description")

# redundant gene ontology description
description_mapping <- c("negative regulation of B cell mediated immunity" = "negative regulation of humoral immune response mediated by circulating immunoglobulin",
  "negative regulation of humoral immune response" = "negative regulation of humoral immune response mediated by circulating immunoglobulin",
  "negative regulation of immunoglobulin mediated immune response" = "negative regulation of humoral immune response mediated by circulating immunoglobulin",
  "negative regulation of complement activation" = "regulation of complement activation",
  "regulation of regulatory T cell differentiation" = "regulatory T cell differentiation",
  "lymphocyte differentiation" = "T cell differentiation",
  "lymphocyte activation involved in immune response" = "T cell activation involved in immune response",
  "regulation of T cell differentiation"  = "T cell differentiation",
  "T cell differentiation involved in immune response" = "T cell differentiation",
  "CD4-positive or CD8-positive, alpha-beta T cell lineage commitment" = "alpha-beta T cell lineage commitment",
   "sphingosine biosynthetic process" = "spingoid metabolic process",   
   "sphingoid biosynthetic process" = "spingoid metabolic process",
  "T-helper cell lineage commitment" = "alpha-beta T cell lineage commitment")


egoMCP2@result <- egoMCP2@result %>%
  mutate(Description = case_when(
    Description == "negative regulation of humoral immune response mediated by circulating immunoglobulin" ~ "negative regulation of humoral immune response",
    Description == "negative regulation of B cell mediated immunity" ~ "negative regulation of humoral immune response",
    Description == "negative regulation of immunoglobulin mediated immune response" ~ "negative regulation of humoral immune response",
    Description == "negative regulation of complement activation" ~ "regulation of complement activation",
    Description == "regulation of regulatory T cell differentiation" ~ "regulatory T cell differentiation",
    Description == "lymphocyte differentiation" ~ "T cell differentiation",
    Description == "cell activation involved in immune response" ~ "leukocyte activation involved in immune response",
    Description == "lymphocyte activation involved in immune response" ~ "T cell activation involved in immune response",
    Description == "CD4-positive or CD8-positive, alpha-beta T cell lineage commitment" ~ "alpha-beta T cell lineage commitment" ,
    Description == "regulation of T cell differentiation"  ~ "T cell differentiation",
    Description == "sphingosine biosynthetic process" ~ "spingoid metabolic process",   
    Description == "sphingoid biosynthetic process" ~ "spingoid metabolic process",
    Description == "T-helper cell lineage commitment" ~ "alpha-beta T cell lineage commitment",
    TRUE ~ Description
  ))

# NA values
na_rows <- egoMCP2@result %>%
  filter(is.na(Description))

egoMCP2@result$Description <- replace(
  egoMCP2@result$Description, 
  egoMCP2@result$Description %in% names(description_mapping), 
  description_mapping[egoMCP2@result$Description])

#print(egoMCP2@result$Description)

egoMCP2_revised <- egoMCP2@result %>%
  group_by(Description) %>%
  summarise(
    Count = sum(Count),
    GeneRatio = sum(as.numeric(GeneRatio)),
    p.adjust = min(p.adjust),  
    .groups = 'drop'
  )

egoMCP2_revised <- egoMCP2_revised %>%
  mutate(Zscore = (Count - mean(Count)) / sd(Count))

#View(egoMCP2_revised)


egoMCP2_df_revised <- egoMCP2_revised 
ggplot(egoMCP2_df_revised[1:20, ], aes(x = reorder(Description, Zscore), y = Zscore)) +
  geom_bar(stat = "identity", aes(fill = Zscore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(
    title = "Macrophage Type 2 Differential Expression by Z-score for T Cell High Condition", 
    y = "Z-score", 
    x = "Description")


# NA 
na_rows <- egoMCP2@result %>%
  filter(is.na(Description))

```


```{r Inflammatory Macrophages (MCP2) cluster for Tcell Intermediate}

DEGenes_MCP2_TCint <- DEGenesMCP2 %>% filter(cluster == "Tcell Intermediate")
egoMCP2_int <- enrichGO(gene = unique(DEGenes_MCP2_TCint$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)
egoMCP2_int <- setReadable(egoMCP2_int, OrgDb = org.Hs.eg.db)

#View(egoMCP2_int@result)

#write.table(egoMCP2_int@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoMCP2 for TC intermediate 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

egoMCP2_int_df <- as.data.frame(egoMCP2_int@result)
ggplot(egoMCP2_int@result[1:10, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Macrophage type 2 Differential Expression by Z-score for T cell Intermediate Condition", y = "Z-score", x = "Description")



```



```{r Inflammatory Macrophages (MCP2) cluster for Tcell Low}

DEGenes_MCP2_TClow <- DEGenesMCP2 %>% filter(cluster == "Tcell Low")
egoMCP2_low <- enrichGO(gene = unique(DEGenes_MCP2_TClow$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)
egoMCP2_low <- setReadable(egoMCP2_low, OrgDb = org.Hs.eg.db)

#View(egoMCP2_low@result)

# write.table(egoMCP2_low@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoMCP2 for TC low 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

egoMCP2_low_df <- as.data.frame(egoMCP2_low@result)
ggplot(egoMCP2_low_df[1:7, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Macrophage type 2 Differential Expression by Z-score for T cell Intermediate Condition", y = "Z-score", x = "Description")

```


```{r Inflammatory Macrophages (MCP2) cluster for Tcell Very Low}

DEGenes_MCP2_TCvlow <- DEGenesMCP2 %>% filter(cluster == "Tcell Very Low")
egoMCP2_vlow <- enrichGO(gene = unique(DEGenes_MCP2_TCvlow$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)
egoMCP2_vlow <- setReadable(egoMCP2_vlow, OrgDb = org.Hs.eg.db)

#View(egoMCP2_vlow@result)

#write.table (egoMCP2_vlow@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoMCP2 for TC vlow 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

egoMCP2_vlow_df <- as.data.frame(egoMCP2_vlow@result)
#by zScore
ggplot(egoMCP2_vlow_df[1:8, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Macrophage type 2 Differential Expression by Z-score for T cell Intermediate Condition", y = "Z-score", x = "Description")

```



```{r Subset Cardiomyocyte Cluster for DE and GSEA}

seuratCM <- subset(seuratM, clusterName %in% "Cardiomyocyte")
table(seuratCM$clusterName)
table(seuratCM$patient)
table(seuratCM$pat_sub)

Idents(seuratCM) <- seuratCM$TC

DimPlot(seuratCM, reduction = "umap")

levels(seuratCM)

table(seuratCM$TC)


DEGenesCM <- FindAllMarkers(seuratCM, only.pos = T, logfc.threshold = 0.2) 

head(DEGenesCM)

#View(DEGenesCM)

#write.table (DEGenesCM, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DE Genes CM for all TCell conditions", sep="\t",quote=F,row.names=F,col.names=T)

```

```{r Cardiomyocyte T cell High Condition}

##adjust table
DEGenesCM <- DEGenesCM %>% 
  mutate(Gene=gene) %>%
  mutate(gene=gsub("^.*\\.", "", Gene))  %>%
  mutate(EnsID=gsub("\\..*","", Gene))

##GSEA Cardiomyocyte
DEGenes_CM_TChigh <- DEGenesCM %>% filter(cluster == "Tcell High")
egoCM <- enrichGO(gene = unique(DEGenes_CM_TChigh$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)

egoCM <- setReadable(egoCM, OrgDb = org.Hs.eg.db)

#write.table(egoCM@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoCM for TC high 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

#print(egoCM@result)

### Gene Ontology results for T cell High Condition

# mitosis, replication
# response IL-7 and repair: STIP1/CRKL/IL7R/IL2RG/P4HB/RAD23B/GIPC1/ATIC/FOSL2/PDIA3/BTK
# alpha beta T cell differentation: RIPK2/CD3E/TGFBR2/LGALS9/CD81/CD55 STAT4/LEF1/GATA3/RHOA/RIPK2/HMGB1/KMT2A/RC3H2/SLC4A2/SOCS1/ATP7A/JUNB/ZBTB16/TGFBR2/ENTPD7/MALT1/ITPKB/NFKBIZ/ZFPM1/IFNG/IL2RG/LY9/FOXP3/LGALS9/BCL3/RARA/ARMC5/SMAD7/TNFSF8/BRD2/ADA/JAK1/IL18/CD83/IL18R1/RPL22/SOCS5/PNP/FOSL2/BCL6/TNFSF4/SYK/GPR18/CTSL
# Response IL-12: STAT4/RIPK2/JAK2/P4HB/TYK2/IL12RB2/PLCB1
# somatic recombination and immune responses: STAT4/LEF1/GATA3/RHOA/RIPK2/HMGB1/KMT2A/RC3H2/SLC4A2/SOCS1/ATP7A/JUNB/ZBTB16/TGFBR2/ENTPD7/MALT1/ITPKB/NFKBIZ/ZFPM1/IFNG/IL2RG/LY9/FOXP3/LGALS9/BCL3/RARA/ARMC5/SMAD7/TNFSF8/BRD2/ADA/JAK1/IL18/CD83/IL18R1/RPL22/SOCS5/PNP/FOSL2/BCL6/TNFSF4/SYK/GPR18/CTSL
# interleukin-15-mediated signaling pathway: IL2RB/IL15/IL2RG/JAK1/PLCB1/IL15RA
# Tcell extravasation IL27RA/CD99L2/CRKL/F11R/MED23/FADD/RIPK3/ICAM1

summary(egoCM@result$p.adjust)
summary(egoCM@result$pvalue)


egoCM_df <- as.data.frame(egoCM@result)
egoCM_df <- egoCM_df %>%
  arrange(desc(zScore))

#by zScore
ggplot(egoCM_df[1:20, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Cardiomyocyte Differential Expression by zScore for T cell High Condition", y = "z Score", x = "Description")


egoCM_df <- egoCM_df %>%
  arrange(desc(RichFactor))

# By RichFactor
ggplot(egoCM_df[1:21, ], aes(x = reorder(Description, RichFactor), y = RichFactor)) +
  geom_bar(stat = "identity", aes(fill = RichFactor)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Cardiomyocyte Differential Expression by Rich Factor for T cell High Condition", y = "RichFactor", x = "Description")
```


`
```{r Cardiomyocyte Cluster T cell Intermediate}

##GSEA Cardiomyocyte
DEGenes_CM_TCint <- DEGenesCM %>% filter(cluster == "Tcell Intermediate")
egoCM_int <- enrichGO(gene = unique(DEGenes_CM_TCint$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)

egoCM_int <- setReadable(egoCM_int, OrgDb = org.Hs.eg.db)
head(egoCM_int)

#View(egoCM_int@result)

#write.table(egoCM_int@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoCM for TC int 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

# Fc receptor mediated stimulatory signaling pathway:NR4A3/CD47
# positive regulation of cardiac muscle hypertrophy: 	NR4A3/MTPN
# IL-12 response: CD47
# chemokine (C-C motif) ligand 5 production, toll-like receptor 7 signaling pathway: DDX3X
# positive regulation of monocyte extravasation: CD47
# regulation of CD8-positive, alpha-beta T cell differentian: SOCS1:Essential negative regulator of type I and type II interferon (IFN) signaling
# respiratory burst involved in defense response: RPS19

egoCM_int_df <- as.data.frame(egoCM_int@result)

egoCM_int_df <- egoCM_int_df %>%
 arrange(desc(zScore))

# by zScore
ggplot(egoCM_int_df[1:21, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Cardiomyocyte Differential Expression by zScore for T cell Intermediate Condition", y = "z Score", x = "Description")


egoCM_int_df <- egoCM_int_df %>%
  arrange(desc(RichFactor))

#By RichFactor
ggplot(egoCM_int_df[1:21, ], aes(x = reorder(Description, RichFactor), y = RichFactor)) +
  geom_bar(stat = "identity", aes(fill = RichFactor)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Cardiomyocyte Differential Expression by Rich Factor for T cell Intermediate Condition", y = "RichFactor", x = "Description")


```


```{r Cardiomyocyte Cluster T cell Low Condition}
##GSEA Cardiomyocyte T cell low

DEGenes_CM_TClow <- DEGenesCM %>% filter(cluster == "Tcell Low")
egoCM_low <- enrichGO(gene = unique(DEGenes_CM_TClow$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)

egoCM_low <- setReadable(egoCM_low, OrgDb = org.Hs.eg.db)
head(egoCM_low)

#write.table(egoCM_low@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoCM for TC low 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)


#View(egoCM_low@result)

# response HGF: GCLC
# positive regulation of T cell activation: NCK1
# response to transforming growth factor beta: INTS9
# B cell mediated immunity: FCMR
# positive regulation of canonical NF-kappaB signal transduction: EDNRA


egoCM_low_df <- as.data.frame(egoCM_low@result)


egoCM_low_df <- egoCM_low_df %>%
 arrange(desc(zScore))

# by zScore
ggplot(egoCM_low_df[1:21, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Cardiomyocyte Differential Expression by zScore for T cell Low Condition", y = "z Score", x = "Description")


egoCM_low_df <- egoCM_low_df %>%
  arrange(desc(RichFactor))

#by RichFactor
ggplot(egoCM_low_df[1:21, ], aes(x = reorder(Description, RichFactor), y = RichFactor)) +
  geom_bar(stat = "identity", aes(fill = RichFactor)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Cardiomyocyte Differential Expression by Rich Factor for T cell Low Condition", y = "RichFactor", x = "Description")

```


```{r Cardiomyocyte Cluster T cell Very Low Condition}


##GSEA Cardiomyocyte T cell Very Low
DEGenes_CM_TCvlow <- DEGenesCM %>% filter(cluster == "Tcell Very Low")
egoCM_vlow <- enrichGO(gene = unique(DEGenes_CM_TCvlow$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)

egoCM_vlow <- setReadable(egoCM_vlow, OrgDb = org.Hs.eg.db)
head(egoCM_vlow)

#write.table(egoCM_vlow@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoCM for TC very low 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)


#View(egoCM_vlow@result)

# positive regulation of dendritic cell cytokine production, positive regulation of T-helper 17: CLEC7A/PLCG2
# regulation of cellular response to osmotic stress: MICU1/CLN3
# phagosome-lysosome fusion: SPG11/CLN3. 
# positive regulation of interleukin-2 production: CLEC7A/PLCG2
# positive regulation of granulocyte differentiation:	HCLS1
# ERBB3 signaling pathway: RTN4
# NK T cell differentiation: ATF2
# positive regulation of interleukin-18 production: CASP1
# viral life cycle SMPD1

egoCM_vlow_df <- as.data.frame(egoCM_vlow@result)


egoCM_vlow_df <- egoCM_vlow_df %>%
 arrange(desc(zScore))

ggplot(egoCM_vlow_df[1:21, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Cardiomyocyte Differential Expression by zScore for T cell Very Low Condition", y = "z Score", x = "Description")


egoCM_vlow_df <- egoCM_vlow_df %>%
  arrange(desc(RichFactor))

ggplot(egoCM_low_df[1:21, ], aes(x = reorder(Description, RichFactor), y = RichFactor)) +
  geom_bar(stat = "identity", aes(fill = RichFactor)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Cardiomyocyte Differential Expression by Rich Factor for T cell Very Low Condition", y = "RichFactor", x = "Description")

```


```{r}
seuratTcell <- subset(seuratM, clusterName %in% "TC")
table(seuratTcell$clusterName)
table(seuratTcell$patient)
table(seuratTcell$pat_sub)

Idents(seuratTcell) <- seuratTcell$TC

DimPlot(seuratTcell, reduction = "umap")

levels(seuratTcell)

table(seuratTcell$TC)

# for Tcell cluster TC High N = 608, Tcell Intermediate N = 468, Tcell Low = 1385, Tcell Very Low = 2375; N = 4836

DEGenesTC <- FindAllMarkers(seuratTcell, only.pos = T, logfc.threshold = 0.2) 
head(DEGenesTC)

DEGenesTC_adj <- FindAllMarkers(seuratTcell, only.pos = T, logfc.threshold = 0.2) %>% 
  dplyr::filter(p_val_adj < 0.05)
head(DEGenesTC_adj)


#View(DEGenesTC)

#write.table (DEGenesTC, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/DE Genes Tcell cluster for all TCell conditions", sep="\t",quote=F,row.names=F,col.names=T)

# TKT, KRBOX4, SCAMP5 (positive regulation of cytokine production), ZNF492, HOMER2(dendritic proteins), UBB (activation of the transcription factor NF-kappa-B), CXCL16 (responds to IFNy)

```

```{r T cell cluster: Tcell High condition}
##adjust table
DEGenesTC <- DEGenesTC %>% 
  mutate(Gene=gene) %>%
  mutate(gene=gsub("^.*\\.", "", Gene))  %>%
  mutate(EnsID=gsub("\\..*","", Gene))

##GSEA Tcell
DEGenesTC_high <- DEGenesTC %>% filter(cluster == "Tcell High")
egoTC <- enrichGO(gene = unique(DEGenesTC_high$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)

egoTC <- setReadable(egoTC, OrgDb = org.Hs.eg.db)

#write.table(egoTC@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoTC for TC high 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)
#View(egoTC@result)

# ERBB3 signaling pathway MAPK1, TP53/MECP2
# BCL10/TP53/RAPGEF2

egoTC_df <- as.data.frame(egoTC@result)


egoTC_df <- egoTC_df %>%
 arrange(desc(zScore))

ggplot(egoTC_df [1:20, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "T cell Differential Expression by zScore for T cell High Condition", y = "z Score", x = "Description")


egoTC_df <- egoTC_df %>%
  arrange(desc(RichFactor))

ggplot(egoTC_df[1:21, ], aes(x = reorder(Description, RichFactor), y = RichFactor)) +
  geom_bar(stat = "identity", aes(fill = RichFactor)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "T cell Differential Expression by Rich Factor for T cell High Condition", y = "RichFactor", x = "Description")

```

```{r T cell cluster: Tcell Intermediate condition}
##GSEA Tcell
DEGenesTC_int <- DEGenesTC %>% filter(cluster == "Tcell Intermediate")
egoTC_int <- enrichGO(gene = unique(DEGenesTC_int$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)

egoTC_int <- setReadable(egoTC_int, OrgDb = org.Hs.eg.db)

# View(egoTC_int@result)

#write.table(egoTC_int@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoTC for TC intermediate 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)


# lymphocyte differentiation: NRARP/HDAC9/PPP2R3C/CD3D/HMGB1/TCIRG1
# positive regulation of ERK1 and ERK2 cascade:GPBAR1/MFHAS1/CIB1/HMGB1/ARRB1
# MAP2/MARK1/UBA6/NCK2
# T cell differentiation: NRARP/CD3D/HMGB1/TCIRG1
# NRARP/PPP2R3C/LRRC17/HMGB1
# Wnt signaling pathway: NRARP/MARK1/MKS1/TAX1BP3
# negative regulation of Notch signaling pathway: NRARP/NIBAN2/ARRB1
# B cell diff: HDAC9/PPP2R3C/TCIRG1
# BBC3/NCK2/NACC2
# BMPR2/MKS1
# Tcell homeostasis PPP2R3C/TCIRG1
# negative regulation NFKB RWDD3/ARRB1


egoTC_df <- as.data.frame(egoTC_int@result)


egoTC_int_df <- egoTC_df %>%
 arrange(desc(zScore))

ggplot(egoTC_int_df[1:21, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "T cell Differential Expression by zScore for T cell Intermediate Condition", y = "z Score", x = "Description")


egoTC_int_df <- egoTC_int_df %>%
  arrange(desc(RichFactor))

ggplot(egoTC_int_df[1:21, ], aes(x = reorder(Description, RichFactor), y = RichFactor)) +
  geom_bar(stat = "identity", aes(fill = RichFactor)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "T cell Differential Expression by Rich Factor for T cell Intermediate Condition", y = "RichFactor", x = "Description")
```



```{r T cell cluster: Tcell Low condition}
##GSEA Tcell

DEGenesTC_low <- DEGenesTC %>% filter(cluster == "Tcell Low")
egoTC_low <- enrichGO(gene = unique(DEGenesTC_low$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)

egoTC_low <- setReadable(egoTC_low, OrgDb = org.Hs.eg.db)

#View(egoTC_low@result)

#write.table(egoTC_low@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoTC for TC low 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

egoTC_df <- as.data.frame(egoTC_low@result)


egoTC_low_df <- egoTC_df %>%
 arrange(desc(zScore))

ggplot(egoTC_low_df[1:21, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "T cell Differential Expression by zScore for T cell Low Condition", y = "z Score", x = "Description")


egoTC_low_df <- egoTC_low_df %>%
  arrange(desc(RichFactor))

ggplot(egoTC_low_df[1:21, ], aes(x = reorder(Description, RichFactor), y = RichFactor)) +
  geom_bar(stat = "identity", aes(fill = RichFactor)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "T cell Differential Expression by Rich Factor for T cell Low Condition", y = "RichFactor", x = "Description")

```

```{r T cell cluster: Tcell Very Low condition}

##GSEA Tcell

DEGenesTC_vlow <- DEGenesTC %>% filter(cluster == "Tcell Very Low")
egoTC_vlow <- enrichGO(gene = unique(DEGenesTC_vlow$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)

egoTC_vlow <- setReadable(egoTC_vlow, OrgDb = org.Hs.eg.db)

#View(egoTC_vlow@result)

# Tcell selection: SRF/RHOA
# RNMT/MAT2A
# SRF Together with MRTFA transcription coactivator, controls expression of genes regulating the cytoskeleton during development, morphogenesis and cell migration.
# regulation of canonical NF-kappaB signal transduction: RHOA/ZMYND11/ZFAND6 (Involved in regulation of TNF-alpha induced NF-kappa-B activation and apoptosis.)
# leukemia inhibitory factor RNMT/MAT2A

#write.table(egoTC_vlow@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoTC for TC very low 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

egoTC_df <- as.data.frame(egoTC_vlow@result)


egoTC_vlow_df <- egoTC_df %>%
 arrange(desc(zScore))

ggplot(egoTC_vlow_df[1:21, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "T cell Differential Expression by zScore for T cell Very Low Condition", y = "z Score", x = "Description")


egoTC_vlow_df <- egoTC_vlow_df %>%
  arrange(desc(RichFactor))

ggplot(egoTC_vlow_df[1:21, ], aes(x = reorder(Description, RichFactor), y = RichFactor)) +
  geom_bar(stat = "identity", aes(fill = RichFactor)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "T cell Differential Expression by Rich Factor for T cell Very Low Condition", y = "RichFactor", x = "Description")




```


```{r Fb Clusters stratified by Tcell condition}

seuratFb <- subset(seuratM, clusterName %in% c("Fb1", "Fb2", "Fb3"))
table(seuratFb$clusterName)
table(seuratFb$patient)
table(seuratFb$pat_sub)

Idents(seuratFb) <- seuratFb$TC

DimPlot(seuratFb, reduction = "umap")

levels(seuratFb)

table(seuratFb$TC)

# for Fb cluster TC High N = 2311, Tcell Intermediate N = 1835, Tcell Low = 5461, Tcell Very Low = 9166;

DEGenesFb <- FindAllMarkers(seuratFb, only.pos = T, logfc.threshold = 0.2) 
head(DEGenesTC)

DEGenesFb_adj <- FindAllMarkers(seuratFb, only.pos = T, logfc.threshold = 0.2) %>% 
  dplyr::filter(p_val_adj < 0.05)
head(DEGenesFb_adj)
```


```{r Activated likely inflammatory Fb Clusters}


DEGenesFb <- DEGenesFb %>% 
  mutate(Gene=gene) %>%
  mutate(gene=gsub("^.*\\.", "", Gene))  %>%
  mutate(EnsID=gsub("\\..*","", Gene))

##GSEA Fibroblast Group
DEGenes_Fb_TChigh <- DEGenesFb %>% filter(cluster == "Tcell High")
egoFb <- enrichGO(gene = unique(DEGenes_Fb_TChigh$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                        

egoFb <- setReadable(egoFb, OrgDb = org.Hs.eg.db)

#write.table(egoFb@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoFb for TC high 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

#View (egoFb@result)

# negative regulation of T-helper 17 cell differentiation: RC3H2;  In resting or LPS-stimulated macrophages, controls inflammation by suppressing TNF expression. 
# chemokine (C-C motif) ligand 5 production TRPV4
# NRARP, PTPN6 (IFN pathway, also Integrins), PLEKHA4
# positive regulation of steroid process: CES1/ERRFI1
# PLEKHM2
# regulation of defense response to virus: APOBEC3G
# TNF pathway COMMD7

egoFb_df <- as.data.frame(egoFb@result)


egoTCFb_df <- egoFb_df %>%
 arrange(desc(zScore))


ggplot(egoFb_df[1:21, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Fibroblast Differential Expression by zScore for T cell High Condition", y = "z Score", x = "Description")


egoFb_df <- egoFb_df %>%
  arrange(desc(RichFactor))

ggplot(egoFb_df[1:18, ], aes(x = reorder(Description, RichFactor), y = RichFactor)) +
  geom_bar(stat = "identity", aes(fill = RichFactor)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Fibroblast Differential Expression by Rich Factor for T cell High Condition", y = "RichFactor", x = "Description")


```



```{r}

##GSEA Fibroblast
DEGenes_Fb_TCvlow <- DEGenesFb %>% filter(cluster == "Tcell Very Low")
egoFb_vlow <- enrichGO(gene = unique(DEGenes_Fb_TCvlow$EnsID),
                          OrgDb = org.Hs.eg.db,
                          keyType = 'ENSEMBL',
                          ont = "BP",
                          pAdjustMethod = "BH")
                          #pvalueCutoff = 0.05,
                          #qvalueCutoff = 0.05)

egoFb_vlow <- setReadable(egoFb_vlow, OrgDb = org.Hs.eg.db)

#write.table(egoFb_vlow@result, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers/egoFb for TC very low 12.10.2024", sep="\t",quote=F,row.names=F,col.names=T)

#View (egoFb_vlow@result)

# negative regulation of T-helper 17 cell differentiation: RC3H2;  In resting or LPS-stimulated macrophages, controls inflammation by suppressing TNF expression. 
# chemokine (C-C motif) ligand 5 production TRPV4
# NRARP, PTPN6 (IFN pathway, also Integrins), PLEKHA4
# positive regulation of steroid process: CES1/ERRFI1
# PLEKHM2
# regulation of defense response to virus: APOBEC3G
# TNF pathway COMMD7

egoFb_vlow_df <- as.data.frame(egoFb_vlow@result)


egoFb_vlow_df <- egoFb_df %>%
 arrange(desc(zScore))


ggplot(egoFb_vlow_df[1:21, ], aes(x = reorder(Description, zScore), y = zScore)) +
  geom_bar(stat = "identity", aes(fill = zScore)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "Fibroblast Differential Expression by zScore for T cell Very Low Condition", y = "z Score", x = "Description")

egoFb_vlow_df <- egoFb_vlow_df %>%
  arrange(desc(RichFactor))

ggplot(egoTC_vlow_df[1:30, ], aes(x = reorder(Description, RichFactor), y = RichFactor)) +
  geom_bar(stat = "identity", aes(fill = RichFactor)) +
  coord_flip() +
  theme_classic() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  labs(title = "T cell Differential Expression by Rich Factor for T cell Very Low Condition", y = "RichFactor", x = "Description")

```





```{r Tcell high vs. low accross all clusters}
table(seuratM$clusterName)
table(seuratM$patient)
table(seuratM$pat_sub)

Idents(seuratM) <- seuratM$TC

DimPlot(seuratM, reduction = "umap")

levels(seuratTcell)

table(seuratM$TC)

# Total: TC High N = 9034, Tcell Intermediate N = 7227, Tcell Low = 21679, Tcell Very Low = 36130$

DEGenes_all <- FindAllMarkers(seuratM, only.pos = T, logfc.threshold = 0.2) 
head(DEGenes_all)

DEGenes_adj <- FindAllMarkers(seuratM, only.pos = T, logfc.threshold = 0.2) %>% 
  dplyr::filter(p_val_adj < 0.05)


#View(DEGenes_all)
```


```{r save RDS with Tcell high vs. low included in metadata}


#saveRDS(seuratM, file="/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_clusters_TCmetadata_27.09.2024.rds")

#write.table(dat_all, file= "/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/docs/Markers_dat_all_clusters_Tcell_high_lowV2", sep = "\t",quote=F,row.names=F,col.names=T)

#saveRDS(seuratM, file="/Users/immbio/Desktop/Emily/Myocarditis snRNAseq/data/01 Merged Seurat files/Myocarditis_clusters_Tc_condition_26.09.24.rds")

```


```{r}
date()
sessionInfo()
```


